<!DOCTYPE html>
<html lang="en">
<head>
<meta content="text/html; charset=utf-8" http-equiv="content-type"/>
<title>Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity</title>
<!--Generated on Fri Oct  4 04:25:14 2024 by LaTeXML (version 0.8.8) http://dlmf.nist.gov/LaTeXML/.-->
<meta content="width=device-width, initial-scale=1, shrink-to-fit=no" name="viewport"/>
<link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0/dist/css/bootstrap.min.css" rel="stylesheet" type="text/css"/>
<link href="/static/browse/0.3.4/css/ar5iv.0.7.9.min.css" rel="stylesheet" type="text/css"/>
<link href="/static/browse/0.3.4/css/ar5iv-fonts.0.7.9.min.css" rel="stylesheet" type="text/css"/>
<link href="/static/browse/0.3.4/css/latexml_styles.css" rel="stylesheet" type="text/css"/>
<script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0/dist/js/bootstrap.bundle.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/html2canvas/1.3.3/html2canvas.min.js"></script>
<script src="/static/browse/0.3.4/js/addons_new.js"></script>
<script src="/static/browse/0.3.4/js/feedbackOverlay.js"></script>
<base href="/html/2410.03138v1/"/></head>
<body>
<nav class="ltx_page_navbar">
<nav class="ltx_TOC">
<ol class="ltx_toclist">
<li class="ltx_tocentry ltx_tocentry_section"><a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S1" title="In Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_title"><span class="ltx_tag ltx_tag_ref">1 </span>Introduction</span></a></li>
<li class="ltx_tocentry ltx_tocentry_section"><a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S2" title="In Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_title"><span class="ltx_tag ltx_tag_ref">2 </span>Related work</span></a></li>
<li class="ltx_tocentry ltx_tocentry_section">
<a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S3" title="In Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_title"><span class="ltx_tag ltx_tag_ref">3 </span>Method</span></a>
<ol class="ltx_toclist ltx_toclist_section">
<li class="ltx_tocentry ltx_tocentry_subsection"><a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S3.SS1" title="In 3 Method ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_title"><span class="ltx_tag ltx_tag_ref">3.1 </span>Supervised fine-tuning</span></a></li>
<li class="ltx_tocentry ltx_tocentry_subsection"><a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S3.SS2" title="In 3 Method ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_title"><span class="ltx_tag ltx_tag_ref">3.2 </span>Reinforcement learning</span></a></li>
</ol>
</li>
<li class="ltx_tocentry ltx_tocentry_section">
<a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4" title="In Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_title"><span class="ltx_tag ltx_tag_ref">4 </span>Experiment</span></a>
<ol class="ltx_toclist ltx_toclist_section">
<li class="ltx_tocentry ltx_tocentry_subsection"><a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.SS1" title="In 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_title"><span class="ltx_tag ltx_tag_ref">4.1 </span>Comparison with decoding schemes</span></a></li>
<li class="ltx_tocentry ltx_tocentry_subsection"><a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.SS2" title="In 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_title"><span class="ltx_tag ltx_tag_ref">4.2 </span>Comparison with existing LLMs</span></a></li>
<li class="ltx_tocentry ltx_tocentry_subsection"><a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.SS3" title="In 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_title"><span class="ltx_tag ltx_tag_ref">4.3 </span>Fine-tuning generalist LLMs</span></a></li>
<li class="ltx_tocentry ltx_tocentry_subsection"><a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.SS4" title="In 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_title"><span class="ltx_tag ltx_tag_ref">4.4 </span>Ablation studies</span></a></li>
</ol>
</li>
<li class="ltx_tocentry ltx_tocentry_section"><a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S5" title="In Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_title"><span class="ltx_tag ltx_tag_ref">5 </span>Conclusion</span></a></li>
<li class="ltx_tocentry ltx_tocentry_appendix">
<a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#A1" title="In Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_title"><span class="ltx_tag ltx_tag_ref">A </span>Detailed implementations and training</span></a>
<ol class="ltx_toclist ltx_toclist_appendix">
<li class="ltx_tocentry ltx_tocentry_subsection"><a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#A1.SS1" title="In Appendix A Detailed implementations and training ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_title"><span class="ltx_tag ltx_tag_ref">A.1 </span>Supervised fine-tuning</span></a></li>
<li class="ltx_tocentry ltx_tocentry_subsection"><a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#A1.SS2" title="In Appendix A Detailed implementations and training ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_title"><span class="ltx_tag ltx_tag_ref">A.2 </span>Reinforcement learning</span></a></li>
</ol>
</li>
<li class="ltx_tocentry ltx_tocentry_appendix"><a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#A2" title="In Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_title"><span class="ltx_tag ltx_tag_ref">B </span>Detailed experimental settings</span></a></li>
<li class="ltx_tocentry ltx_tocentry_appendix"><a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#A3" title="In Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_title"><span class="ltx_tag ltx_tag_ref">C </span>Additional results</span></a></li>
</ol></nav>
</nav>
<div class="ltx_page_main">
<div class="ltx_page_content">
<article class="ltx_document ltx_authors_1line">
<h1 class="ltx_title ltx_title_document">Can LLMs Generate Diverse Molecules? 
<br class="ltx_break"/>Towards Alignment with Structural Diversity</h1>
<div class="ltx_authors">
<span class="ltx_creator ltx_role_author">
<span class="ltx_personname">Hyosoon Jang<sup class="ltx_sup" id="id7.7.id1">1</sup>, Yunhui Jang<sup class="ltx_sup" id="id8.8.id2">1</sup>, Jaehyung Kim<sup class="ltx_sup" id="id9.9.id3">2</sup>, Sungsoo Ahn<sup class="ltx_sup" id="id10.10.id4">1</sup>
<br class="ltx_break"/><sup class="ltx_sup" id="id11.11.id5">1</sup>POSTECH  <sup class="ltx_sup" id="id12.12.id6">2</sup>Yonsei University 
<br class="ltx_break"/><span class="ltx_text ltx_font_typewriter" id="id13.13.id7">{hsjang1205,uni5510,sungsoo.ahn}@postech.ac.kr</span>
<br class="ltx_break"/><span class="ltx_text ltx_font_typewriter" id="id14.14.id8">jaehyungk@yonsei.ac.kr</span>
</span></span>
</div>
<div class="ltx_abstract">
<h6 class="ltx_title ltx_title_abstract">Abstract</h6>
<p class="ltx_p" id="id15.id1">Recent advancements in large language models (LLMs) have demonstrated impressive performance in generating molecular structures as drug candidates, which offers significant potential to accelerate drug discovery. However, the current LLMs overlook a critical requirement for drug discovery: proposing a diverse set of molecules. This diversity is essential for improving the chances of finding a viable drug, as it provides alternative molecules that may succeed where others fail in wet-lab or clinical validations. Despite such a need for diversity, the LLMs often output structurally similar molecules from a given prompt. While decoding schemes like beam search may enhance textual diversity, this often does not align with molecular structural diversity. In response, we propose a new method for fine-tuning molecular generative LLMs to <span class="ltx_text ltx_font_italic" id="id15.id1.1">autoregressively generate a set of structurally diverse molecules</span>, where each molecule is generated by conditioning on the previously generated molecules. Our approach consists of two stages: (1) supervised fine-tuning to adapt LLMs to autoregressively generate molecules in a sequence and (2) reinforcement learning to maximize structural diversity within the generated molecules. Our experiments show that (1) our fine-tuning approach enables the LLMs to better discover diverse molecules compared to existing decoding schemes and (2) our fine-tuned model outperforms other representative LLMs in generating diverse molecules, including the ones fine-tuned on chemical domains.</p>
</div>
<section class="ltx_section" id="S1">
<h2 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">1 </span>Introduction</h2>
<figure class="ltx_figure ltx_align_floatright" id="S1.F1"><img alt="Refer to caption" class="ltx_graphics ltx_centering ltx_img_landscape" height="414" id="S1.F1.g1" src="x1.png" width="830"/>
<figcaption class="ltx_caption ltx_centering"><span class="ltx_tag ltx_tag_figure">Figure 1: </span><span class="ltx_text ltx_font_bold" id="S1.F1.2.1">LLMs generate a molecule with SMILES representation.</span></figcaption>
</figure>
<div class="ltx_para ltx_noindent" id="S1.p1">
<p class="ltx_p" id="S1.p1.1">Recent advances in large language models (LLMs) have demonstrated remarkable potential to accelerate scientific discovery by leveraging their language processing capabilities. This progress has been particularly impactful for candidate design problems such as drug discovery <cite class="ltx_cite ltx_citemacro_citep">(Pei et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib26" title="">2024</a>)</cite>, protein sequence design <cite class="ltx_cite ltx_citemacro_citep">(Zhuo et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib45" title="">2024</a>)</cite>, and material design <cite class="ltx_cite ltx_citemacro_citep">(Gruver et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib11" title="">2024</a>)</cite>. In particular, with expansive biomolecular datasets and molecular string representations, e.g., SMILES <cite class="ltx_cite ltx_citemacro_citep">(Weininger, <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib37" title="">1988</a>)</cite> or SELFIES <cite class="ltx_cite ltx_citemacro_citep">(Krenn et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib17" title="">2020</a>)</cite>, LLMs have demonstrated impressive abilities to generate molecules from textual descriptions, e.g., molecular properties <cite class="ltx_cite ltx_citemacro_citep">(Edwards et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib7" title="">2022</a>; Ye et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib41" title="">2023</a>; Pei et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib26" title="">2024</a>)</cite>, as illustrated in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S1.F1" title="In 1 Introduction ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Figure</span> <span class="ltx_text ltx_ref_tag">1</span></a>.</p>
</div>
<div class="ltx_para ltx_noindent" id="S1.p2">
<p class="ltx_p" id="S1.p2.1">However, current LLM-based molecular generation approaches <cite class="ltx_cite ltx_citemacro_citep">(Edwards et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib7" title="">2022</a>; Ye et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib41" title="">2023</a>; Pei et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib26" title="">2024</a>)</cite> often overlook a critical requirement for drug discovery:<span class="ltx_text ltx_font_italic" id="S1.p2.1.1"> proposing a diverse set of molecules</span>. In computer-aided drug discovery pipelines, identifying a single molecule with a desired property does not guarantee success in real-world pipelines that require additional cell-based studies and clinical trials <cite class="ltx_cite ltx_citemacro_citep">(Vamathevan et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib34" title="">2019</a>)</cite>. Therefore, drug discovery requires a collection of structurally diverse molecules, as illustrated in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S1.F2" title="In 1 Introduction ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Figure</span> <span class="ltx_text ltx_ref_tag">2</span></a>. The generation of structurally diverse molecules increases the chances of finding a viable drug candidate <cite class="ltx_cite ltx_citemacro_citep">(Xie et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib40" title="">2023</a>)</cite>, as different molecules may succeed where others fail. This diversity is essential to enhance the robustness and success of the drug discovery process <cite class="ltx_cite ltx_citemacro_citep">(Krantz, <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib16" title="">1998</a>; Hong et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib13" title="">2020</a>; Sadybekov &amp; Katritch, <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib28" title="">2023</a>)</cite>.</p>
</div>
<figure class="ltx_figure" id="S1.F2"><img alt="Refer to caption" class="ltx_graphics ltx_centering ltx_img_landscape" height="111" id="S1.F2.g1" src="x2.png" width="715"/>
<figcaption class="ltx_caption ltx_centering"><span class="ltx_tag ltx_tag_figure">Figure 2: </span><span class="ltx_text ltx_font_bold" id="S1.F2.2.1">The compute-aided drug design.</span> A collection of structurally diverse molecules is required to increase the chance of identifying viable drug candidates in the real world.</figcaption>
</figure>
<div class="ltx_para ltx_noindent" id="S1.p3">
<p class="ltx_p" id="S1.p3.1">In response, we explore the use of LLMs for diverse molecular generation. We begin by identifying the limitations of recent LLMs <cite class="ltx_cite ltx_citemacro_citep">(Ye et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib41" title="">2023</a>; OpenAI, <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib21" title="">2023</a>)</cite> and decoding schemes <cite class="ltx_cite ltx_citemacro_citep">(Vijayakumar et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib35" title="">2018</a>; Su et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib31" title="">2022</a>)</cite> in generating diverse molecules. Then, we present a new method for fine-tuning LLMs to generate diverse molecules. Our approach can be broadly applied to other LLM-based candidate design problems, e.g., computer-aided design <cite class="ltx_cite ltx_citemacro_citep">(Wu et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib38" title="">2023a</a>)</cite>.</p>
</div>
<div class="ltx_para ltx_noindent" id="S1.p4">
<p class="ltx_p" id="S1.p4.1"><span class="ltx_text ltx_font_bold" id="S1.p4.1.1">Existing LLMs have limitations in generating diverse molecules.</span> To obtain diverse molecules, one may consider querying the recently developed generalist LLMs, e.g., Llama <cite class="ltx_cite ltx_citemacro_citep">(Touvron et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib33" title="">2023</a>)</cite> or ChatGPT <cite class="ltx_cite ltx_citemacro_citep">(OpenAI, <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib21" title="">2023</a>)</cite>. However, our empirical observation in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S1.F3.sf1" title="In Figure 3 ‣ 1 Introduction ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Figure</span> <span class="ltx_text ltx_ref_tag">3(a)</span></a> reveals that even the most recent models produce structurally identical or highly similar molecules from the given prompt.<span class="ltx_note ltx_role_footnote" id="footnote1"><sup class="ltx_note_mark">1</sup><span class="ltx_note_outer"><span class="ltx_note_content"><sup class="ltx_note_mark">1</sup><span class="ltx_tag ltx_tag_note">1</span>ChatGPT-4o <cite class="ltx_cite ltx_citemacro_citep">(OpenAI, <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib21" title="">2023</a>)</cite> generates different SMILES strings that map to an identical molecule.</span></span></span> This observation aligns with previous observations that have shown LLMs may fail to generate diverse outputs <cite class="ltx_cite ltx_citemacro_citep">(Kirk et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib15" title="">2024</a>)</cite> for general text-based domains.</p>
</div>
<div class="ltx_para ltx_noindent" id="S1.p5">
<p class="ltx_p" id="S1.p5.1"><span class="ltx_text ltx_font_bold" id="S1.p5.1.1">Decoding schemes for diversified generation do not align with molecular diversity.</span> We also acknowledge the existence of decoding schemes, e.g., diverse beam search <cite class="ltx_cite ltx_citemacro_citep">(Vijayakumar et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib35" title="">2018</a>)</cite> or contrastive beam search <cite class="ltx_cite ltx_citemacro_citep">(Su et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib31" title="">2022</a>)</cite>, which have been proposed to improve the diversity of output sequences generated by LLMs. However, these decoding schemes are limited to improving the textual diversity which often does not correspond to molecular structural diversity, e.g., there exist many SMILES or SELFIES strings that correspond to the same molecule, as illustrated in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S1.F3.sf2" title="In Figure 3 ‣ 1 Introduction ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Figure</span> <span class="ltx_text ltx_ref_tag">3(b)</span></a>.</p>
</div>
<figure class="ltx_figure" id="S1.F3">
<div class="ltx_flex_figure">
<div class="ltx_flex_cell ltx_flex_size_1">
<figure class="ltx_figure ltx_figure_panel" id="S1.F3.sf1"><img alt="Refer to caption" class="ltx_graphics ltx_centering ltx_img_landscape" height="126" id="S1.F3.sf1.g1" src="x3.png" width="780"/>
<figcaption class="ltx_caption ltx_centering"><span class="ltx_tag ltx_tag_figure">(a) </span>Existing LLMs <cite class="ltx_cite ltx_citemacro_citep">(Ye et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib41" title="">2023</a>; OpenAI, <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib21" title="">2023</a>)</cite> lack the ability to generate a diverse set of molecules.</figcaption>
</figure>
</div>
<div class="ltx_flex_break"></div>
<div class="ltx_flex_cell ltx_flex_size_1">
<figure class="ltx_figure ltx_figure_panel ltx_align_center" id="S1.F3.sf2"><img alt="Refer to caption" class="ltx_graphics ltx_centering ltx_img_landscape" height="132" id="S1.F3.sf2.g1" src="x4.png" width="783"/>
<figcaption class="ltx_caption ltx_centering"><span class="ltx_tag ltx_tag_figure">(b) </span><span class="ltx_text ltx_font_bold" id="S1.F3.sf2.3.1">(Left)</span> Diverse output sequences (SMILES) induce the same molecular structures. <span class="ltx_text ltx_font_bold" id="S1.F3.sf2.4.2">(Right)</span> Improved textual diversity via diverse beam search does not enhance molecular diversity in the experiments (<a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.SS1" title="4.1 Comparison with decoding schemes ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Section</span> <span class="ltx_text ltx_ref_tag">4.1</span></a>).</figcaption>
</figure>
</div>
</div>
<figcaption class="ltx_caption ltx_centering"><span class="ltx_tag ltx_tag_figure">Figure 3: </span><span class="ltx_text ltx_font_bold" id="S1.F3.3.1">Existing works on LLMs fail to generate diverse molecules.</span> The existing decoding schemes <cite class="ltx_cite ltx_citemacro_citep">(Vijayakumar et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib35" title="">2018</a>)</cite> for diverse sequence generation and LLMs for chemical tasks fail to capture the molecular diversity, and may induce <span class="ltx_text" id="S1.F3.4.2" style="background-color:#F79494;">structurally identical molecules</span>.</figcaption>
</figure>
<div class="ltx_para ltx_noindent" id="S1.p6">
<p class="ltx_p" id="S1.p6.1"><span class="ltx_text ltx_font_bold" id="S1.p6.1.1">Our approach.</span> We repurpose existing molecular generative LLMs to autoregressively generate a diverse set of molecules from a single prompt. By enabling the LLMs to generate a new molecule conditioned on previously generated molecules, we expect the LLMs to learn to enhance the structural diversity between the generated molecules. To this end, we propose a two-stage approach to fine-tune LLMs: (a) a supervised fine-tuning stage to repurpose LLMs to autoregressively generate a sequence of multiple molecules and (b) a reinforcement learning stage to maximize the molecular structural diversity. Note that both stages do not require external diverse molecular datasets and are purely based on self-improvement procedures, where the LLMs train on the samples generated by themselves.</p>
</div>
<div class="ltx_para ltx_noindent" id="S1.p7">
<p class="ltx_p" id="S1.p7.1">In the supervised training stage, we train the LLMs to autoregressively generate a set of molecules in a single sequence. The training dataset, i.e., a set of molecules, is collected from the LLMs themselves through iterative sampling, and then filtered to enhance the quality, e.g., removing invalid molecules. However, this stage does not necessarily incorporate molecular diversity, as the training may not involve sufficiently distinct molecules (e.g., limitations in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S1.F3.sf2" title="In Figure 3 ‣ 1 Introduction ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Figure</span> <span class="ltx_text ltx_ref_tag">3(b)</span></a>). To tackle this, we subsequently apply reinforcement learning with exploration towards discovering diverse molecules.</p>
</div>
<div class="ltx_para ltx_noindent" id="S1.p8">
<p class="ltx_p" id="S1.p8.1">Next, in the reinforcement learning stage, we train LLMs to maximize the diversity of molecules within a generated sequence. However, for our task, conventional sequence-wise reinforcement learning <cite class="ltx_cite ltx_citemacro_citep">(Ouyang et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib23" title="">2022</a>)</cite> suffers from the credit assignment problem <cite class="ltx_cite ltx_citemacro_citep">(Zhou et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib44" title="">2024</a>)</cite>: the challenges in identifying and promoting the generation of molecules responsible for increasing diversity, among a larger set of molecules in the sequence. To resolve this issue, we solve multi-stage molecule generation problems for a sequence of molecules, where the generation of each molecule aims to maximize the diversity with respect to the previously generated molecules. We train LLMs to maximize the associated rewards using proximal policy optimization <cite class="ltx_cite ltx_citemacro_citep">(Schulman et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib29" title="">2017</a>)</cite>.</p>
</div>
<div class="ltx_para ltx_noindent" id="S1.p9">
<p class="ltx_p" id="S1.p9.1">We compare our method with the decoding schemes for diversified generation <cite class="ltx_cite ltx_citemacro_citep">(Vijayakumar et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib36" title="">2016</a>; Su et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib31" title="">2022</a>)</cite> and other representative LLMs, including chemical-task specialists <cite class="ltx_cite ltx_citemacro_citep">(Edwards et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib7" title="">2022</a>; Christofidellis et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib5" title="">2023</a>; Pei et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib25" title="">2023</a>; <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib26" title="">2024</a>)</cite>, fine-tuned generalists on chemical domains <cite class="ltx_cite ltx_citemacro_citep">(Fang et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib8" title="">2024</a>; Yu et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib42" title="">2024</a>)</cite>, and the ChatGPT series <cite class="ltx_cite ltx_citemacro_citep">(OpenAI, <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib21" title="">2023</a>; <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib22" title="">2024</a>)</cite>. We observe that (1) our fine-tuning approach enables LLMs to better discover diverse molecules compared to existing decoding schemes and (2) our fine-tuned LLM outperforms other recent LLMs.</p>
</div>
<div class="ltx_para ltx_noindent" id="S1.p10">
<p class="ltx_p" id="S1.p10.1">To conclude, our contributions can be summarized as follows:</p>
<ul class="ltx_itemize" id="S1.I1">
<li class="ltx_item" id="S1.I1.i1" style="list-style-type:none;">
<span class="ltx_tag ltx_tag_item">•</span>
<div class="ltx_para" id="S1.I1.i1.p1">
<p class="ltx_p" id="S1.I1.i1.p1.1">We are the first to explore the use of LLMs for generating diverse molecules.</p>
</div>
</li>
<li class="ltx_item" id="S1.I1.i2" style="list-style-type:none;">
<span class="ltx_tag ltx_tag_item">•</span>
<div class="ltx_para" id="S1.I1.i2.p1">
<p class="ltx_p" id="S1.I1.i2.p1.1">We first propose a fine-tuning approach for LLMs to generate diverse solutions, which presents a new direction distinct from existing approaches focused on the decoding scheme.</p>
</div>
</li>
<li class="ltx_item" id="S1.I1.i3" style="list-style-type:none;">
<span class="ltx_tag ltx_tag_item">•</span>
<div class="ltx_para ltx_noindent" id="S1.I1.i3.p1">
<p class="ltx_p" id="S1.I1.i3.p1.1">Experimentally, our method outperforms the baselines in generating diverse molecular structures.</p>
</div>
</li>
</ul>
</div>
</section>
<section class="ltx_section" id="S2">
<h2 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">2 </span>Related work</h2>
<div class="ltx_para ltx_noindent" id="S2.p1">
<p class="ltx_p" id="S2.p1.1"><span class="ltx_text ltx_font_bold" id="S2.p1.1.1">Large language models (LLMs) for molecular generation.</span> Recent advancements in LLMs have shown increasing promise in scientific applications, especially for molecular generation <cite class="ltx_cite ltx_citemacro_citep">(Edwards et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib7" title="">2022</a>; Pei et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib25" title="">2023</a>; Fang et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib8" title="">2024</a>; Pei et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib26" title="">2024</a>)</cite>. First, <cite class="ltx_cite ltx_citemacro_cite">Edwards et al. (<a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib7" title="">2022</a>)</cite> proposed MolT5 which translates between SMILES <cite class="ltx_cite ltx_citemacro_citep">(Weininger, <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib37" title="">1988</a>)</cite> and molecular text descriptions. Text+Chem T5 <cite class="ltx_cite ltx_citemacro_citep">(Christofidellis et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib5" title="">2023</a>)</cite> is a model pre-trained on both the chemical and natural language domains. Next, BioT5 <cite class="ltx_cite ltx_citemacro_citep">(Pei et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib25" title="">2023</a>)</cite> considers T5 models pre-trained on datasets including bio-text, protein sequences, and molecules. Additionally, <cite class="ltx_cite ltx_citemacro_citet">Ye et al. (<a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib41" title="">2023</a>)</cite>, <cite class="ltx_cite ltx_citemacro_citet">Fang et al. (<a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib8" title="">2024</a>)</cite>, and <cite class="ltx_cite ltx_citemacro_citet">Yu et al. (<a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib42" title="">2024</a>)</cite> fine-tuned generalist LLMs, e.g., Llama <cite class="ltx_cite ltx_citemacro_citep">(Touvron et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib33" title="">2023</a>)</cite>, through biological instructions, molecular modifications, and large-scale molecular datasets, respectively.</p>
</div>
<div class="ltx_para ltx_noindent" id="S2.p2">
<p class="ltx_p" id="S2.p2.1"><span class="ltx_text ltx_font_bold" id="S2.p2.1.1">Decoding schemes for generating diverse output sequences.</span> To generate diverse and high-quality solution candidates from LLMs, existing literature on LLMs has studied improving decoding schemes. To acquire multiple distinct sequences with high likelihoods, one can consider employing beam search, which jointly decodes multiple distinct outputs <cite class="ltx_cite ltx_citemacro_citep">(Och, <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib20" title="">2003</a>)</cite>. To enhance diversity between sequences, <cite class="ltx_cite ltx_citemacro_citet">Vijayakumar et al. (<a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib36" title="">2016</a>; <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib35" title="">2018</a>)</cite> incorporated token-wise differences between generated sequences in the beam search. Furthermore, <cite class="ltx_cite ltx_citemacro_citet">Su et al. (<a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib31" title="">2022</a>)</cite> considered the contrast between the candidate sequences. In addition, <cite class="ltx_cite ltx_citemacro_cite">Holtzman et al. (<a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib12" title="">2020</a>)</cite> proposed nucleus sampling, which enhances random sampling by balancing the quality and the diversity.</p>
</div>
<div class="ltx_para ltx_noindent" id="S2.p3">
<p class="ltx_p" id="S2.p3.1"><span class="ltx_text ltx_font_bold" id="S2.p3.1.1">Reinforcement learning (RL) for fine-tuning LLMs.</span> RL has been effectively applied to fine-tune LLMs, aligning them with desired behaviors expressed through reward signals. One notable example is RL from human feedback to align LLMs with human preference <cite class="ltx_cite ltx_citemacro_citep">(Ouyang et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib23" title="">2022</a>)</cite>. In addition, there has been a surge in research on devising RL for LLMs as well, such as addressing multi-turn settings <cite class="ltx_cite ltx_citemacro_citep">(Shani et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib30" title="">2024</a>)</cite> and incorporating multiple fine-grained reward signals <cite class="ltx_cite ltx_citemacro_citep">(Wu et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib39" title="">2023b</a>)</cite>. For molecular generation, <cite class="ltx_cite ltx_citemacro_citet">Ghugare et al. (<a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib10" title="">2024</a>)</cite> proposed RL-based fine-tuning to generate a molecule satisfying target properties. However, to the best of our knowledge, there exist no prior RL-based approaches that aim to increase the diversity of LLM-generated outputs.</p>
</div>
<figure class="ltx_figure" id="S2.F4">
<div class="ltx_flex_figure">
<div class="ltx_flex_cell ltx_flex_size_1">
<figure class="ltx_figure ltx_figure_panel ltx_align_center" id="S2.F4.sf1"><img alt="Refer to caption" class="ltx_graphics ltx_centering ltx_img_landscape" height="120" id="S2.F4.sf1.g1" src="x5.png" width="789"/>
<figcaption class="ltx_caption ltx_centering"><span class="ltx_tag ltx_tag_figure">(a) </span><span class="ltx_text ltx_font_bold" id="S2.F4.sf1.2.1">Supervised fine-tuning:</span> enables generating a sequence of multiple molecules.</figcaption>
</figure>
</div>
<div class="ltx_flex_break"></div>
<div class="ltx_flex_cell ltx_flex_size_1">
<figure class="ltx_figure ltx_figure_panel ltx_align_center" id="S2.F4.sf2"><img alt="Refer to caption" class="ltx_graphics ltx_centering ltx_img_landscape" height="105" id="S2.F4.sf2.g1" src="x6.png" width="789"/>
<figcaption class="ltx_caption ltx_centering"><span class="ltx_tag ltx_tag_figure">(b) </span><span class="ltx_text ltx_font_bold" id="S2.F4.sf2.2.1">Reinforcement learning:</span> maximizes the reward of generated molecules.</figcaption>
</figure>
</div>
<div class="ltx_flex_break"></div>
<div class="ltx_flex_cell ltx_flex_size_1">
<figure class="ltx_figure ltx_figure_panel ltx_align_center" id="S2.F4.sf3"><img alt="Refer to caption" class="ltx_graphics ltx_centering ltx_img_landscape" height="118" id="S2.F4.sf3.g1" src="x7.png" width="789"/>
<figcaption class="ltx_caption ltx_centering"><span class="ltx_tag ltx_tag_figure">(c) </span><span class="ltx_text ltx_font_bold" id="S2.F4.sf3.2.1">Reward computation:</span> captures the structural diversity.</figcaption>
</figure>
</div>
</div>
<figcaption class="ltx_caption ltx_centering"><span class="ltx_tag ltx_tag_figure">Figure 4: </span><span class="ltx_text ltx_font_bold" id="S2.F4.2.1">Illustration of proposed fine-tuning approaches.</span> We consider two stages for fine-tuning LLMs: a supervised fine-tuning <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S2.F4.sf1" title="In Figure 4 ‣ 2 Related work ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Figure</span> <span class="ltx_text ltx_ref_tag">4(a)</span></a> and a reinforcement learning <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S2.F4.sf2" title="In Figure 4 ‣ 2 Related work ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Figure</span> <span class="ltx_text ltx_ref_tag">4(b)</span></a>. The prompts are simplified for explanatory purposes, and the actual prompts are provided in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#A2" title="Appendix B Detailed experimental settings ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Appendix</span> <span class="ltx_text ltx_ref_tag">B</span></a>.</figcaption>
</figure>
</section>
<section class="ltx_section" id="S3">
<h2 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">3 </span>Method</h2>
<div class="ltx_para ltx_noindent" id="S3.p1">
<p class="ltx_p" id="S3.p1.1">In this section, we present our method for fine-tuning LLMs to generate diverse molecules. Specifically, we consider applying fine-tuning to existing molecular generative LLMs that produce molecular representations such as SMILES or SELFIES. Importantly, our approach is versatile and can be broadly applied to other domains, e.g., protein sequence <cite class="ltx_cite ltx_citemacro_citep">(Zhuo et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib45" title="">2024</a>)</cite> or computer-aided design <cite class="ltx_cite ltx_citemacro_citep">(Wu et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib38" title="">2023a</a>)</cite>. Furthermore, it leverages self-improvement techniques and does not require additional datasets containing diverse molecules.</p>
</div>
<div class="ltx_para ltx_noindent" id="S3.p2">
<p class="ltx_p" id="S3.p2.1"><span class="ltx_text ltx_font_bold" id="S3.p2.1.1">Overview.</span> Our goal is to generate a sequence of structurally diverse molecules from a given prompt by producing them in a single concatenated output. To achieve this, we fine-tune the LLMs in two stages: (a) a supervised fine-tuning phase that repurposes the LLMs to generate a sequence of molecules rather than a single one, and (b) a reinforcement learning phase aimed at further enhancing the structural diversity among the generated molecules.</p>
</div>
<div class="ltx_para ltx_noindent" id="S3.p3">
<p class="ltx_p" id="S3.p3.3"><span class="ltx_text ltx_font_bold" id="S3.p3.3.1">Task details.</span> In detail, we consider generating molecules from a prompt <math alttext="p_{\text{desc}}" class="ltx_Math" display="inline" id="S3.p3.1.m1.1"><semantics id="S3.p3.1.m1.1a"><msub id="S3.p3.1.m1.1.1" xref="S3.p3.1.m1.1.1.cmml"><mi id="S3.p3.1.m1.1.1.2" xref="S3.p3.1.m1.1.1.2.cmml">p</mi><mtext id="S3.p3.1.m1.1.1.3" xref="S3.p3.1.m1.1.1.3a.cmml">desc</mtext></msub><annotation-xml encoding="MathML-Content" id="S3.p3.1.m1.1b"><apply id="S3.p3.1.m1.1.1.cmml" xref="S3.p3.1.m1.1.1"><csymbol cd="ambiguous" id="S3.p3.1.m1.1.1.1.cmml" xref="S3.p3.1.m1.1.1">subscript</csymbol><ci id="S3.p3.1.m1.1.1.2.cmml" xref="S3.p3.1.m1.1.1.2">𝑝</ci><ci id="S3.p3.1.m1.1.1.3a.cmml" xref="S3.p3.1.m1.1.1.3"><mtext id="S3.p3.1.m1.1.1.3.cmml" mathsize="70%" xref="S3.p3.1.m1.1.1.3">desc</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.p3.1.m1.1c">p_{\text{desc}}</annotation><annotation encoding="application/x-llamapun" id="S3.p3.1.m1.1d">italic_p start_POSTSUBSCRIPT desc end_POSTSUBSCRIPT</annotation></semantics></math>, where the prompt describes a molecular property that the generated molecules should possess. Here, we aim to generate diverse molecules that satisfy the given description <math alttext="p_{\text{desc}}" class="ltx_Math" display="inline" id="S3.p3.2.m2.1"><semantics id="S3.p3.2.m2.1a"><msub id="S3.p3.2.m2.1.1" xref="S3.p3.2.m2.1.1.cmml"><mi id="S3.p3.2.m2.1.1.2" xref="S3.p3.2.m2.1.1.2.cmml">p</mi><mtext id="S3.p3.2.m2.1.1.3" xref="S3.p3.2.m2.1.1.3a.cmml">desc</mtext></msub><annotation-xml encoding="MathML-Content" id="S3.p3.2.m2.1b"><apply id="S3.p3.2.m2.1.1.cmml" xref="S3.p3.2.m2.1.1"><csymbol cd="ambiguous" id="S3.p3.2.m2.1.1.1.cmml" xref="S3.p3.2.m2.1.1">subscript</csymbol><ci id="S3.p3.2.m2.1.1.2.cmml" xref="S3.p3.2.m2.1.1.2">𝑝</ci><ci id="S3.p3.2.m2.1.1.3a.cmml" xref="S3.p3.2.m2.1.1.3"><mtext id="S3.p3.2.m2.1.1.3.cmml" mathsize="70%" xref="S3.p3.2.m2.1.1.3">desc</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.p3.2.m2.1c">p_{\text{desc}}</annotation><annotation encoding="application/x-llamapun" id="S3.p3.2.m2.1d">italic_p start_POSTSUBSCRIPT desc end_POSTSUBSCRIPT</annotation></semantics></math>. We let <math alttext="\mathcal{P}" class="ltx_Math" display="inline" id="S3.p3.3.m3.1"><semantics id="S3.p3.3.m3.1a"><mi class="ltx_font_mathcaligraphic" id="S3.p3.3.m3.1.1" xref="S3.p3.3.m3.1.1.cmml">𝒫</mi><annotation-xml encoding="MathML-Content" id="S3.p3.3.m3.1b"><ci id="S3.p3.3.m3.1.1.cmml" xref="S3.p3.3.m3.1.1">𝒫</ci></annotation-xml><annotation encoding="application/x-tex" id="S3.p3.3.m3.1c">\mathcal{P}</annotation><annotation encoding="application/x-llamapun" id="S3.p3.3.m3.1d">caligraphic_P</annotation></semantics></math> denote the prompts used for training, which does not intersect with the ones used for evaluation.</p>
</div>
<figure class="ltx_figure ltx_align_floatright" id="S3.fig1">
<figure class="ltx_float ltx_minipage ltx_align_middle ltx_framed ltx_framed_top" id="alg1" style="width:182.8pt;">
<figcaption class="ltx_caption"><span class="ltx_tag ltx_tag_float"><span class="ltx_text ltx_font_bold" id="alg1.2.1.1">Algorithm 1</span> </span> Supervised fine-tuning</figcaption>
<div class="ltx_listing ltx_listing" id="alg1.3">
<div class="ltx_listingline" id="alg1.l1">
<span class="ltx_tag ltx_tag_listingline">1:</span>  Initialize <math alttext="\pi_{\text{SFT}}" class="ltx_Math" display="inline" id="alg1.l1.m1.1"><semantics id="alg1.l1.m1.1a"><msub id="alg1.l1.m1.1.1" xref="alg1.l1.m1.1.1.cmml"><mi id="alg1.l1.m1.1.1.2" xref="alg1.l1.m1.1.1.2.cmml">π</mi><mtext id="alg1.l1.m1.1.1.3" xref="alg1.l1.m1.1.1.3a.cmml">SFT</mtext></msub><annotation-xml encoding="MathML-Content" id="alg1.l1.m1.1b"><apply id="alg1.l1.m1.1.1.cmml" xref="alg1.l1.m1.1.1"><csymbol cd="ambiguous" id="alg1.l1.m1.1.1.1.cmml" xref="alg1.l1.m1.1.1">subscript</csymbol><ci id="alg1.l1.m1.1.1.2.cmml" xref="alg1.l1.m1.1.1.2">𝜋</ci><ci id="alg1.l1.m1.1.1.3a.cmml" xref="alg1.l1.m1.1.1.3"><mtext id="alg1.l1.m1.1.1.3.cmml" mathsize="70%" xref="alg1.l1.m1.1.1.3">SFT</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="alg1.l1.m1.1c">\pi_{\text{SFT}}</annotation><annotation encoding="application/x-llamapun" id="alg1.l1.m1.1d">italic_π start_POSTSUBSCRIPT SFT end_POSTSUBSCRIPT</annotation></semantics></math> with <math alttext="\pi_{\text{pre}}" class="ltx_Math" display="inline" id="alg1.l1.m2.1"><semantics id="alg1.l1.m2.1a"><msub id="alg1.l1.m2.1.1" xref="alg1.l1.m2.1.1.cmml"><mi id="alg1.l1.m2.1.1.2" xref="alg1.l1.m2.1.1.2.cmml">π</mi><mtext id="alg1.l1.m2.1.1.3" xref="alg1.l1.m2.1.1.3a.cmml">pre</mtext></msub><annotation-xml encoding="MathML-Content" id="alg1.l1.m2.1b"><apply id="alg1.l1.m2.1.1.cmml" xref="alg1.l1.m2.1.1"><csymbol cd="ambiguous" id="alg1.l1.m2.1.1.1.cmml" xref="alg1.l1.m2.1.1">subscript</csymbol><ci id="alg1.l1.m2.1.1.2.cmml" xref="alg1.l1.m2.1.1.2">𝜋</ci><ci id="alg1.l1.m2.1.1.3a.cmml" xref="alg1.l1.m2.1.1.3"><mtext id="alg1.l1.m2.1.1.3.cmml" mathsize="70%" xref="alg1.l1.m2.1.1.3">pre</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="alg1.l1.m2.1c">\pi_{\text{pre}}</annotation><annotation encoding="application/x-llamapun" id="alg1.l1.m2.1d">italic_π start_POSTSUBSCRIPT pre end_POSTSUBSCRIPT</annotation></semantics></math>
</div>
<div class="ltx_listingline" id="alg1.l2">
<span class="ltx_tag ltx_tag_listingline">2:</span>  <span class="ltx_text ltx_font_bold" id="alg1.l2.1">repeat</span>
</div>
<div class="ltx_listingline" id="alg1.l3">
<span class="ltx_tag ltx_tag_listingline">3:</span>     Sample prompt <math alttext="p_{\text{desc}}\sim\mathcal{P}" class="ltx_Math" display="inline" id="alg1.l3.m1.1"><semantics id="alg1.l3.m1.1a"><mrow id="alg1.l3.m1.1.1" xref="alg1.l3.m1.1.1.cmml"><msub id="alg1.l3.m1.1.1.2" xref="alg1.l3.m1.1.1.2.cmml"><mi id="alg1.l3.m1.1.1.2.2" xref="alg1.l3.m1.1.1.2.2.cmml">p</mi><mtext id="alg1.l3.m1.1.1.2.3" xref="alg1.l3.m1.1.1.2.3a.cmml">desc</mtext></msub><mo id="alg1.l3.m1.1.1.1" xref="alg1.l3.m1.1.1.1.cmml">∼</mo><mi class="ltx_font_mathcaligraphic" id="alg1.l3.m1.1.1.3" xref="alg1.l3.m1.1.1.3.cmml">𝒫</mi></mrow><annotation-xml encoding="MathML-Content" id="alg1.l3.m1.1b"><apply id="alg1.l3.m1.1.1.cmml" xref="alg1.l3.m1.1.1"><csymbol cd="latexml" id="alg1.l3.m1.1.1.1.cmml" xref="alg1.l3.m1.1.1.1">similar-to</csymbol><apply id="alg1.l3.m1.1.1.2.cmml" xref="alg1.l3.m1.1.1.2"><csymbol cd="ambiguous" id="alg1.l3.m1.1.1.2.1.cmml" xref="alg1.l3.m1.1.1.2">subscript</csymbol><ci id="alg1.l3.m1.1.1.2.2.cmml" xref="alg1.l3.m1.1.1.2.2">𝑝</ci><ci id="alg1.l3.m1.1.1.2.3a.cmml" xref="alg1.l3.m1.1.1.2.3"><mtext id="alg1.l3.m1.1.1.2.3.cmml" mathsize="70%" xref="alg1.l3.m1.1.1.2.3">desc</mtext></ci></apply><ci id="alg1.l3.m1.1.1.3.cmml" xref="alg1.l3.m1.1.1.3">𝒫</ci></apply></annotation-xml><annotation encoding="application/x-tex" id="alg1.l3.m1.1c">p_{\text{desc}}\sim\mathcal{P}</annotation><annotation encoding="application/x-llamapun" id="alg1.l3.m1.1d">italic_p start_POSTSUBSCRIPT desc end_POSTSUBSCRIPT ∼ caligraphic_P</annotation></semantics></math>
</div>
<div class="ltx_listingline" id="alg1.l4">
<span class="ltx_tag ltx_tag_listingline">4:</span>     Sample <math alttext="\{m_{i}\}_{i=1}^{T}" class="ltx_Math" display="inline" id="alg1.l4.m1.1"><semantics id="alg1.l4.m1.1a"><msubsup id="alg1.l4.m1.1.1" xref="alg1.l4.m1.1.1.cmml"><mrow id="alg1.l4.m1.1.1.1.1.1" xref="alg1.l4.m1.1.1.1.1.2.cmml"><mo id="alg1.l4.m1.1.1.1.1.1.2" stretchy="false" xref="alg1.l4.m1.1.1.1.1.2.cmml">{</mo><msub id="alg1.l4.m1.1.1.1.1.1.1" xref="alg1.l4.m1.1.1.1.1.1.1.cmml"><mi id="alg1.l4.m1.1.1.1.1.1.1.2" xref="alg1.l4.m1.1.1.1.1.1.1.2.cmml">m</mi><mi id="alg1.l4.m1.1.1.1.1.1.1.3" xref="alg1.l4.m1.1.1.1.1.1.1.3.cmml">i</mi></msub><mo id="alg1.l4.m1.1.1.1.1.1.3" stretchy="false" xref="alg1.l4.m1.1.1.1.1.2.cmml">}</mo></mrow><mrow id="alg1.l4.m1.1.1.1.3" xref="alg1.l4.m1.1.1.1.3.cmml"><mi id="alg1.l4.m1.1.1.1.3.2" xref="alg1.l4.m1.1.1.1.3.2.cmml">i</mi><mo id="alg1.l4.m1.1.1.1.3.1" xref="alg1.l4.m1.1.1.1.3.1.cmml">=</mo><mn id="alg1.l4.m1.1.1.1.3.3" xref="alg1.l4.m1.1.1.1.3.3.cmml">1</mn></mrow><mi id="alg1.l4.m1.1.1.3" xref="alg1.l4.m1.1.1.3.cmml">T</mi></msubsup><annotation-xml encoding="MathML-Content" id="alg1.l4.m1.1b"><apply id="alg1.l4.m1.1.1.cmml" xref="alg1.l4.m1.1.1"><csymbol cd="ambiguous" id="alg1.l4.m1.1.1.2.cmml" xref="alg1.l4.m1.1.1">superscript</csymbol><apply id="alg1.l4.m1.1.1.1.cmml" xref="alg1.l4.m1.1.1"><csymbol cd="ambiguous" id="alg1.l4.m1.1.1.1.2.cmml" xref="alg1.l4.m1.1.1">subscript</csymbol><set id="alg1.l4.m1.1.1.1.1.2.cmml" xref="alg1.l4.m1.1.1.1.1.1"><apply id="alg1.l4.m1.1.1.1.1.1.1.cmml" xref="alg1.l4.m1.1.1.1.1.1.1"><csymbol cd="ambiguous" id="alg1.l4.m1.1.1.1.1.1.1.1.cmml" xref="alg1.l4.m1.1.1.1.1.1.1">subscript</csymbol><ci id="alg1.l4.m1.1.1.1.1.1.1.2.cmml" xref="alg1.l4.m1.1.1.1.1.1.1.2">𝑚</ci><ci id="alg1.l4.m1.1.1.1.1.1.1.3.cmml" xref="alg1.l4.m1.1.1.1.1.1.1.3">𝑖</ci></apply></set><apply id="alg1.l4.m1.1.1.1.3.cmml" xref="alg1.l4.m1.1.1.1.3"><eq id="alg1.l4.m1.1.1.1.3.1.cmml" xref="alg1.l4.m1.1.1.1.3.1"></eq><ci id="alg1.l4.m1.1.1.1.3.2.cmml" xref="alg1.l4.m1.1.1.1.3.2">𝑖</ci><cn id="alg1.l4.m1.1.1.1.3.3.cmml" type="integer" xref="alg1.l4.m1.1.1.1.3.3">1</cn></apply></apply><ci id="alg1.l4.m1.1.1.3.cmml" xref="alg1.l4.m1.1.1.3">𝑇</ci></apply></annotation-xml><annotation encoding="application/x-tex" id="alg1.l4.m1.1c">\{m_{i}\}_{i=1}^{T}</annotation><annotation encoding="application/x-llamapun" id="alg1.l4.m1.1d">{ italic_m start_POSTSUBSCRIPT italic_i end_POSTSUBSCRIPT } start_POSTSUBSCRIPT italic_i = 1 end_POSTSUBSCRIPT start_POSTSUPERSCRIPT italic_T end_POSTSUPERSCRIPT</annotation></semantics></math> from <math alttext="\pi_{\text{pre}}(m\;|\;p_{\text{desc}})" class="ltx_Math" display="inline" id="alg1.l4.m2.1"><semantics id="alg1.l4.m2.1a"><mrow id="alg1.l4.m2.1.1" xref="alg1.l4.m2.1.1.cmml"><msub id="alg1.l4.m2.1.1.3" xref="alg1.l4.m2.1.1.3.cmml"><mi id="alg1.l4.m2.1.1.3.2" xref="alg1.l4.m2.1.1.3.2.cmml">π</mi><mtext id="alg1.l4.m2.1.1.3.3" xref="alg1.l4.m2.1.1.3.3a.cmml">pre</mtext></msub><mo id="alg1.l4.m2.1.1.2" xref="alg1.l4.m2.1.1.2.cmml">⁢</mo><mrow id="alg1.l4.m2.1.1.1.1" xref="alg1.l4.m2.1.1.1.1.1.cmml"><mo id="alg1.l4.m2.1.1.1.1.2" stretchy="false" xref="alg1.l4.m2.1.1.1.1.1.cmml">(</mo><mrow id="alg1.l4.m2.1.1.1.1.1" xref="alg1.l4.m2.1.1.1.1.1.cmml"><mi id="alg1.l4.m2.1.1.1.1.1.2" xref="alg1.l4.m2.1.1.1.1.1.2.cmml">m</mi><mo fence="false" id="alg1.l4.m2.1.1.1.1.1.1" lspace="0.558em" rspace="0.558em" xref="alg1.l4.m2.1.1.1.1.1.1.cmml">|</mo><msub id="alg1.l4.m2.1.1.1.1.1.3" xref="alg1.l4.m2.1.1.1.1.1.3.cmml"><mi id="alg1.l4.m2.1.1.1.1.1.3.2" xref="alg1.l4.m2.1.1.1.1.1.3.2.cmml">p</mi><mtext id="alg1.l4.m2.1.1.1.1.1.3.3" xref="alg1.l4.m2.1.1.1.1.1.3.3a.cmml">desc</mtext></msub></mrow><mo id="alg1.l4.m2.1.1.1.1.3" stretchy="false" xref="alg1.l4.m2.1.1.1.1.1.cmml">)</mo></mrow></mrow><annotation-xml encoding="MathML-Content" id="alg1.l4.m2.1b"><apply id="alg1.l4.m2.1.1.cmml" xref="alg1.l4.m2.1.1"><times id="alg1.l4.m2.1.1.2.cmml" xref="alg1.l4.m2.1.1.2"></times><apply id="alg1.l4.m2.1.1.3.cmml" xref="alg1.l4.m2.1.1.3"><csymbol cd="ambiguous" id="alg1.l4.m2.1.1.3.1.cmml" xref="alg1.l4.m2.1.1.3">subscript</csymbol><ci id="alg1.l4.m2.1.1.3.2.cmml" xref="alg1.l4.m2.1.1.3.2">𝜋</ci><ci id="alg1.l4.m2.1.1.3.3a.cmml" xref="alg1.l4.m2.1.1.3.3"><mtext id="alg1.l4.m2.1.1.3.3.cmml" mathsize="70%" xref="alg1.l4.m2.1.1.3.3">pre</mtext></ci></apply><apply id="alg1.l4.m2.1.1.1.1.1.cmml" xref="alg1.l4.m2.1.1.1.1"><csymbol cd="latexml" id="alg1.l4.m2.1.1.1.1.1.1.cmml" xref="alg1.l4.m2.1.1.1.1.1.1">conditional</csymbol><ci id="alg1.l4.m2.1.1.1.1.1.2.cmml" xref="alg1.l4.m2.1.1.1.1.1.2">𝑚</ci><apply id="alg1.l4.m2.1.1.1.1.1.3.cmml" xref="alg1.l4.m2.1.1.1.1.1.3"><csymbol cd="ambiguous" id="alg1.l4.m2.1.1.1.1.1.3.1.cmml" xref="alg1.l4.m2.1.1.1.1.1.3">subscript</csymbol><ci id="alg1.l4.m2.1.1.1.1.1.3.2.cmml" xref="alg1.l4.m2.1.1.1.1.1.3.2">𝑝</ci><ci id="alg1.l4.m2.1.1.1.1.1.3.3a.cmml" xref="alg1.l4.m2.1.1.1.1.1.3.3"><mtext id="alg1.l4.m2.1.1.1.1.1.3.3.cmml" mathsize="70%" xref="alg1.l4.m2.1.1.1.1.1.3.3">desc</mtext></ci></apply></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="alg1.l4.m2.1c">\pi_{\text{pre}}(m\;|\;p_{\text{desc}})</annotation><annotation encoding="application/x-llamapun" id="alg1.l4.m2.1d">italic_π start_POSTSUBSCRIPT pre end_POSTSUBSCRIPT ( italic_m | italic_p start_POSTSUBSCRIPT desc end_POSTSUBSCRIPT )</annotation></semantics></math>
</div>
<div class="ltx_listingline" id="alg1.l5">
<span class="ltx_tag ltx_tag_listingline">5:</span>     Update <math alttext="\{m_{i}\}^{K}_{i=1}\leftarrow\text{Filter}(\{m_{i}\}^{T}_{i=1})" class="ltx_Math" display="inline" id="alg1.l5.m1.2"><semantics id="alg1.l5.m1.2a"><mrow id="alg1.l5.m1.2.2" xref="alg1.l5.m1.2.2.cmml"><msubsup id="alg1.l5.m1.1.1.1" xref="alg1.l5.m1.1.1.1.cmml"><mrow id="alg1.l5.m1.1.1.1.1.1.1" xref="alg1.l5.m1.1.1.1.1.1.2.cmml"><mo id="alg1.l5.m1.1.1.1.1.1.1.2" stretchy="false" xref="alg1.l5.m1.1.1.1.1.1.2.cmml">{</mo><msub id="alg1.l5.m1.1.1.1.1.1.1.1" xref="alg1.l5.m1.1.1.1.1.1.1.1.cmml"><mi id="alg1.l5.m1.1.1.1.1.1.1.1.2" xref="alg1.l5.m1.1.1.1.1.1.1.1.2.cmml">m</mi><mi id="alg1.l5.m1.1.1.1.1.1.1.1.3" xref="alg1.l5.m1.1.1.1.1.1.1.1.3.cmml">i</mi></msub><mo id="alg1.l5.m1.1.1.1.1.1.1.3" stretchy="false" xref="alg1.l5.m1.1.1.1.1.1.2.cmml">}</mo></mrow><mrow id="alg1.l5.m1.1.1.1.3" xref="alg1.l5.m1.1.1.1.3.cmml"><mi id="alg1.l5.m1.1.1.1.3.2" xref="alg1.l5.m1.1.1.1.3.2.cmml">i</mi><mo id="alg1.l5.m1.1.1.1.3.1" xref="alg1.l5.m1.1.1.1.3.1.cmml">=</mo><mn id="alg1.l5.m1.1.1.1.3.3" xref="alg1.l5.m1.1.1.1.3.3.cmml">1</mn></mrow><mi id="alg1.l5.m1.1.1.1.1.3" xref="alg1.l5.m1.1.1.1.1.3.cmml">K</mi></msubsup><mo id="alg1.l5.m1.2.2.3" stretchy="false" xref="alg1.l5.m1.2.2.3.cmml">←</mo><mrow id="alg1.l5.m1.2.2.2" xref="alg1.l5.m1.2.2.2.cmml"><mtext id="alg1.l5.m1.2.2.2.3" xref="alg1.l5.m1.2.2.2.3a.cmml">Filter</mtext><mo id="alg1.l5.m1.2.2.2.2" xref="alg1.l5.m1.2.2.2.2.cmml">⁢</mo><mrow id="alg1.l5.m1.2.2.2.1.1" xref="alg1.l5.m1.2.2.2.1.1.1.cmml"><mo id="alg1.l5.m1.2.2.2.1.1.2" stretchy="false" xref="alg1.l5.m1.2.2.2.1.1.1.cmml">(</mo><msubsup id="alg1.l5.m1.2.2.2.1.1.1" xref="alg1.l5.m1.2.2.2.1.1.1.cmml"><mrow id="alg1.l5.m1.2.2.2.1.1.1.1.1.1" xref="alg1.l5.m1.2.2.2.1.1.1.1.1.2.cmml"><mo id="alg1.l5.m1.2.2.2.1.1.1.1.1.1.2" stretchy="false" xref="alg1.l5.m1.2.2.2.1.1.1.1.1.2.cmml">{</mo><msub id="alg1.l5.m1.2.2.2.1.1.1.1.1.1.1" xref="alg1.l5.m1.2.2.2.1.1.1.1.1.1.1.cmml"><mi id="alg1.l5.m1.2.2.2.1.1.1.1.1.1.1.2" xref="alg1.l5.m1.2.2.2.1.1.1.1.1.1.1.2.cmml">m</mi><mi id="alg1.l5.m1.2.2.2.1.1.1.1.1.1.1.3" xref="alg1.l5.m1.2.2.2.1.1.1.1.1.1.1.3.cmml">i</mi></msub><mo id="alg1.l5.m1.2.2.2.1.1.1.1.1.1.3" stretchy="false" xref="alg1.l5.m1.2.2.2.1.1.1.1.1.2.cmml">}</mo></mrow><mrow id="alg1.l5.m1.2.2.2.1.1.1.3" xref="alg1.l5.m1.2.2.2.1.1.1.3.cmml"><mi id="alg1.l5.m1.2.2.2.1.1.1.3.2" xref="alg1.l5.m1.2.2.2.1.1.1.3.2.cmml">i</mi><mo id="alg1.l5.m1.2.2.2.1.1.1.3.1" xref="alg1.l5.m1.2.2.2.1.1.1.3.1.cmml">=</mo><mn id="alg1.l5.m1.2.2.2.1.1.1.3.3" xref="alg1.l5.m1.2.2.2.1.1.1.3.3.cmml">1</mn></mrow><mi id="alg1.l5.m1.2.2.2.1.1.1.1.3" xref="alg1.l5.m1.2.2.2.1.1.1.1.3.cmml">T</mi></msubsup><mo id="alg1.l5.m1.2.2.2.1.1.3" stretchy="false" xref="alg1.l5.m1.2.2.2.1.1.1.cmml">)</mo></mrow></mrow></mrow><annotation-xml encoding="MathML-Content" id="alg1.l5.m1.2b"><apply id="alg1.l5.m1.2.2.cmml" xref="alg1.l5.m1.2.2"><ci id="alg1.l5.m1.2.2.3.cmml" xref="alg1.l5.m1.2.2.3">←</ci><apply id="alg1.l5.m1.1.1.1.cmml" xref="alg1.l5.m1.1.1.1"><csymbol cd="ambiguous" id="alg1.l5.m1.1.1.1.2.cmml" xref="alg1.l5.m1.1.1.1">subscript</csymbol><apply id="alg1.l5.m1.1.1.1.1.cmml" xref="alg1.l5.m1.1.1.1"><csymbol cd="ambiguous" id="alg1.l5.m1.1.1.1.1.2.cmml" xref="alg1.l5.m1.1.1.1">superscript</csymbol><set id="alg1.l5.m1.1.1.1.1.1.2.cmml" xref="alg1.l5.m1.1.1.1.1.1.1"><apply id="alg1.l5.m1.1.1.1.1.1.1.1.cmml" xref="alg1.l5.m1.1.1.1.1.1.1.1"><csymbol cd="ambiguous" id="alg1.l5.m1.1.1.1.1.1.1.1.1.cmml" xref="alg1.l5.m1.1.1.1.1.1.1.1">subscript</csymbol><ci id="alg1.l5.m1.1.1.1.1.1.1.1.2.cmml" xref="alg1.l5.m1.1.1.1.1.1.1.1.2">𝑚</ci><ci id="alg1.l5.m1.1.1.1.1.1.1.1.3.cmml" xref="alg1.l5.m1.1.1.1.1.1.1.1.3">𝑖</ci></apply></set><ci id="alg1.l5.m1.1.1.1.1.3.cmml" xref="alg1.l5.m1.1.1.1.1.3">𝐾</ci></apply><apply id="alg1.l5.m1.1.1.1.3.cmml" xref="alg1.l5.m1.1.1.1.3"><eq id="alg1.l5.m1.1.1.1.3.1.cmml" xref="alg1.l5.m1.1.1.1.3.1"></eq><ci id="alg1.l5.m1.1.1.1.3.2.cmml" xref="alg1.l5.m1.1.1.1.3.2">𝑖</ci><cn id="alg1.l5.m1.1.1.1.3.3.cmml" type="integer" xref="alg1.l5.m1.1.1.1.3.3">1</cn></apply></apply><apply id="alg1.l5.m1.2.2.2.cmml" xref="alg1.l5.m1.2.2.2"><times id="alg1.l5.m1.2.2.2.2.cmml" xref="alg1.l5.m1.2.2.2.2"></times><ci id="alg1.l5.m1.2.2.2.3a.cmml" xref="alg1.l5.m1.2.2.2.3"><mtext id="alg1.l5.m1.2.2.2.3.cmml" xref="alg1.l5.m1.2.2.2.3">Filter</mtext></ci><apply id="alg1.l5.m1.2.2.2.1.1.1.cmml" xref="alg1.l5.m1.2.2.2.1.1"><csymbol cd="ambiguous" id="alg1.l5.m1.2.2.2.1.1.1.2.cmml" xref="alg1.l5.m1.2.2.2.1.1">subscript</csymbol><apply id="alg1.l5.m1.2.2.2.1.1.1.1.cmml" xref="alg1.l5.m1.2.2.2.1.1"><csymbol cd="ambiguous" id="alg1.l5.m1.2.2.2.1.1.1.1.2.cmml" xref="alg1.l5.m1.2.2.2.1.1">superscript</csymbol><set id="alg1.l5.m1.2.2.2.1.1.1.1.1.2.cmml" xref="alg1.l5.m1.2.2.2.1.1.1.1.1.1"><apply id="alg1.l5.m1.2.2.2.1.1.1.1.1.1.1.cmml" xref="alg1.l5.m1.2.2.2.1.1.1.1.1.1.1"><csymbol cd="ambiguous" id="alg1.l5.m1.2.2.2.1.1.1.1.1.1.1.1.cmml" xref="alg1.l5.m1.2.2.2.1.1.1.1.1.1.1">subscript</csymbol><ci id="alg1.l5.m1.2.2.2.1.1.1.1.1.1.1.2.cmml" xref="alg1.l5.m1.2.2.2.1.1.1.1.1.1.1.2">𝑚</ci><ci id="alg1.l5.m1.2.2.2.1.1.1.1.1.1.1.3.cmml" xref="alg1.l5.m1.2.2.2.1.1.1.1.1.1.1.3">𝑖</ci></apply></set><ci id="alg1.l5.m1.2.2.2.1.1.1.1.3.cmml" xref="alg1.l5.m1.2.2.2.1.1.1.1.3">𝑇</ci></apply><apply id="alg1.l5.m1.2.2.2.1.1.1.3.cmml" xref="alg1.l5.m1.2.2.2.1.1.1.3"><eq id="alg1.l5.m1.2.2.2.1.1.1.3.1.cmml" xref="alg1.l5.m1.2.2.2.1.1.1.3.1"></eq><ci id="alg1.l5.m1.2.2.2.1.1.1.3.2.cmml" xref="alg1.l5.m1.2.2.2.1.1.1.3.2">𝑖</ci><cn id="alg1.l5.m1.2.2.2.1.1.1.3.3.cmml" type="integer" xref="alg1.l5.m1.2.2.2.1.1.1.3.3">1</cn></apply></apply></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="alg1.l5.m1.2c">\{m_{i}\}^{K}_{i=1}\leftarrow\text{Filter}(\{m_{i}\}^{T}_{i=1})</annotation><annotation encoding="application/x-llamapun" id="alg1.l5.m1.2d">{ italic_m start_POSTSUBSCRIPT italic_i end_POSTSUBSCRIPT } start_POSTSUPERSCRIPT italic_K end_POSTSUPERSCRIPT start_POSTSUBSCRIPT italic_i = 1 end_POSTSUBSCRIPT ← Filter ( { italic_m start_POSTSUBSCRIPT italic_i end_POSTSUBSCRIPT } start_POSTSUPERSCRIPT italic_T end_POSTSUPERSCRIPT start_POSTSUBSCRIPT italic_i = 1 end_POSTSUBSCRIPT )</annotation></semantics></math>
</div>
<div class="ltx_listingline" id="alg1.l6">
<span class="ltx_tag ltx_tag_listingline">6:</span>     Maximize <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S3.E1" title="In 3.1 Supervised fine-tuning ‣ 3 Method ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Equation</span> <span class="ltx_text ltx_ref_tag">1</span></a> with <math alttext="\{m_{i}\}^{K}_{i=1}" class="ltx_Math" display="inline" id="alg1.l6.m1.1"><semantics id="alg1.l6.m1.1a"><msubsup id="alg1.l6.m1.1.1" xref="alg1.l6.m1.1.1.cmml"><mrow id="alg1.l6.m1.1.1.1.1.1" xref="alg1.l6.m1.1.1.1.1.2.cmml"><mo id="alg1.l6.m1.1.1.1.1.1.2" stretchy="false" xref="alg1.l6.m1.1.1.1.1.2.cmml">{</mo><msub id="alg1.l6.m1.1.1.1.1.1.1" xref="alg1.l6.m1.1.1.1.1.1.1.cmml"><mi id="alg1.l6.m1.1.1.1.1.1.1.2" xref="alg1.l6.m1.1.1.1.1.1.1.2.cmml">m</mi><mi id="alg1.l6.m1.1.1.1.1.1.1.3" xref="alg1.l6.m1.1.1.1.1.1.1.3.cmml">i</mi></msub><mo id="alg1.l6.m1.1.1.1.1.1.3" stretchy="false" xref="alg1.l6.m1.1.1.1.1.2.cmml">}</mo></mrow><mrow id="alg1.l6.m1.1.1.3" xref="alg1.l6.m1.1.1.3.cmml"><mi id="alg1.l6.m1.1.1.3.2" xref="alg1.l6.m1.1.1.3.2.cmml">i</mi><mo id="alg1.l6.m1.1.1.3.1" xref="alg1.l6.m1.1.1.3.1.cmml">=</mo><mn id="alg1.l6.m1.1.1.3.3" xref="alg1.l6.m1.1.1.3.3.cmml">1</mn></mrow><mi id="alg1.l6.m1.1.1.1.3" xref="alg1.l6.m1.1.1.1.3.cmml">K</mi></msubsup><annotation-xml encoding="MathML-Content" id="alg1.l6.m1.1b"><apply id="alg1.l6.m1.1.1.cmml" xref="alg1.l6.m1.1.1"><csymbol cd="ambiguous" id="alg1.l6.m1.1.1.2.cmml" xref="alg1.l6.m1.1.1">subscript</csymbol><apply id="alg1.l6.m1.1.1.1.cmml" xref="alg1.l6.m1.1.1"><csymbol cd="ambiguous" id="alg1.l6.m1.1.1.1.2.cmml" xref="alg1.l6.m1.1.1">superscript</csymbol><set id="alg1.l6.m1.1.1.1.1.2.cmml" xref="alg1.l6.m1.1.1.1.1.1"><apply id="alg1.l6.m1.1.1.1.1.1.1.cmml" xref="alg1.l6.m1.1.1.1.1.1.1"><csymbol cd="ambiguous" id="alg1.l6.m1.1.1.1.1.1.1.1.cmml" xref="alg1.l6.m1.1.1.1.1.1.1">subscript</csymbol><ci id="alg1.l6.m1.1.1.1.1.1.1.2.cmml" xref="alg1.l6.m1.1.1.1.1.1.1.2">𝑚</ci><ci id="alg1.l6.m1.1.1.1.1.1.1.3.cmml" xref="alg1.l6.m1.1.1.1.1.1.1.3">𝑖</ci></apply></set><ci id="alg1.l6.m1.1.1.1.3.cmml" xref="alg1.l6.m1.1.1.1.3">𝐾</ci></apply><apply id="alg1.l6.m1.1.1.3.cmml" xref="alg1.l6.m1.1.1.3"><eq id="alg1.l6.m1.1.1.3.1.cmml" xref="alg1.l6.m1.1.1.3.1"></eq><ci id="alg1.l6.m1.1.1.3.2.cmml" xref="alg1.l6.m1.1.1.3.2">𝑖</ci><cn id="alg1.l6.m1.1.1.3.3.cmml" type="integer" xref="alg1.l6.m1.1.1.3.3">1</cn></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="alg1.l6.m1.1c">\{m_{i}\}^{K}_{i=1}</annotation><annotation encoding="application/x-llamapun" id="alg1.l6.m1.1d">{ italic_m start_POSTSUBSCRIPT italic_i end_POSTSUBSCRIPT } start_POSTSUPERSCRIPT italic_K end_POSTSUPERSCRIPT start_POSTSUBSCRIPT italic_i = 1 end_POSTSUBSCRIPT</annotation></semantics></math>
</div>
<div class="ltx_listingline" id="alg1.l7">
<span class="ltx_tag ltx_tag_listingline">7:</span>  <span class="ltx_text ltx_font_bold" id="alg1.l7.1">until</span> Converged

</div>
<div class="ltx_listingline" id="alg1.l8">
<span class="ltx_tag ltx_tag_listingline">8:</span>  <span class="ltx_text ltx_font_bold" id="alg1.l8.1">Output:</span> fine-tuned <math alttext="\pi_{\text{SFT}}" class="ltx_Math" display="inline" id="alg1.l8.m1.1"><semantics id="alg1.l8.m1.1a"><msub id="alg1.l8.m1.1.1" xref="alg1.l8.m1.1.1.cmml"><mi id="alg1.l8.m1.1.1.2" xref="alg1.l8.m1.1.1.2.cmml">π</mi><mtext id="alg1.l8.m1.1.1.3" xref="alg1.l8.m1.1.1.3a.cmml">SFT</mtext></msub><annotation-xml encoding="MathML-Content" id="alg1.l8.m1.1b"><apply id="alg1.l8.m1.1.1.cmml" xref="alg1.l8.m1.1.1"><csymbol cd="ambiguous" id="alg1.l8.m1.1.1.1.cmml" xref="alg1.l8.m1.1.1">subscript</csymbol><ci id="alg1.l8.m1.1.1.2.cmml" xref="alg1.l8.m1.1.1.2">𝜋</ci><ci id="alg1.l8.m1.1.1.3a.cmml" xref="alg1.l8.m1.1.1.3"><mtext id="alg1.l8.m1.1.1.3.cmml" mathsize="70%" xref="alg1.l8.m1.1.1.3">SFT</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="alg1.l8.m1.1c">\pi_{\text{SFT}}</annotation><annotation encoding="application/x-llamapun" id="alg1.l8.m1.1d">italic_π start_POSTSUBSCRIPT SFT end_POSTSUBSCRIPT</annotation></semantics></math>
</div>
</div>
</figure>
</figure>
<section class="ltx_subsection" id="S3.SS1">
<h3 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">3.1 </span>Supervised fine-tuning</h3>
<div class="ltx_para ltx_noindent" id="S3.SS1.p1">
<p class="ltx_p" id="S3.SS1.p1.2">We first describe our supervised fine-tuning process for repurposing the pre-trained LLMs to autoregressively generate multiple molecules in a sequence. This involves collecting a dataset of molecules from a pre-trained LLM <math alttext="\pi_{\text{pre}}" class="ltx_Math" display="inline" id="S3.SS1.p1.1.m1.1"><semantics id="S3.SS1.p1.1.m1.1a"><msub id="S3.SS1.p1.1.m1.1.1" xref="S3.SS1.p1.1.m1.1.1.cmml"><mi id="S3.SS1.p1.1.m1.1.1.2" xref="S3.SS1.p1.1.m1.1.1.2.cmml">π</mi><mtext id="S3.SS1.p1.1.m1.1.1.3" xref="S3.SS1.p1.1.m1.1.1.3a.cmml">pre</mtext></msub><annotation-xml encoding="MathML-Content" id="S3.SS1.p1.1.m1.1b"><apply id="S3.SS1.p1.1.m1.1.1.cmml" xref="S3.SS1.p1.1.m1.1.1"><csymbol cd="ambiguous" id="S3.SS1.p1.1.m1.1.1.1.cmml" xref="S3.SS1.p1.1.m1.1.1">subscript</csymbol><ci id="S3.SS1.p1.1.m1.1.1.2.cmml" xref="S3.SS1.p1.1.m1.1.1.2">𝜋</ci><ci id="S3.SS1.p1.1.m1.1.1.3a.cmml" xref="S3.SS1.p1.1.m1.1.1.3"><mtext id="S3.SS1.p1.1.m1.1.1.3.cmml" mathsize="70%" xref="S3.SS1.p1.1.m1.1.1.3">pre</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS1.p1.1.m1.1c">\pi_{\text{pre}}</annotation><annotation encoding="application/x-llamapun" id="S3.SS1.p1.1.m1.1d">italic_π start_POSTSUBSCRIPT pre end_POSTSUBSCRIPT</annotation></semantics></math>, and then fine-tuning the LLM <math alttext="\pi_{\text{SFT}}" class="ltx_Math" display="inline" id="S3.SS1.p1.2.m2.1"><semantics id="S3.SS1.p1.2.m2.1a"><msub id="S3.SS1.p1.2.m2.1.1" xref="S3.SS1.p1.2.m2.1.1.cmml"><mi id="S3.SS1.p1.2.m2.1.1.2" xref="S3.SS1.p1.2.m2.1.1.2.cmml">π</mi><mtext id="S3.SS1.p1.2.m2.1.1.3" xref="S3.SS1.p1.2.m2.1.1.3a.cmml">SFT</mtext></msub><annotation-xml encoding="MathML-Content" id="S3.SS1.p1.2.m2.1b"><apply id="S3.SS1.p1.2.m2.1.1.cmml" xref="S3.SS1.p1.2.m2.1.1"><csymbol cd="ambiguous" id="S3.SS1.p1.2.m2.1.1.1.cmml" xref="S3.SS1.p1.2.m2.1.1">subscript</csymbol><ci id="S3.SS1.p1.2.m2.1.1.2.cmml" xref="S3.SS1.p1.2.m2.1.1.2">𝜋</ci><ci id="S3.SS1.p1.2.m2.1.1.3a.cmml" xref="S3.SS1.p1.2.m2.1.1.3"><mtext id="S3.SS1.p1.2.m2.1.1.3.cmml" mathsize="70%" xref="S3.SS1.p1.2.m2.1.1.3">SFT</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS1.p1.2.m2.1c">\pi_{\text{SFT}}</annotation><annotation encoding="application/x-llamapun" id="S3.SS1.p1.2.m2.1d">italic_π start_POSTSUBSCRIPT SFT end_POSTSUBSCRIPT</annotation></semantics></math> on the collected dataset. We describe the process in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S2.F4.sf1" title="In Figure 4 ‣ 2 Related work ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Figure</span> <span class="ltx_text ltx_ref_tag">4(a)</span></a> and <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#alg1" title="In 3 Method ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Algorithm</span> <span class="ltx_text ltx_ref_tag">1</span></a>.</p>
</div>
<div class="ltx_para ltx_noindent" id="S3.SS1.p2">
<p class="ltx_p" id="S3.SS1.p2.3"><span class="ltx_text ltx_font_bold" id="S3.SS1.p2.3.1">Dataset collection.</span> The dataset collection and supervised training processes are conducted with a set of training prompts <math alttext="\mathcal{P}" class="ltx_Math" display="inline" id="S3.SS1.p2.1.m1.1"><semantics id="S3.SS1.p2.1.m1.1a"><mi class="ltx_font_mathcaligraphic" id="S3.SS1.p2.1.m1.1.1" xref="S3.SS1.p2.1.m1.1.1.cmml">𝒫</mi><annotation-xml encoding="MathML-Content" id="S3.SS1.p2.1.m1.1b"><ci id="S3.SS1.p2.1.m1.1.1.cmml" xref="S3.SS1.p2.1.m1.1.1">𝒫</ci></annotation-xml><annotation encoding="application/x-tex" id="S3.SS1.p2.1.m1.1c">\mathcal{P}</annotation><annotation encoding="application/x-llamapun" id="S3.SS1.p2.1.m1.1d">caligraphic_P</annotation></semantics></math>. Initially, the pre-trained LLM <math alttext="\pi_{\text{pre}}" class="ltx_Math" display="inline" id="S3.SS1.p2.2.m2.1"><semantics id="S3.SS1.p2.2.m2.1a"><msub id="S3.SS1.p2.2.m2.1.1" xref="S3.SS1.p2.2.m2.1.1.cmml"><mi id="S3.SS1.p2.2.m2.1.1.2" xref="S3.SS1.p2.2.m2.1.1.2.cmml">π</mi><mtext id="S3.SS1.p2.2.m2.1.1.3" xref="S3.SS1.p2.2.m2.1.1.3a.cmml">pre</mtext></msub><annotation-xml encoding="MathML-Content" id="S3.SS1.p2.2.m2.1b"><apply id="S3.SS1.p2.2.m2.1.1.cmml" xref="S3.SS1.p2.2.m2.1.1"><csymbol cd="ambiguous" id="S3.SS1.p2.2.m2.1.1.1.cmml" xref="S3.SS1.p2.2.m2.1.1">subscript</csymbol><ci id="S3.SS1.p2.2.m2.1.1.2.cmml" xref="S3.SS1.p2.2.m2.1.1.2">𝜋</ci><ci id="S3.SS1.p2.2.m2.1.1.3a.cmml" xref="S3.SS1.p2.2.m2.1.1.3"><mtext id="S3.SS1.p2.2.m2.1.1.3.cmml" mathsize="70%" xref="S3.SS1.p2.2.m2.1.1.3">pre</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS1.p2.2.m2.1c">\pi_{\text{pre}}</annotation><annotation encoding="application/x-llamapun" id="S3.SS1.p2.2.m2.1d">italic_π start_POSTSUBSCRIPT pre end_POSTSUBSCRIPT</annotation></semantics></math> produces a set of molecules by iterative sampling molecules for a given prompt <math alttext="p_{\text{desc}}\in\mathcal{P}" class="ltx_Math" display="inline" id="S3.SS1.p2.3.m3.1"><semantics id="S3.SS1.p2.3.m3.1a"><mrow id="S3.SS1.p2.3.m3.1.1" xref="S3.SS1.p2.3.m3.1.1.cmml"><msub id="S3.SS1.p2.3.m3.1.1.2" xref="S3.SS1.p2.3.m3.1.1.2.cmml"><mi id="S3.SS1.p2.3.m3.1.1.2.2" xref="S3.SS1.p2.3.m3.1.1.2.2.cmml">p</mi><mtext id="S3.SS1.p2.3.m3.1.1.2.3" xref="S3.SS1.p2.3.m3.1.1.2.3a.cmml">desc</mtext></msub><mo id="S3.SS1.p2.3.m3.1.1.1" xref="S3.SS1.p2.3.m3.1.1.1.cmml">∈</mo><mi class="ltx_font_mathcaligraphic" id="S3.SS1.p2.3.m3.1.1.3" xref="S3.SS1.p2.3.m3.1.1.3.cmml">𝒫</mi></mrow><annotation-xml encoding="MathML-Content" id="S3.SS1.p2.3.m3.1b"><apply id="S3.SS1.p2.3.m3.1.1.cmml" xref="S3.SS1.p2.3.m3.1.1"><in id="S3.SS1.p2.3.m3.1.1.1.cmml" xref="S3.SS1.p2.3.m3.1.1.1"></in><apply id="S3.SS1.p2.3.m3.1.1.2.cmml" xref="S3.SS1.p2.3.m3.1.1.2"><csymbol cd="ambiguous" id="S3.SS1.p2.3.m3.1.1.2.1.cmml" xref="S3.SS1.p2.3.m3.1.1.2">subscript</csymbol><ci id="S3.SS1.p2.3.m3.1.1.2.2.cmml" xref="S3.SS1.p2.3.m3.1.1.2.2">𝑝</ci><ci id="S3.SS1.p2.3.m3.1.1.2.3a.cmml" xref="S3.SS1.p2.3.m3.1.1.2.3"><mtext id="S3.SS1.p2.3.m3.1.1.2.3.cmml" mathsize="70%" xref="S3.SS1.p2.3.m3.1.1.2.3">desc</mtext></ci></apply><ci id="S3.SS1.p2.3.m3.1.1.3.cmml" xref="S3.SS1.p2.3.m3.1.1.3">𝒫</ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS1.p2.3.m3.1c">p_{\text{desc}}\in\mathcal{P}</annotation><annotation encoding="application/x-llamapun" id="S3.SS1.p2.3.m3.1d">italic_p start_POSTSUBSCRIPT desc end_POSTSUBSCRIPT ∈ caligraphic_P</annotation></semantics></math> as follows:</p>
<table class="ltx_equation ltx_eqn_table" id="S3.Ex1">
<tbody><tr class="ltx_equation ltx_eqn_row ltx_align_baseline">
<td class="ltx_eqn_cell ltx_eqn_center_padleft"></td>
<td class="ltx_eqn_cell ltx_align_center"><math alttext="m_{i}\sim\pi_{\text{pre}}\left(m_{i}\middle|p_{\text{desc}}\right)\quad\text{%
for}\;i=1,\ldots,T," class="ltx_math_unparsed" display="block" id="S3.Ex1.m1.1"><semantics id="S3.Ex1.m1.1a"><mrow id="S3.Ex1.m1.1b"><msub id="S3.Ex1.m1.1.1"><mi id="S3.Ex1.m1.1.1.2">m</mi><mi id="S3.Ex1.m1.1.1.3">i</mi></msub><mo id="S3.Ex1.m1.1.2">∼</mo><msub id="S3.Ex1.m1.1.3"><mi id="S3.Ex1.m1.1.3.2">π</mi><mtext id="S3.Ex1.m1.1.3.3">pre</mtext></msub><mrow id="S3.Ex1.m1.1.4"><mo id="S3.Ex1.m1.1.4.1">(</mo><msub id="S3.Ex1.m1.1.4.2"><mi id="S3.Ex1.m1.1.4.2.2">m</mi><mi id="S3.Ex1.m1.1.4.2.3">i</mi></msub><mo id="S3.Ex1.m1.1.4.3" lspace="0em" rspace="0em" stretchy="true">|</mo><msub id="S3.Ex1.m1.1.4.4"><mi id="S3.Ex1.m1.1.4.4.2">p</mi><mtext id="S3.Ex1.m1.1.4.4.3">desc</mtext></msub><mo id="S3.Ex1.m1.1.4.5">)</mo></mrow><mspace id="S3.Ex1.m1.1.5" width="1em"></mspace><mtext id="S3.Ex1.m1.1.6">for</mtext><mi id="S3.Ex1.m1.1.7">i</mi><mo id="S3.Ex1.m1.1.8">=</mo><mn id="S3.Ex1.m1.1.9">1</mn><mo id="S3.Ex1.m1.1.10">,</mo><mi id="S3.Ex1.m1.1.11" mathvariant="normal">…</mi><mo id="S3.Ex1.m1.1.12">,</mo><mi id="S3.Ex1.m1.1.13">T</mi><mo id="S3.Ex1.m1.1.14">,</mo></mrow><annotation encoding="application/x-tex" id="S3.Ex1.m1.1c">m_{i}\sim\pi_{\text{pre}}\left(m_{i}\middle|p_{\text{desc}}\right)\quad\text{%
for}\;i=1,\ldots,T,</annotation><annotation encoding="application/x-llamapun" id="S3.Ex1.m1.1d">italic_m start_POSTSUBSCRIPT italic_i end_POSTSUBSCRIPT ∼ italic_π start_POSTSUBSCRIPT pre end_POSTSUBSCRIPT ( italic_m start_POSTSUBSCRIPT italic_i end_POSTSUBSCRIPT | italic_p start_POSTSUBSCRIPT desc end_POSTSUBSCRIPT ) for italic_i = 1 , … , italic_T ,</annotation></semantics></math></td>
<td class="ltx_eqn_cell ltx_eqn_center_padright"></td>
</tr></tbody>
</table>
<p class="ltx_p" id="S3.SS1.p2.8">where <math alttext="m_{i}" class="ltx_Math" display="inline" id="S3.SS1.p2.4.m1.1"><semantics id="S3.SS1.p2.4.m1.1a"><msub id="S3.SS1.p2.4.m1.1.1" xref="S3.SS1.p2.4.m1.1.1.cmml"><mi id="S3.SS1.p2.4.m1.1.1.2" xref="S3.SS1.p2.4.m1.1.1.2.cmml">m</mi><mi id="S3.SS1.p2.4.m1.1.1.3" xref="S3.SS1.p2.4.m1.1.1.3.cmml">i</mi></msub><annotation-xml encoding="MathML-Content" id="S3.SS1.p2.4.m1.1b"><apply id="S3.SS1.p2.4.m1.1.1.cmml" xref="S3.SS1.p2.4.m1.1.1"><csymbol cd="ambiguous" id="S3.SS1.p2.4.m1.1.1.1.cmml" xref="S3.SS1.p2.4.m1.1.1">subscript</csymbol><ci id="S3.SS1.p2.4.m1.1.1.2.cmml" xref="S3.SS1.p2.4.m1.1.1.2">𝑚</ci><ci id="S3.SS1.p2.4.m1.1.1.3.cmml" xref="S3.SS1.p2.4.m1.1.1.3">𝑖</ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS1.p2.4.m1.1c">m_{i}</annotation><annotation encoding="application/x-llamapun" id="S3.SS1.p2.4.m1.1d">italic_m start_POSTSUBSCRIPT italic_i end_POSTSUBSCRIPT</annotation></semantics></math> denotes the string representation of the molecule. In practice, we employ beam search to collect the set of molecules <math alttext="\{m_{i}\}^{T}_{i=1}" class="ltx_Math" display="inline" id="S3.SS1.p2.5.m2.1"><semantics id="S3.SS1.p2.5.m2.1a"><msubsup id="S3.SS1.p2.5.m2.1.1" xref="S3.SS1.p2.5.m2.1.1.cmml"><mrow id="S3.SS1.p2.5.m2.1.1.1.1.1" xref="S3.SS1.p2.5.m2.1.1.1.1.2.cmml"><mo id="S3.SS1.p2.5.m2.1.1.1.1.1.2" stretchy="false" xref="S3.SS1.p2.5.m2.1.1.1.1.2.cmml">{</mo><msub id="S3.SS1.p2.5.m2.1.1.1.1.1.1" xref="S3.SS1.p2.5.m2.1.1.1.1.1.1.cmml"><mi id="S3.SS1.p2.5.m2.1.1.1.1.1.1.2" xref="S3.SS1.p2.5.m2.1.1.1.1.1.1.2.cmml">m</mi><mi id="S3.SS1.p2.5.m2.1.1.1.1.1.1.3" xref="S3.SS1.p2.5.m2.1.1.1.1.1.1.3.cmml">i</mi></msub><mo id="S3.SS1.p2.5.m2.1.1.1.1.1.3" stretchy="false" xref="S3.SS1.p2.5.m2.1.1.1.1.2.cmml">}</mo></mrow><mrow id="S3.SS1.p2.5.m2.1.1.3" xref="S3.SS1.p2.5.m2.1.1.3.cmml"><mi id="S3.SS1.p2.5.m2.1.1.3.2" xref="S3.SS1.p2.5.m2.1.1.3.2.cmml">i</mi><mo id="S3.SS1.p2.5.m2.1.1.3.1" xref="S3.SS1.p2.5.m2.1.1.3.1.cmml">=</mo><mn id="S3.SS1.p2.5.m2.1.1.3.3" xref="S3.SS1.p2.5.m2.1.1.3.3.cmml">1</mn></mrow><mi id="S3.SS1.p2.5.m2.1.1.1.3" xref="S3.SS1.p2.5.m2.1.1.1.3.cmml">T</mi></msubsup><annotation-xml encoding="MathML-Content" id="S3.SS1.p2.5.m2.1b"><apply id="S3.SS1.p2.5.m2.1.1.cmml" xref="S3.SS1.p2.5.m2.1.1"><csymbol cd="ambiguous" id="S3.SS1.p2.5.m2.1.1.2.cmml" xref="S3.SS1.p2.5.m2.1.1">subscript</csymbol><apply id="S3.SS1.p2.5.m2.1.1.1.cmml" xref="S3.SS1.p2.5.m2.1.1"><csymbol cd="ambiguous" id="S3.SS1.p2.5.m2.1.1.1.2.cmml" xref="S3.SS1.p2.5.m2.1.1">superscript</csymbol><set id="S3.SS1.p2.5.m2.1.1.1.1.2.cmml" xref="S3.SS1.p2.5.m2.1.1.1.1.1"><apply id="S3.SS1.p2.5.m2.1.1.1.1.1.1.cmml" xref="S3.SS1.p2.5.m2.1.1.1.1.1.1"><csymbol cd="ambiguous" id="S3.SS1.p2.5.m2.1.1.1.1.1.1.1.cmml" xref="S3.SS1.p2.5.m2.1.1.1.1.1.1">subscript</csymbol><ci id="S3.SS1.p2.5.m2.1.1.1.1.1.1.2.cmml" xref="S3.SS1.p2.5.m2.1.1.1.1.1.1.2">𝑚</ci><ci id="S3.SS1.p2.5.m2.1.1.1.1.1.1.3.cmml" xref="S3.SS1.p2.5.m2.1.1.1.1.1.1.3">𝑖</ci></apply></set><ci id="S3.SS1.p2.5.m2.1.1.1.3.cmml" xref="S3.SS1.p2.5.m2.1.1.1.3">𝑇</ci></apply><apply id="S3.SS1.p2.5.m2.1.1.3.cmml" xref="S3.SS1.p2.5.m2.1.1.3"><eq id="S3.SS1.p2.5.m2.1.1.3.1.cmml" xref="S3.SS1.p2.5.m2.1.1.3.1"></eq><ci id="S3.SS1.p2.5.m2.1.1.3.2.cmml" xref="S3.SS1.p2.5.m2.1.1.3.2">𝑖</ci><cn id="S3.SS1.p2.5.m2.1.1.3.3.cmml" type="integer" xref="S3.SS1.p2.5.m2.1.1.3.3">1</cn></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS1.p2.5.m2.1c">\{m_{i}\}^{T}_{i=1}</annotation><annotation encoding="application/x-llamapun" id="S3.SS1.p2.5.m2.1d">{ italic_m start_POSTSUBSCRIPT italic_i end_POSTSUBSCRIPT } start_POSTSUPERSCRIPT italic_T end_POSTSUPERSCRIPT start_POSTSUBSCRIPT italic_i = 1 end_POSTSUBSCRIPT</annotation></semantics></math>. Then, we filter out the invalid string representations, duplicate molecules, and molecules that do not satisfy the given prompt <math alttext="p_{\text{desc}}" class="ltx_Math" display="inline" id="S3.SS1.p2.6.m3.1"><semantics id="S3.SS1.p2.6.m3.1a"><msub id="S3.SS1.p2.6.m3.1.1" xref="S3.SS1.p2.6.m3.1.1.cmml"><mi id="S3.SS1.p2.6.m3.1.1.2" xref="S3.SS1.p2.6.m3.1.1.2.cmml">p</mi><mtext id="S3.SS1.p2.6.m3.1.1.3" xref="S3.SS1.p2.6.m3.1.1.3a.cmml">desc</mtext></msub><annotation-xml encoding="MathML-Content" id="S3.SS1.p2.6.m3.1b"><apply id="S3.SS1.p2.6.m3.1.1.cmml" xref="S3.SS1.p2.6.m3.1.1"><csymbol cd="ambiguous" id="S3.SS1.p2.6.m3.1.1.1.cmml" xref="S3.SS1.p2.6.m3.1.1">subscript</csymbol><ci id="S3.SS1.p2.6.m3.1.1.2.cmml" xref="S3.SS1.p2.6.m3.1.1.2">𝑝</ci><ci id="S3.SS1.p2.6.m3.1.1.3a.cmml" xref="S3.SS1.p2.6.m3.1.1.3"><mtext id="S3.SS1.p2.6.m3.1.1.3.cmml" mathsize="70%" xref="S3.SS1.p2.6.m3.1.1.3">desc</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS1.p2.6.m3.1c">p_{\text{desc}}</annotation><annotation encoding="application/x-llamapun" id="S3.SS1.p2.6.m3.1d">italic_p start_POSTSUBSCRIPT desc end_POSTSUBSCRIPT</annotation></semantics></math>. This results in reducing the set of molecules from <math alttext="\{m_{i}\}^{T}_{i=1}" class="ltx_Math" display="inline" id="S3.SS1.p2.7.m4.1"><semantics id="S3.SS1.p2.7.m4.1a"><msubsup id="S3.SS1.p2.7.m4.1.1" xref="S3.SS1.p2.7.m4.1.1.cmml"><mrow id="S3.SS1.p2.7.m4.1.1.1.1.1" xref="S3.SS1.p2.7.m4.1.1.1.1.2.cmml"><mo id="S3.SS1.p2.7.m4.1.1.1.1.1.2" stretchy="false" xref="S3.SS1.p2.7.m4.1.1.1.1.2.cmml">{</mo><msub id="S3.SS1.p2.7.m4.1.1.1.1.1.1" xref="S3.SS1.p2.7.m4.1.1.1.1.1.1.cmml"><mi id="S3.SS1.p2.7.m4.1.1.1.1.1.1.2" xref="S3.SS1.p2.7.m4.1.1.1.1.1.1.2.cmml">m</mi><mi id="S3.SS1.p2.7.m4.1.1.1.1.1.1.3" xref="S3.SS1.p2.7.m4.1.1.1.1.1.1.3.cmml">i</mi></msub><mo id="S3.SS1.p2.7.m4.1.1.1.1.1.3" stretchy="false" xref="S3.SS1.p2.7.m4.1.1.1.1.2.cmml">}</mo></mrow><mrow id="S3.SS1.p2.7.m4.1.1.3" xref="S3.SS1.p2.7.m4.1.1.3.cmml"><mi id="S3.SS1.p2.7.m4.1.1.3.2" xref="S3.SS1.p2.7.m4.1.1.3.2.cmml">i</mi><mo id="S3.SS1.p2.7.m4.1.1.3.1" xref="S3.SS1.p2.7.m4.1.1.3.1.cmml">=</mo><mn id="S3.SS1.p2.7.m4.1.1.3.3" xref="S3.SS1.p2.7.m4.1.1.3.3.cmml">1</mn></mrow><mi id="S3.SS1.p2.7.m4.1.1.1.3" xref="S3.SS1.p2.7.m4.1.1.1.3.cmml">T</mi></msubsup><annotation-xml encoding="MathML-Content" id="S3.SS1.p2.7.m4.1b"><apply id="S3.SS1.p2.7.m4.1.1.cmml" xref="S3.SS1.p2.7.m4.1.1"><csymbol cd="ambiguous" id="S3.SS1.p2.7.m4.1.1.2.cmml" xref="S3.SS1.p2.7.m4.1.1">subscript</csymbol><apply id="S3.SS1.p2.7.m4.1.1.1.cmml" xref="S3.SS1.p2.7.m4.1.1"><csymbol cd="ambiguous" id="S3.SS1.p2.7.m4.1.1.1.2.cmml" xref="S3.SS1.p2.7.m4.1.1">superscript</csymbol><set id="S3.SS1.p2.7.m4.1.1.1.1.2.cmml" xref="S3.SS1.p2.7.m4.1.1.1.1.1"><apply id="S3.SS1.p2.7.m4.1.1.1.1.1.1.cmml" xref="S3.SS1.p2.7.m4.1.1.1.1.1.1"><csymbol cd="ambiguous" id="S3.SS1.p2.7.m4.1.1.1.1.1.1.1.cmml" xref="S3.SS1.p2.7.m4.1.1.1.1.1.1">subscript</csymbol><ci id="S3.SS1.p2.7.m4.1.1.1.1.1.1.2.cmml" xref="S3.SS1.p2.7.m4.1.1.1.1.1.1.2">𝑚</ci><ci id="S3.SS1.p2.7.m4.1.1.1.1.1.1.3.cmml" xref="S3.SS1.p2.7.m4.1.1.1.1.1.1.3">𝑖</ci></apply></set><ci id="S3.SS1.p2.7.m4.1.1.1.3.cmml" xref="S3.SS1.p2.7.m4.1.1.1.3">𝑇</ci></apply><apply id="S3.SS1.p2.7.m4.1.1.3.cmml" xref="S3.SS1.p2.7.m4.1.1.3"><eq id="S3.SS1.p2.7.m4.1.1.3.1.cmml" xref="S3.SS1.p2.7.m4.1.1.3.1"></eq><ci id="S3.SS1.p2.7.m4.1.1.3.2.cmml" xref="S3.SS1.p2.7.m4.1.1.3.2">𝑖</ci><cn id="S3.SS1.p2.7.m4.1.1.3.3.cmml" type="integer" xref="S3.SS1.p2.7.m4.1.1.3.3">1</cn></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS1.p2.7.m4.1c">\{m_{i}\}^{T}_{i=1}</annotation><annotation encoding="application/x-llamapun" id="S3.SS1.p2.7.m4.1d">{ italic_m start_POSTSUBSCRIPT italic_i end_POSTSUBSCRIPT } start_POSTSUPERSCRIPT italic_T end_POSTSUPERSCRIPT start_POSTSUBSCRIPT italic_i = 1 end_POSTSUBSCRIPT</annotation></semantics></math> to <math alttext="\{m_{i}\}^{K}_{i=1}" class="ltx_Math" display="inline" id="S3.SS1.p2.8.m5.1"><semantics id="S3.SS1.p2.8.m5.1a"><msubsup id="S3.SS1.p2.8.m5.1.1" xref="S3.SS1.p2.8.m5.1.1.cmml"><mrow id="S3.SS1.p2.8.m5.1.1.1.1.1" xref="S3.SS1.p2.8.m5.1.1.1.1.2.cmml"><mo id="S3.SS1.p2.8.m5.1.1.1.1.1.2" stretchy="false" xref="S3.SS1.p2.8.m5.1.1.1.1.2.cmml">{</mo><msub id="S3.SS1.p2.8.m5.1.1.1.1.1.1" xref="S3.SS1.p2.8.m5.1.1.1.1.1.1.cmml"><mi id="S3.SS1.p2.8.m5.1.1.1.1.1.1.2" xref="S3.SS1.p2.8.m5.1.1.1.1.1.1.2.cmml">m</mi><mi id="S3.SS1.p2.8.m5.1.1.1.1.1.1.3" xref="S3.SS1.p2.8.m5.1.1.1.1.1.1.3.cmml">i</mi></msub><mo id="S3.SS1.p2.8.m5.1.1.1.1.1.3" stretchy="false" xref="S3.SS1.p2.8.m5.1.1.1.1.2.cmml">}</mo></mrow><mrow id="S3.SS1.p2.8.m5.1.1.3" xref="S3.SS1.p2.8.m5.1.1.3.cmml"><mi id="S3.SS1.p2.8.m5.1.1.3.2" xref="S3.SS1.p2.8.m5.1.1.3.2.cmml">i</mi><mo id="S3.SS1.p2.8.m5.1.1.3.1" xref="S3.SS1.p2.8.m5.1.1.3.1.cmml">=</mo><mn id="S3.SS1.p2.8.m5.1.1.3.3" xref="S3.SS1.p2.8.m5.1.1.3.3.cmml">1</mn></mrow><mi id="S3.SS1.p2.8.m5.1.1.1.3" xref="S3.SS1.p2.8.m5.1.1.1.3.cmml">K</mi></msubsup><annotation-xml encoding="MathML-Content" id="S3.SS1.p2.8.m5.1b"><apply id="S3.SS1.p2.8.m5.1.1.cmml" xref="S3.SS1.p2.8.m5.1.1"><csymbol cd="ambiguous" id="S3.SS1.p2.8.m5.1.1.2.cmml" xref="S3.SS1.p2.8.m5.1.1">subscript</csymbol><apply id="S3.SS1.p2.8.m5.1.1.1.cmml" xref="S3.SS1.p2.8.m5.1.1"><csymbol cd="ambiguous" id="S3.SS1.p2.8.m5.1.1.1.2.cmml" xref="S3.SS1.p2.8.m5.1.1">superscript</csymbol><set id="S3.SS1.p2.8.m5.1.1.1.1.2.cmml" xref="S3.SS1.p2.8.m5.1.1.1.1.1"><apply id="S3.SS1.p2.8.m5.1.1.1.1.1.1.cmml" xref="S3.SS1.p2.8.m5.1.1.1.1.1.1"><csymbol cd="ambiguous" id="S3.SS1.p2.8.m5.1.1.1.1.1.1.1.cmml" xref="S3.SS1.p2.8.m5.1.1.1.1.1.1">subscript</csymbol><ci id="S3.SS1.p2.8.m5.1.1.1.1.1.1.2.cmml" xref="S3.SS1.p2.8.m5.1.1.1.1.1.1.2">𝑚</ci><ci id="S3.SS1.p2.8.m5.1.1.1.1.1.1.3.cmml" xref="S3.SS1.p2.8.m5.1.1.1.1.1.1.3">𝑖</ci></apply></set><ci id="S3.SS1.p2.8.m5.1.1.1.3.cmml" xref="S3.SS1.p2.8.m5.1.1.1.3">𝐾</ci></apply><apply id="S3.SS1.p2.8.m5.1.1.3.cmml" xref="S3.SS1.p2.8.m5.1.1.3"><eq id="S3.SS1.p2.8.m5.1.1.3.1.cmml" xref="S3.SS1.p2.8.m5.1.1.3.1"></eq><ci id="S3.SS1.p2.8.m5.1.1.3.2.cmml" xref="S3.SS1.p2.8.m5.1.1.3.2">𝑖</ci><cn id="S3.SS1.p2.8.m5.1.1.3.3.cmml" type="integer" xref="S3.SS1.p2.8.m5.1.1.3.3">1</cn></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS1.p2.8.m5.1c">\{m_{i}\}^{K}_{i=1}</annotation><annotation encoding="application/x-llamapun" id="S3.SS1.p2.8.m5.1d">{ italic_m start_POSTSUBSCRIPT italic_i end_POSTSUBSCRIPT } start_POSTSUPERSCRIPT italic_K end_POSTSUPERSCRIPT start_POSTSUBSCRIPT italic_i = 1 end_POSTSUBSCRIPT</annotation></semantics></math>. The details are described in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#A1" title="Appendix A Detailed implementations and training ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Appendix</span> <span class="ltx_text ltx_ref_tag">A</span></a>.</p>
</div>
<div class="ltx_para ltx_noindent" id="S3.SS1.p3">
<p class="ltx_p" id="S3.SS1.p3.6"><span class="ltx_text ltx_font_bold" id="S3.SS1.p3.6.1">Supervised training.</span> Given the filtered set of molecules <math alttext="\{m_{i}\}^{K}_{i=1}" class="ltx_Math" display="inline" id="S3.SS1.p3.1.m1.1"><semantics id="S3.SS1.p3.1.m1.1a"><msubsup id="S3.SS1.p3.1.m1.1.1" xref="S3.SS1.p3.1.m1.1.1.cmml"><mrow id="S3.SS1.p3.1.m1.1.1.1.1.1" xref="S3.SS1.p3.1.m1.1.1.1.1.2.cmml"><mo id="S3.SS1.p3.1.m1.1.1.1.1.1.2" stretchy="false" xref="S3.SS1.p3.1.m1.1.1.1.1.2.cmml">{</mo><msub id="S3.SS1.p3.1.m1.1.1.1.1.1.1" xref="S3.SS1.p3.1.m1.1.1.1.1.1.1.cmml"><mi id="S3.SS1.p3.1.m1.1.1.1.1.1.1.2" xref="S3.SS1.p3.1.m1.1.1.1.1.1.1.2.cmml">m</mi><mi id="S3.SS1.p3.1.m1.1.1.1.1.1.1.3" xref="S3.SS1.p3.1.m1.1.1.1.1.1.1.3.cmml">i</mi></msub><mo id="S3.SS1.p3.1.m1.1.1.1.1.1.3" stretchy="false" xref="S3.SS1.p3.1.m1.1.1.1.1.2.cmml">}</mo></mrow><mrow id="S3.SS1.p3.1.m1.1.1.3" xref="S3.SS1.p3.1.m1.1.1.3.cmml"><mi id="S3.SS1.p3.1.m1.1.1.3.2" xref="S3.SS1.p3.1.m1.1.1.3.2.cmml">i</mi><mo id="S3.SS1.p3.1.m1.1.1.3.1" xref="S3.SS1.p3.1.m1.1.1.3.1.cmml">=</mo><mn id="S3.SS1.p3.1.m1.1.1.3.3" xref="S3.SS1.p3.1.m1.1.1.3.3.cmml">1</mn></mrow><mi id="S3.SS1.p3.1.m1.1.1.1.3" xref="S3.SS1.p3.1.m1.1.1.1.3.cmml">K</mi></msubsup><annotation-xml encoding="MathML-Content" id="S3.SS1.p3.1.m1.1b"><apply id="S3.SS1.p3.1.m1.1.1.cmml" xref="S3.SS1.p3.1.m1.1.1"><csymbol cd="ambiguous" id="S3.SS1.p3.1.m1.1.1.2.cmml" xref="S3.SS1.p3.1.m1.1.1">subscript</csymbol><apply id="S3.SS1.p3.1.m1.1.1.1.cmml" xref="S3.SS1.p3.1.m1.1.1"><csymbol cd="ambiguous" id="S3.SS1.p3.1.m1.1.1.1.2.cmml" xref="S3.SS1.p3.1.m1.1.1">superscript</csymbol><set id="S3.SS1.p3.1.m1.1.1.1.1.2.cmml" xref="S3.SS1.p3.1.m1.1.1.1.1.1"><apply id="S3.SS1.p3.1.m1.1.1.1.1.1.1.cmml" xref="S3.SS1.p3.1.m1.1.1.1.1.1.1"><csymbol cd="ambiguous" id="S3.SS1.p3.1.m1.1.1.1.1.1.1.1.cmml" xref="S3.SS1.p3.1.m1.1.1.1.1.1.1">subscript</csymbol><ci id="S3.SS1.p3.1.m1.1.1.1.1.1.1.2.cmml" xref="S3.SS1.p3.1.m1.1.1.1.1.1.1.2">𝑚</ci><ci id="S3.SS1.p3.1.m1.1.1.1.1.1.1.3.cmml" xref="S3.SS1.p3.1.m1.1.1.1.1.1.1.3">𝑖</ci></apply></set><ci id="S3.SS1.p3.1.m1.1.1.1.3.cmml" xref="S3.SS1.p3.1.m1.1.1.1.3">𝐾</ci></apply><apply id="S3.SS1.p3.1.m1.1.1.3.cmml" xref="S3.SS1.p3.1.m1.1.1.3"><eq id="S3.SS1.p3.1.m1.1.1.3.1.cmml" xref="S3.SS1.p3.1.m1.1.1.3.1"></eq><ci id="S3.SS1.p3.1.m1.1.1.3.2.cmml" xref="S3.SS1.p3.1.m1.1.1.3.2">𝑖</ci><cn id="S3.SS1.p3.1.m1.1.1.3.3.cmml" type="integer" xref="S3.SS1.p3.1.m1.1.1.3.3">1</cn></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS1.p3.1.m1.1c">\{m_{i}\}^{K}_{i=1}</annotation><annotation encoding="application/x-llamapun" id="S3.SS1.p3.1.m1.1d">{ italic_m start_POSTSUBSCRIPT italic_i end_POSTSUBSCRIPT } start_POSTSUPERSCRIPT italic_K end_POSTSUPERSCRIPT start_POSTSUBSCRIPT italic_i = 1 end_POSTSUBSCRIPT</annotation></semantics></math>, we train the LLM <math alttext="\pi_{\text{SFT}}" class="ltx_Math" display="inline" id="S3.SS1.p3.2.m2.1"><semantics id="S3.SS1.p3.2.m2.1a"><msub id="S3.SS1.p3.2.m2.1.1" xref="S3.SS1.p3.2.m2.1.1.cmml"><mi id="S3.SS1.p3.2.m2.1.1.2" xref="S3.SS1.p3.2.m2.1.1.2.cmml">π</mi><mtext id="S3.SS1.p3.2.m2.1.1.3" xref="S3.SS1.p3.2.m2.1.1.3a.cmml">SFT</mtext></msub><annotation-xml encoding="MathML-Content" id="S3.SS1.p3.2.m2.1b"><apply id="S3.SS1.p3.2.m2.1.1.cmml" xref="S3.SS1.p3.2.m2.1.1"><csymbol cd="ambiguous" id="S3.SS1.p3.2.m2.1.1.1.cmml" xref="S3.SS1.p3.2.m2.1.1">subscript</csymbol><ci id="S3.SS1.p3.2.m2.1.1.2.cmml" xref="S3.SS1.p3.2.m2.1.1.2">𝜋</ci><ci id="S3.SS1.p3.2.m2.1.1.3a.cmml" xref="S3.SS1.p3.2.m2.1.1.3"><mtext id="S3.SS1.p3.2.m2.1.1.3.cmml" mathsize="70%" xref="S3.SS1.p3.2.m2.1.1.3">SFT</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS1.p3.2.m2.1c">\pi_{\text{SFT}}</annotation><annotation encoding="application/x-llamapun" id="S3.SS1.p3.2.m2.1d">italic_π start_POSTSUBSCRIPT SFT end_POSTSUBSCRIPT</annotation></semantics></math>, which is initialized with <math alttext="\pi_{\text{pre}}" class="ltx_Math" display="inline" id="S3.SS1.p3.3.m3.1"><semantics id="S3.SS1.p3.3.m3.1a"><msub id="S3.SS1.p3.3.m3.1.1" xref="S3.SS1.p3.3.m3.1.1.cmml"><mi id="S3.SS1.p3.3.m3.1.1.2" xref="S3.SS1.p3.3.m3.1.1.2.cmml">π</mi><mtext id="S3.SS1.p3.3.m3.1.1.3" xref="S3.SS1.p3.3.m3.1.1.3a.cmml">pre</mtext></msub><annotation-xml encoding="MathML-Content" id="S3.SS1.p3.3.m3.1b"><apply id="S3.SS1.p3.3.m3.1.1.cmml" xref="S3.SS1.p3.3.m3.1.1"><csymbol cd="ambiguous" id="S3.SS1.p3.3.m3.1.1.1.cmml" xref="S3.SS1.p3.3.m3.1.1">subscript</csymbol><ci id="S3.SS1.p3.3.m3.1.1.2.cmml" xref="S3.SS1.p3.3.m3.1.1.2">𝜋</ci><ci id="S3.SS1.p3.3.m3.1.1.3a.cmml" xref="S3.SS1.p3.3.m3.1.1.3"><mtext id="S3.SS1.p3.3.m3.1.1.3.cmml" mathsize="70%" xref="S3.SS1.p3.3.m3.1.1.3">pre</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS1.p3.3.m3.1c">\pi_{\text{pre}}</annotation><annotation encoding="application/x-llamapun" id="S3.SS1.p3.3.m3.1d">italic_π start_POSTSUBSCRIPT pre end_POSTSUBSCRIPT</annotation></semantics></math>, to generate them as a single concatenated sequence. We denote this sequence by <math alttext="\mathcal{M}_{1:K}=m_{1}||\cdots||m_{K}" class="ltx_Math" display="inline" id="S3.SS1.p3.4.m4.1"><semantics id="S3.SS1.p3.4.m4.1a"><mrow id="S3.SS1.p3.4.m4.1.2" xref="S3.SS1.p3.4.m4.1.2.cmml"><msub id="S3.SS1.p3.4.m4.1.2.2" xref="S3.SS1.p3.4.m4.1.2.2.cmml"><mi class="ltx_font_mathcaligraphic" id="S3.SS1.p3.4.m4.1.2.2.2" xref="S3.SS1.p3.4.m4.1.2.2.2.cmml">ℳ</mi><mrow id="S3.SS1.p3.4.m4.1.2.2.3" xref="S3.SS1.p3.4.m4.1.2.2.3.cmml"><mn id="S3.SS1.p3.4.m4.1.2.2.3.2" xref="S3.SS1.p3.4.m4.1.2.2.3.2.cmml">1</mn><mo id="S3.SS1.p3.4.m4.1.2.2.3.1" lspace="0.278em" rspace="0.278em" xref="S3.SS1.p3.4.m4.1.2.2.3.1.cmml">:</mo><mi id="S3.SS1.p3.4.m4.1.2.2.3.3" xref="S3.SS1.p3.4.m4.1.2.2.3.3.cmml">K</mi></mrow></msub><mo id="S3.SS1.p3.4.m4.1.2.1" xref="S3.SS1.p3.4.m4.1.2.1.cmml">=</mo><mrow id="S3.SS1.p3.4.m4.1.2.3" xref="S3.SS1.p3.4.m4.1.2.3.cmml"><msub id="S3.SS1.p3.4.m4.1.2.3.2" xref="S3.SS1.p3.4.m4.1.2.3.2.cmml"><mi id="S3.SS1.p3.4.m4.1.2.3.2.2" xref="S3.SS1.p3.4.m4.1.2.3.2.2.cmml">m</mi><mn id="S3.SS1.p3.4.m4.1.2.3.2.3" xref="S3.SS1.p3.4.m4.1.2.3.2.3.cmml">1</mn></msub><mo id="S3.SS1.p3.4.m4.1.2.3.1" xref="S3.SS1.p3.4.m4.1.2.3.1.cmml">⁢</mo><mrow id="S3.SS1.p3.4.m4.1.2.3.3.2" xref="S3.SS1.p3.4.m4.1.2.3.3.1.cmml"><mo id="S3.SS1.p3.4.m4.1.2.3.3.2.1" stretchy="false" xref="S3.SS1.p3.4.m4.1.2.3.3.1.1.cmml">‖</mo><mi id="S3.SS1.p3.4.m4.1.1" mathvariant="normal" xref="S3.SS1.p3.4.m4.1.1.cmml">⋯</mi><mo id="S3.SS1.p3.4.m4.1.2.3.3.2.2" stretchy="false" xref="S3.SS1.p3.4.m4.1.2.3.3.1.1.cmml">‖</mo></mrow><mo id="S3.SS1.p3.4.m4.1.2.3.1a" xref="S3.SS1.p3.4.m4.1.2.3.1.cmml">⁢</mo><msub id="S3.SS1.p3.4.m4.1.2.3.4" xref="S3.SS1.p3.4.m4.1.2.3.4.cmml"><mi id="S3.SS1.p3.4.m4.1.2.3.4.2" xref="S3.SS1.p3.4.m4.1.2.3.4.2.cmml">m</mi><mi id="S3.SS1.p3.4.m4.1.2.3.4.3" xref="S3.SS1.p3.4.m4.1.2.3.4.3.cmml">K</mi></msub></mrow></mrow><annotation-xml encoding="MathML-Content" id="S3.SS1.p3.4.m4.1b"><apply id="S3.SS1.p3.4.m4.1.2.cmml" xref="S3.SS1.p3.4.m4.1.2"><eq id="S3.SS1.p3.4.m4.1.2.1.cmml" xref="S3.SS1.p3.4.m4.1.2.1"></eq><apply id="S3.SS1.p3.4.m4.1.2.2.cmml" xref="S3.SS1.p3.4.m4.1.2.2"><csymbol cd="ambiguous" id="S3.SS1.p3.4.m4.1.2.2.1.cmml" xref="S3.SS1.p3.4.m4.1.2.2">subscript</csymbol><ci id="S3.SS1.p3.4.m4.1.2.2.2.cmml" xref="S3.SS1.p3.4.m4.1.2.2.2">ℳ</ci><apply id="S3.SS1.p3.4.m4.1.2.2.3.cmml" xref="S3.SS1.p3.4.m4.1.2.2.3"><ci id="S3.SS1.p3.4.m4.1.2.2.3.1.cmml" xref="S3.SS1.p3.4.m4.1.2.2.3.1">:</ci><cn id="S3.SS1.p3.4.m4.1.2.2.3.2.cmml" type="integer" xref="S3.SS1.p3.4.m4.1.2.2.3.2">1</cn><ci id="S3.SS1.p3.4.m4.1.2.2.3.3.cmml" xref="S3.SS1.p3.4.m4.1.2.2.3.3">𝐾</ci></apply></apply><apply id="S3.SS1.p3.4.m4.1.2.3.cmml" xref="S3.SS1.p3.4.m4.1.2.3"><times id="S3.SS1.p3.4.m4.1.2.3.1.cmml" xref="S3.SS1.p3.4.m4.1.2.3.1"></times><apply id="S3.SS1.p3.4.m4.1.2.3.2.cmml" xref="S3.SS1.p3.4.m4.1.2.3.2"><csymbol cd="ambiguous" id="S3.SS1.p3.4.m4.1.2.3.2.1.cmml" xref="S3.SS1.p3.4.m4.1.2.3.2">subscript</csymbol><ci id="S3.SS1.p3.4.m4.1.2.3.2.2.cmml" xref="S3.SS1.p3.4.m4.1.2.3.2.2">𝑚</ci><cn id="S3.SS1.p3.4.m4.1.2.3.2.3.cmml" type="integer" xref="S3.SS1.p3.4.m4.1.2.3.2.3">1</cn></apply><apply id="S3.SS1.p3.4.m4.1.2.3.3.1.cmml" xref="S3.SS1.p3.4.m4.1.2.3.3.2"><csymbol cd="latexml" id="S3.SS1.p3.4.m4.1.2.3.3.1.1.cmml" xref="S3.SS1.p3.4.m4.1.2.3.3.2.1">norm</csymbol><ci id="S3.SS1.p3.4.m4.1.1.cmml" xref="S3.SS1.p3.4.m4.1.1">⋯</ci></apply><apply id="S3.SS1.p3.4.m4.1.2.3.4.cmml" xref="S3.SS1.p3.4.m4.1.2.3.4"><csymbol cd="ambiguous" id="S3.SS1.p3.4.m4.1.2.3.4.1.cmml" xref="S3.SS1.p3.4.m4.1.2.3.4">subscript</csymbol><ci id="S3.SS1.p3.4.m4.1.2.3.4.2.cmml" xref="S3.SS1.p3.4.m4.1.2.3.4.2">𝑚</ci><ci id="S3.SS1.p3.4.m4.1.2.3.4.3.cmml" xref="S3.SS1.p3.4.m4.1.2.3.4.3">𝐾</ci></apply></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS1.p3.4.m4.1c">\mathcal{M}_{1:K}=m_{1}||\cdots||m_{K}</annotation><annotation encoding="application/x-llamapun" id="S3.SS1.p3.4.m4.1d">caligraphic_M start_POSTSUBSCRIPT 1 : italic_K end_POSTSUBSCRIPT = italic_m start_POSTSUBSCRIPT 1 end_POSTSUBSCRIPT | | ⋯ | | italic_m start_POSTSUBSCRIPT italic_K end_POSTSUBSCRIPT</annotation></semantics></math>, where <math alttext="||" class="ltx_math_unparsed" display="inline" id="S3.SS1.p3.5.m5.1"><semantics id="S3.SS1.p3.5.m5.1a"><mrow id="S3.SS1.p3.5.m5.1b"><mo fence="false" id="S3.SS1.p3.5.m5.1.1" rspace="0.167em" stretchy="false">|</mo><mo fence="false" id="S3.SS1.p3.5.m5.1.2" stretchy="false">|</mo></mrow><annotation encoding="application/x-tex" id="S3.SS1.p3.5.m5.1c">||</annotation><annotation encoding="application/x-llamapun" id="S3.SS1.p3.5.m5.1d">| |</annotation></semantics></math> denotes the concatenation of the molecular string representations. Specifically, given a modified prompt <math alttext="p_{\text{desc+div}}" class="ltx_Math" display="inline" id="S3.SS1.p3.6.m6.1"><semantics id="S3.SS1.p3.6.m6.1a"><msub id="S3.SS1.p3.6.m6.1.1" xref="S3.SS1.p3.6.m6.1.1.cmml"><mi id="S3.SS1.p3.6.m6.1.1.2" xref="S3.SS1.p3.6.m6.1.1.2.cmml">p</mi><mtext id="S3.SS1.p3.6.m6.1.1.3" xref="S3.SS1.p3.6.m6.1.1.3a.cmml">desc+div</mtext></msub><annotation-xml encoding="MathML-Content" id="S3.SS1.p3.6.m6.1b"><apply id="S3.SS1.p3.6.m6.1.1.cmml" xref="S3.SS1.p3.6.m6.1.1"><csymbol cd="ambiguous" id="S3.SS1.p3.6.m6.1.1.1.cmml" xref="S3.SS1.p3.6.m6.1.1">subscript</csymbol><ci id="S3.SS1.p3.6.m6.1.1.2.cmml" xref="S3.SS1.p3.6.m6.1.1.2">𝑝</ci><ci id="S3.SS1.p3.6.m6.1.1.3a.cmml" xref="S3.SS1.p3.6.m6.1.1.3"><mtext id="S3.SS1.p3.6.m6.1.1.3.cmml" mathsize="70%" xref="S3.SS1.p3.6.m6.1.1.3">desc+div</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS1.p3.6.m6.1c">p_{\text{desc+div}}</annotation><annotation encoding="application/x-llamapun" id="S3.SS1.p3.6.m6.1d">italic_p start_POSTSUBSCRIPT desc+div end_POSTSUBSCRIPT</annotation></semantics></math> describing the target property with a request for generating diverse molecules, we train the LLM to maximize the log-likelihood:</p>
<table class="ltx_equation ltx_eqn_table" id="S3.E1">
<tbody><tr class="ltx_equation ltx_eqn_row ltx_align_baseline">
<td class="ltx_eqn_cell ltx_eqn_center_padleft"></td>
<td class="ltx_eqn_cell ltx_align_center"><math alttext="\log\pi_{\text{SFT}}\left(\mathcal{M}_{1:K}\;\middle|\;p_{\text{desc+div}}%
\right)." class="ltx_math_unparsed" display="block" id="S3.E1.m1.1"><semantics id="S3.E1.m1.1a"><mrow id="S3.E1.m1.1b"><mi id="S3.E1.m1.1.1">log</mi><msub id="S3.E1.m1.1.2"><mi id="S3.E1.m1.1.2.2">π</mi><mtext id="S3.E1.m1.1.2.3">SFT</mtext></msub><mrow id="S3.E1.m1.1.3"><mo id="S3.E1.m1.1.3.1">(</mo><msub id="S3.E1.m1.1.3.2"><mi class="ltx_font_mathcaligraphic" id="S3.E1.m1.1.3.2.2">ℳ</mi><mrow id="S3.E1.m1.1.3.2.3"><mn id="S3.E1.m1.1.3.2.3.2">1</mn><mo id="S3.E1.m1.1.3.2.3.1" lspace="0.278em" rspace="0.278em">:</mo><mi id="S3.E1.m1.1.3.2.3.3">K</mi></mrow></msub><mo id="S3.E1.m1.1.3.3" lspace="0em" stretchy="true">|</mo><msub id="S3.E1.m1.1.3.4"><mi id="S3.E1.m1.1.3.4.2">p</mi><mtext id="S3.E1.m1.1.3.4.3">desc+div</mtext></msub><mo id="S3.E1.m1.1.3.5">)</mo></mrow><mo id="S3.E1.m1.1.4" lspace="0em">.</mo></mrow><annotation encoding="application/x-tex" id="S3.E1.m1.1c">\log\pi_{\text{SFT}}\left(\mathcal{M}_{1:K}\;\middle|\;p_{\text{desc+div}}%
\right).</annotation><annotation encoding="application/x-llamapun" id="S3.E1.m1.1d">roman_log italic_π start_POSTSUBSCRIPT SFT end_POSTSUBSCRIPT ( caligraphic_M start_POSTSUBSCRIPT 1 : italic_K end_POSTSUBSCRIPT | italic_p start_POSTSUBSCRIPT desc+div end_POSTSUBSCRIPT ) .</annotation></semantics></math></td>
<td class="ltx_eqn_cell ltx_eqn_center_padright"></td>
<td class="ltx_eqn_cell ltx_eqn_eqno ltx_align_middle ltx_align_right" rowspan="1"><span class="ltx_tag ltx_tag_equation ltx_align_right">(1)</span></td>
</tr></tbody>
</table>
<p class="ltx_p" id="S3.SS1.p3.9">However, the policy <math alttext="\pi_{\text{SFT}}" class="ltx_Math" display="inline" id="S3.SS1.p3.7.m1.1"><semantics id="S3.SS1.p3.7.m1.1a"><msub id="S3.SS1.p3.7.m1.1.1" xref="S3.SS1.p3.7.m1.1.1.cmml"><mi id="S3.SS1.p3.7.m1.1.1.2" xref="S3.SS1.p3.7.m1.1.1.2.cmml">π</mi><mtext id="S3.SS1.p3.7.m1.1.1.3" xref="S3.SS1.p3.7.m1.1.1.3a.cmml">SFT</mtext></msub><annotation-xml encoding="MathML-Content" id="S3.SS1.p3.7.m1.1b"><apply id="S3.SS1.p3.7.m1.1.1.cmml" xref="S3.SS1.p3.7.m1.1.1"><csymbol cd="ambiguous" id="S3.SS1.p3.7.m1.1.1.1.cmml" xref="S3.SS1.p3.7.m1.1.1">subscript</csymbol><ci id="S3.SS1.p3.7.m1.1.1.2.cmml" xref="S3.SS1.p3.7.m1.1.1.2">𝜋</ci><ci id="S3.SS1.p3.7.m1.1.1.3a.cmml" xref="S3.SS1.p3.7.m1.1.1.3"><mtext id="S3.SS1.p3.7.m1.1.1.3.cmml" mathsize="70%" xref="S3.SS1.p3.7.m1.1.1.3">SFT</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS1.p3.7.m1.1c">\pi_{\text{SFT}}</annotation><annotation encoding="application/x-llamapun" id="S3.SS1.p3.7.m1.1d">italic_π start_POSTSUBSCRIPT SFT end_POSTSUBSCRIPT</annotation></semantics></math> does not necessarily incorporate a molecular structural diversity, as the set of molecules <math alttext="\{m_{i}\}^{K}_{i=1}" class="ltx_Math" display="inline" id="S3.SS1.p3.8.m2.1"><semantics id="S3.SS1.p3.8.m2.1a"><msubsup id="S3.SS1.p3.8.m2.1.1" xref="S3.SS1.p3.8.m2.1.1.cmml"><mrow id="S3.SS1.p3.8.m2.1.1.1.1.1" xref="S3.SS1.p3.8.m2.1.1.1.1.2.cmml"><mo id="S3.SS1.p3.8.m2.1.1.1.1.1.2" stretchy="false" xref="S3.SS1.p3.8.m2.1.1.1.1.2.cmml">{</mo><msub id="S3.SS1.p3.8.m2.1.1.1.1.1.1" xref="S3.SS1.p3.8.m2.1.1.1.1.1.1.cmml"><mi id="S3.SS1.p3.8.m2.1.1.1.1.1.1.2" xref="S3.SS1.p3.8.m2.1.1.1.1.1.1.2.cmml">m</mi><mi id="S3.SS1.p3.8.m2.1.1.1.1.1.1.3" xref="S3.SS1.p3.8.m2.1.1.1.1.1.1.3.cmml">i</mi></msub><mo id="S3.SS1.p3.8.m2.1.1.1.1.1.3" stretchy="false" xref="S3.SS1.p3.8.m2.1.1.1.1.2.cmml">}</mo></mrow><mrow id="S3.SS1.p3.8.m2.1.1.3" xref="S3.SS1.p3.8.m2.1.1.3.cmml"><mi id="S3.SS1.p3.8.m2.1.1.3.2" xref="S3.SS1.p3.8.m2.1.1.3.2.cmml">i</mi><mo id="S3.SS1.p3.8.m2.1.1.3.1" xref="S3.SS1.p3.8.m2.1.1.3.1.cmml">=</mo><mn id="S3.SS1.p3.8.m2.1.1.3.3" xref="S3.SS1.p3.8.m2.1.1.3.3.cmml">1</mn></mrow><mi id="S3.SS1.p3.8.m2.1.1.1.3" xref="S3.SS1.p3.8.m2.1.1.1.3.cmml">K</mi></msubsup><annotation-xml encoding="MathML-Content" id="S3.SS1.p3.8.m2.1b"><apply id="S3.SS1.p3.8.m2.1.1.cmml" xref="S3.SS1.p3.8.m2.1.1"><csymbol cd="ambiguous" id="S3.SS1.p3.8.m2.1.1.2.cmml" xref="S3.SS1.p3.8.m2.1.1">subscript</csymbol><apply id="S3.SS1.p3.8.m2.1.1.1.cmml" xref="S3.SS1.p3.8.m2.1.1"><csymbol cd="ambiguous" id="S3.SS1.p3.8.m2.1.1.1.2.cmml" xref="S3.SS1.p3.8.m2.1.1">superscript</csymbol><set id="S3.SS1.p3.8.m2.1.1.1.1.2.cmml" xref="S3.SS1.p3.8.m2.1.1.1.1.1"><apply id="S3.SS1.p3.8.m2.1.1.1.1.1.1.cmml" xref="S3.SS1.p3.8.m2.1.1.1.1.1.1"><csymbol cd="ambiguous" id="S3.SS1.p3.8.m2.1.1.1.1.1.1.1.cmml" xref="S3.SS1.p3.8.m2.1.1.1.1.1.1">subscript</csymbol><ci id="S3.SS1.p3.8.m2.1.1.1.1.1.1.2.cmml" xref="S3.SS1.p3.8.m2.1.1.1.1.1.1.2">𝑚</ci><ci id="S3.SS1.p3.8.m2.1.1.1.1.1.1.3.cmml" xref="S3.SS1.p3.8.m2.1.1.1.1.1.1.3">𝑖</ci></apply></set><ci id="S3.SS1.p3.8.m2.1.1.1.3.cmml" xref="S3.SS1.p3.8.m2.1.1.1.3">𝐾</ci></apply><apply id="S3.SS1.p3.8.m2.1.1.3.cmml" xref="S3.SS1.p3.8.m2.1.1.3"><eq id="S3.SS1.p3.8.m2.1.1.3.1.cmml" xref="S3.SS1.p3.8.m2.1.1.3.1"></eq><ci id="S3.SS1.p3.8.m2.1.1.3.2.cmml" xref="S3.SS1.p3.8.m2.1.1.3.2">𝑖</ci><cn id="S3.SS1.p3.8.m2.1.1.3.3.cmml" type="integer" xref="S3.SS1.p3.8.m2.1.1.3.3">1</cn></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS1.p3.8.m2.1c">\{m_{i}\}^{K}_{i=1}</annotation><annotation encoding="application/x-llamapun" id="S3.SS1.p3.8.m2.1d">{ italic_m start_POSTSUBSCRIPT italic_i end_POSTSUBSCRIPT } start_POSTSUPERSCRIPT italic_K end_POSTSUPERSCRIPT start_POSTSUBSCRIPT italic_i = 1 end_POSTSUBSCRIPT</annotation></semantics></math> collected from <math alttext="\pi_{\text{pre}}" class="ltx_Math" display="inline" id="S3.SS1.p3.9.m3.1"><semantics id="S3.SS1.p3.9.m3.1a"><msub id="S3.SS1.p3.9.m3.1.1" xref="S3.SS1.p3.9.m3.1.1.cmml"><mi id="S3.SS1.p3.9.m3.1.1.2" xref="S3.SS1.p3.9.m3.1.1.2.cmml">π</mi><mtext id="S3.SS1.p3.9.m3.1.1.3" xref="S3.SS1.p3.9.m3.1.1.3a.cmml">pre</mtext></msub><annotation-xml encoding="MathML-Content" id="S3.SS1.p3.9.m3.1b"><apply id="S3.SS1.p3.9.m3.1.1.cmml" xref="S3.SS1.p3.9.m3.1.1"><csymbol cd="ambiguous" id="S3.SS1.p3.9.m3.1.1.1.cmml" xref="S3.SS1.p3.9.m3.1.1">subscript</csymbol><ci id="S3.SS1.p3.9.m3.1.1.2.cmml" xref="S3.SS1.p3.9.m3.1.1.2">𝜋</ci><ci id="S3.SS1.p3.9.m3.1.1.3a.cmml" xref="S3.SS1.p3.9.m3.1.1.3"><mtext id="S3.SS1.p3.9.m3.1.1.3.cmml" mathsize="70%" xref="S3.SS1.p3.9.m3.1.1.3">pre</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS1.p3.9.m3.1c">\pi_{\text{pre}}</annotation><annotation encoding="application/x-llamapun" id="S3.SS1.p3.9.m3.1d">italic_π start_POSTSUBSCRIPT pre end_POSTSUBSCRIPT</annotation></semantics></math> may insufficiently involve diverse molecular structures (e.g., due to limitations in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S1.F3.sf2" title="In Figure 3 ‣ 1 Introduction ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Figure</span> <span class="ltx_text ltx_ref_tag">3(b)</span></a>). To tackle this, we next introduce an online reinforcement learning stage with exploration towards discovering diverse molecules.</p>
</div>
</section>
<section class="ltx_subsection" id="S3.SS2">
<h3 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">3.2 </span>Reinforcement learning</h3>
<figure class="ltx_figure ltx_align_floatright" id="S3.SS2.fig1">
<figure class="ltx_float ltx_minipage ltx_align_middle ltx_framed ltx_framed_top" id="alg2" style="width:206.7pt;">
<figcaption class="ltx_caption"><span class="ltx_tag ltx_tag_float"><span class="ltx_text ltx_font_bold" id="alg2.2.1.1">Algorithm 2</span> </span> Multi-stage RL fine-tuning</figcaption>
<div class="ltx_listing ltx_listing" id="alg2.3">
<div class="ltx_listingline" id="alg2.l1">
<span class="ltx_tag ltx_tag_listingline">1:</span>  Sample <math alttext="p_{\text{desc}}\sim\mathcal{P}" class="ltx_Math" display="inline" id="alg2.l1.m1.1"><semantics id="alg2.l1.m1.1a"><mrow id="alg2.l1.m1.1.1" xref="alg2.l1.m1.1.1.cmml"><msub id="alg2.l1.m1.1.1.2" xref="alg2.l1.m1.1.1.2.cmml"><mi id="alg2.l1.m1.1.1.2.2" xref="alg2.l1.m1.1.1.2.2.cmml">p</mi><mtext id="alg2.l1.m1.1.1.2.3" xref="alg2.l1.m1.1.1.2.3a.cmml">desc</mtext></msub><mo id="alg2.l1.m1.1.1.1" xref="alg2.l1.m1.1.1.1.cmml">∼</mo><mi class="ltx_font_mathcaligraphic" id="alg2.l1.m1.1.1.3" xref="alg2.l1.m1.1.1.3.cmml">𝒫</mi></mrow><annotation-xml encoding="MathML-Content" id="alg2.l1.m1.1b"><apply id="alg2.l1.m1.1.1.cmml" xref="alg2.l1.m1.1.1"><csymbol cd="latexml" id="alg2.l1.m1.1.1.1.cmml" xref="alg2.l1.m1.1.1.1">similar-to</csymbol><apply id="alg2.l1.m1.1.1.2.cmml" xref="alg2.l1.m1.1.1.2"><csymbol cd="ambiguous" id="alg2.l1.m1.1.1.2.1.cmml" xref="alg2.l1.m1.1.1.2">subscript</csymbol><ci id="alg2.l1.m1.1.1.2.2.cmml" xref="alg2.l1.m1.1.1.2.2">𝑝</ci><ci id="alg2.l1.m1.1.1.2.3a.cmml" xref="alg2.l1.m1.1.1.2.3"><mtext id="alg2.l1.m1.1.1.2.3.cmml" mathsize="70%" xref="alg2.l1.m1.1.1.2.3">desc</mtext></ci></apply><ci id="alg2.l1.m1.1.1.3.cmml" xref="alg2.l1.m1.1.1.3">𝒫</ci></apply></annotation-xml><annotation encoding="application/x-tex" id="alg2.l1.m1.1c">p_{\text{desc}}\sim\mathcal{P}</annotation><annotation encoding="application/x-llamapun" id="alg2.l1.m1.1d">italic_p start_POSTSUBSCRIPT desc end_POSTSUBSCRIPT ∼ caligraphic_P</annotation></semantics></math>
</div>
<div class="ltx_listingline" id="alg2.l2">
<span class="ltx_tag ltx_tag_listingline">2:</span>  Sample <math alttext="m_{1}\sim\pi_{\text{RL}}(m_{1}\mid p_{\text{desc+div}})" class="ltx_Math" display="inline" id="alg2.l2.m1.1"><semantics id="alg2.l2.m1.1a"><mrow id="alg2.l2.m1.1.1" xref="alg2.l2.m1.1.1.cmml"><msub id="alg2.l2.m1.1.1.3" xref="alg2.l2.m1.1.1.3.cmml"><mi id="alg2.l2.m1.1.1.3.2" xref="alg2.l2.m1.1.1.3.2.cmml">m</mi><mn id="alg2.l2.m1.1.1.3.3" xref="alg2.l2.m1.1.1.3.3.cmml">1</mn></msub><mo id="alg2.l2.m1.1.1.2" xref="alg2.l2.m1.1.1.2.cmml">∼</mo><mrow id="alg2.l2.m1.1.1.1" xref="alg2.l2.m1.1.1.1.cmml"><msub id="alg2.l2.m1.1.1.1.3" xref="alg2.l2.m1.1.1.1.3.cmml"><mi id="alg2.l2.m1.1.1.1.3.2" xref="alg2.l2.m1.1.1.1.3.2.cmml">π</mi><mtext id="alg2.l2.m1.1.1.1.3.3" xref="alg2.l2.m1.1.1.1.3.3a.cmml">RL</mtext></msub><mo id="alg2.l2.m1.1.1.1.2" xref="alg2.l2.m1.1.1.1.2.cmml">⁢</mo><mrow id="alg2.l2.m1.1.1.1.1.1" xref="alg2.l2.m1.1.1.1.1.1.1.cmml"><mo id="alg2.l2.m1.1.1.1.1.1.2" stretchy="false" xref="alg2.l2.m1.1.1.1.1.1.1.cmml">(</mo><mrow id="alg2.l2.m1.1.1.1.1.1.1" xref="alg2.l2.m1.1.1.1.1.1.1.cmml"><msub id="alg2.l2.m1.1.1.1.1.1.1.2" xref="alg2.l2.m1.1.1.1.1.1.1.2.cmml"><mi id="alg2.l2.m1.1.1.1.1.1.1.2.2" xref="alg2.l2.m1.1.1.1.1.1.1.2.2.cmml">m</mi><mn id="alg2.l2.m1.1.1.1.1.1.1.2.3" xref="alg2.l2.m1.1.1.1.1.1.1.2.3.cmml">1</mn></msub><mo id="alg2.l2.m1.1.1.1.1.1.1.1" xref="alg2.l2.m1.1.1.1.1.1.1.1.cmml">∣</mo><msub id="alg2.l2.m1.1.1.1.1.1.1.3" xref="alg2.l2.m1.1.1.1.1.1.1.3.cmml"><mi id="alg2.l2.m1.1.1.1.1.1.1.3.2" xref="alg2.l2.m1.1.1.1.1.1.1.3.2.cmml">p</mi><mtext id="alg2.l2.m1.1.1.1.1.1.1.3.3" xref="alg2.l2.m1.1.1.1.1.1.1.3.3a.cmml">desc+div</mtext></msub></mrow><mo id="alg2.l2.m1.1.1.1.1.1.3" stretchy="false" xref="alg2.l2.m1.1.1.1.1.1.1.cmml">)</mo></mrow></mrow></mrow><annotation-xml encoding="MathML-Content" id="alg2.l2.m1.1b"><apply id="alg2.l2.m1.1.1.cmml" xref="alg2.l2.m1.1.1"><csymbol cd="latexml" id="alg2.l2.m1.1.1.2.cmml" xref="alg2.l2.m1.1.1.2">similar-to</csymbol><apply id="alg2.l2.m1.1.1.3.cmml" xref="alg2.l2.m1.1.1.3"><csymbol cd="ambiguous" id="alg2.l2.m1.1.1.3.1.cmml" xref="alg2.l2.m1.1.1.3">subscript</csymbol><ci id="alg2.l2.m1.1.1.3.2.cmml" xref="alg2.l2.m1.1.1.3.2">𝑚</ci><cn id="alg2.l2.m1.1.1.3.3.cmml" type="integer" xref="alg2.l2.m1.1.1.3.3">1</cn></apply><apply id="alg2.l2.m1.1.1.1.cmml" xref="alg2.l2.m1.1.1.1"><times id="alg2.l2.m1.1.1.1.2.cmml" xref="alg2.l2.m1.1.1.1.2"></times><apply id="alg2.l2.m1.1.1.1.3.cmml" xref="alg2.l2.m1.1.1.1.3"><csymbol cd="ambiguous" id="alg2.l2.m1.1.1.1.3.1.cmml" xref="alg2.l2.m1.1.1.1.3">subscript</csymbol><ci id="alg2.l2.m1.1.1.1.3.2.cmml" xref="alg2.l2.m1.1.1.1.3.2">𝜋</ci><ci id="alg2.l2.m1.1.1.1.3.3a.cmml" xref="alg2.l2.m1.1.1.1.3.3"><mtext id="alg2.l2.m1.1.1.1.3.3.cmml" mathsize="70%" xref="alg2.l2.m1.1.1.1.3.3">RL</mtext></ci></apply><apply id="alg2.l2.m1.1.1.1.1.1.1.cmml" xref="alg2.l2.m1.1.1.1.1.1"><csymbol cd="latexml" id="alg2.l2.m1.1.1.1.1.1.1.1.cmml" xref="alg2.l2.m1.1.1.1.1.1.1.1">conditional</csymbol><apply id="alg2.l2.m1.1.1.1.1.1.1.2.cmml" xref="alg2.l2.m1.1.1.1.1.1.1.2"><csymbol cd="ambiguous" id="alg2.l2.m1.1.1.1.1.1.1.2.1.cmml" xref="alg2.l2.m1.1.1.1.1.1.1.2">subscript</csymbol><ci id="alg2.l2.m1.1.1.1.1.1.1.2.2.cmml" xref="alg2.l2.m1.1.1.1.1.1.1.2.2">𝑚</ci><cn id="alg2.l2.m1.1.1.1.1.1.1.2.3.cmml" type="integer" xref="alg2.l2.m1.1.1.1.1.1.1.2.3">1</cn></apply><apply id="alg2.l2.m1.1.1.1.1.1.1.3.cmml" xref="alg2.l2.m1.1.1.1.1.1.1.3"><csymbol cd="ambiguous" id="alg2.l2.m1.1.1.1.1.1.1.3.1.cmml" xref="alg2.l2.m1.1.1.1.1.1.1.3">subscript</csymbol><ci id="alg2.l2.m1.1.1.1.1.1.1.3.2.cmml" xref="alg2.l2.m1.1.1.1.1.1.1.3.2">𝑝</ci><ci id="alg2.l2.m1.1.1.1.1.1.1.3.3a.cmml" xref="alg2.l2.m1.1.1.1.1.1.1.3.3"><mtext id="alg2.l2.m1.1.1.1.1.1.1.3.3.cmml" mathsize="70%" xref="alg2.l2.m1.1.1.1.1.1.1.3.3">desc+div</mtext></ci></apply></apply></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="alg2.l2.m1.1c">m_{1}\sim\pi_{\text{RL}}(m_{1}\mid p_{\text{desc+div}})</annotation><annotation encoding="application/x-llamapun" id="alg2.l2.m1.1d">italic_m start_POSTSUBSCRIPT 1 end_POSTSUBSCRIPT ∼ italic_π start_POSTSUBSCRIPT RL end_POSTSUBSCRIPT ( italic_m start_POSTSUBSCRIPT 1 end_POSTSUBSCRIPT ∣ italic_p start_POSTSUBSCRIPT desc+div end_POSTSUBSCRIPT )</annotation></semantics></math>
</div>
<div class="ltx_listingline" id="alg2.l3">
<span class="ltx_tag ltx_tag_listingline">3:</span>  Update <math alttext="\pi_{\text{RL}}" class="ltx_Math" display="inline" id="alg2.l3.m1.1"><semantics id="alg2.l3.m1.1a"><msub id="alg2.l3.m1.1.1" xref="alg2.l3.m1.1.1.cmml"><mi id="alg2.l3.m1.1.1.2" xref="alg2.l3.m1.1.1.2.cmml">π</mi><mtext id="alg2.l3.m1.1.1.3" xref="alg2.l3.m1.1.1.3a.cmml">RL</mtext></msub><annotation-xml encoding="MathML-Content" id="alg2.l3.m1.1b"><apply id="alg2.l3.m1.1.1.cmml" xref="alg2.l3.m1.1.1"><csymbol cd="ambiguous" id="alg2.l3.m1.1.1.1.cmml" xref="alg2.l3.m1.1.1">subscript</csymbol><ci id="alg2.l3.m1.1.1.2.cmml" xref="alg2.l3.m1.1.1.2">𝜋</ci><ci id="alg2.l3.m1.1.1.3a.cmml" xref="alg2.l3.m1.1.1.3"><mtext id="alg2.l3.m1.1.1.3.cmml" mathsize="70%" xref="alg2.l3.m1.1.1.3">RL</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="alg2.l3.m1.1c">\pi_{\text{RL}}</annotation><annotation encoding="application/x-llamapun" id="alg2.l3.m1.1d">italic_π start_POSTSUBSCRIPT RL end_POSTSUBSCRIPT</annotation></semantics></math> with PPO to maximize <math alttext="r(m_{1})" class="ltx_Math" display="inline" id="alg2.l3.m2.1"><semantics id="alg2.l3.m2.1a"><mrow id="alg2.l3.m2.1.1" xref="alg2.l3.m2.1.1.cmml"><mi id="alg2.l3.m2.1.1.3" xref="alg2.l3.m2.1.1.3.cmml">r</mi><mo id="alg2.l3.m2.1.1.2" xref="alg2.l3.m2.1.1.2.cmml">⁢</mo><mrow id="alg2.l3.m2.1.1.1.1" xref="alg2.l3.m2.1.1.1.1.1.cmml"><mo id="alg2.l3.m2.1.1.1.1.2" stretchy="false" xref="alg2.l3.m2.1.1.1.1.1.cmml">(</mo><msub id="alg2.l3.m2.1.1.1.1.1" xref="alg2.l3.m2.1.1.1.1.1.cmml"><mi id="alg2.l3.m2.1.1.1.1.1.2" xref="alg2.l3.m2.1.1.1.1.1.2.cmml">m</mi><mn id="alg2.l3.m2.1.1.1.1.1.3" xref="alg2.l3.m2.1.1.1.1.1.3.cmml">1</mn></msub><mo id="alg2.l3.m2.1.1.1.1.3" stretchy="false" xref="alg2.l3.m2.1.1.1.1.1.cmml">)</mo></mrow></mrow><annotation-xml encoding="MathML-Content" id="alg2.l3.m2.1b"><apply id="alg2.l3.m2.1.1.cmml" xref="alg2.l3.m2.1.1"><times id="alg2.l3.m2.1.1.2.cmml" xref="alg2.l3.m2.1.1.2"></times><ci id="alg2.l3.m2.1.1.3.cmml" xref="alg2.l3.m2.1.1.3">𝑟</ci><apply id="alg2.l3.m2.1.1.1.1.1.cmml" xref="alg2.l3.m2.1.1.1.1"><csymbol cd="ambiguous" id="alg2.l3.m2.1.1.1.1.1.1.cmml" xref="alg2.l3.m2.1.1.1.1">subscript</csymbol><ci id="alg2.l3.m2.1.1.1.1.1.2.cmml" xref="alg2.l3.m2.1.1.1.1.1.2">𝑚</ci><cn id="alg2.l3.m2.1.1.1.1.1.3.cmml" type="integer" xref="alg2.l3.m2.1.1.1.1.1.3">1</cn></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="alg2.l3.m2.1c">r(m_{1})</annotation><annotation encoding="application/x-llamapun" id="alg2.l3.m2.1d">italic_r ( italic_m start_POSTSUBSCRIPT 1 end_POSTSUBSCRIPT )</annotation></semantics></math>
</div>
<div class="ltx_listingline" id="alg2.l4">
<span class="ltx_tag ltx_tag_listingline">4:</span>  <span class="ltx_text ltx_font_bold" id="alg2.l4.1">for</span> <math alttext="k=2,\ldots,K" class="ltx_Math" display="inline" id="alg2.l4.m1.3"><semantics id="alg2.l4.m1.3a"><mrow id="alg2.l4.m1.3.4" xref="alg2.l4.m1.3.4.cmml"><mi id="alg2.l4.m1.3.4.2" xref="alg2.l4.m1.3.4.2.cmml">k</mi><mo id="alg2.l4.m1.3.4.1" xref="alg2.l4.m1.3.4.1.cmml">=</mo><mrow id="alg2.l4.m1.3.4.3.2" xref="alg2.l4.m1.3.4.3.1.cmml"><mn id="alg2.l4.m1.1.1" xref="alg2.l4.m1.1.1.cmml">2</mn><mo id="alg2.l4.m1.3.4.3.2.1" xref="alg2.l4.m1.3.4.3.1.cmml">,</mo><mi id="alg2.l4.m1.2.2" mathvariant="normal" xref="alg2.l4.m1.2.2.cmml">…</mi><mo id="alg2.l4.m1.3.4.3.2.2" xref="alg2.l4.m1.3.4.3.1.cmml">,</mo><mi id="alg2.l4.m1.3.3" xref="alg2.l4.m1.3.3.cmml">K</mi></mrow></mrow><annotation-xml encoding="MathML-Content" id="alg2.l4.m1.3b"><apply id="alg2.l4.m1.3.4.cmml" xref="alg2.l4.m1.3.4"><eq id="alg2.l4.m1.3.4.1.cmml" xref="alg2.l4.m1.3.4.1"></eq><ci id="alg2.l4.m1.3.4.2.cmml" xref="alg2.l4.m1.3.4.2">𝑘</ci><list id="alg2.l4.m1.3.4.3.1.cmml" xref="alg2.l4.m1.3.4.3.2"><cn id="alg2.l4.m1.1.1.cmml" type="integer" xref="alg2.l4.m1.1.1">2</cn><ci id="alg2.l4.m1.2.2.cmml" xref="alg2.l4.m1.2.2">…</ci><ci id="alg2.l4.m1.3.3.cmml" xref="alg2.l4.m1.3.3">𝐾</ci></list></apply></annotation-xml><annotation encoding="application/x-tex" id="alg2.l4.m1.3c">k=2,\ldots,K</annotation><annotation encoding="application/x-llamapun" id="alg2.l4.m1.3d">italic_k = 2 , … , italic_K</annotation></semantics></math> <span class="ltx_text ltx_font_bold" id="alg2.l4.2">do</span>
</div>
<div class="ltx_listingline" id="alg2.l5">
<span class="ltx_tag ltx_tag_listingline">5:</span>     Sample <math alttext="m_{k}\sim\pi_{\text{RL}}(m_{k}\mid\mathcal{M}_{1:k-1},p_{\text{desc+div}})" class="ltx_Math" display="inline" id="alg2.l5.m1.1"><semantics id="alg2.l5.m1.1a"><mrow id="alg2.l5.m1.1.1" xref="alg2.l5.m1.1.1.cmml"><msub id="alg2.l5.m1.1.1.3" xref="alg2.l5.m1.1.1.3.cmml"><mi id="alg2.l5.m1.1.1.3.2" xref="alg2.l5.m1.1.1.3.2.cmml">m</mi><mi id="alg2.l5.m1.1.1.3.3" xref="alg2.l5.m1.1.1.3.3.cmml">k</mi></msub><mo id="alg2.l5.m1.1.1.2" xref="alg2.l5.m1.1.1.2.cmml">∼</mo><mrow id="alg2.l5.m1.1.1.1" xref="alg2.l5.m1.1.1.1.cmml"><msub id="alg2.l5.m1.1.1.1.3" xref="alg2.l5.m1.1.1.1.3.cmml"><mi id="alg2.l5.m1.1.1.1.3.2" xref="alg2.l5.m1.1.1.1.3.2.cmml">π</mi><mtext id="alg2.l5.m1.1.1.1.3.3" xref="alg2.l5.m1.1.1.1.3.3a.cmml">RL</mtext></msub><mo id="alg2.l5.m1.1.1.1.2" xref="alg2.l5.m1.1.1.1.2.cmml">⁢</mo><mrow id="alg2.l5.m1.1.1.1.1.1" xref="alg2.l5.m1.1.1.1.1.1.1.cmml"><mo id="alg2.l5.m1.1.1.1.1.1.2" stretchy="false" xref="alg2.l5.m1.1.1.1.1.1.1.cmml">(</mo><mrow id="alg2.l5.m1.1.1.1.1.1.1" xref="alg2.l5.m1.1.1.1.1.1.1.cmml"><msub id="alg2.l5.m1.1.1.1.1.1.1.4" xref="alg2.l5.m1.1.1.1.1.1.1.4.cmml"><mi id="alg2.l5.m1.1.1.1.1.1.1.4.2" xref="alg2.l5.m1.1.1.1.1.1.1.4.2.cmml">m</mi><mi id="alg2.l5.m1.1.1.1.1.1.1.4.3" xref="alg2.l5.m1.1.1.1.1.1.1.4.3.cmml">k</mi></msub><mo id="alg2.l5.m1.1.1.1.1.1.1.3" xref="alg2.l5.m1.1.1.1.1.1.1.3.cmml">∣</mo><mrow id="alg2.l5.m1.1.1.1.1.1.1.2.2" xref="alg2.l5.m1.1.1.1.1.1.1.2.3.cmml"><msub id="alg2.l5.m1.1.1.1.1.1.1.1.1.1" xref="alg2.l5.m1.1.1.1.1.1.1.1.1.1.cmml"><mi class="ltx_font_mathcaligraphic" id="alg2.l5.m1.1.1.1.1.1.1.1.1.1.2" xref="alg2.l5.m1.1.1.1.1.1.1.1.1.1.2.cmml">ℳ</mi><mrow id="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3" xref="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3.cmml"><mn id="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3.2" xref="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3.2.cmml">1</mn><mo id="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3.1" lspace="0.278em" rspace="0.278em" xref="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3.1.cmml">:</mo><mrow id="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3.3" xref="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3.3.cmml"><mi id="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3.3.2" xref="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3.3.2.cmml">k</mi><mo id="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3.3.1" xref="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3.3.1.cmml">−</mo><mn id="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3.3.3" xref="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3.3.3.cmml">1</mn></mrow></mrow></msub><mo id="alg2.l5.m1.1.1.1.1.1.1.2.2.3" xref="alg2.l5.m1.1.1.1.1.1.1.2.3.cmml">,</mo><msub id="alg2.l5.m1.1.1.1.1.1.1.2.2.2" xref="alg2.l5.m1.1.1.1.1.1.1.2.2.2.cmml"><mi id="alg2.l5.m1.1.1.1.1.1.1.2.2.2.2" xref="alg2.l5.m1.1.1.1.1.1.1.2.2.2.2.cmml">p</mi><mtext id="alg2.l5.m1.1.1.1.1.1.1.2.2.2.3" xref="alg2.l5.m1.1.1.1.1.1.1.2.2.2.3a.cmml">desc+div</mtext></msub></mrow></mrow><mo id="alg2.l5.m1.1.1.1.1.1.3" stretchy="false" xref="alg2.l5.m1.1.1.1.1.1.1.cmml">)</mo></mrow></mrow></mrow><annotation-xml encoding="MathML-Content" id="alg2.l5.m1.1b"><apply id="alg2.l5.m1.1.1.cmml" xref="alg2.l5.m1.1.1"><csymbol cd="latexml" id="alg2.l5.m1.1.1.2.cmml" xref="alg2.l5.m1.1.1.2">similar-to</csymbol><apply id="alg2.l5.m1.1.1.3.cmml" xref="alg2.l5.m1.1.1.3"><csymbol cd="ambiguous" id="alg2.l5.m1.1.1.3.1.cmml" xref="alg2.l5.m1.1.1.3">subscript</csymbol><ci id="alg2.l5.m1.1.1.3.2.cmml" xref="alg2.l5.m1.1.1.3.2">𝑚</ci><ci id="alg2.l5.m1.1.1.3.3.cmml" xref="alg2.l5.m1.1.1.3.3">𝑘</ci></apply><apply id="alg2.l5.m1.1.1.1.cmml" xref="alg2.l5.m1.1.1.1"><times id="alg2.l5.m1.1.1.1.2.cmml" xref="alg2.l5.m1.1.1.1.2"></times><apply id="alg2.l5.m1.1.1.1.3.cmml" xref="alg2.l5.m1.1.1.1.3"><csymbol cd="ambiguous" id="alg2.l5.m1.1.1.1.3.1.cmml" xref="alg2.l5.m1.1.1.1.3">subscript</csymbol><ci id="alg2.l5.m1.1.1.1.3.2.cmml" xref="alg2.l5.m1.1.1.1.3.2">𝜋</ci><ci id="alg2.l5.m1.1.1.1.3.3a.cmml" xref="alg2.l5.m1.1.1.1.3.3"><mtext id="alg2.l5.m1.1.1.1.3.3.cmml" mathsize="70%" xref="alg2.l5.m1.1.1.1.3.3">RL</mtext></ci></apply><apply id="alg2.l5.m1.1.1.1.1.1.1.cmml" xref="alg2.l5.m1.1.1.1.1.1"><csymbol cd="latexml" id="alg2.l5.m1.1.1.1.1.1.1.3.cmml" xref="alg2.l5.m1.1.1.1.1.1.1.3">conditional</csymbol><apply id="alg2.l5.m1.1.1.1.1.1.1.4.cmml" xref="alg2.l5.m1.1.1.1.1.1.1.4"><csymbol cd="ambiguous" id="alg2.l5.m1.1.1.1.1.1.1.4.1.cmml" xref="alg2.l5.m1.1.1.1.1.1.1.4">subscript</csymbol><ci id="alg2.l5.m1.1.1.1.1.1.1.4.2.cmml" xref="alg2.l5.m1.1.1.1.1.1.1.4.2">𝑚</ci><ci id="alg2.l5.m1.1.1.1.1.1.1.4.3.cmml" xref="alg2.l5.m1.1.1.1.1.1.1.4.3">𝑘</ci></apply><list id="alg2.l5.m1.1.1.1.1.1.1.2.3.cmml" xref="alg2.l5.m1.1.1.1.1.1.1.2.2"><apply id="alg2.l5.m1.1.1.1.1.1.1.1.1.1.cmml" xref="alg2.l5.m1.1.1.1.1.1.1.1.1.1"><csymbol cd="ambiguous" id="alg2.l5.m1.1.1.1.1.1.1.1.1.1.1.cmml" xref="alg2.l5.m1.1.1.1.1.1.1.1.1.1">subscript</csymbol><ci id="alg2.l5.m1.1.1.1.1.1.1.1.1.1.2.cmml" xref="alg2.l5.m1.1.1.1.1.1.1.1.1.1.2">ℳ</ci><apply id="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3.cmml" xref="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3"><ci id="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3.1.cmml" xref="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3.1">:</ci><cn id="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3.2.cmml" type="integer" xref="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3.2">1</cn><apply id="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3.3.cmml" xref="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3.3"><minus id="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3.3.1.cmml" xref="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3.3.1"></minus><ci id="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3.3.2.cmml" xref="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3.3.2">𝑘</ci><cn id="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3.3.3.cmml" type="integer" xref="alg2.l5.m1.1.1.1.1.1.1.1.1.1.3.3.3">1</cn></apply></apply></apply><apply id="alg2.l5.m1.1.1.1.1.1.1.2.2.2.cmml" xref="alg2.l5.m1.1.1.1.1.1.1.2.2.2"><csymbol cd="ambiguous" id="alg2.l5.m1.1.1.1.1.1.1.2.2.2.1.cmml" xref="alg2.l5.m1.1.1.1.1.1.1.2.2.2">subscript</csymbol><ci id="alg2.l5.m1.1.1.1.1.1.1.2.2.2.2.cmml" xref="alg2.l5.m1.1.1.1.1.1.1.2.2.2.2">𝑝</ci><ci id="alg2.l5.m1.1.1.1.1.1.1.2.2.2.3a.cmml" xref="alg2.l5.m1.1.1.1.1.1.1.2.2.2.3"><mtext id="alg2.l5.m1.1.1.1.1.1.1.2.2.2.3.cmml" mathsize="70%" xref="alg2.l5.m1.1.1.1.1.1.1.2.2.2.3">desc+div</mtext></ci></apply></list></apply></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="alg2.l5.m1.1c">m_{k}\sim\pi_{\text{RL}}(m_{k}\mid\mathcal{M}_{1:k-1},p_{\text{desc+div}})</annotation><annotation encoding="application/x-llamapun" id="alg2.l5.m1.1d">italic_m start_POSTSUBSCRIPT italic_k end_POSTSUBSCRIPT ∼ italic_π start_POSTSUBSCRIPT RL end_POSTSUBSCRIPT ( italic_m start_POSTSUBSCRIPT italic_k end_POSTSUBSCRIPT ∣ caligraphic_M start_POSTSUBSCRIPT 1 : italic_k - 1 end_POSTSUBSCRIPT , italic_p start_POSTSUBSCRIPT desc+div end_POSTSUBSCRIPT )</annotation></semantics></math>
</div>
<div class="ltx_listingline" id="alg2.l6">
<span class="ltx_tag ltx_tag_listingline">6:</span>     Update <math alttext="\pi_{\text{RL}}" class="ltx_Math" display="inline" id="alg2.l6.m1.1"><semantics id="alg2.l6.m1.1a"><msub id="alg2.l6.m1.1.1" xref="alg2.l6.m1.1.1.cmml"><mi id="alg2.l6.m1.1.1.2" xref="alg2.l6.m1.1.1.2.cmml">π</mi><mtext id="alg2.l6.m1.1.1.3" xref="alg2.l6.m1.1.1.3a.cmml">RL</mtext></msub><annotation-xml encoding="MathML-Content" id="alg2.l6.m1.1b"><apply id="alg2.l6.m1.1.1.cmml" xref="alg2.l6.m1.1.1"><csymbol cd="ambiguous" id="alg2.l6.m1.1.1.1.cmml" xref="alg2.l6.m1.1.1">subscript</csymbol><ci id="alg2.l6.m1.1.1.2.cmml" xref="alg2.l6.m1.1.1.2">𝜋</ci><ci id="alg2.l6.m1.1.1.3a.cmml" xref="alg2.l6.m1.1.1.3"><mtext id="alg2.l6.m1.1.1.3.cmml" mathsize="70%" xref="alg2.l6.m1.1.1.3">RL</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="alg2.l6.m1.1c">\pi_{\text{RL}}</annotation><annotation encoding="application/x-llamapun" id="alg2.l6.m1.1d">italic_π start_POSTSUBSCRIPT RL end_POSTSUBSCRIPT</annotation></semantics></math> with PPO to maximize <math alttext="r(m_{k})" class="ltx_Math" display="inline" id="alg2.l6.m2.1"><semantics id="alg2.l6.m2.1a"><mrow id="alg2.l6.m2.1.1" xref="alg2.l6.m2.1.1.cmml"><mi id="alg2.l6.m2.1.1.3" xref="alg2.l6.m2.1.1.3.cmml">r</mi><mo id="alg2.l6.m2.1.1.2" xref="alg2.l6.m2.1.1.2.cmml">⁢</mo><mrow id="alg2.l6.m2.1.1.1.1" xref="alg2.l6.m2.1.1.1.1.1.cmml"><mo id="alg2.l6.m2.1.1.1.1.2" stretchy="false" xref="alg2.l6.m2.1.1.1.1.1.cmml">(</mo><msub id="alg2.l6.m2.1.1.1.1.1" xref="alg2.l6.m2.1.1.1.1.1.cmml"><mi id="alg2.l6.m2.1.1.1.1.1.2" xref="alg2.l6.m2.1.1.1.1.1.2.cmml">m</mi><mi id="alg2.l6.m2.1.1.1.1.1.3" xref="alg2.l6.m2.1.1.1.1.1.3.cmml">k</mi></msub><mo id="alg2.l6.m2.1.1.1.1.3" stretchy="false" xref="alg2.l6.m2.1.1.1.1.1.cmml">)</mo></mrow></mrow><annotation-xml encoding="MathML-Content" id="alg2.l6.m2.1b"><apply id="alg2.l6.m2.1.1.cmml" xref="alg2.l6.m2.1.1"><times id="alg2.l6.m2.1.1.2.cmml" xref="alg2.l6.m2.1.1.2"></times><ci id="alg2.l6.m2.1.1.3.cmml" xref="alg2.l6.m2.1.1.3">𝑟</ci><apply id="alg2.l6.m2.1.1.1.1.1.cmml" xref="alg2.l6.m2.1.1.1.1"><csymbol cd="ambiguous" id="alg2.l6.m2.1.1.1.1.1.1.cmml" xref="alg2.l6.m2.1.1.1.1">subscript</csymbol><ci id="alg2.l6.m2.1.1.1.1.1.2.cmml" xref="alg2.l6.m2.1.1.1.1.1.2">𝑚</ci><ci id="alg2.l6.m2.1.1.1.1.1.3.cmml" xref="alg2.l6.m2.1.1.1.1.1.3">𝑘</ci></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="alg2.l6.m2.1c">r(m_{k})</annotation><annotation encoding="application/x-llamapun" id="alg2.l6.m2.1d">italic_r ( italic_m start_POSTSUBSCRIPT italic_k end_POSTSUBSCRIPT )</annotation></semantics></math>
</div>
<div class="ltx_listingline" id="alg2.l7">
<span class="ltx_tag ltx_tag_listingline">7:</span>  <span class="ltx_text ltx_font_bold" id="alg2.l7.1">end</span> <span class="ltx_text ltx_font_bold" id="alg2.l7.2">for</span>
</div>
</div>
</figure>
</figure>
<div class="ltx_para ltx_noindent" id="S3.SS2.p1">
<p class="ltx_p" id="S3.SS2.p1.2">We apply reinforcement learning to maximize the diversity of the generated molecules within a sequence. However, when applied to a sequence of molecules <math alttext="\mathcal{M}_{1:K}" class="ltx_Math" display="inline" id="S3.SS2.p1.1.m1.1"><semantics id="S3.SS2.p1.1.m1.1a"><msub id="S3.SS2.p1.1.m1.1.1" xref="S3.SS2.p1.1.m1.1.1.cmml"><mi class="ltx_font_mathcaligraphic" id="S3.SS2.p1.1.m1.1.1.2" xref="S3.SS2.p1.1.m1.1.1.2.cmml">ℳ</mi><mrow id="S3.SS2.p1.1.m1.1.1.3" xref="S3.SS2.p1.1.m1.1.1.3.cmml"><mn id="S3.SS2.p1.1.m1.1.1.3.2" xref="S3.SS2.p1.1.m1.1.1.3.2.cmml">1</mn><mo id="S3.SS2.p1.1.m1.1.1.3.1" lspace="0.278em" rspace="0.278em" xref="S3.SS2.p1.1.m1.1.1.3.1.cmml">:</mo><mi id="S3.SS2.p1.1.m1.1.1.3.3" xref="S3.SS2.p1.1.m1.1.1.3.3.cmml">K</mi></mrow></msub><annotation-xml encoding="MathML-Content" id="S3.SS2.p1.1.m1.1b"><apply id="S3.SS2.p1.1.m1.1.1.cmml" xref="S3.SS2.p1.1.m1.1.1"><csymbol cd="ambiguous" id="S3.SS2.p1.1.m1.1.1.1.cmml" xref="S3.SS2.p1.1.m1.1.1">subscript</csymbol><ci id="S3.SS2.p1.1.m1.1.1.2.cmml" xref="S3.SS2.p1.1.m1.1.1.2">ℳ</ci><apply id="S3.SS2.p1.1.m1.1.1.3.cmml" xref="S3.SS2.p1.1.m1.1.1.3"><ci id="S3.SS2.p1.1.m1.1.1.3.1.cmml" xref="S3.SS2.p1.1.m1.1.1.3.1">:</ci><cn id="S3.SS2.p1.1.m1.1.1.3.2.cmml" type="integer" xref="S3.SS2.p1.1.m1.1.1.3.2">1</cn><ci id="S3.SS2.p1.1.m1.1.1.3.3.cmml" xref="S3.SS2.p1.1.m1.1.1.3.3">𝐾</ci></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS2.p1.1.m1.1c">\mathcal{M}_{1:K}</annotation><annotation encoding="application/x-llamapun" id="S3.SS2.p1.1.m1.1d">caligraphic_M start_POSTSUBSCRIPT 1 : italic_K end_POSTSUBSCRIPT</annotation></semantics></math>, conventional sequence-wise reinforcement learning <cite class="ltx_cite ltx_citemacro_citep">(Ouyang et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib23" title="">2022</a>)</cite> suffers from the credit assignment problem <cite class="ltx_cite ltx_citemacro_citep">(Zhou et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib44" title="">2024</a>)</cite>: the challenge in identifying and promoting the generation of molecules responsible for increasing diversity, among a set of molecules <math alttext="\{m_{i}\}^{K}_{i=1}" class="ltx_Math" display="inline" id="S3.SS2.p1.2.m2.1"><semantics id="S3.SS2.p1.2.m2.1a"><msubsup id="S3.SS2.p1.2.m2.1.1" xref="S3.SS2.p1.2.m2.1.1.cmml"><mrow id="S3.SS2.p1.2.m2.1.1.1.1.1" xref="S3.SS2.p1.2.m2.1.1.1.1.2.cmml"><mo id="S3.SS2.p1.2.m2.1.1.1.1.1.2" stretchy="false" xref="S3.SS2.p1.2.m2.1.1.1.1.2.cmml">{</mo><msub id="S3.SS2.p1.2.m2.1.1.1.1.1.1" xref="S3.SS2.p1.2.m2.1.1.1.1.1.1.cmml"><mi id="S3.SS2.p1.2.m2.1.1.1.1.1.1.2" xref="S3.SS2.p1.2.m2.1.1.1.1.1.1.2.cmml">m</mi><mi id="S3.SS2.p1.2.m2.1.1.1.1.1.1.3" xref="S3.SS2.p1.2.m2.1.1.1.1.1.1.3.cmml">i</mi></msub><mo id="S3.SS2.p1.2.m2.1.1.1.1.1.3" stretchy="false" xref="S3.SS2.p1.2.m2.1.1.1.1.2.cmml">}</mo></mrow><mrow id="S3.SS2.p1.2.m2.1.1.3" xref="S3.SS2.p1.2.m2.1.1.3.cmml"><mi id="S3.SS2.p1.2.m2.1.1.3.2" xref="S3.SS2.p1.2.m2.1.1.3.2.cmml">i</mi><mo id="S3.SS2.p1.2.m2.1.1.3.1" xref="S3.SS2.p1.2.m2.1.1.3.1.cmml">=</mo><mn id="S3.SS2.p1.2.m2.1.1.3.3" xref="S3.SS2.p1.2.m2.1.1.3.3.cmml">1</mn></mrow><mi id="S3.SS2.p1.2.m2.1.1.1.3" xref="S3.SS2.p1.2.m2.1.1.1.3.cmml">K</mi></msubsup><annotation-xml encoding="MathML-Content" id="S3.SS2.p1.2.m2.1b"><apply id="S3.SS2.p1.2.m2.1.1.cmml" xref="S3.SS2.p1.2.m2.1.1"><csymbol cd="ambiguous" id="S3.SS2.p1.2.m2.1.1.2.cmml" xref="S3.SS2.p1.2.m2.1.1">subscript</csymbol><apply id="S3.SS2.p1.2.m2.1.1.1.cmml" xref="S3.SS2.p1.2.m2.1.1"><csymbol cd="ambiguous" id="S3.SS2.p1.2.m2.1.1.1.2.cmml" xref="S3.SS2.p1.2.m2.1.1">superscript</csymbol><set id="S3.SS2.p1.2.m2.1.1.1.1.2.cmml" xref="S3.SS2.p1.2.m2.1.1.1.1.1"><apply id="S3.SS2.p1.2.m2.1.1.1.1.1.1.cmml" xref="S3.SS2.p1.2.m2.1.1.1.1.1.1"><csymbol cd="ambiguous" id="S3.SS2.p1.2.m2.1.1.1.1.1.1.1.cmml" xref="S3.SS2.p1.2.m2.1.1.1.1.1.1">subscript</csymbol><ci id="S3.SS2.p1.2.m2.1.1.1.1.1.1.2.cmml" xref="S3.SS2.p1.2.m2.1.1.1.1.1.1.2">𝑚</ci><ci id="S3.SS2.p1.2.m2.1.1.1.1.1.1.3.cmml" xref="S3.SS2.p1.2.m2.1.1.1.1.1.1.3">𝑖</ci></apply></set><ci id="S3.SS2.p1.2.m2.1.1.1.3.cmml" xref="S3.SS2.p1.2.m2.1.1.1.3">𝐾</ci></apply><apply id="S3.SS2.p1.2.m2.1.1.3.cmml" xref="S3.SS2.p1.2.m2.1.1.3"><eq id="S3.SS2.p1.2.m2.1.1.3.1.cmml" xref="S3.SS2.p1.2.m2.1.1.3.1"></eq><ci id="S3.SS2.p1.2.m2.1.1.3.2.cmml" xref="S3.SS2.p1.2.m2.1.1.3.2">𝑖</ci><cn id="S3.SS2.p1.2.m2.1.1.3.3.cmml" type="integer" xref="S3.SS2.p1.2.m2.1.1.3.3">1</cn></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS2.p1.2.m2.1c">\{m_{i}\}^{K}_{i=1}</annotation><annotation encoding="application/x-llamapun" id="S3.SS2.p1.2.m2.1d">{ italic_m start_POSTSUBSCRIPT italic_i end_POSTSUBSCRIPT } start_POSTSUPERSCRIPT italic_K end_POSTSUPERSCRIPT start_POSTSUBSCRIPT italic_i = 1 end_POSTSUBSCRIPT</annotation></semantics></math>. To circumvent this, we introduce a molecule-wise reinforcement learning.<span class="ltx_note ltx_role_footnote" id="footnote2"><sup class="ltx_note_mark">2</sup><span class="ltx_note_outer"><span class="ltx_note_content"><sup class="ltx_note_mark">2</sup><span class="ltx_tag ltx_tag_note">2</span>We compare both approaches in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.T4" title="In 4.4 Ablation studies ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Table</span> <span class="ltx_text ltx_ref_tag">4</span></a> of <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.SS4" title="4.4 Ablation studies ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Section</span> <span class="ltx_text ltx_ref_tag">4.4</span></a>.</span></span></span></p>
</div>
<div class="ltx_para ltx_noindent" id="S3.SS2.p2">
<p class="ltx_p" id="S3.SS2.p2.10">Specifically, we consider reinforcement learning on a sequence of molecules <math alttext="\mathcal{M}_{1:K}" class="ltx_Math" display="inline" id="S3.SS2.p2.1.m1.1"><semantics id="S3.SS2.p2.1.m1.1a"><msub id="S3.SS2.p2.1.m1.1.1" xref="S3.SS2.p2.1.m1.1.1.cmml"><mi class="ltx_font_mathcaligraphic" id="S3.SS2.p2.1.m1.1.1.2" xref="S3.SS2.p2.1.m1.1.1.2.cmml">ℳ</mi><mrow id="S3.SS2.p2.1.m1.1.1.3" xref="S3.SS2.p2.1.m1.1.1.3.cmml"><mn id="S3.SS2.p2.1.m1.1.1.3.2" xref="S3.SS2.p2.1.m1.1.1.3.2.cmml">1</mn><mo id="S3.SS2.p2.1.m1.1.1.3.1" lspace="0.278em" rspace="0.278em" xref="S3.SS2.p2.1.m1.1.1.3.1.cmml">:</mo><mi id="S3.SS2.p2.1.m1.1.1.3.3" xref="S3.SS2.p2.1.m1.1.1.3.3.cmml">K</mi></mrow></msub><annotation-xml encoding="MathML-Content" id="S3.SS2.p2.1.m1.1b"><apply id="S3.SS2.p2.1.m1.1.1.cmml" xref="S3.SS2.p2.1.m1.1.1"><csymbol cd="ambiguous" id="S3.SS2.p2.1.m1.1.1.1.cmml" xref="S3.SS2.p2.1.m1.1.1">subscript</csymbol><ci id="S3.SS2.p2.1.m1.1.1.2.cmml" xref="S3.SS2.p2.1.m1.1.1.2">ℳ</ci><apply id="S3.SS2.p2.1.m1.1.1.3.cmml" xref="S3.SS2.p2.1.m1.1.1.3"><ci id="S3.SS2.p2.1.m1.1.1.3.1.cmml" xref="S3.SS2.p2.1.m1.1.1.3.1">:</ci><cn id="S3.SS2.p2.1.m1.1.1.3.2.cmml" type="integer" xref="S3.SS2.p2.1.m1.1.1.3.2">1</cn><ci id="S3.SS2.p2.1.m1.1.1.3.3.cmml" xref="S3.SS2.p2.1.m1.1.1.3.3">𝐾</ci></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS2.p2.1.m1.1c">\mathcal{M}_{1:K}</annotation><annotation encoding="application/x-llamapun" id="S3.SS2.p2.1.m1.1d">caligraphic_M start_POSTSUBSCRIPT 1 : italic_K end_POSTSUBSCRIPT</annotation></semantics></math> as learning in <math alttext="K" class="ltx_Math" display="inline" id="S3.SS2.p2.2.m2.1"><semantics id="S3.SS2.p2.2.m2.1a"><mi id="S3.SS2.p2.2.m2.1.1" xref="S3.SS2.p2.2.m2.1.1.cmml">K</mi><annotation-xml encoding="MathML-Content" id="S3.SS2.p2.2.m2.1b"><ci id="S3.SS2.p2.2.m2.1.1.cmml" xref="S3.SS2.p2.2.m2.1.1">𝐾</ci></annotation-xml><annotation encoding="application/x-tex" id="S3.SS2.p2.2.m2.1c">K</annotation><annotation encoding="application/x-llamapun" id="S3.SS2.p2.2.m2.1d">italic_K</annotation></semantics></math> individual stages. Each stage corresponds to generating a molecule <math alttext="m_{k}" class="ltx_Math" display="inline" id="S3.SS2.p2.3.m3.1"><semantics id="S3.SS2.p2.3.m3.1a"><msub id="S3.SS2.p2.3.m3.1.1" xref="S3.SS2.p2.3.m3.1.1.cmml"><mi id="S3.SS2.p2.3.m3.1.1.2" xref="S3.SS2.p2.3.m3.1.1.2.cmml">m</mi><mi id="S3.SS2.p2.3.m3.1.1.3" xref="S3.SS2.p2.3.m3.1.1.3.cmml">k</mi></msub><annotation-xml encoding="MathML-Content" id="S3.SS2.p2.3.m3.1b"><apply id="S3.SS2.p2.3.m3.1.1.cmml" xref="S3.SS2.p2.3.m3.1.1"><csymbol cd="ambiguous" id="S3.SS2.p2.3.m3.1.1.1.cmml" xref="S3.SS2.p2.3.m3.1.1">subscript</csymbol><ci id="S3.SS2.p2.3.m3.1.1.2.cmml" xref="S3.SS2.p2.3.m3.1.1.2">𝑚</ci><ci id="S3.SS2.p2.3.m3.1.1.3.cmml" xref="S3.SS2.p2.3.m3.1.1.3">𝑘</ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS2.p2.3.m3.1c">m_{k}</annotation><annotation encoding="application/x-llamapun" id="S3.SS2.p2.3.m3.1d">italic_m start_POSTSUBSCRIPT italic_k end_POSTSUBSCRIPT</annotation></semantics></math> conditioned on a sequence of previously generated molecules <math alttext="\mathcal{M}_{1:k-1}" class="ltx_Math" display="inline" id="S3.SS2.p2.4.m4.1"><semantics id="S3.SS2.p2.4.m4.1a"><msub id="S3.SS2.p2.4.m4.1.1" xref="S3.SS2.p2.4.m4.1.1.cmml"><mi class="ltx_font_mathcaligraphic" id="S3.SS2.p2.4.m4.1.1.2" xref="S3.SS2.p2.4.m4.1.1.2.cmml">ℳ</mi><mrow id="S3.SS2.p2.4.m4.1.1.3" xref="S3.SS2.p2.4.m4.1.1.3.cmml"><mn id="S3.SS2.p2.4.m4.1.1.3.2" xref="S3.SS2.p2.4.m4.1.1.3.2.cmml">1</mn><mo id="S3.SS2.p2.4.m4.1.1.3.1" lspace="0.278em" rspace="0.278em" xref="S3.SS2.p2.4.m4.1.1.3.1.cmml">:</mo><mrow id="S3.SS2.p2.4.m4.1.1.3.3" xref="S3.SS2.p2.4.m4.1.1.3.3.cmml"><mi id="S3.SS2.p2.4.m4.1.1.3.3.2" xref="S3.SS2.p2.4.m4.1.1.3.3.2.cmml">k</mi><mo id="S3.SS2.p2.4.m4.1.1.3.3.1" xref="S3.SS2.p2.4.m4.1.1.3.3.1.cmml">−</mo><mn id="S3.SS2.p2.4.m4.1.1.3.3.3" xref="S3.SS2.p2.4.m4.1.1.3.3.3.cmml">1</mn></mrow></mrow></msub><annotation-xml encoding="MathML-Content" id="S3.SS2.p2.4.m4.1b"><apply id="S3.SS2.p2.4.m4.1.1.cmml" xref="S3.SS2.p2.4.m4.1.1"><csymbol cd="ambiguous" id="S3.SS2.p2.4.m4.1.1.1.cmml" xref="S3.SS2.p2.4.m4.1.1">subscript</csymbol><ci id="S3.SS2.p2.4.m4.1.1.2.cmml" xref="S3.SS2.p2.4.m4.1.1.2">ℳ</ci><apply id="S3.SS2.p2.4.m4.1.1.3.cmml" xref="S3.SS2.p2.4.m4.1.1.3"><ci id="S3.SS2.p2.4.m4.1.1.3.1.cmml" xref="S3.SS2.p2.4.m4.1.1.3.1">:</ci><cn id="S3.SS2.p2.4.m4.1.1.3.2.cmml" type="integer" xref="S3.SS2.p2.4.m4.1.1.3.2">1</cn><apply id="S3.SS2.p2.4.m4.1.1.3.3.cmml" xref="S3.SS2.p2.4.m4.1.1.3.3"><minus id="S3.SS2.p2.4.m4.1.1.3.3.1.cmml" xref="S3.SS2.p2.4.m4.1.1.3.3.1"></minus><ci id="S3.SS2.p2.4.m4.1.1.3.3.2.cmml" xref="S3.SS2.p2.4.m4.1.1.3.3.2">𝑘</ci><cn id="S3.SS2.p2.4.m4.1.1.3.3.3.cmml" type="integer" xref="S3.SS2.p2.4.m4.1.1.3.3.3">1</cn></apply></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS2.p2.4.m4.1c">\mathcal{M}_{1:k-1}</annotation><annotation encoding="application/x-llamapun" id="S3.SS2.p2.4.m4.1d">caligraphic_M start_POSTSUBSCRIPT 1 : italic_k - 1 end_POSTSUBSCRIPT</annotation></semantics></math>. Then, the LLM <math alttext="\pi_{\text{RL}}" class="ltx_Math" display="inline" id="S3.SS2.p2.5.m5.1"><semantics id="S3.SS2.p2.5.m5.1a"><msub id="S3.SS2.p2.5.m5.1.1" xref="S3.SS2.p2.5.m5.1.1.cmml"><mi id="S3.SS2.p2.5.m5.1.1.2" xref="S3.SS2.p2.5.m5.1.1.2.cmml">π</mi><mtext id="S3.SS2.p2.5.m5.1.1.3" xref="S3.SS2.p2.5.m5.1.1.3a.cmml">RL</mtext></msub><annotation-xml encoding="MathML-Content" id="S3.SS2.p2.5.m5.1b"><apply id="S3.SS2.p2.5.m5.1.1.cmml" xref="S3.SS2.p2.5.m5.1.1"><csymbol cd="ambiguous" id="S3.SS2.p2.5.m5.1.1.1.cmml" xref="S3.SS2.p2.5.m5.1.1">subscript</csymbol><ci id="S3.SS2.p2.5.m5.1.1.2.cmml" xref="S3.SS2.p2.5.m5.1.1.2">𝜋</ci><ci id="S3.SS2.p2.5.m5.1.1.3a.cmml" xref="S3.SS2.p2.5.m5.1.1.3"><mtext id="S3.SS2.p2.5.m5.1.1.3.cmml" mathsize="70%" xref="S3.SS2.p2.5.m5.1.1.3">RL</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS2.p2.5.m5.1c">\pi_{\text{RL}}</annotation><annotation encoding="application/x-llamapun" id="S3.SS2.p2.5.m5.1d">italic_π start_POSTSUBSCRIPT RL end_POSTSUBSCRIPT</annotation></semantics></math> is trained to maximize the return of each stage, which is defined by the reward of the generated molecule <math alttext="m_{k}" class="ltx_Math" display="inline" id="S3.SS2.p2.6.m6.1"><semantics id="S3.SS2.p2.6.m6.1a"><msub id="S3.SS2.p2.6.m6.1.1" xref="S3.SS2.p2.6.m6.1.1.cmml"><mi id="S3.SS2.p2.6.m6.1.1.2" xref="S3.SS2.p2.6.m6.1.1.2.cmml">m</mi><mi id="S3.SS2.p2.6.m6.1.1.3" xref="S3.SS2.p2.6.m6.1.1.3.cmml">k</mi></msub><annotation-xml encoding="MathML-Content" id="S3.SS2.p2.6.m6.1b"><apply id="S3.SS2.p2.6.m6.1.1.cmml" xref="S3.SS2.p2.6.m6.1.1"><csymbol cd="ambiguous" id="S3.SS2.p2.6.m6.1.1.1.cmml" xref="S3.SS2.p2.6.m6.1.1">subscript</csymbol><ci id="S3.SS2.p2.6.m6.1.1.2.cmml" xref="S3.SS2.p2.6.m6.1.1.2">𝑚</ci><ci id="S3.SS2.p2.6.m6.1.1.3.cmml" xref="S3.SS2.p2.6.m6.1.1.3">𝑘</ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS2.p2.6.m6.1c">m_{k}</annotation><annotation encoding="application/x-llamapun" id="S3.SS2.p2.6.m6.1d">italic_m start_POSTSUBSCRIPT italic_k end_POSTSUBSCRIPT</annotation></semantics></math>.
The reward is defined as the diversity between the previously generated molecules <math alttext="\{m_{i}\}_{i=1}^{k-1}" class="ltx_Math" display="inline" id="S3.SS2.p2.7.m7.1"><semantics id="S3.SS2.p2.7.m7.1a"><msubsup id="S3.SS2.p2.7.m7.1.1" xref="S3.SS2.p2.7.m7.1.1.cmml"><mrow id="S3.SS2.p2.7.m7.1.1.1.1.1" xref="S3.SS2.p2.7.m7.1.1.1.1.2.cmml"><mo id="S3.SS2.p2.7.m7.1.1.1.1.1.2" stretchy="false" xref="S3.SS2.p2.7.m7.1.1.1.1.2.cmml">{</mo><msub id="S3.SS2.p2.7.m7.1.1.1.1.1.1" xref="S3.SS2.p2.7.m7.1.1.1.1.1.1.cmml"><mi id="S3.SS2.p2.7.m7.1.1.1.1.1.1.2" xref="S3.SS2.p2.7.m7.1.1.1.1.1.1.2.cmml">m</mi><mi id="S3.SS2.p2.7.m7.1.1.1.1.1.1.3" xref="S3.SS2.p2.7.m7.1.1.1.1.1.1.3.cmml">i</mi></msub><mo id="S3.SS2.p2.7.m7.1.1.1.1.1.3" stretchy="false" xref="S3.SS2.p2.7.m7.1.1.1.1.2.cmml">}</mo></mrow><mrow id="S3.SS2.p2.7.m7.1.1.1.3" xref="S3.SS2.p2.7.m7.1.1.1.3.cmml"><mi id="S3.SS2.p2.7.m7.1.1.1.3.2" xref="S3.SS2.p2.7.m7.1.1.1.3.2.cmml">i</mi><mo id="S3.SS2.p2.7.m7.1.1.1.3.1" xref="S3.SS2.p2.7.m7.1.1.1.3.1.cmml">=</mo><mn id="S3.SS2.p2.7.m7.1.1.1.3.3" xref="S3.SS2.p2.7.m7.1.1.1.3.3.cmml">1</mn></mrow><mrow id="S3.SS2.p2.7.m7.1.1.3" xref="S3.SS2.p2.7.m7.1.1.3.cmml"><mi id="S3.SS2.p2.7.m7.1.1.3.2" xref="S3.SS2.p2.7.m7.1.1.3.2.cmml">k</mi><mo id="S3.SS2.p2.7.m7.1.1.3.1" xref="S3.SS2.p2.7.m7.1.1.3.1.cmml">−</mo><mn id="S3.SS2.p2.7.m7.1.1.3.3" xref="S3.SS2.p2.7.m7.1.1.3.3.cmml">1</mn></mrow></msubsup><annotation-xml encoding="MathML-Content" id="S3.SS2.p2.7.m7.1b"><apply id="S3.SS2.p2.7.m7.1.1.cmml" xref="S3.SS2.p2.7.m7.1.1"><csymbol cd="ambiguous" id="S3.SS2.p2.7.m7.1.1.2.cmml" xref="S3.SS2.p2.7.m7.1.1">superscript</csymbol><apply id="S3.SS2.p2.7.m7.1.1.1.cmml" xref="S3.SS2.p2.7.m7.1.1"><csymbol cd="ambiguous" id="S3.SS2.p2.7.m7.1.1.1.2.cmml" xref="S3.SS2.p2.7.m7.1.1">subscript</csymbol><set id="S3.SS2.p2.7.m7.1.1.1.1.2.cmml" xref="S3.SS2.p2.7.m7.1.1.1.1.1"><apply id="S3.SS2.p2.7.m7.1.1.1.1.1.1.cmml" xref="S3.SS2.p2.7.m7.1.1.1.1.1.1"><csymbol cd="ambiguous" id="S3.SS2.p2.7.m7.1.1.1.1.1.1.1.cmml" xref="S3.SS2.p2.7.m7.1.1.1.1.1.1">subscript</csymbol><ci id="S3.SS2.p2.7.m7.1.1.1.1.1.1.2.cmml" xref="S3.SS2.p2.7.m7.1.1.1.1.1.1.2">𝑚</ci><ci id="S3.SS2.p2.7.m7.1.1.1.1.1.1.3.cmml" xref="S3.SS2.p2.7.m7.1.1.1.1.1.1.3">𝑖</ci></apply></set><apply id="S3.SS2.p2.7.m7.1.1.1.3.cmml" xref="S3.SS2.p2.7.m7.1.1.1.3"><eq id="S3.SS2.p2.7.m7.1.1.1.3.1.cmml" xref="S3.SS2.p2.7.m7.1.1.1.3.1"></eq><ci id="S3.SS2.p2.7.m7.1.1.1.3.2.cmml" xref="S3.SS2.p2.7.m7.1.1.1.3.2">𝑖</ci><cn id="S3.SS2.p2.7.m7.1.1.1.3.3.cmml" type="integer" xref="S3.SS2.p2.7.m7.1.1.1.3.3">1</cn></apply></apply><apply id="S3.SS2.p2.7.m7.1.1.3.cmml" xref="S3.SS2.p2.7.m7.1.1.3"><minus id="S3.SS2.p2.7.m7.1.1.3.1.cmml" xref="S3.SS2.p2.7.m7.1.1.3.1"></minus><ci id="S3.SS2.p2.7.m7.1.1.3.2.cmml" xref="S3.SS2.p2.7.m7.1.1.3.2">𝑘</ci><cn id="S3.SS2.p2.7.m7.1.1.3.3.cmml" type="integer" xref="S3.SS2.p2.7.m7.1.1.3.3">1</cn></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS2.p2.7.m7.1c">\{m_{i}\}_{i=1}^{k-1}</annotation><annotation encoding="application/x-llamapun" id="S3.SS2.p2.7.m7.1d">{ italic_m start_POSTSUBSCRIPT italic_i end_POSTSUBSCRIPT } start_POSTSUBSCRIPT italic_i = 1 end_POSTSUBSCRIPT start_POSTSUPERSCRIPT italic_k - 1 end_POSTSUPERSCRIPT</annotation></semantics></math> and the new molecule <math alttext="m_{k}" class="ltx_Math" display="inline" id="S3.SS2.p2.8.m8.1"><semantics id="S3.SS2.p2.8.m8.1a"><msub id="S3.SS2.p2.8.m8.1.1" xref="S3.SS2.p2.8.m8.1.1.cmml"><mi id="S3.SS2.p2.8.m8.1.1.2" xref="S3.SS2.p2.8.m8.1.1.2.cmml">m</mi><mi id="S3.SS2.p2.8.m8.1.1.3" xref="S3.SS2.p2.8.m8.1.1.3.cmml">k</mi></msub><annotation-xml encoding="MathML-Content" id="S3.SS2.p2.8.m8.1b"><apply id="S3.SS2.p2.8.m8.1.1.cmml" xref="S3.SS2.p2.8.m8.1.1"><csymbol cd="ambiguous" id="S3.SS2.p2.8.m8.1.1.1.cmml" xref="S3.SS2.p2.8.m8.1.1">subscript</csymbol><ci id="S3.SS2.p2.8.m8.1.1.2.cmml" xref="S3.SS2.p2.8.m8.1.1.2">𝑚</ci><ci id="S3.SS2.p2.8.m8.1.1.3.cmml" xref="S3.SS2.p2.8.m8.1.1.3">𝑘</ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS2.p2.8.m8.1c">m_{k}</annotation><annotation encoding="application/x-llamapun" id="S3.SS2.p2.8.m8.1d">italic_m start_POSTSUBSCRIPT italic_k end_POSTSUBSCRIPT</annotation></semantics></math>. We also incorporate an auxiliary reward to ensure that the molecule <math alttext="m_{k}" class="ltx_Math" display="inline" id="S3.SS2.p2.9.m9.1"><semantics id="S3.SS2.p2.9.m9.1a"><msub id="S3.SS2.p2.9.m9.1.1" xref="S3.SS2.p2.9.m9.1.1.cmml"><mi id="S3.SS2.p2.9.m9.1.1.2" xref="S3.SS2.p2.9.m9.1.1.2.cmml">m</mi><mi id="S3.SS2.p2.9.m9.1.1.3" xref="S3.SS2.p2.9.m9.1.1.3.cmml">k</mi></msub><annotation-xml encoding="MathML-Content" id="S3.SS2.p2.9.m9.1b"><apply id="S3.SS2.p2.9.m9.1.1.cmml" xref="S3.SS2.p2.9.m9.1.1"><csymbol cd="ambiguous" id="S3.SS2.p2.9.m9.1.1.1.cmml" xref="S3.SS2.p2.9.m9.1.1">subscript</csymbol><ci id="S3.SS2.p2.9.m9.1.1.2.cmml" xref="S3.SS2.p2.9.m9.1.1.2">𝑚</ci><ci id="S3.SS2.p2.9.m9.1.1.3.cmml" xref="S3.SS2.p2.9.m9.1.1.3">𝑘</ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS2.p2.9.m9.1c">m_{k}</annotation><annotation encoding="application/x-llamapun" id="S3.SS2.p2.9.m9.1d">italic_m start_POSTSUBSCRIPT italic_k end_POSTSUBSCRIPT</annotation></semantics></math> satisfies the given description <math alttext="p_{\text{desc}}" class="ltx_Math" display="inline" id="S3.SS2.p2.10.m10.1"><semantics id="S3.SS2.p2.10.m10.1a"><msub id="S3.SS2.p2.10.m10.1.1" xref="S3.SS2.p2.10.m10.1.1.cmml"><mi id="S3.SS2.p2.10.m10.1.1.2" xref="S3.SS2.p2.10.m10.1.1.2.cmml">p</mi><mtext id="S3.SS2.p2.10.m10.1.1.3" xref="S3.SS2.p2.10.m10.1.1.3a.cmml">desc</mtext></msub><annotation-xml encoding="MathML-Content" id="S3.SS2.p2.10.m10.1b"><apply id="S3.SS2.p2.10.m10.1.1.cmml" xref="S3.SS2.p2.10.m10.1.1"><csymbol cd="ambiguous" id="S3.SS2.p2.10.m10.1.1.1.cmml" xref="S3.SS2.p2.10.m10.1.1">subscript</csymbol><ci id="S3.SS2.p2.10.m10.1.1.2.cmml" xref="S3.SS2.p2.10.m10.1.1.2">𝑝</ci><ci id="S3.SS2.p2.10.m10.1.1.3a.cmml" xref="S3.SS2.p2.10.m10.1.1.3"><mtext id="S3.SS2.p2.10.m10.1.1.3.cmml" mathsize="70%" xref="S3.SS2.p2.10.m10.1.1.3">desc</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS2.p2.10.m10.1c">p_{\text{desc}}</annotation><annotation encoding="application/x-llamapun" id="S3.SS2.p2.10.m10.1d">italic_p start_POSTSUBSCRIPT desc end_POSTSUBSCRIPT</annotation></semantics></math>. We illustrate our approach in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S2.F4.sf2" title="In Figure 4 ‣ 2 Related work ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Figure</span> <span class="ltx_text ltx_ref_tag">4(b)</span></a> and provide the detailed algorithm in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#alg2" title="In 3.2 Reinforcement learning ‣ 3 Method ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Algorithm</span> <span class="ltx_text ltx_ref_tag">2</span></a>.</p>
</div>
<div class="ltx_para ltx_noindent" id="S3.SS2.p3">
<p class="ltx_p" id="S3.SS2.p3.3"><span class="ltx_text ltx_font_bold" id="S3.SS2.p3.3.1">Reward.</span> The reward evaluates the generated molecule <math alttext="m_{k}" class="ltx_Math" display="inline" id="S3.SS2.p3.1.m1.1"><semantics id="S3.SS2.p3.1.m1.1a"><msub id="S3.SS2.p3.1.m1.1.1" xref="S3.SS2.p3.1.m1.1.1.cmml"><mi id="S3.SS2.p3.1.m1.1.1.2" xref="S3.SS2.p3.1.m1.1.1.2.cmml">m</mi><mi id="S3.SS2.p3.1.m1.1.1.3" xref="S3.SS2.p3.1.m1.1.1.3.cmml">k</mi></msub><annotation-xml encoding="MathML-Content" id="S3.SS2.p3.1.m1.1b"><apply id="S3.SS2.p3.1.m1.1.1.cmml" xref="S3.SS2.p3.1.m1.1.1"><csymbol cd="ambiguous" id="S3.SS2.p3.1.m1.1.1.1.cmml" xref="S3.SS2.p3.1.m1.1.1">subscript</csymbol><ci id="S3.SS2.p3.1.m1.1.1.2.cmml" xref="S3.SS2.p3.1.m1.1.1.2">𝑚</ci><ci id="S3.SS2.p3.1.m1.1.1.3.cmml" xref="S3.SS2.p3.1.m1.1.1.3">𝑘</ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS2.p3.1.m1.1c">m_{k}</annotation><annotation encoding="application/x-llamapun" id="S3.SS2.p3.1.m1.1d">italic_m start_POSTSUBSCRIPT italic_k end_POSTSUBSCRIPT</annotation></semantics></math> with a diversity reward <math alttext="r_{\text{div}}(m_{k},\{m_{i}\}^{k-1}_{i=1})" class="ltx_Math" display="inline" id="S3.SS2.p3.2.m2.2"><semantics id="S3.SS2.p3.2.m2.2a"><mrow id="S3.SS2.p3.2.m2.2.2" xref="S3.SS2.p3.2.m2.2.2.cmml"><msub id="S3.SS2.p3.2.m2.2.2.4" xref="S3.SS2.p3.2.m2.2.2.4.cmml"><mi id="S3.SS2.p3.2.m2.2.2.4.2" xref="S3.SS2.p3.2.m2.2.2.4.2.cmml">r</mi><mtext id="S3.SS2.p3.2.m2.2.2.4.3" xref="S3.SS2.p3.2.m2.2.2.4.3a.cmml">div</mtext></msub><mo id="S3.SS2.p3.2.m2.2.2.3" xref="S3.SS2.p3.2.m2.2.2.3.cmml">⁢</mo><mrow id="S3.SS2.p3.2.m2.2.2.2.2" xref="S3.SS2.p3.2.m2.2.2.2.3.cmml"><mo id="S3.SS2.p3.2.m2.2.2.2.2.3" stretchy="false" xref="S3.SS2.p3.2.m2.2.2.2.3.cmml">(</mo><msub id="S3.SS2.p3.2.m2.1.1.1.1.1" xref="S3.SS2.p3.2.m2.1.1.1.1.1.cmml"><mi id="S3.SS2.p3.2.m2.1.1.1.1.1.2" xref="S3.SS2.p3.2.m2.1.1.1.1.1.2.cmml">m</mi><mi id="S3.SS2.p3.2.m2.1.1.1.1.1.3" xref="S3.SS2.p3.2.m2.1.1.1.1.1.3.cmml">k</mi></msub><mo id="S3.SS2.p3.2.m2.2.2.2.2.4" xref="S3.SS2.p3.2.m2.2.2.2.3.cmml">,</mo><msubsup id="S3.SS2.p3.2.m2.2.2.2.2.2" xref="S3.SS2.p3.2.m2.2.2.2.2.2.cmml"><mrow id="S3.SS2.p3.2.m2.2.2.2.2.2.1.1.1" xref="S3.SS2.p3.2.m2.2.2.2.2.2.1.1.2.cmml"><mo id="S3.SS2.p3.2.m2.2.2.2.2.2.1.1.1.2" stretchy="false" xref="S3.SS2.p3.2.m2.2.2.2.2.2.1.1.2.cmml">{</mo><msub id="S3.SS2.p3.2.m2.2.2.2.2.2.1.1.1.1" xref="S3.SS2.p3.2.m2.2.2.2.2.2.1.1.1.1.cmml"><mi id="S3.SS2.p3.2.m2.2.2.2.2.2.1.1.1.1.2" xref="S3.SS2.p3.2.m2.2.2.2.2.2.1.1.1.1.2.cmml">m</mi><mi id="S3.SS2.p3.2.m2.2.2.2.2.2.1.1.1.1.3" xref="S3.SS2.p3.2.m2.2.2.2.2.2.1.1.1.1.3.cmml">i</mi></msub><mo id="S3.SS2.p3.2.m2.2.2.2.2.2.1.1.1.3" stretchy="false" xref="S3.SS2.p3.2.m2.2.2.2.2.2.1.1.2.cmml">}</mo></mrow><mrow id="S3.SS2.p3.2.m2.2.2.2.2.2.3" xref="S3.SS2.p3.2.m2.2.2.2.2.2.3.cmml"><mi id="S3.SS2.p3.2.m2.2.2.2.2.2.3.2" xref="S3.SS2.p3.2.m2.2.2.2.2.2.3.2.cmml">i</mi><mo id="S3.SS2.p3.2.m2.2.2.2.2.2.3.1" xref="S3.SS2.p3.2.m2.2.2.2.2.2.3.1.cmml">=</mo><mn id="S3.SS2.p3.2.m2.2.2.2.2.2.3.3" xref="S3.SS2.p3.2.m2.2.2.2.2.2.3.3.cmml">1</mn></mrow><mrow id="S3.SS2.p3.2.m2.2.2.2.2.2.1.3" xref="S3.SS2.p3.2.m2.2.2.2.2.2.1.3.cmml"><mi id="S3.SS2.p3.2.m2.2.2.2.2.2.1.3.2" xref="S3.SS2.p3.2.m2.2.2.2.2.2.1.3.2.cmml">k</mi><mo id="S3.SS2.p3.2.m2.2.2.2.2.2.1.3.1" xref="S3.SS2.p3.2.m2.2.2.2.2.2.1.3.1.cmml">−</mo><mn id="S3.SS2.p3.2.m2.2.2.2.2.2.1.3.3" xref="S3.SS2.p3.2.m2.2.2.2.2.2.1.3.3.cmml">1</mn></mrow></msubsup><mo id="S3.SS2.p3.2.m2.2.2.2.2.5" stretchy="false" xref="S3.SS2.p3.2.m2.2.2.2.3.cmml">)</mo></mrow></mrow><annotation-xml encoding="MathML-Content" id="S3.SS2.p3.2.m2.2b"><apply id="S3.SS2.p3.2.m2.2.2.cmml" xref="S3.SS2.p3.2.m2.2.2"><times id="S3.SS2.p3.2.m2.2.2.3.cmml" xref="S3.SS2.p3.2.m2.2.2.3"></times><apply id="S3.SS2.p3.2.m2.2.2.4.cmml" xref="S3.SS2.p3.2.m2.2.2.4"><csymbol cd="ambiguous" id="S3.SS2.p3.2.m2.2.2.4.1.cmml" xref="S3.SS2.p3.2.m2.2.2.4">subscript</csymbol><ci id="S3.SS2.p3.2.m2.2.2.4.2.cmml" xref="S3.SS2.p3.2.m2.2.2.4.2">𝑟</ci><ci id="S3.SS2.p3.2.m2.2.2.4.3a.cmml" xref="S3.SS2.p3.2.m2.2.2.4.3"><mtext id="S3.SS2.p3.2.m2.2.2.4.3.cmml" mathsize="70%" xref="S3.SS2.p3.2.m2.2.2.4.3">div</mtext></ci></apply><interval closure="open" id="S3.SS2.p3.2.m2.2.2.2.3.cmml" xref="S3.SS2.p3.2.m2.2.2.2.2"><apply id="S3.SS2.p3.2.m2.1.1.1.1.1.cmml" xref="S3.SS2.p3.2.m2.1.1.1.1.1"><csymbol cd="ambiguous" id="S3.SS2.p3.2.m2.1.1.1.1.1.1.cmml" xref="S3.SS2.p3.2.m2.1.1.1.1.1">subscript</csymbol><ci id="S3.SS2.p3.2.m2.1.1.1.1.1.2.cmml" xref="S3.SS2.p3.2.m2.1.1.1.1.1.2">𝑚</ci><ci id="S3.SS2.p3.2.m2.1.1.1.1.1.3.cmml" xref="S3.SS2.p3.2.m2.1.1.1.1.1.3">𝑘</ci></apply><apply id="S3.SS2.p3.2.m2.2.2.2.2.2.cmml" xref="S3.SS2.p3.2.m2.2.2.2.2.2"><csymbol cd="ambiguous" id="S3.SS2.p3.2.m2.2.2.2.2.2.2.cmml" xref="S3.SS2.p3.2.m2.2.2.2.2.2">subscript</csymbol><apply id="S3.SS2.p3.2.m2.2.2.2.2.2.1.cmml" xref="S3.SS2.p3.2.m2.2.2.2.2.2"><csymbol cd="ambiguous" id="S3.SS2.p3.2.m2.2.2.2.2.2.1.2.cmml" xref="S3.SS2.p3.2.m2.2.2.2.2.2">superscript</csymbol><set id="S3.SS2.p3.2.m2.2.2.2.2.2.1.1.2.cmml" xref="S3.SS2.p3.2.m2.2.2.2.2.2.1.1.1"><apply id="S3.SS2.p3.2.m2.2.2.2.2.2.1.1.1.1.cmml" xref="S3.SS2.p3.2.m2.2.2.2.2.2.1.1.1.1"><csymbol cd="ambiguous" id="S3.SS2.p3.2.m2.2.2.2.2.2.1.1.1.1.1.cmml" xref="S3.SS2.p3.2.m2.2.2.2.2.2.1.1.1.1">subscript</csymbol><ci id="S3.SS2.p3.2.m2.2.2.2.2.2.1.1.1.1.2.cmml" xref="S3.SS2.p3.2.m2.2.2.2.2.2.1.1.1.1.2">𝑚</ci><ci id="S3.SS2.p3.2.m2.2.2.2.2.2.1.1.1.1.3.cmml" xref="S3.SS2.p3.2.m2.2.2.2.2.2.1.1.1.1.3">𝑖</ci></apply></set><apply id="S3.SS2.p3.2.m2.2.2.2.2.2.1.3.cmml" xref="S3.SS2.p3.2.m2.2.2.2.2.2.1.3"><minus id="S3.SS2.p3.2.m2.2.2.2.2.2.1.3.1.cmml" xref="S3.SS2.p3.2.m2.2.2.2.2.2.1.3.1"></minus><ci id="S3.SS2.p3.2.m2.2.2.2.2.2.1.3.2.cmml" xref="S3.SS2.p3.2.m2.2.2.2.2.2.1.3.2">𝑘</ci><cn id="S3.SS2.p3.2.m2.2.2.2.2.2.1.3.3.cmml" type="integer" xref="S3.SS2.p3.2.m2.2.2.2.2.2.1.3.3">1</cn></apply></apply><apply id="S3.SS2.p3.2.m2.2.2.2.2.2.3.cmml" xref="S3.SS2.p3.2.m2.2.2.2.2.2.3"><eq id="S3.SS2.p3.2.m2.2.2.2.2.2.3.1.cmml" xref="S3.SS2.p3.2.m2.2.2.2.2.2.3.1"></eq><ci id="S3.SS2.p3.2.m2.2.2.2.2.2.3.2.cmml" xref="S3.SS2.p3.2.m2.2.2.2.2.2.3.2">𝑖</ci><cn id="S3.SS2.p3.2.m2.2.2.2.2.2.3.3.cmml" type="integer" xref="S3.SS2.p3.2.m2.2.2.2.2.2.3.3">1</cn></apply></apply></interval></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS2.p3.2.m2.2c">r_{\text{div}}(m_{k},\{m_{i}\}^{k-1}_{i=1})</annotation><annotation encoding="application/x-llamapun" id="S3.SS2.p3.2.m2.2d">italic_r start_POSTSUBSCRIPT div end_POSTSUBSCRIPT ( italic_m start_POSTSUBSCRIPT italic_k end_POSTSUBSCRIPT , { italic_m start_POSTSUBSCRIPT italic_i end_POSTSUBSCRIPT } start_POSTSUPERSCRIPT italic_k - 1 end_POSTSUPERSCRIPT start_POSTSUBSCRIPT italic_i = 1 end_POSTSUBSCRIPT )</annotation></semantics></math> and a description-matching reward <math alttext="r_{\text{match}}(m_{k},p_{\text{desc}})" class="ltx_Math" display="inline" id="S3.SS2.p3.3.m3.2"><semantics id="S3.SS2.p3.3.m3.2a"><mrow id="S3.SS2.p3.3.m3.2.2" xref="S3.SS2.p3.3.m3.2.2.cmml"><msub id="S3.SS2.p3.3.m3.2.2.4" xref="S3.SS2.p3.3.m3.2.2.4.cmml"><mi id="S3.SS2.p3.3.m3.2.2.4.2" xref="S3.SS2.p3.3.m3.2.2.4.2.cmml">r</mi><mtext id="S3.SS2.p3.3.m3.2.2.4.3" xref="S3.SS2.p3.3.m3.2.2.4.3a.cmml">match</mtext></msub><mo id="S3.SS2.p3.3.m3.2.2.3" xref="S3.SS2.p3.3.m3.2.2.3.cmml">⁢</mo><mrow id="S3.SS2.p3.3.m3.2.2.2.2" xref="S3.SS2.p3.3.m3.2.2.2.3.cmml"><mo id="S3.SS2.p3.3.m3.2.2.2.2.3" stretchy="false" xref="S3.SS2.p3.3.m3.2.2.2.3.cmml">(</mo><msub id="S3.SS2.p3.3.m3.1.1.1.1.1" xref="S3.SS2.p3.3.m3.1.1.1.1.1.cmml"><mi id="S3.SS2.p3.3.m3.1.1.1.1.1.2" xref="S3.SS2.p3.3.m3.1.1.1.1.1.2.cmml">m</mi><mi id="S3.SS2.p3.3.m3.1.1.1.1.1.3" xref="S3.SS2.p3.3.m3.1.1.1.1.1.3.cmml">k</mi></msub><mo id="S3.SS2.p3.3.m3.2.2.2.2.4" xref="S3.SS2.p3.3.m3.2.2.2.3.cmml">,</mo><msub id="S3.SS2.p3.3.m3.2.2.2.2.2" xref="S3.SS2.p3.3.m3.2.2.2.2.2.cmml"><mi id="S3.SS2.p3.3.m3.2.2.2.2.2.2" xref="S3.SS2.p3.3.m3.2.2.2.2.2.2.cmml">p</mi><mtext id="S3.SS2.p3.3.m3.2.2.2.2.2.3" xref="S3.SS2.p3.3.m3.2.2.2.2.2.3a.cmml">desc</mtext></msub><mo id="S3.SS2.p3.3.m3.2.2.2.2.5" stretchy="false" xref="S3.SS2.p3.3.m3.2.2.2.3.cmml">)</mo></mrow></mrow><annotation-xml encoding="MathML-Content" id="S3.SS2.p3.3.m3.2b"><apply id="S3.SS2.p3.3.m3.2.2.cmml" xref="S3.SS2.p3.3.m3.2.2"><times id="S3.SS2.p3.3.m3.2.2.3.cmml" xref="S3.SS2.p3.3.m3.2.2.3"></times><apply id="S3.SS2.p3.3.m3.2.2.4.cmml" xref="S3.SS2.p3.3.m3.2.2.4"><csymbol cd="ambiguous" id="S3.SS2.p3.3.m3.2.2.4.1.cmml" xref="S3.SS2.p3.3.m3.2.2.4">subscript</csymbol><ci id="S3.SS2.p3.3.m3.2.2.4.2.cmml" xref="S3.SS2.p3.3.m3.2.2.4.2">𝑟</ci><ci id="S3.SS2.p3.3.m3.2.2.4.3a.cmml" xref="S3.SS2.p3.3.m3.2.2.4.3"><mtext id="S3.SS2.p3.3.m3.2.2.4.3.cmml" mathsize="70%" xref="S3.SS2.p3.3.m3.2.2.4.3">match</mtext></ci></apply><interval closure="open" id="S3.SS2.p3.3.m3.2.2.2.3.cmml" xref="S3.SS2.p3.3.m3.2.2.2.2"><apply id="S3.SS2.p3.3.m3.1.1.1.1.1.cmml" xref="S3.SS2.p3.3.m3.1.1.1.1.1"><csymbol cd="ambiguous" id="S3.SS2.p3.3.m3.1.1.1.1.1.1.cmml" xref="S3.SS2.p3.3.m3.1.1.1.1.1">subscript</csymbol><ci id="S3.SS2.p3.3.m3.1.1.1.1.1.2.cmml" xref="S3.SS2.p3.3.m3.1.1.1.1.1.2">𝑚</ci><ci id="S3.SS2.p3.3.m3.1.1.1.1.1.3.cmml" xref="S3.SS2.p3.3.m3.1.1.1.1.1.3">𝑘</ci></apply><apply id="S3.SS2.p3.3.m3.2.2.2.2.2.cmml" xref="S3.SS2.p3.3.m3.2.2.2.2.2"><csymbol cd="ambiguous" id="S3.SS2.p3.3.m3.2.2.2.2.2.1.cmml" xref="S3.SS2.p3.3.m3.2.2.2.2.2">subscript</csymbol><ci id="S3.SS2.p3.3.m3.2.2.2.2.2.2.cmml" xref="S3.SS2.p3.3.m3.2.2.2.2.2.2">𝑝</ci><ci id="S3.SS2.p3.3.m3.2.2.2.2.2.3a.cmml" xref="S3.SS2.p3.3.m3.2.2.2.2.2.3"><mtext id="S3.SS2.p3.3.m3.2.2.2.2.2.3.cmml" mathsize="70%" xref="S3.SS2.p3.3.m3.2.2.2.2.2.3">desc</mtext></ci></apply></interval></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS2.p3.3.m3.2c">r_{\text{match}}(m_{k},p_{\text{desc}})</annotation><annotation encoding="application/x-llamapun" id="S3.SS2.p3.3.m3.2d">italic_r start_POSTSUBSCRIPT match end_POSTSUBSCRIPT ( italic_m start_POSTSUBSCRIPT italic_k end_POSTSUBSCRIPT , italic_p start_POSTSUBSCRIPT desc end_POSTSUBSCRIPT )</annotation></semantics></math>, as follows:</p>
<table class="ltx_equation ltx_eqn_table" id="S3.Ex2">
<tbody><tr class="ltx_equation ltx_eqn_row ltx_align_baseline">
<td class="ltx_eqn_cell ltx_eqn_center_padleft"></td>
<td class="ltx_eqn_cell ltx_align_center"><math alttext="r(m_{k})=r_{\text{div}}(m_{k},\{m_{i}\}^{k-1}_{i=1})+r_{\text{match}}(m_{k},p_%
{\text{desc}})," class="ltx_Math" display="block" id="S3.Ex2.m1.1"><semantics id="S3.Ex2.m1.1a"><mrow id="S3.Ex2.m1.1.1.1" xref="S3.Ex2.m1.1.1.1.1.cmml"><mrow id="S3.Ex2.m1.1.1.1.1" xref="S3.Ex2.m1.1.1.1.1.cmml"><mrow id="S3.Ex2.m1.1.1.1.1.1" xref="S3.Ex2.m1.1.1.1.1.1.cmml"><mi id="S3.Ex2.m1.1.1.1.1.1.3" xref="S3.Ex2.m1.1.1.1.1.1.3.cmml">r</mi><mo id="S3.Ex2.m1.1.1.1.1.1.2" xref="S3.Ex2.m1.1.1.1.1.1.2.cmml">⁢</mo><mrow id="S3.Ex2.m1.1.1.1.1.1.1.1" xref="S3.Ex2.m1.1.1.1.1.1.1.1.1.cmml"><mo id="S3.Ex2.m1.1.1.1.1.1.1.1.2" stretchy="false" xref="S3.Ex2.m1.1.1.1.1.1.1.1.1.cmml">(</mo><msub id="S3.Ex2.m1.1.1.1.1.1.1.1.1" xref="S3.Ex2.m1.1.1.1.1.1.1.1.1.cmml"><mi id="S3.Ex2.m1.1.1.1.1.1.1.1.1.2" xref="S3.Ex2.m1.1.1.1.1.1.1.1.1.2.cmml">m</mi><mi id="S3.Ex2.m1.1.1.1.1.1.1.1.1.3" xref="S3.Ex2.m1.1.1.1.1.1.1.1.1.3.cmml">k</mi></msub><mo id="S3.Ex2.m1.1.1.1.1.1.1.1.3" stretchy="false" xref="S3.Ex2.m1.1.1.1.1.1.1.1.1.cmml">)</mo></mrow></mrow><mo id="S3.Ex2.m1.1.1.1.1.6" xref="S3.Ex2.m1.1.1.1.1.6.cmml">=</mo><mrow id="S3.Ex2.m1.1.1.1.1.5" xref="S3.Ex2.m1.1.1.1.1.5.cmml"><mrow id="S3.Ex2.m1.1.1.1.1.3.2" xref="S3.Ex2.m1.1.1.1.1.3.2.cmml"><msub id="S3.Ex2.m1.1.1.1.1.3.2.4" xref="S3.Ex2.m1.1.1.1.1.3.2.4.cmml"><mi id="S3.Ex2.m1.1.1.1.1.3.2.4.2" xref="S3.Ex2.m1.1.1.1.1.3.2.4.2.cmml">r</mi><mtext id="S3.Ex2.m1.1.1.1.1.3.2.4.3" xref="S3.Ex2.m1.1.1.1.1.3.2.4.3a.cmml">div</mtext></msub><mo id="S3.Ex2.m1.1.1.1.1.3.2.3" xref="S3.Ex2.m1.1.1.1.1.3.2.3.cmml">⁢</mo><mrow id="S3.Ex2.m1.1.1.1.1.3.2.2.2" xref="S3.Ex2.m1.1.1.1.1.3.2.2.3.cmml"><mo id="S3.Ex2.m1.1.1.1.1.3.2.2.2.3" stretchy="false" xref="S3.Ex2.m1.1.1.1.1.3.2.2.3.cmml">(</mo><msub id="S3.Ex2.m1.1.1.1.1.2.1.1.1.1" xref="S3.Ex2.m1.1.1.1.1.2.1.1.1.1.cmml"><mi id="S3.Ex2.m1.1.1.1.1.2.1.1.1.1.2" xref="S3.Ex2.m1.1.1.1.1.2.1.1.1.1.2.cmml">m</mi><mi id="S3.Ex2.m1.1.1.1.1.2.1.1.1.1.3" xref="S3.Ex2.m1.1.1.1.1.2.1.1.1.1.3.cmml">k</mi></msub><mo id="S3.Ex2.m1.1.1.1.1.3.2.2.2.4" xref="S3.Ex2.m1.1.1.1.1.3.2.2.3.cmml">,</mo><msubsup id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.cmml"><mrow id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.1.1" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.1.2.cmml"><mo id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.1.1.2" stretchy="false" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.1.2.cmml">{</mo><msub id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.1.1.1" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.1.1.1.cmml"><mi id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.1.1.1.2" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.1.1.1.2.cmml">m</mi><mi id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.1.1.1.3" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.1.1.1.3.cmml">i</mi></msub><mo id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.1.1.3" stretchy="false" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.1.2.cmml">}</mo></mrow><mrow id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.3" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.3.cmml"><mi id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.3.2" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.3.2.cmml">i</mi><mo id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.3.1" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.3.1.cmml">=</mo><mn id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.3.3" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.3.3.cmml">1</mn></mrow><mrow id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.3" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.3.cmml"><mi id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.3.2" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.3.2.cmml">k</mi><mo id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.3.1" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.3.1.cmml">−</mo><mn id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.3.3" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.3.3.cmml">1</mn></mrow></msubsup><mo id="S3.Ex2.m1.1.1.1.1.3.2.2.2.5" stretchy="false" xref="S3.Ex2.m1.1.1.1.1.3.2.2.3.cmml">)</mo></mrow></mrow><mo id="S3.Ex2.m1.1.1.1.1.5.5" xref="S3.Ex2.m1.1.1.1.1.5.5.cmml">+</mo><mrow id="S3.Ex2.m1.1.1.1.1.5.4" xref="S3.Ex2.m1.1.1.1.1.5.4.cmml"><msub id="S3.Ex2.m1.1.1.1.1.5.4.4" xref="S3.Ex2.m1.1.1.1.1.5.4.4.cmml"><mi id="S3.Ex2.m1.1.1.1.1.5.4.4.2" xref="S3.Ex2.m1.1.1.1.1.5.4.4.2.cmml">r</mi><mtext id="S3.Ex2.m1.1.1.1.1.5.4.4.3" xref="S3.Ex2.m1.1.1.1.1.5.4.4.3a.cmml">match</mtext></msub><mo id="S3.Ex2.m1.1.1.1.1.5.4.3" xref="S3.Ex2.m1.1.1.1.1.5.4.3.cmml">⁢</mo><mrow id="S3.Ex2.m1.1.1.1.1.5.4.2.2" xref="S3.Ex2.m1.1.1.1.1.5.4.2.3.cmml"><mo id="S3.Ex2.m1.1.1.1.1.5.4.2.2.3" stretchy="false" xref="S3.Ex2.m1.1.1.1.1.5.4.2.3.cmml">(</mo><msub id="S3.Ex2.m1.1.1.1.1.4.3.1.1.1" xref="S3.Ex2.m1.1.1.1.1.4.3.1.1.1.cmml"><mi id="S3.Ex2.m1.1.1.1.1.4.3.1.1.1.2" xref="S3.Ex2.m1.1.1.1.1.4.3.1.1.1.2.cmml">m</mi><mi id="S3.Ex2.m1.1.1.1.1.4.3.1.1.1.3" xref="S3.Ex2.m1.1.1.1.1.4.3.1.1.1.3.cmml">k</mi></msub><mo id="S3.Ex2.m1.1.1.1.1.5.4.2.2.4" xref="S3.Ex2.m1.1.1.1.1.5.4.2.3.cmml">,</mo><msub id="S3.Ex2.m1.1.1.1.1.5.4.2.2.2" xref="S3.Ex2.m1.1.1.1.1.5.4.2.2.2.cmml"><mi id="S3.Ex2.m1.1.1.1.1.5.4.2.2.2.2" xref="S3.Ex2.m1.1.1.1.1.5.4.2.2.2.2.cmml">p</mi><mtext id="S3.Ex2.m1.1.1.1.1.5.4.2.2.2.3" xref="S3.Ex2.m1.1.1.1.1.5.4.2.2.2.3a.cmml">desc</mtext></msub><mo id="S3.Ex2.m1.1.1.1.1.5.4.2.2.5" stretchy="false" xref="S3.Ex2.m1.1.1.1.1.5.4.2.3.cmml">)</mo></mrow></mrow></mrow></mrow><mo id="S3.Ex2.m1.1.1.1.2" xref="S3.Ex2.m1.1.1.1.1.cmml">,</mo></mrow><annotation-xml encoding="MathML-Content" id="S3.Ex2.m1.1b"><apply id="S3.Ex2.m1.1.1.1.1.cmml" xref="S3.Ex2.m1.1.1.1"><eq id="S3.Ex2.m1.1.1.1.1.6.cmml" xref="S3.Ex2.m1.1.1.1.1.6"></eq><apply id="S3.Ex2.m1.1.1.1.1.1.cmml" xref="S3.Ex2.m1.1.1.1.1.1"><times id="S3.Ex2.m1.1.1.1.1.1.2.cmml" xref="S3.Ex2.m1.1.1.1.1.1.2"></times><ci id="S3.Ex2.m1.1.1.1.1.1.3.cmml" xref="S3.Ex2.m1.1.1.1.1.1.3">𝑟</ci><apply id="S3.Ex2.m1.1.1.1.1.1.1.1.1.cmml" xref="S3.Ex2.m1.1.1.1.1.1.1.1"><csymbol cd="ambiguous" id="S3.Ex2.m1.1.1.1.1.1.1.1.1.1.cmml" xref="S3.Ex2.m1.1.1.1.1.1.1.1">subscript</csymbol><ci id="S3.Ex2.m1.1.1.1.1.1.1.1.1.2.cmml" xref="S3.Ex2.m1.1.1.1.1.1.1.1.1.2">𝑚</ci><ci id="S3.Ex2.m1.1.1.1.1.1.1.1.1.3.cmml" xref="S3.Ex2.m1.1.1.1.1.1.1.1.1.3">𝑘</ci></apply></apply><apply id="S3.Ex2.m1.1.1.1.1.5.cmml" xref="S3.Ex2.m1.1.1.1.1.5"><plus id="S3.Ex2.m1.1.1.1.1.5.5.cmml" xref="S3.Ex2.m1.1.1.1.1.5.5"></plus><apply id="S3.Ex2.m1.1.1.1.1.3.2.cmml" xref="S3.Ex2.m1.1.1.1.1.3.2"><times id="S3.Ex2.m1.1.1.1.1.3.2.3.cmml" xref="S3.Ex2.m1.1.1.1.1.3.2.3"></times><apply id="S3.Ex2.m1.1.1.1.1.3.2.4.cmml" xref="S3.Ex2.m1.1.1.1.1.3.2.4"><csymbol cd="ambiguous" id="S3.Ex2.m1.1.1.1.1.3.2.4.1.cmml" xref="S3.Ex2.m1.1.1.1.1.3.2.4">subscript</csymbol><ci id="S3.Ex2.m1.1.1.1.1.3.2.4.2.cmml" xref="S3.Ex2.m1.1.1.1.1.3.2.4.2">𝑟</ci><ci id="S3.Ex2.m1.1.1.1.1.3.2.4.3a.cmml" xref="S3.Ex2.m1.1.1.1.1.3.2.4.3"><mtext id="S3.Ex2.m1.1.1.1.1.3.2.4.3.cmml" mathsize="70%" xref="S3.Ex2.m1.1.1.1.1.3.2.4.3">div</mtext></ci></apply><interval closure="open" id="S3.Ex2.m1.1.1.1.1.3.2.2.3.cmml" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2"><apply id="S3.Ex2.m1.1.1.1.1.2.1.1.1.1.cmml" xref="S3.Ex2.m1.1.1.1.1.2.1.1.1.1"><csymbol cd="ambiguous" id="S3.Ex2.m1.1.1.1.1.2.1.1.1.1.1.cmml" xref="S3.Ex2.m1.1.1.1.1.2.1.1.1.1">subscript</csymbol><ci id="S3.Ex2.m1.1.1.1.1.2.1.1.1.1.2.cmml" xref="S3.Ex2.m1.1.1.1.1.2.1.1.1.1.2">𝑚</ci><ci id="S3.Ex2.m1.1.1.1.1.2.1.1.1.1.3.cmml" xref="S3.Ex2.m1.1.1.1.1.2.1.1.1.1.3">𝑘</ci></apply><apply id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.cmml" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2"><csymbol cd="ambiguous" id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.2.cmml" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2">subscript</csymbol><apply id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.cmml" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2"><csymbol cd="ambiguous" id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.2.cmml" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2">superscript</csymbol><set id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.1.2.cmml" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.1.1"><apply id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.1.1.1.cmml" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.1.1.1"><csymbol cd="ambiguous" id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.1.1.1.1.cmml" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.1.1.1">subscript</csymbol><ci id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.1.1.1.2.cmml" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.1.1.1.2">𝑚</ci><ci id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.1.1.1.3.cmml" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.1.1.1.3">𝑖</ci></apply></set><apply id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.3.cmml" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.3"><minus id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.3.1.cmml" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.3.1"></minus><ci id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.3.2.cmml" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.3.2">𝑘</ci><cn id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.3.3.cmml" type="integer" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.1.3.3">1</cn></apply></apply><apply id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.3.cmml" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.3"><eq id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.3.1.cmml" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.3.1"></eq><ci id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.3.2.cmml" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.3.2">𝑖</ci><cn id="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.3.3.cmml" type="integer" xref="S3.Ex2.m1.1.1.1.1.3.2.2.2.2.3.3">1</cn></apply></apply></interval></apply><apply id="S3.Ex2.m1.1.1.1.1.5.4.cmml" xref="S3.Ex2.m1.1.1.1.1.5.4"><times id="S3.Ex2.m1.1.1.1.1.5.4.3.cmml" xref="S3.Ex2.m1.1.1.1.1.5.4.3"></times><apply id="S3.Ex2.m1.1.1.1.1.5.4.4.cmml" xref="S3.Ex2.m1.1.1.1.1.5.4.4"><csymbol cd="ambiguous" id="S3.Ex2.m1.1.1.1.1.5.4.4.1.cmml" xref="S3.Ex2.m1.1.1.1.1.5.4.4">subscript</csymbol><ci id="S3.Ex2.m1.1.1.1.1.5.4.4.2.cmml" xref="S3.Ex2.m1.1.1.1.1.5.4.4.2">𝑟</ci><ci id="S3.Ex2.m1.1.1.1.1.5.4.4.3a.cmml" xref="S3.Ex2.m1.1.1.1.1.5.4.4.3"><mtext id="S3.Ex2.m1.1.1.1.1.5.4.4.3.cmml" mathsize="70%" xref="S3.Ex2.m1.1.1.1.1.5.4.4.3">match</mtext></ci></apply><interval closure="open" id="S3.Ex2.m1.1.1.1.1.5.4.2.3.cmml" xref="S3.Ex2.m1.1.1.1.1.5.4.2.2"><apply id="S3.Ex2.m1.1.1.1.1.4.3.1.1.1.cmml" xref="S3.Ex2.m1.1.1.1.1.4.3.1.1.1"><csymbol cd="ambiguous" id="S3.Ex2.m1.1.1.1.1.4.3.1.1.1.1.cmml" xref="S3.Ex2.m1.1.1.1.1.4.3.1.1.1">subscript</csymbol><ci id="S3.Ex2.m1.1.1.1.1.4.3.1.1.1.2.cmml" xref="S3.Ex2.m1.1.1.1.1.4.3.1.1.1.2">𝑚</ci><ci id="S3.Ex2.m1.1.1.1.1.4.3.1.1.1.3.cmml" xref="S3.Ex2.m1.1.1.1.1.4.3.1.1.1.3">𝑘</ci></apply><apply id="S3.Ex2.m1.1.1.1.1.5.4.2.2.2.cmml" xref="S3.Ex2.m1.1.1.1.1.5.4.2.2.2"><csymbol cd="ambiguous" id="S3.Ex2.m1.1.1.1.1.5.4.2.2.2.1.cmml" xref="S3.Ex2.m1.1.1.1.1.5.4.2.2.2">subscript</csymbol><ci id="S3.Ex2.m1.1.1.1.1.5.4.2.2.2.2.cmml" xref="S3.Ex2.m1.1.1.1.1.5.4.2.2.2.2">𝑝</ci><ci id="S3.Ex2.m1.1.1.1.1.5.4.2.2.2.3a.cmml" xref="S3.Ex2.m1.1.1.1.1.5.4.2.2.2.3"><mtext id="S3.Ex2.m1.1.1.1.1.5.4.2.2.2.3.cmml" mathsize="70%" xref="S3.Ex2.m1.1.1.1.1.5.4.2.2.2.3">desc</mtext></ci></apply></interval></apply></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.Ex2.m1.1c">r(m_{k})=r_{\text{div}}(m_{k},\{m_{i}\}^{k-1}_{i=1})+r_{\text{match}}(m_{k},p_%
{\text{desc}}),</annotation><annotation encoding="application/x-llamapun" id="S3.Ex2.m1.1d">italic_r ( italic_m start_POSTSUBSCRIPT italic_k end_POSTSUBSCRIPT ) = italic_r start_POSTSUBSCRIPT div end_POSTSUBSCRIPT ( italic_m start_POSTSUBSCRIPT italic_k end_POSTSUBSCRIPT , { italic_m start_POSTSUBSCRIPT italic_i end_POSTSUBSCRIPT } start_POSTSUPERSCRIPT italic_k - 1 end_POSTSUPERSCRIPT start_POSTSUBSCRIPT italic_i = 1 end_POSTSUBSCRIPT ) + italic_r start_POSTSUBSCRIPT match end_POSTSUBSCRIPT ( italic_m start_POSTSUBSCRIPT italic_k end_POSTSUBSCRIPT , italic_p start_POSTSUBSCRIPT desc end_POSTSUBSCRIPT ) ,</annotation></semantics></math></td>
<td class="ltx_eqn_cell ltx_eqn_center_padright"></td>
</tr></tbody>
</table>
<p class="ltx_p" id="S3.SS2.p3.12">where the diversity reward <math alttext="r_{\text{div}}" class="ltx_Math" display="inline" id="S3.SS2.p3.4.m1.1"><semantics id="S3.SS2.p3.4.m1.1a"><msub id="S3.SS2.p3.4.m1.1.1" xref="S3.SS2.p3.4.m1.1.1.cmml"><mi id="S3.SS2.p3.4.m1.1.1.2" xref="S3.SS2.p3.4.m1.1.1.2.cmml">r</mi><mtext id="S3.SS2.p3.4.m1.1.1.3" xref="S3.SS2.p3.4.m1.1.1.3a.cmml">div</mtext></msub><annotation-xml encoding="MathML-Content" id="S3.SS2.p3.4.m1.1b"><apply id="S3.SS2.p3.4.m1.1.1.cmml" xref="S3.SS2.p3.4.m1.1.1"><csymbol cd="ambiguous" id="S3.SS2.p3.4.m1.1.1.1.cmml" xref="S3.SS2.p3.4.m1.1.1">subscript</csymbol><ci id="S3.SS2.p3.4.m1.1.1.2.cmml" xref="S3.SS2.p3.4.m1.1.1.2">𝑟</ci><ci id="S3.SS2.p3.4.m1.1.1.3a.cmml" xref="S3.SS2.p3.4.m1.1.1.3"><mtext id="S3.SS2.p3.4.m1.1.1.3.cmml" mathsize="70%" xref="S3.SS2.p3.4.m1.1.1.3">div</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS2.p3.4.m1.1c">r_{\text{div}}</annotation><annotation encoding="application/x-llamapun" id="S3.SS2.p3.4.m1.1d">italic_r start_POSTSUBSCRIPT div end_POSTSUBSCRIPT</annotation></semantics></math> evaluates structural differences between the molecule <math alttext="m_{k}" class="ltx_Math" display="inline" id="S3.SS2.p3.5.m2.1"><semantics id="S3.SS2.p3.5.m2.1a"><msub id="S3.SS2.p3.5.m2.1.1" xref="S3.SS2.p3.5.m2.1.1.cmml"><mi id="S3.SS2.p3.5.m2.1.1.2" xref="S3.SS2.p3.5.m2.1.1.2.cmml">m</mi><mi id="S3.SS2.p3.5.m2.1.1.3" xref="S3.SS2.p3.5.m2.1.1.3.cmml">k</mi></msub><annotation-xml encoding="MathML-Content" id="S3.SS2.p3.5.m2.1b"><apply id="S3.SS2.p3.5.m2.1.1.cmml" xref="S3.SS2.p3.5.m2.1.1"><csymbol cd="ambiguous" id="S3.SS2.p3.5.m2.1.1.1.cmml" xref="S3.SS2.p3.5.m2.1.1">subscript</csymbol><ci id="S3.SS2.p3.5.m2.1.1.2.cmml" xref="S3.SS2.p3.5.m2.1.1.2">𝑚</ci><ci id="S3.SS2.p3.5.m2.1.1.3.cmml" xref="S3.SS2.p3.5.m2.1.1.3">𝑘</ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS2.p3.5.m2.1c">m_{k}</annotation><annotation encoding="application/x-llamapun" id="S3.SS2.p3.5.m2.1d">italic_m start_POSTSUBSCRIPT italic_k end_POSTSUBSCRIPT</annotation></semantics></math> and the previously generated molecules <math alttext="\{m_{i}\}^{k-1}_{i=1}" class="ltx_Math" display="inline" id="S3.SS2.p3.6.m3.1"><semantics id="S3.SS2.p3.6.m3.1a"><msubsup id="S3.SS2.p3.6.m3.1.1" xref="S3.SS2.p3.6.m3.1.1.cmml"><mrow id="S3.SS2.p3.6.m3.1.1.1.1.1" xref="S3.SS2.p3.6.m3.1.1.1.1.2.cmml"><mo id="S3.SS2.p3.6.m3.1.1.1.1.1.2" stretchy="false" xref="S3.SS2.p3.6.m3.1.1.1.1.2.cmml">{</mo><msub id="S3.SS2.p3.6.m3.1.1.1.1.1.1" xref="S3.SS2.p3.6.m3.1.1.1.1.1.1.cmml"><mi id="S3.SS2.p3.6.m3.1.1.1.1.1.1.2" xref="S3.SS2.p3.6.m3.1.1.1.1.1.1.2.cmml">m</mi><mi id="S3.SS2.p3.6.m3.1.1.1.1.1.1.3" xref="S3.SS2.p3.6.m3.1.1.1.1.1.1.3.cmml">i</mi></msub><mo id="S3.SS2.p3.6.m3.1.1.1.1.1.3" stretchy="false" xref="S3.SS2.p3.6.m3.1.1.1.1.2.cmml">}</mo></mrow><mrow id="S3.SS2.p3.6.m3.1.1.3" xref="S3.SS2.p3.6.m3.1.1.3.cmml"><mi id="S3.SS2.p3.6.m3.1.1.3.2" xref="S3.SS2.p3.6.m3.1.1.3.2.cmml">i</mi><mo id="S3.SS2.p3.6.m3.1.1.3.1" xref="S3.SS2.p3.6.m3.1.1.3.1.cmml">=</mo><mn id="S3.SS2.p3.6.m3.1.1.3.3" xref="S3.SS2.p3.6.m3.1.1.3.3.cmml">1</mn></mrow><mrow id="S3.SS2.p3.6.m3.1.1.1.3" xref="S3.SS2.p3.6.m3.1.1.1.3.cmml"><mi id="S3.SS2.p3.6.m3.1.1.1.3.2" xref="S3.SS2.p3.6.m3.1.1.1.3.2.cmml">k</mi><mo id="S3.SS2.p3.6.m3.1.1.1.3.1" xref="S3.SS2.p3.6.m3.1.1.1.3.1.cmml">−</mo><mn id="S3.SS2.p3.6.m3.1.1.1.3.3" xref="S3.SS2.p3.6.m3.1.1.1.3.3.cmml">1</mn></mrow></msubsup><annotation-xml encoding="MathML-Content" id="S3.SS2.p3.6.m3.1b"><apply id="S3.SS2.p3.6.m3.1.1.cmml" xref="S3.SS2.p3.6.m3.1.1"><csymbol cd="ambiguous" id="S3.SS2.p3.6.m3.1.1.2.cmml" xref="S3.SS2.p3.6.m3.1.1">subscript</csymbol><apply id="S3.SS2.p3.6.m3.1.1.1.cmml" xref="S3.SS2.p3.6.m3.1.1"><csymbol cd="ambiguous" id="S3.SS2.p3.6.m3.1.1.1.2.cmml" xref="S3.SS2.p3.6.m3.1.1">superscript</csymbol><set id="S3.SS2.p3.6.m3.1.1.1.1.2.cmml" xref="S3.SS2.p3.6.m3.1.1.1.1.1"><apply id="S3.SS2.p3.6.m3.1.1.1.1.1.1.cmml" xref="S3.SS2.p3.6.m3.1.1.1.1.1.1"><csymbol cd="ambiguous" id="S3.SS2.p3.6.m3.1.1.1.1.1.1.1.cmml" xref="S3.SS2.p3.6.m3.1.1.1.1.1.1">subscript</csymbol><ci id="S3.SS2.p3.6.m3.1.1.1.1.1.1.2.cmml" xref="S3.SS2.p3.6.m3.1.1.1.1.1.1.2">𝑚</ci><ci id="S3.SS2.p3.6.m3.1.1.1.1.1.1.3.cmml" xref="S3.SS2.p3.6.m3.1.1.1.1.1.1.3">𝑖</ci></apply></set><apply id="S3.SS2.p3.6.m3.1.1.1.3.cmml" xref="S3.SS2.p3.6.m3.1.1.1.3"><minus id="S3.SS2.p3.6.m3.1.1.1.3.1.cmml" xref="S3.SS2.p3.6.m3.1.1.1.3.1"></minus><ci id="S3.SS2.p3.6.m3.1.1.1.3.2.cmml" xref="S3.SS2.p3.6.m3.1.1.1.3.2">𝑘</ci><cn id="S3.SS2.p3.6.m3.1.1.1.3.3.cmml" type="integer" xref="S3.SS2.p3.6.m3.1.1.1.3.3">1</cn></apply></apply><apply id="S3.SS2.p3.6.m3.1.1.3.cmml" xref="S3.SS2.p3.6.m3.1.1.3"><eq id="S3.SS2.p3.6.m3.1.1.3.1.cmml" xref="S3.SS2.p3.6.m3.1.1.3.1"></eq><ci id="S3.SS2.p3.6.m3.1.1.3.2.cmml" xref="S3.SS2.p3.6.m3.1.1.3.2">𝑖</ci><cn id="S3.SS2.p3.6.m3.1.1.3.3.cmml" type="integer" xref="S3.SS2.p3.6.m3.1.1.3.3">1</cn></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS2.p3.6.m3.1c">\{m_{i}\}^{k-1}_{i=1}</annotation><annotation encoding="application/x-llamapun" id="S3.SS2.p3.6.m3.1d">{ italic_m start_POSTSUBSCRIPT italic_i end_POSTSUBSCRIPT } start_POSTSUPERSCRIPT italic_k - 1 end_POSTSUPERSCRIPT start_POSTSUBSCRIPT italic_i = 1 end_POSTSUBSCRIPT</annotation></semantics></math> by assessing their true molecular structures. Note that <math alttext="r_{\text{div}}(m_{1})" class="ltx_Math" display="inline" id="S3.SS2.p3.7.m4.1"><semantics id="S3.SS2.p3.7.m4.1a"><mrow id="S3.SS2.p3.7.m4.1.1" xref="S3.SS2.p3.7.m4.1.1.cmml"><msub id="S3.SS2.p3.7.m4.1.1.3" xref="S3.SS2.p3.7.m4.1.1.3.cmml"><mi id="S3.SS2.p3.7.m4.1.1.3.2" xref="S3.SS2.p3.7.m4.1.1.3.2.cmml">r</mi><mtext id="S3.SS2.p3.7.m4.1.1.3.3" xref="S3.SS2.p3.7.m4.1.1.3.3a.cmml">div</mtext></msub><mo id="S3.SS2.p3.7.m4.1.1.2" xref="S3.SS2.p3.7.m4.1.1.2.cmml">⁢</mo><mrow id="S3.SS2.p3.7.m4.1.1.1.1" xref="S3.SS2.p3.7.m4.1.1.1.1.1.cmml"><mo id="S3.SS2.p3.7.m4.1.1.1.1.2" stretchy="false" xref="S3.SS2.p3.7.m4.1.1.1.1.1.cmml">(</mo><msub id="S3.SS2.p3.7.m4.1.1.1.1.1" xref="S3.SS2.p3.7.m4.1.1.1.1.1.cmml"><mi id="S3.SS2.p3.7.m4.1.1.1.1.1.2" xref="S3.SS2.p3.7.m4.1.1.1.1.1.2.cmml">m</mi><mn id="S3.SS2.p3.7.m4.1.1.1.1.1.3" xref="S3.SS2.p3.7.m4.1.1.1.1.1.3.cmml">1</mn></msub><mo id="S3.SS2.p3.7.m4.1.1.1.1.3" stretchy="false" xref="S3.SS2.p3.7.m4.1.1.1.1.1.cmml">)</mo></mrow></mrow><annotation-xml encoding="MathML-Content" id="S3.SS2.p3.7.m4.1b"><apply id="S3.SS2.p3.7.m4.1.1.cmml" xref="S3.SS2.p3.7.m4.1.1"><times id="S3.SS2.p3.7.m4.1.1.2.cmml" xref="S3.SS2.p3.7.m4.1.1.2"></times><apply id="S3.SS2.p3.7.m4.1.1.3.cmml" xref="S3.SS2.p3.7.m4.1.1.3"><csymbol cd="ambiguous" id="S3.SS2.p3.7.m4.1.1.3.1.cmml" xref="S3.SS2.p3.7.m4.1.1.3">subscript</csymbol><ci id="S3.SS2.p3.7.m4.1.1.3.2.cmml" xref="S3.SS2.p3.7.m4.1.1.3.2">𝑟</ci><ci id="S3.SS2.p3.7.m4.1.1.3.3a.cmml" xref="S3.SS2.p3.7.m4.1.1.3.3"><mtext id="S3.SS2.p3.7.m4.1.1.3.3.cmml" mathsize="70%" xref="S3.SS2.p3.7.m4.1.1.3.3">div</mtext></ci></apply><apply id="S3.SS2.p3.7.m4.1.1.1.1.1.cmml" xref="S3.SS2.p3.7.m4.1.1.1.1"><csymbol cd="ambiguous" id="S3.SS2.p3.7.m4.1.1.1.1.1.1.cmml" xref="S3.SS2.p3.7.m4.1.1.1.1">subscript</csymbol><ci id="S3.SS2.p3.7.m4.1.1.1.1.1.2.cmml" xref="S3.SS2.p3.7.m4.1.1.1.1.1.2">𝑚</ci><cn id="S3.SS2.p3.7.m4.1.1.1.1.1.3.cmml" type="integer" xref="S3.SS2.p3.7.m4.1.1.1.1.1.3">1</cn></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS2.p3.7.m4.1c">r_{\text{div}}(m_{1})</annotation><annotation encoding="application/x-llamapun" id="S3.SS2.p3.7.m4.1d">italic_r start_POSTSUBSCRIPT div end_POSTSUBSCRIPT ( italic_m start_POSTSUBSCRIPT 1 end_POSTSUBSCRIPT )</annotation></semantics></math> is zero. The description-matching reward <math alttext="r_{\text{match}}" class="ltx_Math" display="inline" id="S3.SS2.p3.8.m5.1"><semantics id="S3.SS2.p3.8.m5.1a"><msub id="S3.SS2.p3.8.m5.1.1" xref="S3.SS2.p3.8.m5.1.1.cmml"><mi id="S3.SS2.p3.8.m5.1.1.2" xref="S3.SS2.p3.8.m5.1.1.2.cmml">r</mi><mtext id="S3.SS2.p3.8.m5.1.1.3" xref="S3.SS2.p3.8.m5.1.1.3a.cmml">match</mtext></msub><annotation-xml encoding="MathML-Content" id="S3.SS2.p3.8.m5.1b"><apply id="S3.SS2.p3.8.m5.1.1.cmml" xref="S3.SS2.p3.8.m5.1.1"><csymbol cd="ambiguous" id="S3.SS2.p3.8.m5.1.1.1.cmml" xref="S3.SS2.p3.8.m5.1.1">subscript</csymbol><ci id="S3.SS2.p3.8.m5.1.1.2.cmml" xref="S3.SS2.p3.8.m5.1.1.2">𝑟</ci><ci id="S3.SS2.p3.8.m5.1.1.3a.cmml" xref="S3.SS2.p3.8.m5.1.1.3"><mtext id="S3.SS2.p3.8.m5.1.1.3.cmml" mathsize="70%" xref="S3.SS2.p3.8.m5.1.1.3">match</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS2.p3.8.m5.1c">r_{\text{match}}</annotation><annotation encoding="application/x-llamapun" id="S3.SS2.p3.8.m5.1d">italic_r start_POSTSUBSCRIPT match end_POSTSUBSCRIPT</annotation></semantics></math> evaluates whether the molecule <math alttext="m_{k}" class="ltx_Math" display="inline" id="S3.SS2.p3.9.m6.1"><semantics id="S3.SS2.p3.9.m6.1a"><msub id="S3.SS2.p3.9.m6.1.1" xref="S3.SS2.p3.9.m6.1.1.cmml"><mi id="S3.SS2.p3.9.m6.1.1.2" xref="S3.SS2.p3.9.m6.1.1.2.cmml">m</mi><mi id="S3.SS2.p3.9.m6.1.1.3" xref="S3.SS2.p3.9.m6.1.1.3.cmml">k</mi></msub><annotation-xml encoding="MathML-Content" id="S3.SS2.p3.9.m6.1b"><apply id="S3.SS2.p3.9.m6.1.1.cmml" xref="S3.SS2.p3.9.m6.1.1"><csymbol cd="ambiguous" id="S3.SS2.p3.9.m6.1.1.1.cmml" xref="S3.SS2.p3.9.m6.1.1">subscript</csymbol><ci id="S3.SS2.p3.9.m6.1.1.2.cmml" xref="S3.SS2.p3.9.m6.1.1.2">𝑚</ci><ci id="S3.SS2.p3.9.m6.1.1.3.cmml" xref="S3.SS2.p3.9.m6.1.1.3">𝑘</ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS2.p3.9.m6.1c">m_{k}</annotation><annotation encoding="application/x-llamapun" id="S3.SS2.p3.9.m6.1d">italic_m start_POSTSUBSCRIPT italic_k end_POSTSUBSCRIPT</annotation></semantics></math> satisfies the given description <math alttext="p_{\text{desc}}" class="ltx_Math" display="inline" id="S3.SS2.p3.10.m7.1"><semantics id="S3.SS2.p3.10.m7.1a"><msub id="S3.SS2.p3.10.m7.1.1" xref="S3.SS2.p3.10.m7.1.1.cmml"><mi id="S3.SS2.p3.10.m7.1.1.2" xref="S3.SS2.p3.10.m7.1.1.2.cmml">p</mi><mtext id="S3.SS2.p3.10.m7.1.1.3" xref="S3.SS2.p3.10.m7.1.1.3a.cmml">desc</mtext></msub><annotation-xml encoding="MathML-Content" id="S3.SS2.p3.10.m7.1b"><apply id="S3.SS2.p3.10.m7.1.1.cmml" xref="S3.SS2.p3.10.m7.1.1"><csymbol cd="ambiguous" id="S3.SS2.p3.10.m7.1.1.1.cmml" xref="S3.SS2.p3.10.m7.1.1">subscript</csymbol><ci id="S3.SS2.p3.10.m7.1.1.2.cmml" xref="S3.SS2.p3.10.m7.1.1.2">𝑝</ci><ci id="S3.SS2.p3.10.m7.1.1.3a.cmml" xref="S3.SS2.p3.10.m7.1.1.3"><mtext id="S3.SS2.p3.10.m7.1.1.3.cmml" mathsize="70%" xref="S3.SS2.p3.10.m7.1.1.3">desc</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS2.p3.10.m7.1c">p_{\text{desc}}</annotation><annotation encoding="application/x-llamapun" id="S3.SS2.p3.10.m7.1d">italic_p start_POSTSUBSCRIPT desc end_POSTSUBSCRIPT</annotation></semantics></math>. In a practical implementation, the final action that completes the molecule <math alttext="m_{k}" class="ltx_Math" display="inline" id="S3.SS2.p3.11.m8.1"><semantics id="S3.SS2.p3.11.m8.1a"><msub id="S3.SS2.p3.11.m8.1.1" xref="S3.SS2.p3.11.m8.1.1.cmml"><mi id="S3.SS2.p3.11.m8.1.1.2" xref="S3.SS2.p3.11.m8.1.1.2.cmml">m</mi><mi id="S3.SS2.p3.11.m8.1.1.3" xref="S3.SS2.p3.11.m8.1.1.3.cmml">k</mi></msub><annotation-xml encoding="MathML-Content" id="S3.SS2.p3.11.m8.1b"><apply id="S3.SS2.p3.11.m8.1.1.cmml" xref="S3.SS2.p3.11.m8.1.1"><csymbol cd="ambiguous" id="S3.SS2.p3.11.m8.1.1.1.cmml" xref="S3.SS2.p3.11.m8.1.1">subscript</csymbol><ci id="S3.SS2.p3.11.m8.1.1.2.cmml" xref="S3.SS2.p3.11.m8.1.1.2">𝑚</ci><ci id="S3.SS2.p3.11.m8.1.1.3.cmml" xref="S3.SS2.p3.11.m8.1.1.3">𝑘</ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS2.p3.11.m8.1c">m_{k}</annotation><annotation encoding="application/x-llamapun" id="S3.SS2.p3.11.m8.1d">italic_m start_POSTSUBSCRIPT italic_k end_POSTSUBSCRIPT</annotation></semantics></math> yields the reward <math alttext="r(m_{k})" class="ltx_Math" display="inline" id="S3.SS2.p3.12.m9.1"><semantics id="S3.SS2.p3.12.m9.1a"><mrow id="S3.SS2.p3.12.m9.1.1" xref="S3.SS2.p3.12.m9.1.1.cmml"><mi id="S3.SS2.p3.12.m9.1.1.3" xref="S3.SS2.p3.12.m9.1.1.3.cmml">r</mi><mo id="S3.SS2.p3.12.m9.1.1.2" xref="S3.SS2.p3.12.m9.1.1.2.cmml">⁢</mo><mrow id="S3.SS2.p3.12.m9.1.1.1.1" xref="S3.SS2.p3.12.m9.1.1.1.1.1.cmml"><mo id="S3.SS2.p3.12.m9.1.1.1.1.2" stretchy="false" xref="S3.SS2.p3.12.m9.1.1.1.1.1.cmml">(</mo><msub id="S3.SS2.p3.12.m9.1.1.1.1.1" xref="S3.SS2.p3.12.m9.1.1.1.1.1.cmml"><mi id="S3.SS2.p3.12.m9.1.1.1.1.1.2" xref="S3.SS2.p3.12.m9.1.1.1.1.1.2.cmml">m</mi><mi id="S3.SS2.p3.12.m9.1.1.1.1.1.3" xref="S3.SS2.p3.12.m9.1.1.1.1.1.3.cmml">k</mi></msub><mo id="S3.SS2.p3.12.m9.1.1.1.1.3" stretchy="false" xref="S3.SS2.p3.12.m9.1.1.1.1.1.cmml">)</mo></mrow></mrow><annotation-xml encoding="MathML-Content" id="S3.SS2.p3.12.m9.1b"><apply id="S3.SS2.p3.12.m9.1.1.cmml" xref="S3.SS2.p3.12.m9.1.1"><times id="S3.SS2.p3.12.m9.1.1.2.cmml" xref="S3.SS2.p3.12.m9.1.1.2"></times><ci id="S3.SS2.p3.12.m9.1.1.3.cmml" xref="S3.SS2.p3.12.m9.1.1.3">𝑟</ci><apply id="S3.SS2.p3.12.m9.1.1.1.1.1.cmml" xref="S3.SS2.p3.12.m9.1.1.1.1"><csymbol cd="ambiguous" id="S3.SS2.p3.12.m9.1.1.1.1.1.1.cmml" xref="S3.SS2.p3.12.m9.1.1.1.1">subscript</csymbol><ci id="S3.SS2.p3.12.m9.1.1.1.1.1.2.cmml" xref="S3.SS2.p3.12.m9.1.1.1.1.1.2">𝑚</ci><ci id="S3.SS2.p3.12.m9.1.1.1.1.1.3.cmml" xref="S3.SS2.p3.12.m9.1.1.1.1.1.3">𝑘</ci></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS2.p3.12.m9.1c">r(m_{k})</annotation><annotation encoding="application/x-llamapun" id="S3.SS2.p3.12.m9.1d">italic_r ( italic_m start_POSTSUBSCRIPT italic_k end_POSTSUBSCRIPT )</annotation></semantics></math>. The detailed reward implementation is described in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#A1" title="Appendix A Detailed implementations and training ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Appendix</span> <span class="ltx_text ltx_ref_tag">A</span></a>.</p>
</div>
<div class="ltx_para ltx_noindent" id="S3.SS2.p4">
<p class="ltx_p" id="S3.SS2.p4.3"><span class="ltx_text ltx_font_bold" id="S3.SS2.p4.3.1">Policy optimization.</span> We optimize the LLM <math alttext="\pi_{\text{RL}}" class="ltx_Math" display="inline" id="S3.SS2.p4.1.m1.1"><semantics id="S3.SS2.p4.1.m1.1a"><msub id="S3.SS2.p4.1.m1.1.1" xref="S3.SS2.p4.1.m1.1.1.cmml"><mi id="S3.SS2.p4.1.m1.1.1.2" xref="S3.SS2.p4.1.m1.1.1.2.cmml">π</mi><mtext id="S3.SS2.p4.1.m1.1.1.3" xref="S3.SS2.p4.1.m1.1.1.3a.cmml">RL</mtext></msub><annotation-xml encoding="MathML-Content" id="S3.SS2.p4.1.m1.1b"><apply id="S3.SS2.p4.1.m1.1.1.cmml" xref="S3.SS2.p4.1.m1.1.1"><csymbol cd="ambiguous" id="S3.SS2.p4.1.m1.1.1.1.cmml" xref="S3.SS2.p4.1.m1.1.1">subscript</csymbol><ci id="S3.SS2.p4.1.m1.1.1.2.cmml" xref="S3.SS2.p4.1.m1.1.1.2">𝜋</ci><ci id="S3.SS2.p4.1.m1.1.1.3a.cmml" xref="S3.SS2.p4.1.m1.1.1.3"><mtext id="S3.SS2.p4.1.m1.1.1.3.cmml" mathsize="70%" xref="S3.SS2.p4.1.m1.1.1.3">RL</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS2.p4.1.m1.1c">\pi_{\text{RL}}</annotation><annotation encoding="application/x-llamapun" id="S3.SS2.p4.1.m1.1d">italic_π start_POSTSUBSCRIPT RL end_POSTSUBSCRIPT</annotation></semantics></math> to maximize the reward using proximal policy optimization <cite class="ltx_cite ltx_citemacro_citep">(PPO; Schulman et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib29" title="">2017</a>; Ouyang et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib23" title="">2022</a>)</cite>. Note that <math alttext="\pi_{\text{RL}}" class="ltx_Math" display="inline" id="S3.SS2.p4.2.m2.1"><semantics id="S3.SS2.p4.2.m2.1a"><msub id="S3.SS2.p4.2.m2.1.1" xref="S3.SS2.p4.2.m2.1.1.cmml"><mi id="S3.SS2.p4.2.m2.1.1.2" xref="S3.SS2.p4.2.m2.1.1.2.cmml">π</mi><mtext id="S3.SS2.p4.2.m2.1.1.3" xref="S3.SS2.p4.2.m2.1.1.3a.cmml">RL</mtext></msub><annotation-xml encoding="MathML-Content" id="S3.SS2.p4.2.m2.1b"><apply id="S3.SS2.p4.2.m2.1.1.cmml" xref="S3.SS2.p4.2.m2.1.1"><csymbol cd="ambiguous" id="S3.SS2.p4.2.m2.1.1.1.cmml" xref="S3.SS2.p4.2.m2.1.1">subscript</csymbol><ci id="S3.SS2.p4.2.m2.1.1.2.cmml" xref="S3.SS2.p4.2.m2.1.1.2">𝜋</ci><ci id="S3.SS2.p4.2.m2.1.1.3a.cmml" xref="S3.SS2.p4.2.m2.1.1.3"><mtext id="S3.SS2.p4.2.m2.1.1.3.cmml" mathsize="70%" xref="S3.SS2.p4.2.m2.1.1.3">RL</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS2.p4.2.m2.1c">\pi_{\text{RL}}</annotation><annotation encoding="application/x-llamapun" id="S3.SS2.p4.2.m2.1d">italic_π start_POSTSUBSCRIPT RL end_POSTSUBSCRIPT</annotation></semantics></math> is initialized with <math alttext="\pi_{\text{SFT}}" class="ltx_Math" display="inline" id="S3.SS2.p4.3.m3.1"><semantics id="S3.SS2.p4.3.m3.1a"><msub id="S3.SS2.p4.3.m3.1.1" xref="S3.SS2.p4.3.m3.1.1.cmml"><mi id="S3.SS2.p4.3.m3.1.1.2" xref="S3.SS2.p4.3.m3.1.1.2.cmml">π</mi><mtext id="S3.SS2.p4.3.m3.1.1.3" xref="S3.SS2.p4.3.m3.1.1.3a.cmml">SFT</mtext></msub><annotation-xml encoding="MathML-Content" id="S3.SS2.p4.3.m3.1b"><apply id="S3.SS2.p4.3.m3.1.1.cmml" xref="S3.SS2.p4.3.m3.1.1"><csymbol cd="ambiguous" id="S3.SS2.p4.3.m3.1.1.1.cmml" xref="S3.SS2.p4.3.m3.1.1">subscript</csymbol><ci id="S3.SS2.p4.3.m3.1.1.2.cmml" xref="S3.SS2.p4.3.m3.1.1.2">𝜋</ci><ci id="S3.SS2.p4.3.m3.1.1.3a.cmml" xref="S3.SS2.p4.3.m3.1.1.3"><mtext id="S3.SS2.p4.3.m3.1.1.3.cmml" mathsize="70%" xref="S3.SS2.p4.3.m3.1.1.3">SFT</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S3.SS2.p4.3.m3.1c">\pi_{\text{SFT}}</annotation><annotation encoding="application/x-llamapun" id="S3.SS2.p4.3.m3.1d">italic_π start_POSTSUBSCRIPT SFT end_POSTSUBSCRIPT</annotation></semantics></math>. In addition, we combine per-token KL penalty from the supervised fine-tuned model at each token following prior studies <cite class="ltx_cite ltx_citemacro_citep">(Ouyang et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib23" title="">2022</a>)</cite>.</p>
</div>
</section>
</section>
<section class="ltx_section" id="S4">
<h2 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">4 </span>Experiment</h2>
<div class="ltx_para ltx_noindent" id="S4.p1">
<p class="ltx_p" id="S4.p1.1">In this section, we validate our supervised fine-tuning and reinforcement learning methods for generating diverse molecules, coined Div-SFT and Div-SFT+RL, respectively. In these experiments, one can observe that (1) our fine-tuning approach enables LLMs to better discover diverse molecules compared to decoding schemes for diversified generation and (2) our fine-tuned LLM outperforms other representative LLMs, including generalist and specialist models for chemical tasks.</p>
</div>
<section class="ltx_subsection" id="S4.SS1">
<h3 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">4.1 </span>Comparison with decoding schemes</h3>
<div class="ltx_para ltx_noindent" id="S4.SS1.p1">
<p class="ltx_p" id="S4.SS1.p1.1">In this experiment, we show that our method enables LLMs to better generate diverse molecules compared to the existing decoding schemes for diverse sequence generation. To validate the consistent improvement, we implement our fine-tuning method and decoding schemes on two models specialized in the chemical domain: BioT5<sup class="ltx_sup" id="S4.SS1.p1.1.1"><span class="ltx_text ltx_font_italic" id="S4.SS1.p1.1.1.1">+</span></sup> <cite class="ltx_cite ltx_citemacro_citep">(Pei et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib26" title="">2024</a>)</cite> and MolT5 <cite class="ltx_cite ltx_citemacro_citep">(Edwards et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib7" title="">2022</a>)</cite>.</p>
</div>
<figure class="ltx_figure" id="S4.F5">
<div class="ltx_flex_figure">
<div class="ltx_flex_cell ltx_flex_size_1">
<figure class="ltx_figure ltx_figure_panel ltx_align_center" id="S4.F5.sf1"><img alt="Refer to caption" class="ltx_graphics ltx_img_landscape" height="270" id="S4.F5.sf1.g1" src="x8.png" width="822"/>
<figcaption class="ltx_caption"><span class="ltx_tag ltx_tag_figure">(a) </span>Comparison with decoding schemes on BioT5<sup class="ltx_sup" id="S4.F5.sf1.4.1"><span class="ltx_text ltx_font_italic" id="S4.F5.sf1.4.1.1">+</span></sup></figcaption>
</figure>
</div>
<div class="ltx_flex_break"></div>
<div class="ltx_flex_cell ltx_flex_size_1">
<figure class="ltx_figure ltx_figure_panel ltx_align_center" id="S4.F5.sf2"><img alt="Refer to caption" class="ltx_graphics ltx_img_landscape" height="270" id="S4.F5.sf2.g1" src="x9.png" width="822"/>
<figcaption class="ltx_caption"><span class="ltx_tag ltx_tag_figure">(b) </span>Comparison with decoding schemes on MolT5</figcaption>
</figure>
</div>
</div>
<figcaption class="ltx_caption ltx_centering"><span class="ltx_tag ltx_tag_figure">Figure 5: </span><span class="ltx_text ltx_font_bold" id="S4.F5.5.1">Comparison with decoding schemes.</span> <span class="ltx_text ltx_font_bold" id="S4.F5.6.2">NCircles</span> represents both quality and diversity-related metric. <span class="ltx_text ltx_font_bold" id="S4.F5.7.3">Top 10 and Accepted &amp; unique</span> represent quality-related metrics. <span class="ltx_text ltx_font_bold" id="S4.F5.8.4">IntDiv.</span> represents a diversity-related metric. Our method generates more diverse and high-quality molecules compared to the baselines. Notably, our method makes a larger gap over the baselines on metrics related to capturing both quality and diversity, i.e., NCircles.</figcaption>
</figure>
<div class="ltx_para ltx_noindent" id="S4.SS1.p2">
<p class="ltx_p" id="S4.SS1.p2.1"><span class="ltx_text ltx_font_bold" id="S4.SS1.p2.1.1">Tasks.</span> We consider description-guided molecule generation using the ChEBI-20 dataset, which includes training and test sets <cite class="ltx_cite ltx_citemacro_citep">(Edwards et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib6" title="">2021</a>)</cite>. Note that the training dataset has also been used to pre-train the base LLMs, BioT5<sup class="ltx_sup" id="S4.SS1.p2.1.2"><span class="ltx_text ltx_font_italic" id="S4.SS1.p2.1.2.1">+</span></sup> and MolT5, i.e., not an external dataset. The test dataset is unobserved during both the pre-training and fine-tuning phases. Each data point consists of a pair of a description and a ground-truth example molecule that satisfies the description.</p>
</div>
<div class="ltx_para ltx_noindent" id="S4.SS1.p3">
<p class="ltx_p" id="S4.SS1.p3.2">In our experiments, we consider generating <math alttext="50" class="ltx_Math" display="inline" id="S4.SS1.p3.1.m1.1"><semantics id="S4.SS1.p3.1.m1.1a"><mn id="S4.SS1.p3.1.m1.1.1" xref="S4.SS1.p3.1.m1.1.1.cmml">50</mn><annotation-xml encoding="MathML-Content" id="S4.SS1.p3.1.m1.1b"><cn id="S4.SS1.p3.1.m1.1.1.cmml" type="integer" xref="S4.SS1.p3.1.m1.1.1">50</cn></annotation-xml><annotation encoding="application/x-tex" id="S4.SS1.p3.1.m1.1c">50</annotation><annotation encoding="application/x-llamapun" id="S4.SS1.p3.1.m1.1d">50</annotation></semantics></math> molecules for each description in the ChEBI-20 test dataset involving <math alttext="3300" class="ltx_Math" display="inline" id="S4.SS1.p3.2.m2.1"><semantics id="S4.SS1.p3.2.m2.1a"><mn id="S4.SS1.p3.2.m2.1.1" xref="S4.SS1.p3.2.m2.1.1.cmml">3300</mn><annotation-xml encoding="MathML-Content" id="S4.SS1.p3.2.m2.1b"><cn id="S4.SS1.p3.2.m2.1.1.cmml" type="integer" xref="S4.SS1.p3.2.m2.1.1">3300</cn></annotation-xml><annotation encoding="application/x-tex" id="S4.SS1.p3.2.m2.1c">3300</annotation><annotation encoding="application/x-llamapun" id="S4.SS1.p3.2.m2.1d">3300</annotation></semantics></math> molecular descriptions. We then evaluate both the structural diversity of the generated molecules and how well they satisfy the given descriptions.</p>
</div>
<div class="ltx_para ltx_noindent" id="S4.SS1.p4">
<p class="ltx_p" id="S4.SS1.p4.1"><span class="ltx_text ltx_font_bold" id="S4.SS1.p4.1.1">Metrics.</span> To measure the similarity between two molecules, we use Tanimoto similarity <cite class="ltx_cite ltx_citemacro_citep">(Bajusz et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib2" title="">2015</a>)</cite>, which based on distance of molecular structures from fingerprints. To be specific, we consider the following metrics for generated molecules:</p>
<ul class="ltx_itemize" id="S4.I1">
<li class="ltx_item" id="S4.I1.i1" style="list-style-type:none;">
<span class="ltx_tag ltx_tag_item">•</span>
<div class="ltx_para" id="S4.I1.i1.p1">
<p class="ltx_p" id="S4.I1.i1.p1.1">The number of accepted and unique molecules (Accepted &amp; unique): This metric measures the number of valid and unique molecules satisfying the given description.<span class="ltx_note ltx_role_footnote" id="footnote3"><sup class="ltx_note_mark">3</sup><span class="ltx_note_outer"><span class="ltx_note_content"><sup class="ltx_note_mark">3</sup><span class="ltx_tag ltx_tag_note">3</span>Following the evaluation metric of prior studies <cite class="ltx_cite ltx_citemacro_citep">(Edwards et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib7" title="">2022</a>; Pei et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib26" title="">2024</a>)</cite>, we evaluate the molecule as satisfying the given description when its BLEU score between a ground-truth sample is high (<math alttext="&gt;0.7" class="ltx_Math" display="inline" id="footnote3.m1.1"><semantics id="footnote3.m1.1b"><mrow id="footnote3.m1.1.1" xref="footnote3.m1.1.1.cmml"><mi id="footnote3.m1.1.1.2" xref="footnote3.m1.1.1.2.cmml"></mi><mo id="footnote3.m1.1.1.1" xref="footnote3.m1.1.1.1.cmml">&gt;</mo><mn id="footnote3.m1.1.1.3" xref="footnote3.m1.1.1.3.cmml">0.7</mn></mrow><annotation-xml encoding="MathML-Content" id="footnote3.m1.1c"><apply id="footnote3.m1.1.1.cmml" xref="footnote3.m1.1.1"><gt id="footnote3.m1.1.1.1.cmml" xref="footnote3.m1.1.1.1"></gt><csymbol cd="latexml" id="footnote3.m1.1.1.2.cmml" xref="footnote3.m1.1.1.2">absent</csymbol><cn id="footnote3.m1.1.1.3.cmml" type="float" xref="footnote3.m1.1.1.3">0.7</cn></apply></annotation-xml><annotation encoding="application/x-tex" id="footnote3.m1.1d">&gt;0.7</annotation><annotation encoding="application/x-llamapun" id="footnote3.m1.1e">&gt; 0.7</annotation></semantics></math>).</span></span></span></p>
</div>
</li>
<li class="ltx_item" id="S4.I1.i2" style="list-style-type:none;">
<span class="ltx_tag ltx_tag_item">•</span>
<div class="ltx_para" id="S4.I1.i2.p1">
<p class="ltx_p" id="S4.I1.i2.p1.1">The number of circles <cite class="ltx_cite ltx_citemacro_citep">(NCircles; Xie et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib40" title="">2023</a>)</cite>: This metric considers both the quality and diversity of generated molecules. For the accepted molecules satisfying the given description, this metric counts the number of maximum distinct clusters with Tanimoto similarity <cite class="ltx_cite ltx_citemacro_citep">(Bajusz et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib2" title="">2015</a>)</cite> below a threshold <math alttext="h" class="ltx_Math" display="inline" id="S4.I1.i2.p1.1.m1.1"><semantics id="S4.I1.i2.p1.1.m1.1a"><mi id="S4.I1.i2.p1.1.m1.1.1" xref="S4.I1.i2.p1.1.m1.1.1.cmml">h</mi><annotation-xml encoding="MathML-Content" id="S4.I1.i2.p1.1.m1.1b"><ci id="S4.I1.i2.p1.1.m1.1.1.cmml" xref="S4.I1.i2.p1.1.m1.1.1">ℎ</ci></annotation-xml><annotation encoding="application/x-tex" id="S4.I1.i2.p1.1.m1.1c">h</annotation><annotation encoding="application/x-llamapun" id="S4.I1.i2.p1.1.m1.1d">italic_h</annotation></semantics></math>, i.e., indicating chemical space coverage (illustrated in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.F6" title="In 4.1 Comparison with decoding schemes ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Figure</span> <span class="ltx_text ltx_ref_tag">6</span></a>).</p>
</div>
</li>
<li class="ltx_item" id="S4.I1.i3" style="list-style-type:none;">
<span class="ltx_tag ltx_tag_item">•</span>
<div class="ltx_para" id="S4.I1.i3.p1">
<p class="ltx_p" id="S4.I1.i3.p1.1">Internal diversity <cite class="ltx_cite ltx_citemacro_citep">(IntDiv.; Polykovskiy et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib27" title="">2020</a>)</cite>: This metric is the complement of the average of pair-wise Tanimoto similarities between the accepted molecules satisfying the given description.</p>
</div>
</li>
<li class="ltx_item" id="S4.I1.i4" style="list-style-type:none;">
<span class="ltx_tag ltx_tag_item">•</span>
<div class="ltx_para ltx_noindent" id="S4.I1.i4.p1">
<p class="ltx_p" id="S4.I1.i4.p1.1">Average of top 10 scores (Top 10): This metric measures the quality of the generated molecules by averaging the BLEU scores <cite class="ltx_cite ltx_citemacro_citep">(Papineni et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib24" title="">2002</a>)</cite> of the top unique <math alttext="10" class="ltx_Math" display="inline" id="S4.I1.i4.p1.1.m1.1"><semantics id="S4.I1.i4.p1.1.m1.1a"><mn id="S4.I1.i4.p1.1.m1.1.1" xref="S4.I1.i4.p1.1.m1.1.1.cmml">10</mn><annotation-xml encoding="MathML-Content" id="S4.I1.i4.p1.1.m1.1b"><cn id="S4.I1.i4.p1.1.m1.1.1.cmml" type="integer" xref="S4.I1.i4.p1.1.m1.1.1">10</cn></annotation-xml><annotation encoding="application/x-tex" id="S4.I1.i4.p1.1.m1.1c">10</annotation><annotation encoding="application/x-llamapun" id="S4.I1.i4.p1.1.m1.1d">10</annotation></semantics></math> molecules yielding high BLEU scores between the ground-truth example.</p>
</div>
</li>
</ul>
</div>
<figure class="ltx_figure" id="S4.F6">
<div class="ltx_flex_figure">
<div class="ltx_flex_cell ltx_flex_size_4">
<figure class="ltx_figure ltx_figure_panel ltx_align_center" id="S4.F6.sf1"><img alt="Refer to caption" class="ltx_graphics ltx_centering ltx_img_square" height="830" id="S4.F6.sf1.g1" src="x10.png" width="830"/>
<figcaption class="ltx_caption ltx_centering"><span class="ltx_tag ltx_tag_figure">(a) </span>BS 
<br class="ltx_break"/>(22,11)</figcaption>
</figure>
</div>
<div class="ltx_flex_cell ltx_flex_size_4">
<figure class="ltx_figure ltx_figure_panel ltx_align_center" id="S4.F6.sf2"><img alt="Refer to caption" class="ltx_graphics ltx_centering ltx_img_square" height="830" id="S4.F6.sf2.g1" src="x11.png" width="830"/>
<figcaption class="ltx_caption ltx_centering"><span class="ltx_tag ltx_tag_figure">(b) </span>Diverse BS 
<br class="ltx_break"/>(7,4)</figcaption>
</figure>
</div>
<div class="ltx_flex_cell ltx_flex_size_4">
<figure class="ltx_figure ltx_figure_panel ltx_align_center" id="S4.F6.sf3"><img alt="Refer to caption" class="ltx_graphics ltx_centering ltx_img_square" height="830" id="S4.F6.sf3.g1" src="x12.png" width="830"/>
<figcaption class="ltx_caption ltx_centering"><span class="ltx_tag ltx_tag_figure">(c) </span>Contrastive BS
<br class="ltx_break"/>(26, 11)</figcaption>
</figure>
</div>
<div class="ltx_flex_cell ltx_flex_size_4">
<figure class="ltx_figure ltx_figure_panel ltx_align_center" id="S4.F6.sf4"><img alt="Refer to caption" class="ltx_graphics ltx_centering ltx_img_square" height="830" id="S4.F6.sf4.g1" src="x13.png" width="830"/>
<figcaption class="ltx_caption ltx_centering"><span class="ltx_tag ltx_tag_figure">(d) </span>Div-SFT+RL
<br class="ltx_break"/>(29,28)</figcaption>
</figure>
</div>
</div>
<figcaption class="ltx_caption ltx_centering"><span class="ltx_tag ltx_tag_figure">Figure 6: </span><span class="ltx_text ltx_font_bold" id="S4.F6.3.1">The distribution of (Accepted &amp; unique, NCircles<sub class="ltx_sub" id="S4.F6.3.1.1"><span class="ltx_text ltx_font_medium ltx_font_italic" id="S4.F6.3.1.1.1">h=0.65</span></sub>) obtained by each method.</span> The molecules are generated from the description in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#A3.T13" title="In Appendix C Additional results ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Table</span> <span class="ltx_text ltx_ref_tag">13</span></a>. The Fruchterman-Reingold force-directed algorithm <cite class="ltx_cite ltx_citemacro_citep">(Fruchterman &amp; Reingold, <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib9" title="">1991</a>)</cite> embeds each molecule in a 2-dimensional space based on Tanimoto similarity. Here, a low distance yields a high Tanimoto similarity. Each dot, shaded circle, and edge represent accepted molecules, distinct clusters (used to measure NCircles), and a pair with similarity above <math alttext="0.65" class="ltx_Math" display="inline" id="S4.F6.4.m1.1"><semantics id="S4.F6.4.m1.1b"><mn id="S4.F6.4.m1.1.1" xref="S4.F6.4.m1.1.1.cmml">0.65</mn><annotation-xml encoding="MathML-Content" id="S4.F6.4.m1.1c"><cn id="S4.F6.4.m1.1.1.cmml" type="float" xref="S4.F6.4.m1.1.1">0.65</cn></annotation-xml><annotation encoding="application/x-tex" id="S4.F6.4.m1.1d">0.65</annotation><annotation encoding="application/x-llamapun" id="S4.F6.4.m1.1e">0.65</annotation></semantics></math>, respectively. Our method better captures the molecular diversity.</figcaption>
</figure>
<figure class="ltx_table" id="S4.T1">
<figcaption class="ltx_caption"><span class="ltx_tag ltx_tag_table">Table 1: </span><span class="ltx_text ltx_font_bold" id="S4.T1.7.1">Visualization of the generated molecules with their diversity.</span> The eight molecules are generated with the description: “The molecule is a primary aliphatic ammonium ion which is obtained from streptothricin F by protonation of the guanidino and amino groups. It has a role as an antimicrobial agent. It is a guanidinium ion and a primary aliphatic ammonium ion. It is a conjugate acid of a streptothricin F’.” The generated molecule with <span class="ltx_text" id="S4.T1.8.2" style="color:#0000FF;">blue line</span> indicates the accepted molecule.</figcaption>
<table class="ltx_tabular ltx_align_middle" id="S4.T1.4">
<tr class="ltx_tr" id="S4.T1.4.5">
<td class="ltx_td ltx_align_left ltx_border_tt" id="S4.T1.4.5.1">
<span class="ltx_text" id="S4.T1.4.5.1.1"></span><span class="ltx_text" id="S4.T1.4.5.1.2">
<span class="ltx_tabular ltx_align_middle" id="S4.T1.4.5.1.2.1">
<span class="ltx_tr" id="S4.T1.4.5.1.2.1.1">
<span class="ltx_td ltx_nopad_r ltx_align_left" id="S4.T1.4.5.1.2.1.1.1">Method</span></span>
<span class="ltx_tr" id="S4.T1.4.5.1.2.1.2">
<span class="ltx_td ltx_nopad_r ltx_align_left" id="S4.T1.4.5.1.2.1.2.1">(IntDiv.)</span></span>
</span></span><span class="ltx_text" id="S4.T1.4.5.1.3"></span></td>
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_tt" id="S4.T1.4.5.2" style="width:314.0pt;">
<span class="ltx_inline-block ltx_align_top" id="S4.T1.4.5.2.1">
<span class="ltx_p" id="S4.T1.4.5.2.1.1">Example of outputs</span>
</span>
</td>
</tr>
<tr class="ltx_tr" id="S4.T1.1.1">
<td class="ltx_td ltx_align_left ltx_border_t" id="S4.T1.1.1.2">
<span class="ltx_text" id="S4.T1.1.1.2.1"></span><span class="ltx_text" id="S4.T1.1.1.2.2">
<span class="ltx_tabular ltx_align_middle" id="S4.T1.1.1.2.2.1">
<span class="ltx_tr" id="S4.T1.1.1.2.2.1.1">
<span class="ltx_td ltx_nopad_r ltx_align_left" id="S4.T1.1.1.2.2.1.1.1">BS</span></span>
<span class="ltx_tr" id="S4.T1.1.1.2.2.1.2">
<span class="ltx_td ltx_nopad_r ltx_align_left" id="S4.T1.1.1.2.2.1.2.1">(0.51)</span></span>
</span></span><span class="ltx_text" id="S4.T1.1.1.2.3"></span></td>
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_t" id="S4.T1.1.1.1" style="width:314.0pt;">
<span class="ltx_inline-block ltx_align_top" id="S4.T1.1.1.1.1"><img alt="[Uncaptioned image]" class="ltx_graphics ltx_img_landscape" height="58" id="S4.T1.1.1.1.1.g1" src="x14.png" width="601"/>
</span>
</td>
</tr>
<tr class="ltx_tr" id="S4.T1.2.2">
<td class="ltx_td ltx_align_left ltx_border_t" id="S4.T1.2.2.2">
<span class="ltx_text" id="S4.T1.2.2.2.1"></span><span class="ltx_text" id="S4.T1.2.2.2.2">
<span class="ltx_tabular ltx_align_middle" id="S4.T1.2.2.2.2.1">
<span class="ltx_tr" id="S4.T1.2.2.2.2.1.1">
<span class="ltx_td ltx_nopad_r ltx_align_left" id="S4.T1.2.2.2.2.1.1.1">Diverse BS</span></span>
<span class="ltx_tr" id="S4.T1.2.2.2.2.1.2">
<span class="ltx_td ltx_nopad_r ltx_align_left" id="S4.T1.2.2.2.2.1.2.1">(0.28)</span></span>
</span></span><span class="ltx_text" id="S4.T1.2.2.2.3"></span></td>
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_t" id="S4.T1.2.2.1" style="width:314.0pt;">
<span class="ltx_inline-block ltx_align_top" id="S4.T1.2.2.1.1"><img alt="[Uncaptioned image]" class="ltx_graphics ltx_img_landscape" height="57" id="S4.T1.2.2.1.1.g1" src="x15.png" width="601"/>
</span>
</td>
</tr>
<tr class="ltx_tr" id="S4.T1.3.3">
<td class="ltx_td ltx_align_left ltx_border_t" id="S4.T1.3.3.2">
<span class="ltx_text" id="S4.T1.3.3.2.1"></span><span class="ltx_text" id="S4.T1.3.3.2.2">
<span class="ltx_tabular ltx_align_middle" id="S4.T1.3.3.2.2.1">
<span class="ltx_tr" id="S4.T1.3.3.2.2.1.1">
<span class="ltx_td ltx_nopad_r ltx_align_left" id="S4.T1.3.3.2.2.1.1.1">Contrastive BS</span></span>
<span class="ltx_tr" id="S4.T1.3.3.2.2.1.2">
<span class="ltx_td ltx_nopad_r ltx_align_left" id="S4.T1.3.3.2.2.1.2.1">(0.53)</span></span>
</span></span><span class="ltx_text" id="S4.T1.3.3.2.3"></span></td>
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_t" id="S4.T1.3.3.1" style="width:314.0pt;">
<span class="ltx_inline-block ltx_align_top" id="S4.T1.3.3.1.1"><img alt="[Uncaptioned image]" class="ltx_graphics ltx_img_landscape" height="58" id="S4.T1.3.3.1.1.g1" src="x16.png" width="601"/>
</span>
</td>
</tr>
<tr class="ltx_tr" id="S4.T1.4.4">
<td class="ltx_td ltx_align_left ltx_border_bb ltx_border_t" id="S4.T1.4.4.2">
<span class="ltx_text" id="S4.T1.4.4.2.1"></span><span class="ltx_text" id="S4.T1.4.4.2.2">
<span class="ltx_tabular ltx_align_middle" id="S4.T1.4.4.2.2.1">
<span class="ltx_tr" id="S4.T1.4.4.2.2.1.1">
<span class="ltx_td ltx_nopad_r ltx_align_left" id="S4.T1.4.4.2.2.1.1.1">Div-SFT+RL</span></span>
<span class="ltx_tr" id="S4.T1.4.4.2.2.1.2">
<span class="ltx_td ltx_nopad_r ltx_align_left" id="S4.T1.4.4.2.2.1.2.1">(0.69)</span></span>
</span></span><span class="ltx_text" id="S4.T1.4.4.2.3"></span></td>
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_bb ltx_border_t" id="S4.T1.4.4.1" style="width:314.0pt;">
<span class="ltx_inline-block ltx_align_top" id="S4.T1.4.4.1.1"><img alt="[Uncaptioned image]" class="ltx_graphics ltx_img_landscape" height="58" id="S4.T1.4.4.1.1.g1" src="x17.png" width="601"/>
</span>
</td>
</tr>
</table>
</figure>
<div class="ltx_para ltx_noindent" id="S4.SS1.p5">
<p class="ltx_p" id="S4.SS1.p5.4"><span class="ltx_text ltx_font_bold" id="S4.SS1.p5.4.1">Implementations.</span> For supervised fine-tuning, we collect a hundred molecules for each molecular description <math alttext="p_{\text{desc}}" class="ltx_Math" display="inline" id="S4.SS1.p5.1.m1.1"><semantics id="S4.SS1.p5.1.m1.1a"><msub id="S4.SS1.p5.1.m1.1.1" xref="S4.SS1.p5.1.m1.1.1.cmml"><mi id="S4.SS1.p5.1.m1.1.1.2" xref="S4.SS1.p5.1.m1.1.1.2.cmml">p</mi><mtext id="S4.SS1.p5.1.m1.1.1.3" xref="S4.SS1.p5.1.m1.1.1.3a.cmml">desc</mtext></msub><annotation-xml encoding="MathML-Content" id="S4.SS1.p5.1.m1.1b"><apply id="S4.SS1.p5.1.m1.1.1.cmml" xref="S4.SS1.p5.1.m1.1.1"><csymbol cd="ambiguous" id="S4.SS1.p5.1.m1.1.1.1.cmml" xref="S4.SS1.p5.1.m1.1.1">subscript</csymbol><ci id="S4.SS1.p5.1.m1.1.1.2.cmml" xref="S4.SS1.p5.1.m1.1.1.2">𝑝</ci><ci id="S4.SS1.p5.1.m1.1.1.3a.cmml" xref="S4.SS1.p5.1.m1.1.1.3"><mtext id="S4.SS1.p5.1.m1.1.1.3.cmml" mathsize="70%" xref="S4.SS1.p5.1.m1.1.1.3">desc</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S4.SS1.p5.1.m1.1c">p_{\text{desc}}</annotation><annotation encoding="application/x-llamapun" id="S4.SS1.p5.1.m1.1d">italic_p start_POSTSUBSCRIPT desc end_POSTSUBSCRIPT</annotation></semantics></math> in ChEBI-20 training sets. Then, they are filtered to remove invalid molecules, duplicated molecules, and unaccepted molecules. The description-matching reward <math alttext="r_{\text{match}}(m_{k})" class="ltx_Math" display="inline" id="S4.SS1.p5.2.m2.1"><semantics id="S4.SS1.p5.2.m2.1a"><mrow id="S4.SS1.p5.2.m2.1.1" xref="S4.SS1.p5.2.m2.1.1.cmml"><msub id="S4.SS1.p5.2.m2.1.1.3" xref="S4.SS1.p5.2.m2.1.1.3.cmml"><mi id="S4.SS1.p5.2.m2.1.1.3.2" xref="S4.SS1.p5.2.m2.1.1.3.2.cmml">r</mi><mtext id="S4.SS1.p5.2.m2.1.1.3.3" xref="S4.SS1.p5.2.m2.1.1.3.3a.cmml">match</mtext></msub><mo id="S4.SS1.p5.2.m2.1.1.2" xref="S4.SS1.p5.2.m2.1.1.2.cmml">⁢</mo><mrow id="S4.SS1.p5.2.m2.1.1.1.1" xref="S4.SS1.p5.2.m2.1.1.1.1.1.cmml"><mo id="S4.SS1.p5.2.m2.1.1.1.1.2" stretchy="false" xref="S4.SS1.p5.2.m2.1.1.1.1.1.cmml">(</mo><msub id="S4.SS1.p5.2.m2.1.1.1.1.1" xref="S4.SS1.p5.2.m2.1.1.1.1.1.cmml"><mi id="S4.SS1.p5.2.m2.1.1.1.1.1.2" xref="S4.SS1.p5.2.m2.1.1.1.1.1.2.cmml">m</mi><mi id="S4.SS1.p5.2.m2.1.1.1.1.1.3" xref="S4.SS1.p5.2.m2.1.1.1.1.1.3.cmml">k</mi></msub><mo id="S4.SS1.p5.2.m2.1.1.1.1.3" stretchy="false" xref="S4.SS1.p5.2.m2.1.1.1.1.1.cmml">)</mo></mrow></mrow><annotation-xml encoding="MathML-Content" id="S4.SS1.p5.2.m2.1b"><apply id="S4.SS1.p5.2.m2.1.1.cmml" xref="S4.SS1.p5.2.m2.1.1"><times id="S4.SS1.p5.2.m2.1.1.2.cmml" xref="S4.SS1.p5.2.m2.1.1.2"></times><apply id="S4.SS1.p5.2.m2.1.1.3.cmml" xref="S4.SS1.p5.2.m2.1.1.3"><csymbol cd="ambiguous" id="S4.SS1.p5.2.m2.1.1.3.1.cmml" xref="S4.SS1.p5.2.m2.1.1.3">subscript</csymbol><ci id="S4.SS1.p5.2.m2.1.1.3.2.cmml" xref="S4.SS1.p5.2.m2.1.1.3.2">𝑟</ci><ci id="S4.SS1.p5.2.m2.1.1.3.3a.cmml" xref="S4.SS1.p5.2.m2.1.1.3.3"><mtext id="S4.SS1.p5.2.m2.1.1.3.3.cmml" mathsize="70%" xref="S4.SS1.p5.2.m2.1.1.3.3">match</mtext></ci></apply><apply id="S4.SS1.p5.2.m2.1.1.1.1.1.cmml" xref="S4.SS1.p5.2.m2.1.1.1.1"><csymbol cd="ambiguous" id="S4.SS1.p5.2.m2.1.1.1.1.1.1.cmml" xref="S4.SS1.p5.2.m2.1.1.1.1">subscript</csymbol><ci id="S4.SS1.p5.2.m2.1.1.1.1.1.2.cmml" xref="S4.SS1.p5.2.m2.1.1.1.1.1.2">𝑚</ci><ci id="S4.SS1.p5.2.m2.1.1.1.1.1.3.cmml" xref="S4.SS1.p5.2.m2.1.1.1.1.1.3">𝑘</ci></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="S4.SS1.p5.2.m2.1c">r_{\text{match}}(m_{k})</annotation><annotation encoding="application/x-llamapun" id="S4.SS1.p5.2.m2.1d">italic_r start_POSTSUBSCRIPT match end_POSTSUBSCRIPT ( italic_m start_POSTSUBSCRIPT italic_k end_POSTSUBSCRIPT )</annotation></semantics></math> is defined as a BLEU score between a ground-truth example. The diversity reward <math alttext="r_{\text{div}}(m_{k},\{m_{i}\}^{k-1}_{i=1})" class="ltx_Math" display="inline" id="S4.SS1.p5.3.m3.2"><semantics id="S4.SS1.p5.3.m3.2a"><mrow id="S4.SS1.p5.3.m3.2.2" xref="S4.SS1.p5.3.m3.2.2.cmml"><msub id="S4.SS1.p5.3.m3.2.2.4" xref="S4.SS1.p5.3.m3.2.2.4.cmml"><mi id="S4.SS1.p5.3.m3.2.2.4.2" xref="S4.SS1.p5.3.m3.2.2.4.2.cmml">r</mi><mtext id="S4.SS1.p5.3.m3.2.2.4.3" xref="S4.SS1.p5.3.m3.2.2.4.3a.cmml">div</mtext></msub><mo id="S4.SS1.p5.3.m3.2.2.3" xref="S4.SS1.p5.3.m3.2.2.3.cmml">⁢</mo><mrow id="S4.SS1.p5.3.m3.2.2.2.2" xref="S4.SS1.p5.3.m3.2.2.2.3.cmml"><mo id="S4.SS1.p5.3.m3.2.2.2.2.3" stretchy="false" xref="S4.SS1.p5.3.m3.2.2.2.3.cmml">(</mo><msub id="S4.SS1.p5.3.m3.1.1.1.1.1" xref="S4.SS1.p5.3.m3.1.1.1.1.1.cmml"><mi id="S4.SS1.p5.3.m3.1.1.1.1.1.2" xref="S4.SS1.p5.3.m3.1.1.1.1.1.2.cmml">m</mi><mi id="S4.SS1.p5.3.m3.1.1.1.1.1.3" xref="S4.SS1.p5.3.m3.1.1.1.1.1.3.cmml">k</mi></msub><mo id="S4.SS1.p5.3.m3.2.2.2.2.4" xref="S4.SS1.p5.3.m3.2.2.2.3.cmml">,</mo><msubsup id="S4.SS1.p5.3.m3.2.2.2.2.2" xref="S4.SS1.p5.3.m3.2.2.2.2.2.cmml"><mrow id="S4.SS1.p5.3.m3.2.2.2.2.2.1.1.1" xref="S4.SS1.p5.3.m3.2.2.2.2.2.1.1.2.cmml"><mo id="S4.SS1.p5.3.m3.2.2.2.2.2.1.1.1.2" stretchy="false" xref="S4.SS1.p5.3.m3.2.2.2.2.2.1.1.2.cmml">{</mo><msub id="S4.SS1.p5.3.m3.2.2.2.2.2.1.1.1.1" xref="S4.SS1.p5.3.m3.2.2.2.2.2.1.1.1.1.cmml"><mi id="S4.SS1.p5.3.m3.2.2.2.2.2.1.1.1.1.2" xref="S4.SS1.p5.3.m3.2.2.2.2.2.1.1.1.1.2.cmml">m</mi><mi id="S4.SS1.p5.3.m3.2.2.2.2.2.1.1.1.1.3" xref="S4.SS1.p5.3.m3.2.2.2.2.2.1.1.1.1.3.cmml">i</mi></msub><mo id="S4.SS1.p5.3.m3.2.2.2.2.2.1.1.1.3" stretchy="false" xref="S4.SS1.p5.3.m3.2.2.2.2.2.1.1.2.cmml">}</mo></mrow><mrow id="S4.SS1.p5.3.m3.2.2.2.2.2.3" xref="S4.SS1.p5.3.m3.2.2.2.2.2.3.cmml"><mi id="S4.SS1.p5.3.m3.2.2.2.2.2.3.2" xref="S4.SS1.p5.3.m3.2.2.2.2.2.3.2.cmml">i</mi><mo id="S4.SS1.p5.3.m3.2.2.2.2.2.3.1" xref="S4.SS1.p5.3.m3.2.2.2.2.2.3.1.cmml">=</mo><mn id="S4.SS1.p5.3.m3.2.2.2.2.2.3.3" xref="S4.SS1.p5.3.m3.2.2.2.2.2.3.3.cmml">1</mn></mrow><mrow id="S4.SS1.p5.3.m3.2.2.2.2.2.1.3" xref="S4.SS1.p5.3.m3.2.2.2.2.2.1.3.cmml"><mi id="S4.SS1.p5.3.m3.2.2.2.2.2.1.3.2" xref="S4.SS1.p5.3.m3.2.2.2.2.2.1.3.2.cmml">k</mi><mo id="S4.SS1.p5.3.m3.2.2.2.2.2.1.3.1" xref="S4.SS1.p5.3.m3.2.2.2.2.2.1.3.1.cmml">−</mo><mn id="S4.SS1.p5.3.m3.2.2.2.2.2.1.3.3" xref="S4.SS1.p5.3.m3.2.2.2.2.2.1.3.3.cmml">1</mn></mrow></msubsup><mo id="S4.SS1.p5.3.m3.2.2.2.2.5" stretchy="false" xref="S4.SS1.p5.3.m3.2.2.2.3.cmml">)</mo></mrow></mrow><annotation-xml encoding="MathML-Content" id="S4.SS1.p5.3.m3.2b"><apply id="S4.SS1.p5.3.m3.2.2.cmml" xref="S4.SS1.p5.3.m3.2.2"><times id="S4.SS1.p5.3.m3.2.2.3.cmml" xref="S4.SS1.p5.3.m3.2.2.3"></times><apply id="S4.SS1.p5.3.m3.2.2.4.cmml" xref="S4.SS1.p5.3.m3.2.2.4"><csymbol cd="ambiguous" id="S4.SS1.p5.3.m3.2.2.4.1.cmml" xref="S4.SS1.p5.3.m3.2.2.4">subscript</csymbol><ci id="S4.SS1.p5.3.m3.2.2.4.2.cmml" xref="S4.SS1.p5.3.m3.2.2.4.2">𝑟</ci><ci id="S4.SS1.p5.3.m3.2.2.4.3a.cmml" xref="S4.SS1.p5.3.m3.2.2.4.3"><mtext id="S4.SS1.p5.3.m3.2.2.4.3.cmml" mathsize="70%" xref="S4.SS1.p5.3.m3.2.2.4.3">div</mtext></ci></apply><interval closure="open" id="S4.SS1.p5.3.m3.2.2.2.3.cmml" xref="S4.SS1.p5.3.m3.2.2.2.2"><apply id="S4.SS1.p5.3.m3.1.1.1.1.1.cmml" xref="S4.SS1.p5.3.m3.1.1.1.1.1"><csymbol cd="ambiguous" id="S4.SS1.p5.3.m3.1.1.1.1.1.1.cmml" xref="S4.SS1.p5.3.m3.1.1.1.1.1">subscript</csymbol><ci id="S4.SS1.p5.3.m3.1.1.1.1.1.2.cmml" xref="S4.SS1.p5.3.m3.1.1.1.1.1.2">𝑚</ci><ci id="S4.SS1.p5.3.m3.1.1.1.1.1.3.cmml" xref="S4.SS1.p5.3.m3.1.1.1.1.1.3">𝑘</ci></apply><apply id="S4.SS1.p5.3.m3.2.2.2.2.2.cmml" xref="S4.SS1.p5.3.m3.2.2.2.2.2"><csymbol cd="ambiguous" id="S4.SS1.p5.3.m3.2.2.2.2.2.2.cmml" xref="S4.SS1.p5.3.m3.2.2.2.2.2">subscript</csymbol><apply id="S4.SS1.p5.3.m3.2.2.2.2.2.1.cmml" xref="S4.SS1.p5.3.m3.2.2.2.2.2"><csymbol cd="ambiguous" id="S4.SS1.p5.3.m3.2.2.2.2.2.1.2.cmml" xref="S4.SS1.p5.3.m3.2.2.2.2.2">superscript</csymbol><set id="S4.SS1.p5.3.m3.2.2.2.2.2.1.1.2.cmml" xref="S4.SS1.p5.3.m3.2.2.2.2.2.1.1.1"><apply id="S4.SS1.p5.3.m3.2.2.2.2.2.1.1.1.1.cmml" xref="S4.SS1.p5.3.m3.2.2.2.2.2.1.1.1.1"><csymbol cd="ambiguous" id="S4.SS1.p5.3.m3.2.2.2.2.2.1.1.1.1.1.cmml" xref="S4.SS1.p5.3.m3.2.2.2.2.2.1.1.1.1">subscript</csymbol><ci id="S4.SS1.p5.3.m3.2.2.2.2.2.1.1.1.1.2.cmml" xref="S4.SS1.p5.3.m3.2.2.2.2.2.1.1.1.1.2">𝑚</ci><ci id="S4.SS1.p5.3.m3.2.2.2.2.2.1.1.1.1.3.cmml" xref="S4.SS1.p5.3.m3.2.2.2.2.2.1.1.1.1.3">𝑖</ci></apply></set><apply id="S4.SS1.p5.3.m3.2.2.2.2.2.1.3.cmml" xref="S4.SS1.p5.3.m3.2.2.2.2.2.1.3"><minus id="S4.SS1.p5.3.m3.2.2.2.2.2.1.3.1.cmml" xref="S4.SS1.p5.3.m3.2.2.2.2.2.1.3.1"></minus><ci id="S4.SS1.p5.3.m3.2.2.2.2.2.1.3.2.cmml" xref="S4.SS1.p5.3.m3.2.2.2.2.2.1.3.2">𝑘</ci><cn id="S4.SS1.p5.3.m3.2.2.2.2.2.1.3.3.cmml" type="integer" xref="S4.SS1.p5.3.m3.2.2.2.2.2.1.3.3">1</cn></apply></apply><apply id="S4.SS1.p5.3.m3.2.2.2.2.2.3.cmml" xref="S4.SS1.p5.3.m3.2.2.2.2.2.3"><eq id="S4.SS1.p5.3.m3.2.2.2.2.2.3.1.cmml" xref="S4.SS1.p5.3.m3.2.2.2.2.2.3.1"></eq><ci id="S4.SS1.p5.3.m3.2.2.2.2.2.3.2.cmml" xref="S4.SS1.p5.3.m3.2.2.2.2.2.3.2">𝑖</ci><cn id="S4.SS1.p5.3.m3.2.2.2.2.2.3.3.cmml" type="integer" xref="S4.SS1.p5.3.m3.2.2.2.2.2.3.3">1</cn></apply></apply></interval></apply></annotation-xml><annotation encoding="application/x-tex" id="S4.SS1.p5.3.m3.2c">r_{\text{div}}(m_{k},\{m_{i}\}^{k-1}_{i=1})</annotation><annotation encoding="application/x-llamapun" id="S4.SS1.p5.3.m3.2d">italic_r start_POSTSUBSCRIPT div end_POSTSUBSCRIPT ( italic_m start_POSTSUBSCRIPT italic_k end_POSTSUBSCRIPT , { italic_m start_POSTSUBSCRIPT italic_i end_POSTSUBSCRIPT } start_POSTSUPERSCRIPT italic_k - 1 end_POSTSUPERSCRIPT start_POSTSUBSCRIPT italic_i = 1 end_POSTSUBSCRIPT )</annotation></semantics></math> is defined as the complement of the maximum Tanimoto similarity between generated molecules <math alttext="\{m_{i}\}^{k-1}_{i=1}" class="ltx_Math" display="inline" id="S4.SS1.p5.4.m4.1"><semantics id="S4.SS1.p5.4.m4.1a"><msubsup id="S4.SS1.p5.4.m4.1.1" xref="S4.SS1.p5.4.m4.1.1.cmml"><mrow id="S4.SS1.p5.4.m4.1.1.1.1.1" xref="S4.SS1.p5.4.m4.1.1.1.1.2.cmml"><mo id="S4.SS1.p5.4.m4.1.1.1.1.1.2" stretchy="false" xref="S4.SS1.p5.4.m4.1.1.1.1.2.cmml">{</mo><msub id="S4.SS1.p5.4.m4.1.1.1.1.1.1" xref="S4.SS1.p5.4.m4.1.1.1.1.1.1.cmml"><mi id="S4.SS1.p5.4.m4.1.1.1.1.1.1.2" xref="S4.SS1.p5.4.m4.1.1.1.1.1.1.2.cmml">m</mi><mi id="S4.SS1.p5.4.m4.1.1.1.1.1.1.3" xref="S4.SS1.p5.4.m4.1.1.1.1.1.1.3.cmml">i</mi></msub><mo id="S4.SS1.p5.4.m4.1.1.1.1.1.3" stretchy="false" xref="S4.SS1.p5.4.m4.1.1.1.1.2.cmml">}</mo></mrow><mrow id="S4.SS1.p5.4.m4.1.1.3" xref="S4.SS1.p5.4.m4.1.1.3.cmml"><mi id="S4.SS1.p5.4.m4.1.1.3.2" xref="S4.SS1.p5.4.m4.1.1.3.2.cmml">i</mi><mo id="S4.SS1.p5.4.m4.1.1.3.1" xref="S4.SS1.p5.4.m4.1.1.3.1.cmml">=</mo><mn id="S4.SS1.p5.4.m4.1.1.3.3" xref="S4.SS1.p5.4.m4.1.1.3.3.cmml">1</mn></mrow><mrow id="S4.SS1.p5.4.m4.1.1.1.3" xref="S4.SS1.p5.4.m4.1.1.1.3.cmml"><mi id="S4.SS1.p5.4.m4.1.1.1.3.2" xref="S4.SS1.p5.4.m4.1.1.1.3.2.cmml">k</mi><mo id="S4.SS1.p5.4.m4.1.1.1.3.1" xref="S4.SS1.p5.4.m4.1.1.1.3.1.cmml">−</mo><mn id="S4.SS1.p5.4.m4.1.1.1.3.3" xref="S4.SS1.p5.4.m4.1.1.1.3.3.cmml">1</mn></mrow></msubsup><annotation-xml encoding="MathML-Content" id="S4.SS1.p5.4.m4.1b"><apply id="S4.SS1.p5.4.m4.1.1.cmml" xref="S4.SS1.p5.4.m4.1.1"><csymbol cd="ambiguous" id="S4.SS1.p5.4.m4.1.1.2.cmml" xref="S4.SS1.p5.4.m4.1.1">subscript</csymbol><apply id="S4.SS1.p5.4.m4.1.1.1.cmml" xref="S4.SS1.p5.4.m4.1.1"><csymbol cd="ambiguous" id="S4.SS1.p5.4.m4.1.1.1.2.cmml" xref="S4.SS1.p5.4.m4.1.1">superscript</csymbol><set id="S4.SS1.p5.4.m4.1.1.1.1.2.cmml" xref="S4.SS1.p5.4.m4.1.1.1.1.1"><apply id="S4.SS1.p5.4.m4.1.1.1.1.1.1.cmml" xref="S4.SS1.p5.4.m4.1.1.1.1.1.1"><csymbol cd="ambiguous" id="S4.SS1.p5.4.m4.1.1.1.1.1.1.1.cmml" xref="S4.SS1.p5.4.m4.1.1.1.1.1.1">subscript</csymbol><ci id="S4.SS1.p5.4.m4.1.1.1.1.1.1.2.cmml" xref="S4.SS1.p5.4.m4.1.1.1.1.1.1.2">𝑚</ci><ci id="S4.SS1.p5.4.m4.1.1.1.1.1.1.3.cmml" xref="S4.SS1.p5.4.m4.1.1.1.1.1.1.3">𝑖</ci></apply></set><apply id="S4.SS1.p5.4.m4.1.1.1.3.cmml" xref="S4.SS1.p5.4.m4.1.1.1.3"><minus id="S4.SS1.p5.4.m4.1.1.1.3.1.cmml" xref="S4.SS1.p5.4.m4.1.1.1.3.1"></minus><ci id="S4.SS1.p5.4.m4.1.1.1.3.2.cmml" xref="S4.SS1.p5.4.m4.1.1.1.3.2">𝑘</ci><cn id="S4.SS1.p5.4.m4.1.1.1.3.3.cmml" type="integer" xref="S4.SS1.p5.4.m4.1.1.1.3.3">1</cn></apply></apply><apply id="S4.SS1.p5.4.m4.1.1.3.cmml" xref="S4.SS1.p5.4.m4.1.1.3"><eq id="S4.SS1.p5.4.m4.1.1.3.1.cmml" xref="S4.SS1.p5.4.m4.1.1.3.1"></eq><ci id="S4.SS1.p5.4.m4.1.1.3.2.cmml" xref="S4.SS1.p5.4.m4.1.1.3.2">𝑖</ci><cn id="S4.SS1.p5.4.m4.1.1.3.3.cmml" type="integer" xref="S4.SS1.p5.4.m4.1.1.3.3">1</cn></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="S4.SS1.p5.4.m4.1c">\{m_{i}\}^{k-1}_{i=1}</annotation><annotation encoding="application/x-llamapun" id="S4.SS1.p5.4.m4.1d">{ italic_m start_POSTSUBSCRIPT italic_i end_POSTSUBSCRIPT } start_POSTSUPERSCRIPT italic_k - 1 end_POSTSUPERSCRIPT start_POSTSUBSCRIPT italic_i = 1 end_POSTSUBSCRIPT</annotation></semantics></math>. The detailed implementations and hyper-parameters follow <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#A1" title="Appendix A Detailed implementations and training ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Appendix</span> <span class="ltx_text ltx_ref_tag">A</span></a>.</p>
</div>
<div class="ltx_para ltx_noindent" id="S4.SS1.p6">
<p class="ltx_p" id="S4.SS1.p6.1"><span class="ltx_text ltx_font_bold" id="S4.SS1.p6.1.1">Baselines.</span> We compare our fine-tuning approach with various decoding schemes. We consider the random sampling with different temperatures <math alttext="\{0.7,1.0,1.5\}" class="ltx_Math" display="inline" id="S4.SS1.p6.1.m1.3"><semantics id="S4.SS1.p6.1.m1.3a"><mrow id="S4.SS1.p6.1.m1.3.4.2" xref="S4.SS1.p6.1.m1.3.4.1.cmml"><mo id="S4.SS1.p6.1.m1.3.4.2.1" stretchy="false" xref="S4.SS1.p6.1.m1.3.4.1.cmml">{</mo><mn id="S4.SS1.p6.1.m1.1.1" xref="S4.SS1.p6.1.m1.1.1.cmml">0.7</mn><mo id="S4.SS1.p6.1.m1.3.4.2.2" xref="S4.SS1.p6.1.m1.3.4.1.cmml">,</mo><mn id="S4.SS1.p6.1.m1.2.2" xref="S4.SS1.p6.1.m1.2.2.cmml">1.0</mn><mo id="S4.SS1.p6.1.m1.3.4.2.3" xref="S4.SS1.p6.1.m1.3.4.1.cmml">,</mo><mn id="S4.SS1.p6.1.m1.3.3" xref="S4.SS1.p6.1.m1.3.3.cmml">1.5</mn><mo id="S4.SS1.p6.1.m1.3.4.2.4" stretchy="false" xref="S4.SS1.p6.1.m1.3.4.1.cmml">}</mo></mrow><annotation-xml encoding="MathML-Content" id="S4.SS1.p6.1.m1.3b"><set id="S4.SS1.p6.1.m1.3.4.1.cmml" xref="S4.SS1.p6.1.m1.3.4.2"><cn id="S4.SS1.p6.1.m1.1.1.cmml" type="float" xref="S4.SS1.p6.1.m1.1.1">0.7</cn><cn id="S4.SS1.p6.1.m1.2.2.cmml" type="float" xref="S4.SS1.p6.1.m1.2.2">1.0</cn><cn id="S4.SS1.p6.1.m1.3.3.cmml" type="float" xref="S4.SS1.p6.1.m1.3.3">1.5</cn></set></annotation-xml><annotation encoding="application/x-tex" id="S4.SS1.p6.1.m1.3c">\{0.7,1.0,1.5\}</annotation><annotation encoding="application/x-llamapun" id="S4.SS1.p6.1.m1.3d">{ 0.7 , 1.0 , 1.5 }</annotation></semantics></math> and nucleus sampling <cite class="ltx_cite ltx_citemacro_citep">(Holtzman et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib12" title="">2020</a>)</cite>. We also consider beam search (BS) and the variants of BS to promote sequence-level diversity between the generated samples: diverse BS <cite class="ltx_cite ltx_citemacro_citep">(Vijayakumar et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib35" title="">2018</a>)</cite> and contrastive BS <cite class="ltx_cite ltx_citemacro_citep">(Su et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib31" title="">2022</a>)</cite>. The detailed settings are described in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#A2" title="Appendix B Detailed experimental settings ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Appendix</span> <span class="ltx_text ltx_ref_tag">B</span></a>.</p>
</div>
<div class="ltx_para ltx_noindent" id="S4.SS1.p7">
<p class="ltx_p" id="S4.SS1.p7.1"><span class="ltx_text ltx_font_bold" id="S4.SS1.p7.1.1">Results.</span> We present the main results on BioT5<sup class="ltx_sup" id="S4.SS1.p7.1.2"><span class="ltx_text ltx_font_italic" id="S4.SS1.p7.1.2.1">+</span></sup> and MolT5 in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.F5.sf1" title="In Figure 5 ‣ 4.1 Comparison with decoding schemes ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Figure</span> <span class="ltx_text ltx_ref_tag">5(a)</span></a> and <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.F5.sf2" title="In Figure 5 ‣ 4.1 Comparison with decoding schemes ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Figure</span> <span class="ltx_text ltx_ref_tag">5(b)</span></a>, respectively. One can see that our approach, i.e., Div-SFT+RL, shows superior performance compared to the considered baselines. Especially, it is worth noting that our approach makes significant improvements in NCircles metrics that require generating diverse molecules satisfying the given description.</p>
</div>
<div class="ltx_para ltx_noindent" id="S4.SS1.p8">
<p class="ltx_p" id="S4.SS1.p8.1">We also provide the qualitative results in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.F6" title="In 4.1 Comparison with decoding schemes ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Figure</span> <span class="ltx_text ltx_ref_tag">6</span></a> and <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#A3.T13" title="In Appendix C Additional results ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Table</span> <span class="ltx_text ltx_ref_tag">13</span></a> with sampled molecules. One can see that variants of beam search for diverse sentence generation, i.e., diverse BS and contrastive BS, do not enhance molecular structural diversity. In contrast, our approach demonstrates significantly higher molecular structural diversity. In <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#A3" title="Appendix C Additional results ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Appendix</span> <span class="ltx_text ltx_ref_tag">C</span></a>, we provide additional examples of molecules generated by the considered baselines and our method.</p>
</div>
<figure class="ltx_table" id="S4.T2">
<figcaption class="ltx_caption"><span class="ltx_tag ltx_tag_table">Table 2: </span><span class="ltx_text ltx_font_bold" id="S4.T2.6.1">Comparison with existing LLMs on description-based molecule generation.</span> Our fine-tuned model shows superior performance compared to the considered baselines. </figcaption>
<div class="ltx_inline-block ltx_align_center ltx_transformed_outer" id="S4.T2.4" style="width:400.0pt;height:234pt;vertical-align:-0.0pt;"><span class="ltx_transformed_inner" style="transform:translate(0.0pt,0.0pt) scale(1.0,1.0) ;">
<table class="ltx_tabular ltx_align_middle" id="S4.T2.4.4">
<tr class="ltx_tr" id="S4.T2.2.2.2">
<td class="ltx_td ltx_align_left ltx_border_tt" id="S4.T2.2.2.2.3">Method</td>
<td class="ltx_td ltx_align_center ltx_border_tt" id="S4.T2.2.2.2.4">Accepted &amp; Unique</td>
<td class="ltx_td ltx_align_center ltx_border_tt" id="S4.T2.1.1.1.1">NCircles<sub class="ltx_sub" id="S4.T2.1.1.1.1.1"><span class="ltx_text ltx_font_italic" id="S4.T2.1.1.1.1.1.1">h=0.75</span></sub>
</td>
<td class="ltx_td ltx_align_center ltx_border_tt" id="S4.T2.2.2.2.2">NCircles<sub class="ltx_sub" id="S4.T2.2.2.2.2.1"><span class="ltx_text ltx_font_italic" id="S4.T2.2.2.2.2.1.1">h=0.65</span></sub>
</td>
<td class="ltx_td ltx_align_center ltx_border_tt" id="S4.T2.2.2.2.5">IntDiv.</td>
<td class="ltx_td ltx_align_center ltx_border_tt" id="S4.T2.2.2.2.6">Top 10</td>
</tr>
<tr class="ltx_tr" id="S4.T2.4.4.5" style="background-color:#F3F3F3;">
<td class="ltx_td ltx_align_left ltx_border_t" colspan="6" id="S4.T2.4.4.5.1"><span class="ltx_text ltx_font_italic" id="S4.T2.4.4.5.1.1" style="background-color:#F3F3F3;">Chemical-task Specialist LLMs</span></td>
</tr>
<tr class="ltx_tr" id="S4.T2.4.4.6">
<td class="ltx_td ltx_align_left" id="S4.T2.4.4.6.1">MolT5</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.6.2">3.12</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.6.3">1.85</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.6.4">1.59</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.6.5">0.29</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.6.6">0.41</td>
</tr>
<tr class="ltx_tr" id="S4.T2.4.4.7">
<td class="ltx_td ltx_align_left" id="S4.T2.4.4.7.1">Text+Chem T5</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.7.2">13.78</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.7.3">4.82</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.7.4">3.29</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.7.5">0.23</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.7.6">0.63</td>
</tr>
<tr class="ltx_tr" id="S4.T2.4.4.8">
<td class="ltx_td ltx_align_left" id="S4.T2.4.4.8.1">BioT5</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.8.2">15.58</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.8.3">8.45</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.8.4">6.07</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.8.5">0.33</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.8.6">0.69</td>
</tr>
<tr class="ltx_tr" id="S4.T2.3.3.3">
<td class="ltx_td ltx_align_left" id="S4.T2.3.3.3.1">BioT5<sup class="ltx_sup" id="S4.T2.3.3.3.1.1"><span class="ltx_text ltx_font_italic" id="S4.T2.3.3.3.1.1.1">+</span></sup>
</td>
<td class="ltx_td ltx_align_center" id="S4.T2.3.3.3.2">16.03</td>
<td class="ltx_td ltx_align_center" id="S4.T2.3.3.3.3">7.93</td>
<td class="ltx_td ltx_align_center" id="S4.T2.3.3.3.4">6.16</td>
<td class="ltx_td ltx_align_center" id="S4.T2.3.3.3.5">0.31</td>
<td class="ltx_td ltx_align_center" id="S4.T2.3.3.3.6">0.63</td>
</tr>
<tr class="ltx_tr" id="S4.T2.4.4.9" style="background-color:#F3F3F3;">
<td class="ltx_td ltx_align_left ltx_border_t" colspan="6" id="S4.T2.4.4.9.1"><span class="ltx_text ltx_font_italic" id="S4.T2.4.4.9.1.1" style="background-color:#F3F3F3;">Generalist LLMs</span></td>
</tr>
<tr class="ltx_tr" id="S4.T2.4.4.10">
<td class="ltx_td ltx_align_left" id="S4.T2.4.4.10.1">Mol-instructions</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.10.2">0.06</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.10.3">0.03</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.10.4">0.03</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.10.5">0.02</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.10.6">0.01</td>
</tr>
<tr class="ltx_tr" id="S4.T2.4.4.11">
<td class="ltx_td ltx_align_left" id="S4.T2.4.4.11.1">LlaSMol</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.11.2">17.68</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.11.3">6.73</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.11.4">4.94</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.11.5">0.26</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.11.6">0.68</td>
</tr>
<tr class="ltx_tr" id="S4.T2.4.4.12">
<td class="ltx_td ltx_align_left" id="S4.T2.4.4.12.1">GPT-3.5-turbo</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.12.2">3.87</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.12.3">2.83</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.12.4">2.49</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.12.5">0.20</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.12.6">0.41</td>
</tr>
<tr class="ltx_tr" id="S4.T2.4.4.13">
<td class="ltx_td ltx_align_left" id="S4.T2.4.4.13.1">GPT-4o</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.13.2">5.74</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.13.3">4.09</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.13.4">3.53</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.13.5">0.22</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.13.6">0.48</td>
</tr>
<tr class="ltx_tr" id="S4.T2.4.4.14">
<td class="ltx_td ltx_align_left" id="S4.T2.4.4.14.1">o1-preview</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.14.2">6.92</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.14.3">3.46</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.14.4">2.36</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.14.5">0.13</td>
<td class="ltx_td ltx_align_center" id="S4.T2.4.4.14.6">0.31</td>
</tr>
<tr class="ltx_tr" id="S4.T2.4.4.4">
<td class="ltx_td ltx_align_left ltx_border_bb ltx_border_t" id="S4.T2.4.4.4.1"><span class="ltx_text ltx_font_bold" id="S4.T2.4.4.4.1.1">BioT5<sup class="ltx_sup" id="S4.T2.4.4.4.1.1.1"><span class="ltx_text ltx_font_medium ltx_font_italic" id="S4.T2.4.4.4.1.1.1.1">+</span></sup>+ours</span></td>
<td class="ltx_td ltx_align_center ltx_border_bb ltx_border_t" id="S4.T2.4.4.4.2"><span class="ltx_text ltx_font_bold" id="S4.T2.4.4.4.2.1">21.98</span></td>
<td class="ltx_td ltx_align_center ltx_border_bb ltx_border_t" id="S4.T2.4.4.4.3"><span class="ltx_text ltx_font_bold" id="S4.T2.4.4.4.3.1">16.98</span></td>
<td class="ltx_td ltx_align_center ltx_border_bb ltx_border_t" id="S4.T2.4.4.4.4"><span class="ltx_text ltx_font_bold" id="S4.T2.4.4.4.4.1">14.35</span></td>
<td class="ltx_td ltx_align_center ltx_border_bb ltx_border_t" id="S4.T2.4.4.4.5"><span class="ltx_text ltx_font_bold" id="S4.T2.4.4.4.5.1">0.45</span></td>
<td class="ltx_td ltx_align_center ltx_border_bb ltx_border_t" id="S4.T2.4.4.4.6"><span class="ltx_text ltx_font_bold" id="S4.T2.4.4.4.6.1">0.74</span></td>
</tr>
</table>
</span></div>
</figure>
</section>
<section class="ltx_subsection" id="S4.SS2">
<h3 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">4.2 </span>Comparison with existing LLMs</h3>
<div class="ltx_para ltx_noindent" id="S4.SS2.p1">
<p class="ltx_p" id="S4.SS2.p1.2">Here, we compare our fine-tuned BioT5<sup class="ltx_sup" id="S4.SS2.p1.2.1"><span class="ltx_text ltx_font_italic" id="S4.SS2.p1.2.1.1">+</span></sup>, presented in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.F5" title="In 4.1 Comparison with decoding schemes ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Figure</span> <span class="ltx_text ltx_ref_tag">5</span></a>, with various recent LLMs, including chemical-task specialists, generalists, and fine-tuned generalist LLMs on chemical domains. The purpose of this experiment is to highlight the limitations of existing LLMs in capturing molecular diversity, whereas our fine-tuned model successfully captures molecular diversity during generation. The tasks and metrics are the same as settings in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.SS1" title="4.1 Comparison with decoding schemes ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Section</span> <span class="ltx_text ltx_ref_tag">4.1</span></a>. We use the first <math alttext="500" class="ltx_Math" display="inline" id="S4.SS2.p1.2.m2.1"><semantics id="S4.SS2.p1.2.m2.1a"><mn id="S4.SS2.p1.2.m2.1.1" xref="S4.SS2.p1.2.m2.1.1.cmml">500</mn><annotation-xml encoding="MathML-Content" id="S4.SS2.p1.2.m2.1b"><cn id="S4.SS2.p1.2.m2.1.1.cmml" type="integer" xref="S4.SS2.p1.2.m2.1.1">500</cn></annotation-xml><annotation encoding="application/x-tex" id="S4.SS2.p1.2.m2.1c">500</annotation><annotation encoding="application/x-llamapun" id="S4.SS2.p1.2.m2.1d">500</annotation></semantics></math> molecular descriptions in the ChEBI-20 test dataset for evaluation.</p>
</div>
<div class="ltx_para ltx_noindent" id="S4.SS2.p2">
<p class="ltx_p" id="S4.SS2.p2.2"><span class="ltx_text ltx_font_bold" id="S4.SS2.p2.2.1">Baselines.</span> We compare our fine-tuned model with various existing LLMs. We consider four LLMs specialized for chemical tasks: Text+Chem T5 <cite class="ltx_cite ltx_citemacro_citep">(Christofidellis et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib5" title="">2023</a>)</cite>, MolT5, BioT5 <cite class="ltx_cite ltx_citemacro_citep">(Pei et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib25" title="">2023</a>)</cite>, and BioT5<sup class="ltx_sup" id="S4.SS2.p2.2.2"><span class="ltx_text ltx_font_italic" id="S4.SS2.p2.2.2.1">+</span></sup>. Next, we consider two generalist LLMs: Mol-instructions <cite class="ltx_cite ltx_citemacro_citep">(Fang et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib8" title="">2024</a>)</cite> fine-tuned from Llama-7B <cite class="ltx_cite ltx_citemacro_citep">(Touvron et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib33" title="">2023</a>)</cite>, and LlaSMol <cite class="ltx_cite ltx_citemacro_citep">(Yu et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib42" title="">2024</a>)</cite> fine-tuned from Mistral-7B <cite class="ltx_cite ltx_citemacro_citep">(AI, <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib1" title="">2023</a>)</cite>, based on description-based molecular generation tasks. For each baseline, we report the best results (highest NCircles<sub class="ltx_sub" id="S4.SS2.p2.2.3"><span class="ltx_text ltx_font_italic" id="S4.SS2.p2.2.3.1">h=0.65</span></sub>) obtained using either random sampling, beam search, diverse beam search, or contrastive beam search. We also consider three strong API-based generalist LLMs: GPT-3.5, GPT-4o, and o1-preview <cite class="ltx_cite ltx_citemacro_citep">(OpenAI, <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib21" title="">2023</a>; <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib22" title="">2024</a>)</cite>.<span class="ltx_note ltx_role_footnote" id="footnote4"><sup class="ltx_note_mark">4</sup><span class="ltx_note_outer"><span class="ltx_note_content"><sup class="ltx_note_mark">4</sup><span class="ltx_tag ltx_tag_note">4</span>We use <span class="ltx_text ltx_font_typewriter" id="footnote4.1">gpt-3.5-turbo-0125</span> and <span class="ltx_text ltx_font_typewriter" id="footnote4.2">gpt-4o-2024-08-06</span>.</span></span></span> The detailed experimental settings and prompts are described in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#A2" title="Appendix B Detailed experimental settings ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Appendix</span> <span class="ltx_text ltx_ref_tag">B</span></a>.</p>
</div>
<div class="ltx_para ltx_noindent" id="S4.SS2.p3">
<p class="ltx_p" id="S4.SS2.p3.1"><span class="ltx_text ltx_font_bold" id="S4.SS2.p3.1.1">Results.</span> We present the results in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.T2" title="In 4.1 Comparison with decoding schemes ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Table</span> <span class="ltx_text ltx_ref_tag">2</span></a>. One can see that our fine-tuned model shows superior performance compared to the considered baselines in discovering diverse and high-quality molecules. Furthermore, it is noteworthy that most existing LLMs yield low NCircles with respect to the number of generated accepted and unique molecules, while our method yields relatively high NCircles.</p>
</div>
</section>
<section class="ltx_subsection" id="S4.SS3">
<h3 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">4.3 </span>Fine-tuning generalist LLMs</h3>
<div class="ltx_para ltx_noindent" id="S4.SS3.p1">
<p class="ltx_p" id="S4.SS3.p1.1">We further validate whether our fine-tuning method improves the generalist LLMs as well in terms of the diversity of generated molecules. Here, as a base LLM for implementing our method, we consider DrugAssist <cite class="ltx_cite ltx_citemacro_citep">(Ye et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib41" title="">2023</a>)</cite> which is fine-tuned from the Llama-7B. As baselines, we apply random sampling and contrastive beam search to DrugAssist.</p>
</div>
<div class="ltx_para ltx_noindent" id="S4.SS3.p2">
<p class="ltx_p" id="S4.SS3.p2.1"><span class="ltx_text ltx_font_bold" id="S4.SS3.p2.1.1">Tasks.</span> In this experiment, we consider prompts based on the four quantitative molecule properties: hydrogen bond (HB) donors, HB acceptors, Bertz complexity <cite class="ltx_cite ltx_citemacro_citep">(Bertz, <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib3" title="">1981</a>)</cite>, and quantitative estimate of drug-likeness (QED) <cite class="ltx_cite ltx_citemacro_citep">(Bickerton et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib4" title="">2012</a>)</cite>. The goal of this task is to generate diverse molecules that satisfy the properties described in the given prompt.</p>
</div>
<div class="ltx_para ltx_noindent" id="S4.SS3.p3">
<p class="ltx_p" id="S4.SS3.p3.2"><span class="ltx_text ltx_font_bold" id="S4.SS3.p3.2.1">Implementations.</span> We apply our supervised fine-tuning and reinforcement learning to enhance DrugAssist. For supervised fine-tuning, we collect multiple molecules for prompts about three properties: HB donors, HB acceptors, and Bertz complexity. The prompts about QED are excluded from training but included in the evaluation to assess generalization to unseen properties. The sampled molecules are filtered to remove invalid molecules, duplicated molecules, and molecules that do not satisfy the given properties. The property-match reward <math alttext="r_{\text{match}}(m_{k})" class="ltx_Math" display="inline" id="S4.SS3.p3.1.m1.1"><semantics id="S4.SS3.p3.1.m1.1a"><mrow id="S4.SS3.p3.1.m1.1.1" xref="S4.SS3.p3.1.m1.1.1.cmml"><msub id="S4.SS3.p3.1.m1.1.1.3" xref="S4.SS3.p3.1.m1.1.1.3.cmml"><mi id="S4.SS3.p3.1.m1.1.1.3.2" xref="S4.SS3.p3.1.m1.1.1.3.2.cmml">r</mi><mtext id="S4.SS3.p3.1.m1.1.1.3.3" xref="S4.SS3.p3.1.m1.1.1.3.3a.cmml">match</mtext></msub><mo id="S4.SS3.p3.1.m1.1.1.2" xref="S4.SS3.p3.1.m1.1.1.2.cmml">⁢</mo><mrow id="S4.SS3.p3.1.m1.1.1.1.1" xref="S4.SS3.p3.1.m1.1.1.1.1.1.cmml"><mo id="S4.SS3.p3.1.m1.1.1.1.1.2" stretchy="false" xref="S4.SS3.p3.1.m1.1.1.1.1.1.cmml">(</mo><msub id="S4.SS3.p3.1.m1.1.1.1.1.1" xref="S4.SS3.p3.1.m1.1.1.1.1.1.cmml"><mi id="S4.SS3.p3.1.m1.1.1.1.1.1.2" xref="S4.SS3.p3.1.m1.1.1.1.1.1.2.cmml">m</mi><mi id="S4.SS3.p3.1.m1.1.1.1.1.1.3" xref="S4.SS3.p3.1.m1.1.1.1.1.1.3.cmml">k</mi></msub><mo id="S4.SS3.p3.1.m1.1.1.1.1.3" stretchy="false" xref="S4.SS3.p3.1.m1.1.1.1.1.1.cmml">)</mo></mrow></mrow><annotation-xml encoding="MathML-Content" id="S4.SS3.p3.1.m1.1b"><apply id="S4.SS3.p3.1.m1.1.1.cmml" xref="S4.SS3.p3.1.m1.1.1"><times id="S4.SS3.p3.1.m1.1.1.2.cmml" xref="S4.SS3.p3.1.m1.1.1.2"></times><apply id="S4.SS3.p3.1.m1.1.1.3.cmml" xref="S4.SS3.p3.1.m1.1.1.3"><csymbol cd="ambiguous" id="S4.SS3.p3.1.m1.1.1.3.1.cmml" xref="S4.SS3.p3.1.m1.1.1.3">subscript</csymbol><ci id="S4.SS3.p3.1.m1.1.1.3.2.cmml" xref="S4.SS3.p3.1.m1.1.1.3.2">𝑟</ci><ci id="S4.SS3.p3.1.m1.1.1.3.3a.cmml" xref="S4.SS3.p3.1.m1.1.1.3.3"><mtext id="S4.SS3.p3.1.m1.1.1.3.3.cmml" mathsize="70%" xref="S4.SS3.p3.1.m1.1.1.3.3">match</mtext></ci></apply><apply id="S4.SS3.p3.1.m1.1.1.1.1.1.cmml" xref="S4.SS3.p3.1.m1.1.1.1.1"><csymbol cd="ambiguous" id="S4.SS3.p3.1.m1.1.1.1.1.1.1.cmml" xref="S4.SS3.p3.1.m1.1.1.1.1">subscript</csymbol><ci id="S4.SS3.p3.1.m1.1.1.1.1.1.2.cmml" xref="S4.SS3.p3.1.m1.1.1.1.1.1.2">𝑚</ci><ci id="S4.SS3.p3.1.m1.1.1.1.1.1.3.cmml" xref="S4.SS3.p3.1.m1.1.1.1.1.1.3">𝑘</ci></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="S4.SS3.p3.1.m1.1c">r_{\text{match}}(m_{k})</annotation><annotation encoding="application/x-llamapun" id="S4.SS3.p3.1.m1.1d">italic_r start_POSTSUBSCRIPT match end_POSTSUBSCRIPT ( italic_m start_POSTSUBSCRIPT italic_k end_POSTSUBSCRIPT )</annotation></semantics></math> yields a non-zero value when satisfying the given property, and the diversity reward <math alttext="r_{\text{div}}(m_{k},\{m_{i}\}^{k-1}_{i=1})" class="ltx_Math" display="inline" id="S4.SS3.p3.2.m2.2"><semantics id="S4.SS3.p3.2.m2.2a"><mrow id="S4.SS3.p3.2.m2.2.2" xref="S4.SS3.p3.2.m2.2.2.cmml"><msub id="S4.SS3.p3.2.m2.2.2.4" xref="S4.SS3.p3.2.m2.2.2.4.cmml"><mi id="S4.SS3.p3.2.m2.2.2.4.2" xref="S4.SS3.p3.2.m2.2.2.4.2.cmml">r</mi><mtext id="S4.SS3.p3.2.m2.2.2.4.3" xref="S4.SS3.p3.2.m2.2.2.4.3a.cmml">div</mtext></msub><mo id="S4.SS3.p3.2.m2.2.2.3" xref="S4.SS3.p3.2.m2.2.2.3.cmml">⁢</mo><mrow id="S4.SS3.p3.2.m2.2.2.2.2" xref="S4.SS3.p3.2.m2.2.2.2.3.cmml"><mo id="S4.SS3.p3.2.m2.2.2.2.2.3" stretchy="false" xref="S4.SS3.p3.2.m2.2.2.2.3.cmml">(</mo><msub id="S4.SS3.p3.2.m2.1.1.1.1.1" xref="S4.SS3.p3.2.m2.1.1.1.1.1.cmml"><mi id="S4.SS3.p3.2.m2.1.1.1.1.1.2" xref="S4.SS3.p3.2.m2.1.1.1.1.1.2.cmml">m</mi><mi id="S4.SS3.p3.2.m2.1.1.1.1.1.3" xref="S4.SS3.p3.2.m2.1.1.1.1.1.3.cmml">k</mi></msub><mo id="S4.SS3.p3.2.m2.2.2.2.2.4" xref="S4.SS3.p3.2.m2.2.2.2.3.cmml">,</mo><msubsup id="S4.SS3.p3.2.m2.2.2.2.2.2" xref="S4.SS3.p3.2.m2.2.2.2.2.2.cmml"><mrow id="S4.SS3.p3.2.m2.2.2.2.2.2.1.1.1" xref="S4.SS3.p3.2.m2.2.2.2.2.2.1.1.2.cmml"><mo id="S4.SS3.p3.2.m2.2.2.2.2.2.1.1.1.2" stretchy="false" xref="S4.SS3.p3.2.m2.2.2.2.2.2.1.1.2.cmml">{</mo><msub id="S4.SS3.p3.2.m2.2.2.2.2.2.1.1.1.1" xref="S4.SS3.p3.2.m2.2.2.2.2.2.1.1.1.1.cmml"><mi id="S4.SS3.p3.2.m2.2.2.2.2.2.1.1.1.1.2" xref="S4.SS3.p3.2.m2.2.2.2.2.2.1.1.1.1.2.cmml">m</mi><mi id="S4.SS3.p3.2.m2.2.2.2.2.2.1.1.1.1.3" xref="S4.SS3.p3.2.m2.2.2.2.2.2.1.1.1.1.3.cmml">i</mi></msub><mo id="S4.SS3.p3.2.m2.2.2.2.2.2.1.1.1.3" stretchy="false" xref="S4.SS3.p3.2.m2.2.2.2.2.2.1.1.2.cmml">}</mo></mrow><mrow id="S4.SS3.p3.2.m2.2.2.2.2.2.3" xref="S4.SS3.p3.2.m2.2.2.2.2.2.3.cmml"><mi id="S4.SS3.p3.2.m2.2.2.2.2.2.3.2" xref="S4.SS3.p3.2.m2.2.2.2.2.2.3.2.cmml">i</mi><mo id="S4.SS3.p3.2.m2.2.2.2.2.2.3.1" xref="S4.SS3.p3.2.m2.2.2.2.2.2.3.1.cmml">=</mo><mn id="S4.SS3.p3.2.m2.2.2.2.2.2.3.3" xref="S4.SS3.p3.2.m2.2.2.2.2.2.3.3.cmml">1</mn></mrow><mrow id="S4.SS3.p3.2.m2.2.2.2.2.2.1.3" xref="S4.SS3.p3.2.m2.2.2.2.2.2.1.3.cmml"><mi id="S4.SS3.p3.2.m2.2.2.2.2.2.1.3.2" xref="S4.SS3.p3.2.m2.2.2.2.2.2.1.3.2.cmml">k</mi><mo id="S4.SS3.p3.2.m2.2.2.2.2.2.1.3.1" xref="S4.SS3.p3.2.m2.2.2.2.2.2.1.3.1.cmml">−</mo><mn id="S4.SS3.p3.2.m2.2.2.2.2.2.1.3.3" xref="S4.SS3.p3.2.m2.2.2.2.2.2.1.3.3.cmml">1</mn></mrow></msubsup><mo id="S4.SS3.p3.2.m2.2.2.2.2.5" stretchy="false" xref="S4.SS3.p3.2.m2.2.2.2.3.cmml">)</mo></mrow></mrow><annotation-xml encoding="MathML-Content" id="S4.SS3.p3.2.m2.2b"><apply id="S4.SS3.p3.2.m2.2.2.cmml" xref="S4.SS3.p3.2.m2.2.2"><times id="S4.SS3.p3.2.m2.2.2.3.cmml" xref="S4.SS3.p3.2.m2.2.2.3"></times><apply id="S4.SS3.p3.2.m2.2.2.4.cmml" xref="S4.SS3.p3.2.m2.2.2.4"><csymbol cd="ambiguous" id="S4.SS3.p3.2.m2.2.2.4.1.cmml" xref="S4.SS3.p3.2.m2.2.2.4">subscript</csymbol><ci id="S4.SS3.p3.2.m2.2.2.4.2.cmml" xref="S4.SS3.p3.2.m2.2.2.4.2">𝑟</ci><ci id="S4.SS3.p3.2.m2.2.2.4.3a.cmml" xref="S4.SS3.p3.2.m2.2.2.4.3"><mtext id="S4.SS3.p3.2.m2.2.2.4.3.cmml" mathsize="70%" xref="S4.SS3.p3.2.m2.2.2.4.3">div</mtext></ci></apply><interval closure="open" id="S4.SS3.p3.2.m2.2.2.2.3.cmml" xref="S4.SS3.p3.2.m2.2.2.2.2"><apply id="S4.SS3.p3.2.m2.1.1.1.1.1.cmml" xref="S4.SS3.p3.2.m2.1.1.1.1.1"><csymbol cd="ambiguous" id="S4.SS3.p3.2.m2.1.1.1.1.1.1.cmml" xref="S4.SS3.p3.2.m2.1.1.1.1.1">subscript</csymbol><ci id="S4.SS3.p3.2.m2.1.1.1.1.1.2.cmml" xref="S4.SS3.p3.2.m2.1.1.1.1.1.2">𝑚</ci><ci id="S4.SS3.p3.2.m2.1.1.1.1.1.3.cmml" xref="S4.SS3.p3.2.m2.1.1.1.1.1.3">𝑘</ci></apply><apply id="S4.SS3.p3.2.m2.2.2.2.2.2.cmml" xref="S4.SS3.p3.2.m2.2.2.2.2.2"><csymbol cd="ambiguous" id="S4.SS3.p3.2.m2.2.2.2.2.2.2.cmml" xref="S4.SS3.p3.2.m2.2.2.2.2.2">subscript</csymbol><apply id="S4.SS3.p3.2.m2.2.2.2.2.2.1.cmml" xref="S4.SS3.p3.2.m2.2.2.2.2.2"><csymbol cd="ambiguous" id="S4.SS3.p3.2.m2.2.2.2.2.2.1.2.cmml" xref="S4.SS3.p3.2.m2.2.2.2.2.2">superscript</csymbol><set id="S4.SS3.p3.2.m2.2.2.2.2.2.1.1.2.cmml" xref="S4.SS3.p3.2.m2.2.2.2.2.2.1.1.1"><apply id="S4.SS3.p3.2.m2.2.2.2.2.2.1.1.1.1.cmml" xref="S4.SS3.p3.2.m2.2.2.2.2.2.1.1.1.1"><csymbol cd="ambiguous" id="S4.SS3.p3.2.m2.2.2.2.2.2.1.1.1.1.1.cmml" xref="S4.SS3.p3.2.m2.2.2.2.2.2.1.1.1.1">subscript</csymbol><ci id="S4.SS3.p3.2.m2.2.2.2.2.2.1.1.1.1.2.cmml" xref="S4.SS3.p3.2.m2.2.2.2.2.2.1.1.1.1.2">𝑚</ci><ci id="S4.SS3.p3.2.m2.2.2.2.2.2.1.1.1.1.3.cmml" xref="S4.SS3.p3.2.m2.2.2.2.2.2.1.1.1.1.3">𝑖</ci></apply></set><apply id="S4.SS3.p3.2.m2.2.2.2.2.2.1.3.cmml" xref="S4.SS3.p3.2.m2.2.2.2.2.2.1.3"><minus id="S4.SS3.p3.2.m2.2.2.2.2.2.1.3.1.cmml" xref="S4.SS3.p3.2.m2.2.2.2.2.2.1.3.1"></minus><ci id="S4.SS3.p3.2.m2.2.2.2.2.2.1.3.2.cmml" xref="S4.SS3.p3.2.m2.2.2.2.2.2.1.3.2">𝑘</ci><cn id="S4.SS3.p3.2.m2.2.2.2.2.2.1.3.3.cmml" type="integer" xref="S4.SS3.p3.2.m2.2.2.2.2.2.1.3.3">1</cn></apply></apply><apply id="S4.SS3.p3.2.m2.2.2.2.2.2.3.cmml" xref="S4.SS3.p3.2.m2.2.2.2.2.2.3"><eq id="S4.SS3.p3.2.m2.2.2.2.2.2.3.1.cmml" xref="S4.SS3.p3.2.m2.2.2.2.2.2.3.1"></eq><ci id="S4.SS3.p3.2.m2.2.2.2.2.2.3.2.cmml" xref="S4.SS3.p3.2.m2.2.2.2.2.2.3.2">𝑖</ci><cn id="S4.SS3.p3.2.m2.2.2.2.2.2.3.3.cmml" type="integer" xref="S4.SS3.p3.2.m2.2.2.2.2.2.3.3">1</cn></apply></apply></interval></apply></annotation-xml><annotation encoding="application/x-tex" id="S4.SS3.p3.2.m2.2c">r_{\text{div}}(m_{k},\{m_{i}\}^{k-1}_{i=1})</annotation><annotation encoding="application/x-llamapun" id="S4.SS3.p3.2.m2.2d">italic_r start_POSTSUBSCRIPT div end_POSTSUBSCRIPT ( italic_m start_POSTSUBSCRIPT italic_k end_POSTSUBSCRIPT , { italic_m start_POSTSUBSCRIPT italic_i end_POSTSUBSCRIPT } start_POSTSUPERSCRIPT italic_k - 1 end_POSTSUPERSCRIPT start_POSTSUBSCRIPT italic_i = 1 end_POSTSUBSCRIPT )</annotation></semantics></math> is defined as same as <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.SS1" title="4.1 Comparison with decoding schemes ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Section</span> <span class="ltx_text ltx_ref_tag">4.1</span></a>. The detailed implementations and hyper-parameters follow <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#A1" title="Appendix A Detailed implementations and training ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Appendix</span> <span class="ltx_text ltx_ref_tag">A</span></a>.</p>
</div>
<figure class="ltx_figure" id="S4.F7">
<div class="ltx_flex_figure">
<div class="ltx_flex_cell ltx_flex_size_4">
<figure class="ltx_figure ltx_figure_panel ltx_align_center" id="S4.F7.sf1"><img alt="Refer to caption" class="ltx_graphics ltx_centering ltx_img_square" height="830" id="S4.F7.sf1.g1" src="x18.png" width="830"/>
<figcaption class="ltx_caption ltx_centering"><span class="ltx_tag ltx_tag_figure">(a) </span>HB donors</figcaption>
</figure>
</div>
<div class="ltx_flex_cell ltx_flex_size_4">
<figure class="ltx_figure ltx_figure_panel ltx_align_center" id="S4.F7.sf2"><img alt="Refer to caption" class="ltx_graphics ltx_centering ltx_img_square" height="830" id="S4.F7.sf2.g1" src="x19.png" width="830"/>
<figcaption class="ltx_caption ltx_centering"><span class="ltx_tag ltx_tag_figure">(b) </span>HB acceptors</figcaption>
</figure>
</div>
<div class="ltx_flex_cell ltx_flex_size_4">
<figure class="ltx_figure ltx_figure_panel ltx_align_center" id="S4.F7.sf3"><img alt="Refer to caption" class="ltx_graphics ltx_centering ltx_img_square" height="830" id="S4.F7.sf3.g1" src="x20.png" width="830"/>
<figcaption class="ltx_caption ltx_centering"><span class="ltx_tag ltx_tag_figure">(c) </span>Bertz complexity</figcaption>
</figure>
</div>
<div class="ltx_flex_cell ltx_flex_size_4">
<figure class="ltx_figure ltx_figure_panel ltx_align_center" id="S4.F7.sf4"><img alt="Refer to caption" class="ltx_graphics ltx_centering ltx_img_square" height="830" id="S4.F7.sf4.g1" src="x21.png" width="830"/>
<figcaption class="ltx_caption ltx_centering"><span class="ltx_tag ltx_tag_figure">(d) </span>QED</figcaption>
</figure>
</div>
</div>
<figcaption class="ltx_caption ltx_centering"><span class="ltx_tag ltx_tag_figure">Figure 7: </span><span class="ltx_text ltx_font_bold" id="S4.F7.2.1">Experiments on DrugAssist.</span> Our method consistently improves the performance for generating diverse and high-quality molecules when implemented on generalist LLMs.</figcaption>
</figure>
<div class="ltx_para ltx_noindent" id="S4.SS3.p4">
<p class="ltx_p" id="S4.SS3.p4.1"><span class="ltx_text ltx_font_bold" id="S4.SS3.p4.1.1">Results.</span> We present the results in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.F7" title="In 4.3 Fine-tuning generalist LLMs ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Figure</span> <span class="ltx_text ltx_ref_tag">7</span></a>. One can see that our approach consistently improves the performance in generating diverse and high-quality molecules when implemented on generalist LLMs. Note that our approach consistently demonstrates superior performance for the unseen prompt, i.e., the prompt about QED in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.F7.sf4" title="In Figure 7 ‣ 4.3 Fine-tuning generalist LLMs ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Figure</span> <span class="ltx_text ltx_ref_tag">7(d)</span></a>.</p>
</div>
<figure class="ltx_table" id="S4.T3">
<figcaption class="ltx_caption"><span class="ltx_tag ltx_tag_table">Table 3: </span><span class="ltx_text ltx_font_bold" id="S4.T3.10.1">Experiments with the large number of samples.</span> Our method discovers more diverse molecules with respect to the (1) the number of generations and (2) time costs.</figcaption>
<table class="ltx_tabular ltx_centering ltx_align_middle" id="S4.T3.8">
<tr class="ltx_tr" id="S4.T3.7.7">
<td class="ltx_td ltx_align_left ltx_border_tt" id="S4.T3.1.1.1">Method<math alttext="{}_{\text{num. of generations}}" class="ltx_Math" display="inline" id="S4.T3.1.1.1.m1.1"><semantics id="S4.T3.1.1.1.m1.1a"><msub id="S4.T3.1.1.1.m1.1.1" xref="S4.T3.1.1.1.m1.1.1.cmml"><mi id="S4.T3.1.1.1.m1.1.1a" xref="S4.T3.1.1.1.m1.1.1.cmml"></mi><mtext id="S4.T3.1.1.1.m1.1.1.1" xref="S4.T3.1.1.1.m1.1.1.1a.cmml">num. of generations</mtext></msub><annotation-xml encoding="MathML-Content" id="S4.T3.1.1.1.m1.1b"><apply id="S4.T3.1.1.1.m1.1.1.cmml" xref="S4.T3.1.1.1.m1.1.1"><ci id="S4.T3.1.1.1.m1.1.1.1a.cmml" xref="S4.T3.1.1.1.m1.1.1.1"><mtext id="S4.T3.1.1.1.m1.1.1.1.cmml" mathsize="70%" xref="S4.T3.1.1.1.m1.1.1.1">num. of generations</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S4.T3.1.1.1.m1.1c">{}_{\text{num. of generations}}</annotation><annotation encoding="application/x-llamapun" id="S4.T3.1.1.1.m1.1d">start_FLOATSUBSCRIPT num. of generations end_FLOATSUBSCRIPT</annotation></semantics></math>
</td>
<td class="ltx_td ltx_align_center ltx_border_tt" id="S4.T3.2.2.2"><math alttext="{\text{BS}_{300}}" class="ltx_Math" display="inline" id="S4.T3.2.2.2.m1.1"><semantics id="S4.T3.2.2.2.m1.1a"><msub id="S4.T3.2.2.2.m1.1.1" xref="S4.T3.2.2.2.m1.1.1.cmml"><mtext id="S4.T3.2.2.2.m1.1.1.2" xref="S4.T3.2.2.2.m1.1.1.2a.cmml">BS</mtext><mn id="S4.T3.2.2.2.m1.1.1.3" xref="S4.T3.2.2.2.m1.1.1.3.cmml">300</mn></msub><annotation-xml encoding="MathML-Content" id="S4.T3.2.2.2.m1.1b"><apply id="S4.T3.2.2.2.m1.1.1.cmml" xref="S4.T3.2.2.2.m1.1.1"><csymbol cd="ambiguous" id="S4.T3.2.2.2.m1.1.1.1.cmml" xref="S4.T3.2.2.2.m1.1.1">subscript</csymbol><ci id="S4.T3.2.2.2.m1.1.1.2a.cmml" xref="S4.T3.2.2.2.m1.1.1.2"><mtext id="S4.T3.2.2.2.m1.1.1.2.cmml" xref="S4.T3.2.2.2.m1.1.1.2">BS</mtext></ci><cn id="S4.T3.2.2.2.m1.1.1.3.cmml" type="integer" xref="S4.T3.2.2.2.m1.1.1.3">300</cn></apply></annotation-xml><annotation encoding="application/x-tex" id="S4.T3.2.2.2.m1.1c">{\text{BS}_{300}}</annotation><annotation encoding="application/x-llamapun" id="S4.T3.2.2.2.m1.1d">BS start_POSTSUBSCRIPT 300 end_POSTSUBSCRIPT</annotation></semantics></math></td>
<td class="ltx_td ltx_align_center ltx_border_tt" id="S4.T3.3.3.3"><math alttext="{\text{BS}_{400}}" class="ltx_Math" display="inline" id="S4.T3.3.3.3.m1.1"><semantics id="S4.T3.3.3.3.m1.1a"><msub id="S4.T3.3.3.3.m1.1.1" xref="S4.T3.3.3.3.m1.1.1.cmml"><mtext id="S4.T3.3.3.3.m1.1.1.2" xref="S4.T3.3.3.3.m1.1.1.2a.cmml">BS</mtext><mn id="S4.T3.3.3.3.m1.1.1.3" xref="S4.T3.3.3.3.m1.1.1.3.cmml">400</mn></msub><annotation-xml encoding="MathML-Content" id="S4.T3.3.3.3.m1.1b"><apply id="S4.T3.3.3.3.m1.1.1.cmml" xref="S4.T3.3.3.3.m1.1.1"><csymbol cd="ambiguous" id="S4.T3.3.3.3.m1.1.1.1.cmml" xref="S4.T3.3.3.3.m1.1.1">subscript</csymbol><ci id="S4.T3.3.3.3.m1.1.1.2a.cmml" xref="S4.T3.3.3.3.m1.1.1.2"><mtext id="S4.T3.3.3.3.m1.1.1.2.cmml" xref="S4.T3.3.3.3.m1.1.1.2">BS</mtext></ci><cn id="S4.T3.3.3.3.m1.1.1.3.cmml" type="integer" xref="S4.T3.3.3.3.m1.1.1.3">400</cn></apply></annotation-xml><annotation encoding="application/x-tex" id="S4.T3.3.3.3.m1.1c">{\text{BS}_{400}}</annotation><annotation encoding="application/x-llamapun" id="S4.T3.3.3.3.m1.1d">BS start_POSTSUBSCRIPT 400 end_POSTSUBSCRIPT</annotation></semantics></math></td>
<td class="ltx_td ltx_align_center ltx_border_tt" id="S4.T3.4.4.4"><math alttext="{\text{BS}_{500}}" class="ltx_Math" display="inline" id="S4.T3.4.4.4.m1.1"><semantics id="S4.T3.4.4.4.m1.1a"><msub id="S4.T3.4.4.4.m1.1.1" xref="S4.T3.4.4.4.m1.1.1.cmml"><mtext id="S4.T3.4.4.4.m1.1.1.2" xref="S4.T3.4.4.4.m1.1.1.2a.cmml">BS</mtext><mn id="S4.T3.4.4.4.m1.1.1.3" xref="S4.T3.4.4.4.m1.1.1.3.cmml">500</mn></msub><annotation-xml encoding="MathML-Content" id="S4.T3.4.4.4.m1.1b"><apply id="S4.T3.4.4.4.m1.1.1.cmml" xref="S4.T3.4.4.4.m1.1.1"><csymbol cd="ambiguous" id="S4.T3.4.4.4.m1.1.1.1.cmml" xref="S4.T3.4.4.4.m1.1.1">subscript</csymbol><ci id="S4.T3.4.4.4.m1.1.1.2a.cmml" xref="S4.T3.4.4.4.m1.1.1.2"><mtext id="S4.T3.4.4.4.m1.1.1.2.cmml" xref="S4.T3.4.4.4.m1.1.1.2">BS</mtext></ci><cn id="S4.T3.4.4.4.m1.1.1.3.cmml" type="integer" xref="S4.T3.4.4.4.m1.1.1.3">500</cn></apply></annotation-xml><annotation encoding="application/x-tex" id="S4.T3.4.4.4.m1.1c">{\text{BS}_{500}}</annotation><annotation encoding="application/x-llamapun" id="S4.T3.4.4.4.m1.1d">BS start_POSTSUBSCRIPT 500 end_POSTSUBSCRIPT</annotation></semantics></math></td>
<td class="ltx_td ltx_align_center ltx_border_tt" id="S4.T3.5.5.5"><math alttext="{\text{Ours}_{85}}" class="ltx_Math" display="inline" id="S4.T3.5.5.5.m1.1"><semantics id="S4.T3.5.5.5.m1.1a"><msub id="S4.T3.5.5.5.m1.1.1" xref="S4.T3.5.5.5.m1.1.1.cmml"><mtext id="S4.T3.5.5.5.m1.1.1.2" xref="S4.T3.5.5.5.m1.1.1.2a.cmml">Ours</mtext><mn id="S4.T3.5.5.5.m1.1.1.3" xref="S4.T3.5.5.5.m1.1.1.3.cmml">85</mn></msub><annotation-xml encoding="MathML-Content" id="S4.T3.5.5.5.m1.1b"><apply id="S4.T3.5.5.5.m1.1.1.cmml" xref="S4.T3.5.5.5.m1.1.1"><csymbol cd="ambiguous" id="S4.T3.5.5.5.m1.1.1.1.cmml" xref="S4.T3.5.5.5.m1.1.1">subscript</csymbol><ci id="S4.T3.5.5.5.m1.1.1.2a.cmml" xref="S4.T3.5.5.5.m1.1.1.2"><mtext id="S4.T3.5.5.5.m1.1.1.2.cmml" xref="S4.T3.5.5.5.m1.1.1.2">Ours</mtext></ci><cn id="S4.T3.5.5.5.m1.1.1.3.cmml" type="integer" xref="S4.T3.5.5.5.m1.1.1.3">85</cn></apply></annotation-xml><annotation encoding="application/x-tex" id="S4.T3.5.5.5.m1.1c">{\text{Ours}_{85}}</annotation><annotation encoding="application/x-llamapun" id="S4.T3.5.5.5.m1.1d">Ours start_POSTSUBSCRIPT 85 end_POSTSUBSCRIPT</annotation></semantics></math></td>
<td class="ltx_td ltx_align_center ltx_border_tt" id="S4.T3.6.6.6"><math alttext="{\text{Ours}_{120}}" class="ltx_Math" display="inline" id="S4.T3.6.6.6.m1.1"><semantics id="S4.T3.6.6.6.m1.1a"><msub id="S4.T3.6.6.6.m1.1.1" xref="S4.T3.6.6.6.m1.1.1.cmml"><mtext id="S4.T3.6.6.6.m1.1.1.2" xref="S4.T3.6.6.6.m1.1.1.2a.cmml">Ours</mtext><mn id="S4.T3.6.6.6.m1.1.1.3" xref="S4.T3.6.6.6.m1.1.1.3.cmml">120</mn></msub><annotation-xml encoding="MathML-Content" id="S4.T3.6.6.6.m1.1b"><apply id="S4.T3.6.6.6.m1.1.1.cmml" xref="S4.T3.6.6.6.m1.1.1"><csymbol cd="ambiguous" id="S4.T3.6.6.6.m1.1.1.1.cmml" xref="S4.T3.6.6.6.m1.1.1">subscript</csymbol><ci id="S4.T3.6.6.6.m1.1.1.2a.cmml" xref="S4.T3.6.6.6.m1.1.1.2"><mtext id="S4.T3.6.6.6.m1.1.1.2.cmml" xref="S4.T3.6.6.6.m1.1.1.2">Ours</mtext></ci><cn id="S4.T3.6.6.6.m1.1.1.3.cmml" type="integer" xref="S4.T3.6.6.6.m1.1.1.3">120</cn></apply></annotation-xml><annotation encoding="application/x-tex" id="S4.T3.6.6.6.m1.1c">{\text{Ours}_{120}}</annotation><annotation encoding="application/x-llamapun" id="S4.T3.6.6.6.m1.1d">Ours start_POSTSUBSCRIPT 120 end_POSTSUBSCRIPT</annotation></semantics></math></td>
<td class="ltx_td ltx_align_center ltx_border_tt" id="S4.T3.7.7.7"><math alttext="{\text{Ours}_{155}}" class="ltx_Math" display="inline" id="S4.T3.7.7.7.m1.1"><semantics id="S4.T3.7.7.7.m1.1a"><msub id="S4.T3.7.7.7.m1.1.1" xref="S4.T3.7.7.7.m1.1.1.cmml"><mtext id="S4.T3.7.7.7.m1.1.1.2" xref="S4.T3.7.7.7.m1.1.1.2a.cmml">Ours</mtext><mn id="S4.T3.7.7.7.m1.1.1.3" xref="S4.T3.7.7.7.m1.1.1.3.cmml">155</mn></msub><annotation-xml encoding="MathML-Content" id="S4.T3.7.7.7.m1.1b"><apply id="S4.T3.7.7.7.m1.1.1.cmml" xref="S4.T3.7.7.7.m1.1.1"><csymbol cd="ambiguous" id="S4.T3.7.7.7.m1.1.1.1.cmml" xref="S4.T3.7.7.7.m1.1.1">subscript</csymbol><ci id="S4.T3.7.7.7.m1.1.1.2a.cmml" xref="S4.T3.7.7.7.m1.1.1.2"><mtext id="S4.T3.7.7.7.m1.1.1.2.cmml" xref="S4.T3.7.7.7.m1.1.1.2">Ours</mtext></ci><cn id="S4.T3.7.7.7.m1.1.1.3.cmml" type="integer" xref="S4.T3.7.7.7.m1.1.1.3">155</cn></apply></annotation-xml><annotation encoding="application/x-tex" id="S4.T3.7.7.7.m1.1c">{\text{Ours}_{155}}</annotation><annotation encoding="application/x-llamapun" id="S4.T3.7.7.7.m1.1d">Ours start_POSTSUBSCRIPT 155 end_POSTSUBSCRIPT</annotation></semantics></math></td>
</tr>
<tr class="ltx_tr" id="S4.T3.8.8">
<td class="ltx_td ltx_align_left ltx_border_t" id="S4.T3.8.8.1">NCircles<sub class="ltx_sub" id="S4.T3.8.8.1.1"><span class="ltx_text ltx_font_italic" id="S4.T3.8.8.1.1.1">h=0.65</span></sub>
</td>
<td class="ltx_td ltx_align_center ltx_border_t" id="S4.T3.8.8.2">18.6</td>
<td class="ltx_td ltx_align_center ltx_border_t" id="S4.T3.8.8.3">20.4</td>
<td class="ltx_td ltx_align_center ltx_border_t" id="S4.T3.8.8.4">21.3</td>
<td class="ltx_td ltx_align_center ltx_border_t" id="S4.T3.8.8.5">20.4</td>
<td class="ltx_td ltx_align_center ltx_border_t" id="S4.T3.8.8.6">23.6</td>
<td class="ltx_td ltx_align_center ltx_border_t" id="S4.T3.8.8.7">25.6</td>
</tr>
<tr class="ltx_tr" id="S4.T3.8.9">
<td class="ltx_td ltx_align_left ltx_border_bb" id="S4.T3.8.9.1">Time (sec.)</td>
<td class="ltx_td ltx_align_center ltx_border_bb" id="S4.T3.8.9.2">323</td>
<td class="ltx_td ltx_align_center ltx_border_bb" id="S4.T3.8.9.3">452</td>
<td class="ltx_td ltx_align_center ltx_border_bb" id="S4.T3.8.9.4">585</td>
<td class="ltx_td ltx_align_center ltx_border_bb" id="S4.T3.8.9.5">65</td>
<td class="ltx_td ltx_align_center ltx_border_bb" id="S4.T3.8.9.6">86</td>
<td class="ltx_td ltx_align_center ltx_border_bb" id="S4.T3.8.9.7">107</td>
</tr>
</table>
</figure>
</section>
<section class="ltx_subsection" id="S4.SS4">
<h3 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">4.4 </span>Ablation studies</h3>
<figure class="ltx_table" id="S4.T4">
<figcaption class="ltx_caption"><span class="ltx_tag ltx_tag_table">Table 4: </span><span class="ltx_text ltx_font_bold" id="S4.T4.6.1">Comparison with variants of implementations.</span> Applying multi-stage reinforcement learning shows superior performance compared to (1) supervised fine-tuning with hard filtering and (2) single-stage reinforcement learning.</figcaption>
<table class="ltx_tabular ltx_centering ltx_align_middle" id="S4.T4.4">
<tr class="ltx_tr" id="S4.T4.2.2">
<td class="ltx_td ltx_align_left ltx_border_tt" id="S4.T4.2.2.3">Method</td>
<td class="ltx_td ltx_align_center ltx_border_tt" id="S4.T4.2.2.4">Accepted &amp; Unique</td>
<td class="ltx_td ltx_align_center ltx_border_tt" id="S4.T4.1.1.1">NCircles<sub class="ltx_sub" id="S4.T4.1.1.1.1"><span class="ltx_text ltx_font_italic" id="S4.T4.1.1.1.1.1">h=0.75</span></sub>
</td>
<td class="ltx_td ltx_align_center ltx_border_tt" id="S4.T4.2.2.2">NCircles<sub class="ltx_sub" id="S4.T4.2.2.2.1"><span class="ltx_text ltx_font_italic" id="S4.T4.2.2.2.1.1">h=0.65</span></sub>
</td>
<td class="ltx_td ltx_align_center ltx_border_tt" id="S4.T4.2.2.5">IntDiv.</td>
<td class="ltx_td ltx_align_center ltx_border_tt" id="S4.T4.2.2.6">Top 10</td>
</tr>
<tr class="ltx_tr" id="S4.T4.3.3">
<td class="ltx_td ltx_align_left ltx_border_t" id="S4.T4.3.3.1">Div-SFT<math alttext="{}_{\text{hard}}" class="ltx_Math" display="inline" id="S4.T4.3.3.1.m1.1"><semantics id="S4.T4.3.3.1.m1.1a"><msub id="S4.T4.3.3.1.m1.1.1" xref="S4.T4.3.3.1.m1.1.1.cmml"><mi id="S4.T4.3.3.1.m1.1.1a" xref="S4.T4.3.3.1.m1.1.1.cmml"></mi><mtext id="S4.T4.3.3.1.m1.1.1.1" xref="S4.T4.3.3.1.m1.1.1.1a.cmml">hard</mtext></msub><annotation-xml encoding="MathML-Content" id="S4.T4.3.3.1.m1.1b"><apply id="S4.T4.3.3.1.m1.1.1.cmml" xref="S4.T4.3.3.1.m1.1.1"><ci id="S4.T4.3.3.1.m1.1.1.1a.cmml" xref="S4.T4.3.3.1.m1.1.1.1"><mtext id="S4.T4.3.3.1.m1.1.1.1.cmml" mathsize="70%" xref="S4.T4.3.3.1.m1.1.1.1">hard</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S4.T4.3.3.1.m1.1c">{}_{\text{hard}}</annotation><annotation encoding="application/x-llamapun" id="S4.T4.3.3.1.m1.1d">start_FLOATSUBSCRIPT hard end_FLOATSUBSCRIPT</annotation></semantics></math>
</td>
<td class="ltx_td ltx_align_center ltx_border_t" id="S4.T4.3.3.2">9.23</td>
<td class="ltx_td ltx_align_center ltx_border_t" id="S4.T4.3.3.3">7.46</td>
<td class="ltx_td ltx_align_center ltx_border_t" id="S4.T4.3.3.4">6.15</td>
<td class="ltx_td ltx_align_center ltx_border_t" id="S4.T4.3.3.5">0.33</td>
<td class="ltx_td ltx_align_center ltx_border_t" id="S4.T4.3.3.6">0.59</td>
</tr>
<tr class="ltx_tr" id="S4.T4.4.4">
<td class="ltx_td ltx_align_left" id="S4.T4.4.4.1">Div-SFT+RL<math alttext="{}_{\text{single}}" class="ltx_Math" display="inline" id="S4.T4.4.4.1.m1.1"><semantics id="S4.T4.4.4.1.m1.1a"><msub id="S4.T4.4.4.1.m1.1.1" xref="S4.T4.4.4.1.m1.1.1.cmml"><mi id="S4.T4.4.4.1.m1.1.1a" xref="S4.T4.4.4.1.m1.1.1.cmml"></mi><mtext id="S4.T4.4.4.1.m1.1.1.1" xref="S4.T4.4.4.1.m1.1.1.1a.cmml">single</mtext></msub><annotation-xml encoding="MathML-Content" id="S4.T4.4.4.1.m1.1b"><apply id="S4.T4.4.4.1.m1.1.1.cmml" xref="S4.T4.4.4.1.m1.1.1"><ci id="S4.T4.4.4.1.m1.1.1.1a.cmml" xref="S4.T4.4.4.1.m1.1.1.1"><mtext id="S4.T4.4.4.1.m1.1.1.1.cmml" mathsize="70%" xref="S4.T4.4.4.1.m1.1.1.1">single</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="S4.T4.4.4.1.m1.1c">{}_{\text{single}}</annotation><annotation encoding="application/x-llamapun" id="S4.T4.4.4.1.m1.1d">start_FLOATSUBSCRIPT single end_FLOATSUBSCRIPT</annotation></semantics></math>
</td>
<td class="ltx_td ltx_align_center" id="S4.T4.4.4.2">14.16</td>
<td class="ltx_td ltx_align_center" id="S4.T4.4.4.3">8.50</td>
<td class="ltx_td ltx_align_center" id="S4.T4.4.4.4">6.49</td>
<td class="ltx_td ltx_align_center" id="S4.T4.4.4.5">0.29</td>
<td class="ltx_td ltx_align_center" id="S4.T4.4.4.6">0.68</td>
</tr>
<tr class="ltx_tr" id="S4.T4.4.5">
<td class="ltx_td ltx_align_left ltx_border_b ltx_border_t" id="S4.T4.4.5.1">Div-SFT+RL</td>
<td class="ltx_td ltx_align_center ltx_border_b ltx_border_t" id="S4.T4.4.5.2">21.98</td>
<td class="ltx_td ltx_align_center ltx_border_b ltx_border_t" id="S4.T4.4.5.3">16.98</td>
<td class="ltx_td ltx_align_center ltx_border_b ltx_border_t" id="S4.T4.4.5.4">14.35</td>
<td class="ltx_td ltx_align_center ltx_border_b ltx_border_t" id="S4.T4.4.5.5">0.45</td>
<td class="ltx_td ltx_align_center ltx_border_b ltx_border_t" id="S4.T4.4.5.6">0.74</td>
</tr>
</table>
</figure>
<div class="ltx_para ltx_noindent" id="S4.SS4.p1">
<p class="ltx_p" id="S4.SS4.p1.7"><span class="ltx_text ltx_font_bold" id="S4.SS4.p1.7.1">Large number of samples vs. performance.</span> We also analyze how well our method discovers diverse molecules with respect to the number of generations. In this experiment, we extend beyond the settings in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.SS1" title="4.1 Comparison with decoding schemes ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Section</span> <span class="ltx_text ltx_ref_tag">4.1</span></a>. We consider our method to generate <math alttext="85" class="ltx_Math" display="inline" id="S4.SS4.p1.1.m1.1"><semantics id="S4.SS4.p1.1.m1.1a"><mn id="S4.SS4.p1.1.m1.1.1" xref="S4.SS4.p1.1.m1.1.1.cmml">85</mn><annotation-xml encoding="MathML-Content" id="S4.SS4.p1.1.m1.1b"><cn id="S4.SS4.p1.1.m1.1.1.cmml" type="integer" xref="S4.SS4.p1.1.m1.1.1">85</cn></annotation-xml><annotation encoding="application/x-tex" id="S4.SS4.p1.1.m1.1c">85</annotation><annotation encoding="application/x-llamapun" id="S4.SS4.p1.1.m1.1d">85</annotation></semantics></math>, <math alttext="120" class="ltx_Math" display="inline" id="S4.SS4.p1.2.m2.1"><semantics id="S4.SS4.p1.2.m2.1a"><mn id="S4.SS4.p1.2.m2.1.1" xref="S4.SS4.p1.2.m2.1.1.cmml">120</mn><annotation-xml encoding="MathML-Content" id="S4.SS4.p1.2.m2.1b"><cn id="S4.SS4.p1.2.m2.1.1.cmml" type="integer" xref="S4.SS4.p1.2.m2.1.1">120</cn></annotation-xml><annotation encoding="application/x-tex" id="S4.SS4.p1.2.m2.1c">120</annotation><annotation encoding="application/x-llamapun" id="S4.SS4.p1.2.m2.1d">120</annotation></semantics></math>, and <math alttext="155" class="ltx_Math" display="inline" id="S4.SS4.p1.3.m3.1"><semantics id="S4.SS4.p1.3.m3.1a"><mn id="S4.SS4.p1.3.m3.1.1" xref="S4.SS4.p1.3.m3.1.1.cmml">155</mn><annotation-xml encoding="MathML-Content" id="S4.SS4.p1.3.m3.1b"><cn id="S4.SS4.p1.3.m3.1.1.cmml" type="integer" xref="S4.SS4.p1.3.m3.1.1">155</cn></annotation-xml><annotation encoding="application/x-tex" id="S4.SS4.p1.3.m3.1c">155</annotation><annotation encoding="application/x-llamapun" id="S4.SS4.p1.3.m3.1d">155</annotation></semantics></math> molecules with a single NVIDIA A100 SXM4 40GB GPU. Next, we consider beam search with beam sizes of <math alttext="300" class="ltx_Math" display="inline" id="S4.SS4.p1.4.m4.1"><semantics id="S4.SS4.p1.4.m4.1a"><mn id="S4.SS4.p1.4.m4.1.1" xref="S4.SS4.p1.4.m4.1.1.cmml">300</mn><annotation-xml encoding="MathML-Content" id="S4.SS4.p1.4.m4.1b"><cn id="S4.SS4.p1.4.m4.1.1.cmml" type="integer" xref="S4.SS4.p1.4.m4.1.1">300</cn></annotation-xml><annotation encoding="application/x-tex" id="S4.SS4.p1.4.m4.1c">300</annotation><annotation encoding="application/x-llamapun" id="S4.SS4.p1.4.m4.1d">300</annotation></semantics></math>, <math alttext="400" class="ltx_Math" display="inline" id="S4.SS4.p1.5.m5.1"><semantics id="S4.SS4.p1.5.m5.1a"><mn id="S4.SS4.p1.5.m5.1.1" xref="S4.SS4.p1.5.m5.1.1.cmml">400</mn><annotation-xml encoding="MathML-Content" id="S4.SS4.p1.5.m5.1b"><cn id="S4.SS4.p1.5.m5.1.1.cmml" type="integer" xref="S4.SS4.p1.5.m5.1.1">400</cn></annotation-xml><annotation encoding="application/x-tex" id="S4.SS4.p1.5.m5.1c">400</annotation><annotation encoding="application/x-llamapun" id="S4.SS4.p1.5.m5.1d">400</annotation></semantics></math>, and <math alttext="500" class="ltx_Math" display="inline" id="S4.SS4.p1.6.m6.1"><semantics id="S4.SS4.p1.6.m6.1a"><mn id="S4.SS4.p1.6.m6.1.1" xref="S4.SS4.p1.6.m6.1.1.cmml">500</mn><annotation-xml encoding="MathML-Content" id="S4.SS4.p1.6.m6.1b"><cn id="S4.SS4.p1.6.m6.1.1.cmml" type="integer" xref="S4.SS4.p1.6.m6.1.1">500</cn></annotation-xml><annotation encoding="application/x-tex" id="S4.SS4.p1.6.m6.1c">500</annotation><annotation encoding="application/x-llamapun" id="S4.SS4.p1.6.m6.1d">500</annotation></semantics></math>, respectively. The beam search is implemented with four NVIDIA A100 SXM4 40GB GPUs due to the memory limitation of a single GPU. Note that this experiment uses <math alttext="250" class="ltx_Math" display="inline" id="S4.SS4.p1.7.m7.1"><semantics id="S4.SS4.p1.7.m7.1a"><mn id="S4.SS4.p1.7.m7.1.1" xref="S4.SS4.p1.7.m7.1.1.cmml">250</mn><annotation-xml encoding="MathML-Content" id="S4.SS4.p1.7.m7.1b"><cn id="S4.SS4.p1.7.m7.1.1.cmml" type="integer" xref="S4.SS4.p1.7.m7.1.1">250</cn></annotation-xml><annotation encoding="application/x-tex" id="S4.SS4.p1.7.m7.1c">250</annotation><annotation encoding="application/x-llamapun" id="S4.SS4.p1.7.m7.1d">250</annotation></semantics></math> molecular descriptions in the ChEBI-20 test set. We present the results in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.T3" title="In 4.3 Fine-tuning generalist LLMs ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Table</span> <span class="ltx_text ltx_ref_tag">3</span></a>. One can see that our method (1) discovers more diverse molecules with respect to the total number of generations and (2) exhibits further performance improvements as the number of generations increases.</p>
</div>
<div class="ltx_para ltx_noindent" id="S4.SS4.p2">
<p class="ltx_p" id="S4.SS4.p2.1"><span class="ltx_text ltx_font_bold" id="S4.SS4.p2.1.1">Time costs vs. performance.</span> In addition, we also analyze how well our method discovers diverse molecules with respect to the time costs. In <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.T3" title="In 4.3 Fine-tuning generalist LLMs ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Table</span> <span class="ltx_text ltx_ref_tag">3</span></a>, we present the time costs for each method. One can see that our method discovers more diverse molecules in a practical time compared to the decoding schemes for diverse sequence generation.</p>
</div>
<div class="ltx_para ltx_noindent" id="S4.SS4.p3">
<p class="ltx_p" id="S4.SS4.p3.1"><span class="ltx_text ltx_font_bold" id="S4.SS4.p3.1.1">SFT with hard filtering vs. RL.</span> To train LLMs to generate diverse molecules, one may also consider supervised fine-tuning on hard-filtered datasets, e.g., using a set of molecules filtered by similarity, as an alternative to reinforcement learning. In this experiment, we additionally perform supervised fine-tuning with distinct molecules, where each pair of molecules yields a Tanimoto similarity below <math alttext="0.65" class="ltx_Math" display="inline" id="S4.SS4.p3.1.m1.1"><semantics id="S4.SS4.p3.1.m1.1a"><mn id="S4.SS4.p3.1.m1.1.1" xref="S4.SS4.p3.1.m1.1.1.cmml">0.65</mn><annotation-xml encoding="MathML-Content" id="S4.SS4.p3.1.m1.1b"><cn id="S4.SS4.p3.1.m1.1.1.cmml" type="float" xref="S4.SS4.p3.1.m1.1.1">0.65</cn></annotation-xml><annotation encoding="application/x-tex" id="S4.SS4.p3.1.m1.1c">0.65</annotation><annotation encoding="application/x-llamapun" id="S4.SS4.p3.1.m1.1d">0.65</annotation></semantics></math>. The results are presented in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.T4" title="In 4.4 Ablation studies ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Table</span> <span class="ltx_text ltx_ref_tag">4</span></a>. One can see that while it yields relatively high NCircles with respect to the number of accepted unique molecules, the overall performance is worse compared to the performance of reinforcement learning.</p>
</div>
<div class="ltx_para ltx_noindent" id="S4.SS4.p4">
<p class="ltx_p" id="S4.SS4.p4.1"><span class="ltx_text ltx_font_bold" id="S4.SS4.p4.1.1">Single-stage vs. multi-stage RL.</span> As mentioned in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S3.SS2" title="3.2 Reinforcement learning ‣ 3 Method ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Section</span> <span class="ltx_text ltx_ref_tag">3.2</span></a>, we consider a multi-stage setting for generating multiple molecules. However, one may also consider a single-stage setting, where the return of a generated sequence is defined as the sum of the rewards from multiple generated molecules. In <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.T4" title="In 4.4 Ablation studies ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Table</span> <span class="ltx_text ltx_ref_tag">4</span></a>, we compare both approaches. One can see that the multi-stage setting significantly outperforms the single-stage setting. We hypothesize that this result stems from credit assignment issues in the single-stage setting. Namely, the single-stage setting lacks signals to capture molecule-wise impacts on diversity among a large set of molecules and fails to promote the generation of molecules responsible for increasing diversity.</p>
</div>
</section>
</section>
<section class="ltx_section" id="S5">
<h2 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">5 </span>Conclusion</h2>
<div class="ltx_para ltx_noindent" id="S5.p1">
<p class="ltx_p" id="S5.p1.1">In this paper, we identify the limitations of large language models (LLMs) for generating diverse molecules. Then, we present new supervised fine-tuning and reinforcement learning methods to adapt existing molecular generative LLMs to generate a diverse set of molecules. Experiments show that our approach enables LLMs to better discover diverse molecules compared to the existing approaches. An interesting avenue for future work is to reduce the space and time complexity in generating the sequence of molecules, for example, based on set encoding techniques <cite class="ltx_cite ltx_citemacro_citep">(Zaheer et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib43" title="">2017</a>)</cite>.</p>
</div>
</section>
<section class="ltx_bibliography" id="bib">
<h2 class="ltx_title ltx_title_bibliography">References</h2>
<ul class="ltx_biblist">
<li class="ltx_bibitem" id="bib.bib1">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">AI (2023)</span>
<span class="ltx_bibblock">
Mistral AI.

</span>
<span class="ltx_bibblock">Mistral 7b: A next-generation LLM.

</span>
<span class="ltx_bibblock"><a class="ltx_ref ltx_url ltx_font_typewriter" href="https://mistral.ai" title="">https://mistral.ai</a>, 2023.

</span>
<span class="ltx_bibblock">Accessed: 2024-09-30.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib2">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Bajusz et al. (2015)</span>
<span class="ltx_bibblock">
Dávid Bajusz, Anita Rácz, and Károly Héberger.

</span>
<span class="ltx_bibblock">Why is tanimoto index an appropriate choice for fingerprint-based
similarity calculations?

</span>
<span class="ltx_bibblock"><em class="ltx_emph ltx_font_italic" id="bib.bib2.1.1">Journal of cheminformatics</em>, 7(1):1–13,
2015.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib3">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Bertz (1981)</span>
<span class="ltx_bibblock">
Steven H Bertz.

</span>
<span class="ltx_bibblock">The first general index of molecular complexity.

</span>
<span class="ltx_bibblock"><em class="ltx_emph ltx_font_italic" id="bib.bib3.1.1">Journal of the American Chemical Society</em>, 103(12):3599–3601, 1981.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib4">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Bickerton et al. (2012)</span>
<span class="ltx_bibblock">
G Richard Bickerton, Gaia V Paolini, Jérémy Besnard, Sorel Muresan, and
Andrew L Hopkins.

</span>
<span class="ltx_bibblock">Quantifying the chemical beauty of drugs.

</span>
<span class="ltx_bibblock"><em class="ltx_emph ltx_font_italic" id="bib.bib4.1.1">Nature chemistry</em>, 4(2):90–98, 2012.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib5">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Christofidellis et al. (2023)</span>
<span class="ltx_bibblock">
Dimitrios Christofidellis, Giorgio Giannone, Jannis Born, Ole Winther, Teodoro
Laino, and Matteo Manica.

</span>
<span class="ltx_bibblock">Unifying molecular and textual representations via multi-task
language modelling.

</span>
<span class="ltx_bibblock">In <em class="ltx_emph ltx_font_italic" id="bib.bib5.1.1">Proceedings of the 40th International Conference on Machine
Learning</em>, volume 202 of <em class="ltx_emph ltx_font_italic" id="bib.bib5.2.2">Proceedings of Machine Learning Research</em>,
pp.  6140–6157. PMLR, 2023.

</span>
<span class="ltx_bibblock">URL <a class="ltx_ref ltx_url ltx_font_typewriter" href="https://proceedings.mlr.press/v202/christofidellis23a.html" title="">https://proceedings.mlr.press/v202/christofidellis23a.html</a>.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib6">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Edwards et al. (2021)</span>
<span class="ltx_bibblock">
Carl Edwards, ChengXiang Zhai, and Heng Ji.

</span>
<span class="ltx_bibblock">Text2Mol: Cross-modal molecule retrieval with natural language
queries.

</span>
<span class="ltx_bibblock">In Marie-Francine Moens, Xuanjing Huang, Lucia Specia, and Scott
Wen-tau Yih (eds.), <em class="ltx_emph ltx_font_italic" id="bib.bib6.1.1">Proceedings of the 2021 Conference on Empirical
Methods in Natural Language Processing</em>, pp.  595–607, Online and Punta
Cana, Dominican Republic, November 2021. Association for Computational
Linguistics.

</span>
<span class="ltx_bibblock">doi: <span class="ltx_ref ltx_nolink ltx_Url ltx_ref_self">10.18653/v1/2021.emnlp-main.47</span>.

</span>
<span class="ltx_bibblock">URL <a class="ltx_ref ltx_url ltx_font_typewriter" href="https://aclanthology.org/2021.emnlp-main.47" title="">https://aclanthology.org/2021.emnlp-main.47</a>.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib7">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Edwards et al. (2022)</span>
<span class="ltx_bibblock">
Carl Edwards, Tuan Lai, Kevin Ros, Garrett Honke, Kyunghyun Cho, and Heng Ji.

</span>
<span class="ltx_bibblock">Translation between molecules and natural language.

</span>
<span class="ltx_bibblock">In <em class="ltx_emph ltx_font_italic" id="bib.bib7.1.1">Proceedings of the 2022 Conference on Empirical Methods in
Natural Language Processing</em>, pp.  375–413, Abu Dhabi, United Arab
Emirates, December 2022. Association for Computational Linguistics.

</span>
<span class="ltx_bibblock">URL <a class="ltx_ref ltx_url ltx_font_typewriter" href="https://aclanthology.org/2022.emnlp-main.26" title="">https://aclanthology.org/2022.emnlp-main.26</a>.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib8">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Fang et al. (2024)</span>
<span class="ltx_bibblock">
Yin Fang, Xiaozhuan Liang, Ningyu Zhang, Kangwei Liu, Rui Huang, Zhuo Chen,
Xiaohui Fan, and Huajun Chen.

</span>
<span class="ltx_bibblock">Mol-instructions: A large-scale biomolecular instruction dataset
for large language models.

</span>
<span class="ltx_bibblock">In <em class="ltx_emph ltx_font_italic" id="bib.bib8.1.1">The Twelfth International Conference on Learning
Representations</em>, 2024.

</span>
<span class="ltx_bibblock">URL <a class="ltx_ref ltx_url ltx_font_typewriter" href="https://openreview.net/pdf?id=Tlsdsb6l9n" title="">https://openreview.net/pdf?id=Tlsdsb6l9n</a>.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib9">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Fruchterman &amp; Reingold (1991)</span>
<span class="ltx_bibblock">
Thomas MJ Fruchterman and Edward M Reingold.

</span>
<span class="ltx_bibblock">Graph drawing by force-directed placement.

</span>
<span class="ltx_bibblock"><em class="ltx_emph ltx_font_italic" id="bib.bib9.1.1">Software: Practice and experience</em>, 21(11):1129–1164, 1991.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib10">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Ghugare et al. (2024)</span>
<span class="ltx_bibblock">
Raj Ghugare, Santiago Miret, Adriana Hugessen, Mariano Phielipp, and Glen
Berseth.

</span>
<span class="ltx_bibblock">Searching for high-value molecules using reinforcement learning and
transformers.

</span>
<span class="ltx_bibblock">In <em class="ltx_emph ltx_font_italic" id="bib.bib10.1.1">The Twelfth International Conference on Learning
Representations</em>, 2024.

</span>
<span class="ltx_bibblock">URL <a class="ltx_ref ltx_url ltx_font_typewriter" href="https://openreview.net/forum?id=nqlymMx42E" title="">https://openreview.net/forum?id=nqlymMx42E</a>.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib11">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Gruver et al. (2024)</span>
<span class="ltx_bibblock">
Nate Gruver, Anuroop Sriram, Andrea Madotto, Andrew Gordon Wilson, C. Lawrence
Zitnick, and Zachary Ward Ulissi.

</span>
<span class="ltx_bibblock">Fine-tuned language models generate stable inorganic materials as
text.

</span>
<span class="ltx_bibblock">In <em class="ltx_emph ltx_font_italic" id="bib.bib11.1.1">The Twelfth International Conference on Learning
Representations</em>, 2024.

</span>
<span class="ltx_bibblock">URL <a class="ltx_ref ltx_url ltx_font_typewriter" href="https://openreview.net/forum?id=vN9fpfqoP1" title="">https://openreview.net/forum?id=vN9fpfqoP1</a>.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib12">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Holtzman et al. (2020)</span>
<span class="ltx_bibblock">
Ari Holtzman, Jan Buys, Li Du, Maxwell Forbes, and Yejin Choi.

</span>
<span class="ltx_bibblock">The curious case of neural text degeneration.

</span>
<span class="ltx_bibblock">In <em class="ltx_emph ltx_font_italic" id="bib.bib12.1.1">The Tenth International Conference on Learning
Representations</em>, 2020.

</span>
<span class="ltx_bibblock">URL <a class="ltx_ref ltx_url ltx_font_typewriter" href="https://openreview.net/forum?id=rygGQyrFvH" title="">https://openreview.net/forum?id=rygGQyrFvH</a>.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib13">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Hong et al. (2020)</span>
<span class="ltx_bibblock">
Benke Hong, Tuoping Luo, and Xiaoguang Lei.

</span>
<span class="ltx_bibblock">Late-stage diversification of natural products.

</span>
<span class="ltx_bibblock"><em class="ltx_emph ltx_font_italic" id="bib.bib13.1.1">ACS central science</em>, 6(5):622–635, 2020.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib14">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Hu et al. (2022)</span>
<span class="ltx_bibblock">
Edward J Hu, yelong shen, Phillip Wallis, Zeyuan Allen-Zhu, Yuanzhi Li, Shean
Wang, Lu Wang, and Weizhu Chen.

</span>
<span class="ltx_bibblock">LoRA: Low-rank adaptation of large language models.

</span>
<span class="ltx_bibblock">In <em class="ltx_emph ltx_font_italic" id="bib.bib14.1.1">The Tenth International Conference on Learning
Representations</em>, 2022.

</span>
<span class="ltx_bibblock">URL <a class="ltx_ref ltx_url ltx_font_typewriter" href="https://openreview.net/forum?id=nZeVKeeFYf9" title="">https://openreview.net/forum?id=nZeVKeeFYf9</a>.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib15">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Kirk et al. (2024)</span>
<span class="ltx_bibblock">
Robert Kirk, Ishita Mediratta, Christoforos Nalmpantis, Jelena Luketina, Eric
Hambro, Edward Grefenstette, and Roberta Raileanu.

</span>
<span class="ltx_bibblock">Understanding the effects of rlhf on LLM generalisation and
diversity.

</span>
<span class="ltx_bibblock">In <em class="ltx_emph ltx_font_italic" id="bib.bib15.1.1">The Twelfth International Conference on Learning
Representations</em>, 2024.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib16">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Krantz (1998)</span>
<span class="ltx_bibblock">
Allen Krantz.

</span>
<span class="ltx_bibblock">Diversification of the drug discovery process.

</span>
<span class="ltx_bibblock"><em class="ltx_emph ltx_font_italic" id="bib.bib16.1.1">Nature biotechnology</em>, 16(13):1294–1294,
1998.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib17">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Krenn et al. (2020)</span>
<span class="ltx_bibblock">
Mario Krenn, Florian Häse, AkshatKumar Nigam, Pascal Friederich, and Alan
Aspuru-Guzik.

</span>
<span class="ltx_bibblock">Self-referencing embedded strings (selfies): A 100% robust molecular
string representation.

</span>
<span class="ltx_bibblock"><em class="ltx_emph ltx_font_italic" id="bib.bib17.1.1">Machine Learning: Science and Technology</em>, 1(4):045024, 2020.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib18">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Landrum (2016)</span>
<span class="ltx_bibblock">
Greg Landrum.

</span>
<span class="ltx_bibblock">Rdkit: Open-source cheminformatics software.

</span>
<span class="ltx_bibblock"><a class="ltx_ref ltx_url ltx_font_typewriter" href="http://www.rdkit.org" title="">http://www.rdkit.org</a>, 2016.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib19">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Li et al. (2024)</span>
<span class="ltx_bibblock">
Jiatong Li, Yunqing Liu, Wenqi Fan, Xiao-Yong Wei, Hui Liu, Jiliang Tang, and
Qing Li.

</span>
<span class="ltx_bibblock">Empowering molecule discovery for molecule-caption translation with
large language models: A chatgpt perspective.

</span>
<span class="ltx_bibblock"><em class="ltx_emph ltx_font_italic" id="bib.bib19.1.1">IEEE Transactions on Knowledge and Data Engineering</em>, 2024.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib20">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Och (2003)</span>
<span class="ltx_bibblock">
Franz Josef Och.

</span>
<span class="ltx_bibblock">Minimum error rate training in statistical machine translation.

</span>
<span class="ltx_bibblock">In <em class="ltx_emph ltx_font_italic" id="bib.bib20.1.1">Proceedings of the 41st Annual Meeting of the Association
for Computational Linguistics</em>, pp.  160–167, Sapporo, Japan, July 2003.
Association for Computational Linguistics.

</span>
<span class="ltx_bibblock">doi: <span class="ltx_ref ltx_nolink ltx_Url ltx_ref_self">10.3115/1075096.1075117</span>.

</span>
<span class="ltx_bibblock">URL <a class="ltx_ref ltx_url ltx_font_typewriter" href="https://aclanthology.org/P03-1021" title="">https://aclanthology.org/P03-1021</a>.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib21">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">OpenAI (2023)</span>
<span class="ltx_bibblock">
OpenAI.

</span>
<span class="ltx_bibblock">Chatgpt: Openai language model.

</span>
<span class="ltx_bibblock"><a class="ltx_ref ltx_url ltx_font_typewriter" href="https://openai.com/chatgpt" title="">https://openai.com/chatgpt</a>, 2023.

</span>
<span class="ltx_bibblock">Accessed: 2023-09-23.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib22">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">OpenAI (2024)</span>
<span class="ltx_bibblock">
OpenAI.

</span>
<span class="ltx_bibblock">Introducing openai o1 preview, 2024.

</span>
<span class="ltx_bibblock">URL <a class="ltx_ref ltx_url ltx_font_typewriter" href="https://openai.com/index/introducing-openai-o1-preview/" title="">https://openai.com/index/introducing-openai-o1-preview/</a>.

</span>
<span class="ltx_bibblock">Accessed: 2024-09-17.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib23">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Ouyang et al. (2022)</span>
<span class="ltx_bibblock">
Long Ouyang, Jeffrey Wu, Xu Jiang, Diogo Almeida, Carroll Wainwright, Pamela
Mishkin, Chong Zhang, Sandhini Agarwal, Katarina Slama, Alex Ray, et al.

</span>
<span class="ltx_bibblock">Training language models to follow instructions with human feedback.

</span>
<span class="ltx_bibblock"><em class="ltx_emph ltx_font_italic" id="bib.bib23.1.1">Advances in neural information processing systems</em>,
35:27730–27744, 2022.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib24">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Papineni et al. (2002)</span>
<span class="ltx_bibblock">
Kishore Papineni, Salim Roukos, Todd Ward, and Wei-Jing Zhu.

</span>
<span class="ltx_bibblock">Bleu: a method for automatic evaluation of machine translation.

</span>
<span class="ltx_bibblock">In Pierre Isabelle, Eugene Charniak, and Dekang Lin (eds.),
<em class="ltx_emph ltx_font_italic" id="bib.bib24.1.1">Proceedings of the 40th Annual Meeting of the Association for
Computational Linguistics</em>, pp.  311–318, Philadelphia, Pennsylvania, USA,
July 2002. Association for Computational Linguistics.

</span>
<span class="ltx_bibblock">doi: <span class="ltx_ref ltx_nolink ltx_Url ltx_ref_self">10.3115/1073083.1073135</span>.

</span>
<span class="ltx_bibblock">URL <a class="ltx_ref ltx_url ltx_font_typewriter" href="https://aclanthology.org/P02-1040" title="">https://aclanthology.org/P02-1040</a>.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib25">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Pei et al. (2023)</span>
<span class="ltx_bibblock">
Qizhi Pei, Wei Zhang, Jinhua Zhu, Kehan Wu, Kaiyuan Gao, Lijun Wu, Yingce Xia,
and Rui Yan.

</span>
<span class="ltx_bibblock">Biot5: Enriching cross-modal integration in biology with chemical
knowledge and natural language associations.

</span>
<span class="ltx_bibblock">In <em class="ltx_emph ltx_font_italic" id="bib.bib25.1.1">Proceedings of the 2023 Conference on Empirical Methods in
Natural Language Processing</em>, pp.  1102–1123. Association for Computational
Linguistics, December 2023.

</span>
<span class="ltx_bibblock">URL <a class="ltx_ref ltx_url ltx_font_typewriter" href="https://aclanthology.org/2023.emnlp-main.70" title="">https://aclanthology.org/2023.emnlp-main.70</a>.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib26">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Pei et al. (2024)</span>
<span class="ltx_bibblock">
Qizhi Pei, Lijun Wu, Kaiyuan Gao, Xiaozhuan Liang, Yin Fang, Jinhua Zhu,
Shufang Xie, Tao Qin, and Rui Yan.

</span>
<span class="ltx_bibblock">BioT5+: Towards generalized biological understanding with IUPAC
integration and multi-task tuning.

</span>
<span class="ltx_bibblock">In Lun-Wei Ku, Andre Martins, and Vivek Srikumar (eds.),
<em class="ltx_emph ltx_font_italic" id="bib.bib26.1.1">Findings of the Association for Computational Linguistics ACL 2024</em>,
pp.  1216–1240, Bangkok, Thailand and virtual meeting, August 2024.
Association for Computational Linguistics.

</span>
<span class="ltx_bibblock">URL <a class="ltx_ref ltx_url ltx_font_typewriter" href="https://aclanthology.org/2024.findings-acl.71" title="">https://aclanthology.org/2024.findings-acl.71</a>.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib27">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Polykovskiy et al. (2020)</span>
<span class="ltx_bibblock">
Daniil Polykovskiy, Alexander Zhebrak, Benjamin Sanchez-Lengeling, Sergey
Golovanov, Oktai Tatanov, Stanislav Belyaev, Rauf Kurbanov, Aleksey
Artamonov, Vladimir Aladinskiy, Mark Veselov, Artur Kadurin, Simon Johansson,
Hongming Chen, Sergey Nikolenko, Alan Aspuru-Guzik, and Alex Zhavoronkov.

</span>
<span class="ltx_bibblock">Molecular Sets (MOSES): A Benchmarking Platform for
Molecular Generation Models.

</span>
<span class="ltx_bibblock"><em class="ltx_emph ltx_font_italic" id="bib.bib27.1.1">Frontiers in Pharmacology</em>, 2020.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib28">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Sadybekov &amp; Katritch (2023)</span>
<span class="ltx_bibblock">
Anastasiia V Sadybekov and Vsevolod Katritch.

</span>
<span class="ltx_bibblock">Computational approaches streamlining drug discovery.

</span>
<span class="ltx_bibblock"><em class="ltx_emph ltx_font_italic" id="bib.bib28.1.1">Nature</em>, 616(7958):673–685, 2023.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib29">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Schulman et al. (2017)</span>
<span class="ltx_bibblock">
John Schulman, Filip Wolski, Prafulla Dhariwal, Alec Radford, and Oleg Klimov.

</span>
<span class="ltx_bibblock">Proximal policy optimization algorithms.

</span>
<span class="ltx_bibblock"><em class="ltx_emph ltx_font_italic" id="bib.bib29.1.1">arXiv preprint arXiv:1707.06347</em>, 2017.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib30">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Shani et al. (2024)</span>
<span class="ltx_bibblock">
Lior Shani, Aviv Rosenberg, Asaf Cassel, Oran Lang, Daniele Calandriello,
Avital Zipori, Hila Noga, Orgad Keller, Bilal Piot, Idan Szpektor, et al.

</span>
<span class="ltx_bibblock">Multi-turn reinforcement learning from preference human feedback.

</span>
<span class="ltx_bibblock"><em class="ltx_emph ltx_font_italic" id="bib.bib30.1.1">arXiv preprint arXiv:2405.14655</em>, 2024.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib31">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Su et al. (2022)</span>
<span class="ltx_bibblock">
Yixuan Su, Tian Lan, Yan Wang, Dani Yogatama, Lingpeng Kong, and Nigel Collier.

</span>
<span class="ltx_bibblock">A contrastive framework for neural text generation.

</span>
<span class="ltx_bibblock"><em class="ltx_emph ltx_font_italic" id="bib.bib31.1.1">Advances in Neural Information Processing Systems</em>,
35:21548–21561, 2022.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib32">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Taylor et al. (2022)</span>
<span class="ltx_bibblock">
Ross Taylor, Marcin Kardas, Guillem Cucurull, Thomas Scialom, Anthony
Hartshorn, Elvis Saravia, Andrew Poulton, Viktor Kerkez, and Robert Stojnic.

</span>
<span class="ltx_bibblock">Galactica: A large language model for science.

</span>
<span class="ltx_bibblock">2022.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib33">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Touvron et al. (2023)</span>
<span class="ltx_bibblock">
Hugo Touvron, Thibaut Lavril, Gautier Izacard, Xavier Martinet, Marie-Anne
Lachaux, Timothée Lacroix, Baptiste Rozière, Naman Goyal, Eric Hambro,
Faisal Azhar, Aurelien Rodriguez, Armand Joulin, Edouard Grave, and Guillaume
Lample.

</span>
<span class="ltx_bibblock">Llama: Open and efficient foundation language models, 2023.

</span>
<span class="ltx_bibblock">URL <a class="ltx_ref ltx_url ltx_font_typewriter" href="https://arxiv.org/abs/2302.13971" title="">https://arxiv.org/abs/2302.13971</a>.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib34">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Vamathevan et al. (2019)</span>
<span class="ltx_bibblock">
Jessica Vamathevan, Dominic Clark, Paul Czodrowski, Ian Dunham, Edgardo Ferran,
George Lee, Bin Li, Anant Madabhushi, Parantu Shah, Michaela Spitzer, et al.

</span>
<span class="ltx_bibblock">Applications of machine learning in drug discovery and development.

</span>
<span class="ltx_bibblock"><em class="ltx_emph ltx_font_italic" id="bib.bib34.1.1">Nature reviews Drug discovery</em>, 18(6):463–477, 2019.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib35">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Vijayakumar et al. (2018)</span>
<span class="ltx_bibblock">
Ashwin Vijayakumar, Michael Cogswell, Ramprasaath Selvaraju, Qing Sun, Stefan
Lee, David Crandall, and Dhruv Batra.

</span>
<span class="ltx_bibblock">Diverse beam search for improved description of complex scenes.

</span>
<span class="ltx_bibblock"><em class="ltx_emph ltx_font_italic" id="bib.bib35.1.1">Proceedings of the AAAI Conference on Artificial Intelligence</em>,
32(1), Apr. 2018.

</span>
<span class="ltx_bibblock">doi: <span class="ltx_ref ltx_nolink ltx_Url ltx_ref_self">10.1609/aaai.v32i1.12340</span>.

</span>
<span class="ltx_bibblock">URL <a class="ltx_ref ltx_url ltx_font_typewriter" href="https://ojs.aaai.org/index.php/AAAI/article/view/12340" title="">https://ojs.aaai.org/index.php/AAAI/article/view/12340</a>.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib36">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Vijayakumar et al. (2016)</span>
<span class="ltx_bibblock">
Ashwin K Vijayakumar, Michael Cogswell, Ramprasath R Selvaraju, Qing Sun,
Stefan Lee, David Crandall, and Dhruv Batra.

</span>
<span class="ltx_bibblock">Diverse beam search: Decoding diverse solutions from neural sequence
models.

</span>
<span class="ltx_bibblock"><em class="ltx_emph ltx_font_italic" id="bib.bib36.1.1">arXiv preprint arXiv:1610.02424</em>, 2016.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib37">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Weininger (1988)</span>
<span class="ltx_bibblock">
David Weininger.

</span>
<span class="ltx_bibblock">Smiles, a chemical language and information system. 1. introduction
to methodology and encoding rules.

</span>
<span class="ltx_bibblock"><em class="ltx_emph ltx_font_italic" id="bib.bib37.1.1">Journal of chemical information and computer sciences</em>,
28(1):31–36, 1988.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib38">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Wu et al. (2023a)</span>
<span class="ltx_bibblock">
Sifan Wu, Amir Khasahmadi, Mor Katz, Pradeep Kumar Jayaraman, Yewen Pu, Karl
Willis, and Bang Liu.

</span>
<span class="ltx_bibblock">Cad-LLM: Large language model for cad generation.

</span>
<span class="ltx_bibblock">In <em class="ltx_emph ltx_font_italic" id="bib.bib38.1.1">NeurIPS 2023 Workshop on Machine Learning for Creativity and
Design</em>. NeurIPS, 2023a.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib39">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Wu et al. (2023b)</span>
<span class="ltx_bibblock">
Zeqiu Wu, Yushi Hu, Weijia Shi, Nouha Dziri, Alane Suhr, Prithviraj
Ammanabrolu, Noah A. Smith, Mari Ostendorf, and Hannaneh Hajishirzi.

</span>
<span class="ltx_bibblock">Fine-grained human feedback gives better rewards for language model
training.

</span>
<span class="ltx_bibblock">In <em class="ltx_emph ltx_font_italic" id="bib.bib39.1.1">Thirty-seventh Conference on Neural Information Processing
Systems</em>, 2023b.

</span>
<span class="ltx_bibblock">URL <a class="ltx_ref ltx_url ltx_font_typewriter" href="https://openreview.net/forum?id=CSbGXyCswu" title="">https://openreview.net/forum?id=CSbGXyCswu</a>.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib40">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Xie et al. (2023)</span>
<span class="ltx_bibblock">
Yutong Xie, Ziqiao Xu, Jiaqi Ma, and Qiaozhu Mei.

</span>
<span class="ltx_bibblock">How much space has been explored? measuring the chemical space
covered by databases and machine-generated molecules.

</span>
<span class="ltx_bibblock">In <em class="ltx_emph ltx_font_italic" id="bib.bib40.1.1">The Eleventh International Conference on Learning
Representations</em>, 2023.

</span>
<span class="ltx_bibblock">URL <a class="ltx_ref ltx_url ltx_font_typewriter" href="https://openreview.net/forum?id=Yo06F8kfMa1" title="">https://openreview.net/forum?id=Yo06F8kfMa1</a>.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib41">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Ye et al. (2023)</span>
<span class="ltx_bibblock">
Geyan Ye, Xibao Cai, Houtim Lai, Xing Wang, Junhong Huang, Longyue Wang, Wei
Liu, and Xiangxiang Zeng.

</span>
<span class="ltx_bibblock">Drugassist: A large language model for molecule optimization.

</span>
<span class="ltx_bibblock"><em class="ltx_emph ltx_font_italic" id="bib.bib41.1.1">arXiv preprint arXiv:2401.10334</em>, 2023.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib42">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Yu et al. (2024)</span>
<span class="ltx_bibblock">
Botao Yu, Frazier N. Baker, Ziqi Chen, Xia Ning, and Huan Sun.

</span>
<span class="ltx_bibblock">LlaSMol: Advancing large language models for chemistry with a
large-scale, comprehensive, high-quality instruction tuning dataset.

</span>
<span class="ltx_bibblock">In <em class="ltx_emph ltx_font_italic" id="bib.bib42.1.1">First Conference on Language Modeling</em>, 2024.

</span>
<span class="ltx_bibblock">URL <a class="ltx_ref ltx_url ltx_font_typewriter" href="https://openreview.net/forum?id=lY6XTF9tPv" title="">https://openreview.net/forum?id=lY6XTF9tPv</a>.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib43">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Zaheer et al. (2017)</span>
<span class="ltx_bibblock">
Manzil Zaheer, Satwik Kottur, Siamak Ravanbakhsh, Barnabas Poczos, Russ R
Salakhutdinov, and Alexander J Smola.

</span>
<span class="ltx_bibblock">Deep sets.

</span>
<span class="ltx_bibblock">In I. Guyon, U. Von Luxburg, S. Bengio, H. Wallach, R. Fergus,
S. Vishwanathan, and R. Garnett (eds.), <em class="ltx_emph ltx_font_italic" id="bib.bib43.1.1">Advances in Neural Information
Processing Systems</em>, volume 30. Curran Associates, Inc., 2017.

</span>
<span class="ltx_bibblock">URL
<a class="ltx_ref ltx_url ltx_font_typewriter" href="https://proceedings.neurips.cc/paper_files/paper/2017/file/f22e4747da1aa27e363d86d40ff442fe-Paper.pdf" title="">https://proceedings.neurips.cc/paper_files/paper/2017/file/f22e4747da1aa27e363d86d40ff442fe-Paper.pdf</a>.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib44">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Zhou et al. (2024)</span>
<span class="ltx_bibblock">
Yifei Zhou, Andrea Zanette, Jiayi Pan, Sergey Levine, and Aviral Kumar.

</span>
<span class="ltx_bibblock">Archer: Training language model agents via hierarchical multi-turn
rl.

</span>
<span class="ltx_bibblock"><em class="ltx_emph ltx_font_italic" id="bib.bib44.1.1">arXiv preprint arXiv:2402.19446</em>, 2024.

</span>
</li>
<li class="ltx_bibitem" id="bib.bib45">
<span class="ltx_tag ltx_role_refnum ltx_tag_bibitem">Zhuo et al. (2024)</span>
<span class="ltx_bibblock">
Le Zhuo, Zewen Chi, Minghao Xu, Heyan Huang, Heqi Zheng, Conghui He, Xian-Ling
Mao, and Wentao Zhang.

</span>
<span class="ltx_bibblock">ProtLLM: An interleaved protein-language LLM with protein-as-word
pre-training.

</span>
<span class="ltx_bibblock"><em class="ltx_emph ltx_font_italic" id="bib.bib45.1.1">arXiv preprint arXiv:2403.07920</em>, 2024.

</span>
</li>
</ul>
</section>
<div class="ltx_pagination ltx_role_newpage"></div>
<section class="ltx_appendix" id="A1">
<h2 class="ltx_title ltx_title_appendix">
<span class="ltx_tag ltx_tag_appendix">Appendix A </span>Detailed implementations and training</h2>
<section class="ltx_subsection" id="A1.SS1">
<h3 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">A.1 </span>Supervised fine-tuning</h3>
<div class="ltx_para ltx_noindent" id="A1.SS1.p1">
<p class="ltx_p" id="A1.SS1.p1.3"><span class="ltx_text ltx_font_bold" id="A1.SS1.p1.3.1">Dataset Collection.</span> To fine-tune BioT5<sup class="ltx_sup" id="A1.SS1.p1.3.2"><span class="ltx_text ltx_font_italic" id="A1.SS1.p1.3.2.1">+</span></sup> and MolT5 (<a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.SS1" title="4.1 Comparison with decoding schemes ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Sections</span> <span class="ltx_text ltx_ref_tag">4.1</span></a> and <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.SS2" title="4.2 Comparison with existing LLMs ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">4.2</span></a>), we collect <math alttext="T=100" class="ltx_Math" display="inline" id="A1.SS1.p1.2.m2.1"><semantics id="A1.SS1.p1.2.m2.1a"><mrow id="A1.SS1.p1.2.m2.1.1" xref="A1.SS1.p1.2.m2.1.1.cmml"><mi id="A1.SS1.p1.2.m2.1.1.2" xref="A1.SS1.p1.2.m2.1.1.2.cmml">T</mi><mo id="A1.SS1.p1.2.m2.1.1.1" xref="A1.SS1.p1.2.m2.1.1.1.cmml">=</mo><mn id="A1.SS1.p1.2.m2.1.1.3" xref="A1.SS1.p1.2.m2.1.1.3.cmml">100</mn></mrow><annotation-xml encoding="MathML-Content" id="A1.SS1.p1.2.m2.1b"><apply id="A1.SS1.p1.2.m2.1.1.cmml" xref="A1.SS1.p1.2.m2.1.1"><eq id="A1.SS1.p1.2.m2.1.1.1.cmml" xref="A1.SS1.p1.2.m2.1.1.1"></eq><ci id="A1.SS1.p1.2.m2.1.1.2.cmml" xref="A1.SS1.p1.2.m2.1.1.2">𝑇</ci><cn id="A1.SS1.p1.2.m2.1.1.3.cmml" type="integer" xref="A1.SS1.p1.2.m2.1.1.3">100</cn></apply></annotation-xml><annotation encoding="application/x-tex" id="A1.SS1.p1.2.m2.1c">T=100</annotation><annotation encoding="application/x-llamapun" id="A1.SS1.p1.2.m2.1d">italic_T = 100</annotation></semantics></math> molecules using beam search for each training molecular description in the ChEBI-20 training dataset. Note that this dataset has been considered in the original BioT5<sup class="ltx_sup" id="A1.SS1.p1.3.3"><span class="ltx_text ltx_font_italic" id="A1.SS1.p1.3.3.1">+</span></sup> and MolT5.</p>
</div>
<div class="ltx_para ltx_noindent" id="A1.SS1.p2">
<p class="ltx_p" id="A1.SS1.p2.2">For the fine-tuning of DrugAssist (<a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.SS3" title="4.3 Fine-tuning generalist LLMs ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Section</span> <span class="ltx_text ltx_ref_tag">4.3</span></a>), we collect <math alttext="T=300" class="ltx_Math" display="inline" id="A1.SS1.p2.1.m1.1"><semantics id="A1.SS1.p2.1.m1.1a"><mrow id="A1.SS1.p2.1.m1.1.1" xref="A1.SS1.p2.1.m1.1.1.cmml"><mi id="A1.SS1.p2.1.m1.1.1.2" xref="A1.SS1.p2.1.m1.1.1.2.cmml">T</mi><mo id="A1.SS1.p2.1.m1.1.1.1" xref="A1.SS1.p2.1.m1.1.1.1.cmml">=</mo><mn id="A1.SS1.p2.1.m1.1.1.3" xref="A1.SS1.p2.1.m1.1.1.3.cmml">300</mn></mrow><annotation-xml encoding="MathML-Content" id="A1.SS1.p2.1.m1.1b"><apply id="A1.SS1.p2.1.m1.1.1.cmml" xref="A1.SS1.p2.1.m1.1.1"><eq id="A1.SS1.p2.1.m1.1.1.1.cmml" xref="A1.SS1.p2.1.m1.1.1.1"></eq><ci id="A1.SS1.p2.1.m1.1.1.2.cmml" xref="A1.SS1.p2.1.m1.1.1.2">𝑇</ci><cn id="A1.SS1.p2.1.m1.1.1.3.cmml" type="integer" xref="A1.SS1.p2.1.m1.1.1.3">300</cn></apply></annotation-xml><annotation encoding="application/x-tex" id="A1.SS1.p2.1.m1.1c">T=300</annotation><annotation encoding="application/x-llamapun" id="A1.SS1.p2.1.m1.1d">italic_T = 300</annotation></semantics></math> molecules using beam search for each training prompt. Note that the collected molecules were filtered to remove invalid string representations, duplicated molecules, and molecules that do not satisfy the given description. The invalid string representations are evaluated with RDKit package <cite class="ltx_cite ltx_citemacro_citep">(Landrum, <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib18" title="">2016</a>)</cite>. Additionally, the collected molecules are concatenated into a single sequence <math alttext="m_{1}||\cdots||m_{K}" class="ltx_Math" display="inline" id="A1.SS1.p2.2.m2.1"><semantics id="A1.SS1.p2.2.m2.1a"><mrow id="A1.SS1.p2.2.m2.1.2" xref="A1.SS1.p2.2.m2.1.2.cmml"><msub id="A1.SS1.p2.2.m2.1.2.2" xref="A1.SS1.p2.2.m2.1.2.2.cmml"><mi id="A1.SS1.p2.2.m2.1.2.2.2" xref="A1.SS1.p2.2.m2.1.2.2.2.cmml">m</mi><mn id="A1.SS1.p2.2.m2.1.2.2.3" xref="A1.SS1.p2.2.m2.1.2.2.3.cmml">1</mn></msub><mo id="A1.SS1.p2.2.m2.1.2.1" xref="A1.SS1.p2.2.m2.1.2.1.cmml">⁢</mo><mrow id="A1.SS1.p2.2.m2.1.2.3.2" xref="A1.SS1.p2.2.m2.1.2.3.1.cmml"><mo id="A1.SS1.p2.2.m2.1.2.3.2.1" stretchy="false" xref="A1.SS1.p2.2.m2.1.2.3.1.1.cmml">‖</mo><mi id="A1.SS1.p2.2.m2.1.1" mathvariant="normal" xref="A1.SS1.p2.2.m2.1.1.cmml">⋯</mi><mo id="A1.SS1.p2.2.m2.1.2.3.2.2" stretchy="false" xref="A1.SS1.p2.2.m2.1.2.3.1.1.cmml">‖</mo></mrow><mo id="A1.SS1.p2.2.m2.1.2.1a" xref="A1.SS1.p2.2.m2.1.2.1.cmml">⁢</mo><msub id="A1.SS1.p2.2.m2.1.2.4" xref="A1.SS1.p2.2.m2.1.2.4.cmml"><mi id="A1.SS1.p2.2.m2.1.2.4.2" xref="A1.SS1.p2.2.m2.1.2.4.2.cmml">m</mi><mi id="A1.SS1.p2.2.m2.1.2.4.3" xref="A1.SS1.p2.2.m2.1.2.4.3.cmml">K</mi></msub></mrow><annotation-xml encoding="MathML-Content" id="A1.SS1.p2.2.m2.1b"><apply id="A1.SS1.p2.2.m2.1.2.cmml" xref="A1.SS1.p2.2.m2.1.2"><times id="A1.SS1.p2.2.m2.1.2.1.cmml" xref="A1.SS1.p2.2.m2.1.2.1"></times><apply id="A1.SS1.p2.2.m2.1.2.2.cmml" xref="A1.SS1.p2.2.m2.1.2.2"><csymbol cd="ambiguous" id="A1.SS1.p2.2.m2.1.2.2.1.cmml" xref="A1.SS1.p2.2.m2.1.2.2">subscript</csymbol><ci id="A1.SS1.p2.2.m2.1.2.2.2.cmml" xref="A1.SS1.p2.2.m2.1.2.2.2">𝑚</ci><cn id="A1.SS1.p2.2.m2.1.2.2.3.cmml" type="integer" xref="A1.SS1.p2.2.m2.1.2.2.3">1</cn></apply><apply id="A1.SS1.p2.2.m2.1.2.3.1.cmml" xref="A1.SS1.p2.2.m2.1.2.3.2"><csymbol cd="latexml" id="A1.SS1.p2.2.m2.1.2.3.1.1.cmml" xref="A1.SS1.p2.2.m2.1.2.3.2.1">norm</csymbol><ci id="A1.SS1.p2.2.m2.1.1.cmml" xref="A1.SS1.p2.2.m2.1.1">⋯</ci></apply><apply id="A1.SS1.p2.2.m2.1.2.4.cmml" xref="A1.SS1.p2.2.m2.1.2.4"><csymbol cd="ambiguous" id="A1.SS1.p2.2.m2.1.2.4.1.cmml" xref="A1.SS1.p2.2.m2.1.2.4">subscript</csymbol><ci id="A1.SS1.p2.2.m2.1.2.4.2.cmml" xref="A1.SS1.p2.2.m2.1.2.4.2">𝑚</ci><ci id="A1.SS1.p2.2.m2.1.2.4.3.cmml" xref="A1.SS1.p2.2.m2.1.2.4.3">𝐾</ci></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="A1.SS1.p2.2.m2.1c">m_{1}||\cdots||m_{K}</annotation><annotation encoding="application/x-llamapun" id="A1.SS1.p2.2.m2.1d">italic_m start_POSTSUBSCRIPT 1 end_POSTSUBSCRIPT | | ⋯ | | italic_m start_POSTSUBSCRIPT italic_K end_POSTSUBSCRIPT</annotation></semantics></math>. Note that two molecules are separated by introducing a newline character (‘\n’).</p>
</div>
<div class="ltx_para ltx_noindent" id="A1.SS1.p3">
<p class="ltx_p" id="A1.SS1.p3.1"><span class="ltx_text ltx_font_bold" id="A1.SS1.p3.1.1">Supervised learning.</span> We consider four NVIDIA A100 GPUs for supervised fine-tuning.</p>
</div>
<div class="ltx_para ltx_noindent" id="A1.SS1.p4">
<ul class="ltx_itemize" id="A1.I1">
<li class="ltx_item" id="A1.I1.i1" style="list-style-type:none;">
<span class="ltx_tag ltx_tag_item">•</span>
<div class="ltx_para" id="A1.I1.i1.p1">
<p class="ltx_p" id="A1.I1.i1.p1.6">For the supervised fine-tuning of BioT5<sup class="ltx_sup" id="A1.I1.i1.p1.6.1"><span class="ltx_text ltx_font_italic" id="A1.I1.i1.p1.6.1.1">+</span></sup> and MolT5 (<a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.SS1" title="4.1 Comparison with decoding schemes ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Sections</span> <span class="ltx_text ltx_ref_tag">4.1</span></a> and <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.SS2" title="4.2 Comparison with existing LLMs ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">4.2</span></a>), we consider <math alttext="80" class="ltx_Math" display="inline" id="A1.I1.i1.p1.2.m2.1"><semantics id="A1.I1.i1.p1.2.m2.1a"><mn id="A1.I1.i1.p1.2.m2.1.1" xref="A1.I1.i1.p1.2.m2.1.1.cmml">80</mn><annotation-xml encoding="MathML-Content" id="A1.I1.i1.p1.2.m2.1b"><cn id="A1.I1.i1.p1.2.m2.1.1.cmml" type="integer" xref="A1.I1.i1.p1.2.m2.1.1">80</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I1.i1.p1.2.m2.1c">80</annotation><annotation encoding="application/x-llamapun" id="A1.I1.i1.p1.2.m2.1d">80</annotation></semantics></math> epochs, <math alttext="8" class="ltx_Math" display="inline" id="A1.I1.i1.p1.3.m3.1"><semantics id="A1.I1.i1.p1.3.m3.1a"><mn id="A1.I1.i1.p1.3.m3.1.1" xref="A1.I1.i1.p1.3.m3.1.1.cmml">8</mn><annotation-xml encoding="MathML-Content" id="A1.I1.i1.p1.3.m3.1b"><cn id="A1.I1.i1.p1.3.m3.1.1.cmml" type="integer" xref="A1.I1.i1.p1.3.m3.1.1">8</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I1.i1.p1.3.m3.1c">8</annotation><annotation encoding="application/x-llamapun" id="A1.I1.i1.p1.3.m3.1d">8</annotation></semantics></math>-batch size, <math alttext="5\textrm{e}-4" class="ltx_Math" display="inline" id="A1.I1.i1.p1.4.m4.1"><semantics id="A1.I1.i1.p1.4.m4.1a"><mrow id="A1.I1.i1.p1.4.m4.1.1" xref="A1.I1.i1.p1.4.m4.1.1.cmml"><mrow id="A1.I1.i1.p1.4.m4.1.1.2" xref="A1.I1.i1.p1.4.m4.1.1.2.cmml"><mn id="A1.I1.i1.p1.4.m4.1.1.2.2" xref="A1.I1.i1.p1.4.m4.1.1.2.2.cmml">5</mn><mo id="A1.I1.i1.p1.4.m4.1.1.2.1" xref="A1.I1.i1.p1.4.m4.1.1.2.1.cmml">⁢</mo><mtext id="A1.I1.i1.p1.4.m4.1.1.2.3" xref="A1.I1.i1.p1.4.m4.1.1.2.3a.cmml">e</mtext></mrow><mo id="A1.I1.i1.p1.4.m4.1.1.1" xref="A1.I1.i1.p1.4.m4.1.1.1.cmml">−</mo><mn id="A1.I1.i1.p1.4.m4.1.1.3" xref="A1.I1.i1.p1.4.m4.1.1.3.cmml">4</mn></mrow><annotation-xml encoding="MathML-Content" id="A1.I1.i1.p1.4.m4.1b"><apply id="A1.I1.i1.p1.4.m4.1.1.cmml" xref="A1.I1.i1.p1.4.m4.1.1"><minus id="A1.I1.i1.p1.4.m4.1.1.1.cmml" xref="A1.I1.i1.p1.4.m4.1.1.1"></minus><apply id="A1.I1.i1.p1.4.m4.1.1.2.cmml" xref="A1.I1.i1.p1.4.m4.1.1.2"><times id="A1.I1.i1.p1.4.m4.1.1.2.1.cmml" xref="A1.I1.i1.p1.4.m4.1.1.2.1"></times><cn id="A1.I1.i1.p1.4.m4.1.1.2.2.cmml" type="integer" xref="A1.I1.i1.p1.4.m4.1.1.2.2">5</cn><ci id="A1.I1.i1.p1.4.m4.1.1.2.3a.cmml" xref="A1.I1.i1.p1.4.m4.1.1.2.3"><mtext id="A1.I1.i1.p1.4.m4.1.1.2.3.cmml" xref="A1.I1.i1.p1.4.m4.1.1.2.3">e</mtext></ci></apply><cn id="A1.I1.i1.p1.4.m4.1.1.3.cmml" type="integer" xref="A1.I1.i1.p1.4.m4.1.1.3">4</cn></apply></annotation-xml><annotation encoding="application/x-tex" id="A1.I1.i1.p1.4.m4.1c">5\textrm{e}-4</annotation><annotation encoding="application/x-llamapun" id="A1.I1.i1.p1.4.m4.1d">5 e - 4</annotation></semantics></math> learning rate, <math alttext="0.05" class="ltx_Math" display="inline" id="A1.I1.i1.p1.5.m5.1"><semantics id="A1.I1.i1.p1.5.m5.1a"><mn id="A1.I1.i1.p1.5.m5.1.1" xref="A1.I1.i1.p1.5.m5.1.1.cmml">0.05</mn><annotation-xml encoding="MathML-Content" id="A1.I1.i1.p1.5.m5.1b"><cn id="A1.I1.i1.p1.5.m5.1.1.cmml" type="float" xref="A1.I1.i1.p1.5.m5.1.1">0.05</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I1.i1.p1.5.m5.1c">0.05</annotation><annotation encoding="application/x-llamapun" id="A1.I1.i1.p1.5.m5.1d">0.05</annotation></semantics></math> warm-up ratio, and apply a cosine learning scheduler. The maximum sequence length in supervised training is limited to <math alttext="2560" class="ltx_Math" display="inline" id="A1.I1.i1.p1.6.m6.1"><semantics id="A1.I1.i1.p1.6.m6.1a"><mn id="A1.I1.i1.p1.6.m6.1.1" xref="A1.I1.i1.p1.6.m6.1.1.cmml">2560</mn><annotation-xml encoding="MathML-Content" id="A1.I1.i1.p1.6.m6.1b"><cn id="A1.I1.i1.p1.6.m6.1.1.cmml" type="integer" xref="A1.I1.i1.p1.6.m6.1.1">2560</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I1.i1.p1.6.m6.1c">2560</annotation><annotation encoding="application/x-llamapun" id="A1.I1.i1.p1.6.m6.1d">2560</annotation></semantics></math> due to memory limitations.</p>
</div>
</li>
<li class="ltx_item" id="A1.I1.i2" style="list-style-type:none;">
<span class="ltx_tag ltx_tag_item">•</span>
<div class="ltx_para ltx_noindent" id="A1.I1.i2.p1">
<p class="ltx_p" id="A1.I1.i2.p1.7">For the supervised fine-tuning of DrugAssist (<a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.SS3" title="4.3 Fine-tuning generalist LLMs ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Section</span> <span class="ltx_text ltx_ref_tag">4.3</span></a>), we consider <math alttext="80" class="ltx_Math" display="inline" id="A1.I1.i2.p1.1.m1.1"><semantics id="A1.I1.i2.p1.1.m1.1a"><mn id="A1.I1.i2.p1.1.m1.1.1" xref="A1.I1.i2.p1.1.m1.1.1.cmml">80</mn><annotation-xml encoding="MathML-Content" id="A1.I1.i2.p1.1.m1.1b"><cn id="A1.I1.i2.p1.1.m1.1.1.cmml" type="integer" xref="A1.I1.i2.p1.1.m1.1.1">80</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I1.i2.p1.1.m1.1c">80</annotation><annotation encoding="application/x-llamapun" id="A1.I1.i2.p1.1.m1.1d">80</annotation></semantics></math> epochs, <math alttext="4" class="ltx_Math" display="inline" id="A1.I1.i2.p1.2.m2.1"><semantics id="A1.I1.i2.p1.2.m2.1a"><mn id="A1.I1.i2.p1.2.m2.1.1" xref="A1.I1.i2.p1.2.m2.1.1.cmml">4</mn><annotation-xml encoding="MathML-Content" id="A1.I1.i2.p1.2.m2.1b"><cn id="A1.I1.i2.p1.2.m2.1.1.cmml" type="integer" xref="A1.I1.i2.p1.2.m2.1.1">4</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I1.i2.p1.2.m2.1c">4</annotation><annotation encoding="application/x-llamapun" id="A1.I1.i2.p1.2.m2.1d">4</annotation></semantics></math>-batch size, <math alttext="3\textrm{e}-5" class="ltx_Math" display="inline" id="A1.I1.i2.p1.3.m3.1"><semantics id="A1.I1.i2.p1.3.m3.1a"><mrow id="A1.I1.i2.p1.3.m3.1.1" xref="A1.I1.i2.p1.3.m3.1.1.cmml"><mrow id="A1.I1.i2.p1.3.m3.1.1.2" xref="A1.I1.i2.p1.3.m3.1.1.2.cmml"><mn id="A1.I1.i2.p1.3.m3.1.1.2.2" xref="A1.I1.i2.p1.3.m3.1.1.2.2.cmml">3</mn><mo id="A1.I1.i2.p1.3.m3.1.1.2.1" xref="A1.I1.i2.p1.3.m3.1.1.2.1.cmml">⁢</mo><mtext id="A1.I1.i2.p1.3.m3.1.1.2.3" xref="A1.I1.i2.p1.3.m3.1.1.2.3a.cmml">e</mtext></mrow><mo id="A1.I1.i2.p1.3.m3.1.1.1" xref="A1.I1.i2.p1.3.m3.1.1.1.cmml">−</mo><mn id="A1.I1.i2.p1.3.m3.1.1.3" xref="A1.I1.i2.p1.3.m3.1.1.3.cmml">5</mn></mrow><annotation-xml encoding="MathML-Content" id="A1.I1.i2.p1.3.m3.1b"><apply id="A1.I1.i2.p1.3.m3.1.1.cmml" xref="A1.I1.i2.p1.3.m3.1.1"><minus id="A1.I1.i2.p1.3.m3.1.1.1.cmml" xref="A1.I1.i2.p1.3.m3.1.1.1"></minus><apply id="A1.I1.i2.p1.3.m3.1.1.2.cmml" xref="A1.I1.i2.p1.3.m3.1.1.2"><times id="A1.I1.i2.p1.3.m3.1.1.2.1.cmml" xref="A1.I1.i2.p1.3.m3.1.1.2.1"></times><cn id="A1.I1.i2.p1.3.m3.1.1.2.2.cmml" type="integer" xref="A1.I1.i2.p1.3.m3.1.1.2.2">3</cn><ci id="A1.I1.i2.p1.3.m3.1.1.2.3a.cmml" xref="A1.I1.i2.p1.3.m3.1.1.2.3"><mtext id="A1.I1.i2.p1.3.m3.1.1.2.3.cmml" xref="A1.I1.i2.p1.3.m3.1.1.2.3">e</mtext></ci></apply><cn id="A1.I1.i2.p1.3.m3.1.1.3.cmml" type="integer" xref="A1.I1.i2.p1.3.m3.1.1.3">5</cn></apply></annotation-xml><annotation encoding="application/x-tex" id="A1.I1.i2.p1.3.m3.1c">3\textrm{e}-5</annotation><annotation encoding="application/x-llamapun" id="A1.I1.i2.p1.3.m3.1d">3 e - 5</annotation></semantics></math> learning rate, <math alttext="0.05" class="ltx_Math" display="inline" id="A1.I1.i2.p1.4.m4.1"><semantics id="A1.I1.i2.p1.4.m4.1a"><mn id="A1.I1.i2.p1.4.m4.1.1" xref="A1.I1.i2.p1.4.m4.1.1.cmml">0.05</mn><annotation-xml encoding="MathML-Content" id="A1.I1.i2.p1.4.m4.1b"><cn id="A1.I1.i2.p1.4.m4.1.1.cmml" type="float" xref="A1.I1.i2.p1.4.m4.1.1">0.05</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I1.i2.p1.4.m4.1c">0.05</annotation><annotation encoding="application/x-llamapun" id="A1.I1.i2.p1.4.m4.1d">0.05</annotation></semantics></math> warm-up ratio, and apply a cosine learning scheduler. The maximum sequence length in supervised training is limited to <math alttext="1024" class="ltx_Math" display="inline" id="A1.I1.i2.p1.5.m5.1"><semantics id="A1.I1.i2.p1.5.m5.1a"><mn id="A1.I1.i2.p1.5.m5.1.1" xref="A1.I1.i2.p1.5.m5.1.1.cmml">1024</mn><annotation-xml encoding="MathML-Content" id="A1.I1.i2.p1.5.m5.1b"><cn id="A1.I1.i2.p1.5.m5.1.1.cmml" type="integer" xref="A1.I1.i2.p1.5.m5.1.1">1024</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I1.i2.p1.5.m5.1c">1024</annotation><annotation encoding="application/x-llamapun" id="A1.I1.i2.p1.5.m5.1d">1024</annotation></semantics></math> due to memory limitations. We also apply LoRA <cite class="ltx_cite ltx_citemacro_citep">(Hu et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib14" title="">2022</a>)</cite>, where the rank and alpha are <math alttext="64" class="ltx_Math" display="inline" id="A1.I1.i2.p1.6.m6.1"><semantics id="A1.I1.i2.p1.6.m6.1a"><mn id="A1.I1.i2.p1.6.m6.1.1" xref="A1.I1.i2.p1.6.m6.1.1.cmml">64</mn><annotation-xml encoding="MathML-Content" id="A1.I1.i2.p1.6.m6.1b"><cn id="A1.I1.i2.p1.6.m6.1.1.cmml" type="integer" xref="A1.I1.i2.p1.6.m6.1.1">64</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I1.i2.p1.6.m6.1c">64</annotation><annotation encoding="application/x-llamapun" id="A1.I1.i2.p1.6.m6.1d">64</annotation></semantics></math> and <math alttext="128" class="ltx_Math" display="inline" id="A1.I1.i2.p1.7.m7.1"><semantics id="A1.I1.i2.p1.7.m7.1a"><mn id="A1.I1.i2.p1.7.m7.1.1" xref="A1.I1.i2.p1.7.m7.1.1.cmml">128</mn><annotation-xml encoding="MathML-Content" id="A1.I1.i2.p1.7.m7.1b"><cn id="A1.I1.i2.p1.7.m7.1.1.cmml" type="integer" xref="A1.I1.i2.p1.7.m7.1.1">128</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I1.i2.p1.7.m7.1c">128</annotation><annotation encoding="application/x-llamapun" id="A1.I1.i2.p1.7.m7.1d">128</annotation></semantics></math>, respectively.</p>
</div>
</li>
</ul>
</div>
</section>
<section class="ltx_subsection" id="A1.SS2">
<h3 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">A.2 </span>Reinforcement learning</h3>
<div class="ltx_para ltx_noindent" id="A1.SS2.p1">
<p class="ltx_p" id="A1.SS2.p1.1"><span class="ltx_text ltx_font_bold" id="A1.SS2.p1.1.1">Reward Design.</span> In experiments with BioT5<sup class="ltx_sup" id="A1.SS2.p1.1.2"><span class="ltx_text ltx_font_italic" id="A1.SS2.p1.1.2.1">+</span></sup> and MolT5 on the ChEBI-20 dataset (<a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.SS1" title="4.1 Comparison with decoding schemes ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Sections</span> <span class="ltx_text ltx_ref_tag">4.1</span></a> and <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.SS2" title="4.2 Comparison with existing LLMs ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">4.2</span></a>), we define the description-matching reward using the BLEU score as follows:</p>
<table class="ltx_equation ltx_eqn_table" id="A1.E2">
<tbody><tr class="ltx_equation ltx_eqn_row ltx_align_baseline">
<td class="ltx_eqn_cell ltx_eqn_center_padleft"></td>
<td class="ltx_eqn_cell ltx_align_center"><math alttext="r_{\text{match}}(m_{k},p_{\text{desc}})=\text{BLEU}(m_{p_{\text{mol}}},m_{k})^%
{\alpha}," class="ltx_Math" display="block" id="A1.E2.m1.1"><semantics id="A1.E2.m1.1a"><mrow id="A1.E2.m1.1.1.1" xref="A1.E2.m1.1.1.1.1.cmml"><mrow id="A1.E2.m1.1.1.1.1" xref="A1.E2.m1.1.1.1.1.cmml"><mrow id="A1.E2.m1.1.1.1.1.2" xref="A1.E2.m1.1.1.1.1.2.cmml"><msub id="A1.E2.m1.1.1.1.1.2.4" xref="A1.E2.m1.1.1.1.1.2.4.cmml"><mi id="A1.E2.m1.1.1.1.1.2.4.2" xref="A1.E2.m1.1.1.1.1.2.4.2.cmml">r</mi><mtext id="A1.E2.m1.1.1.1.1.2.4.3" xref="A1.E2.m1.1.1.1.1.2.4.3a.cmml">match</mtext></msub><mo id="A1.E2.m1.1.1.1.1.2.3" xref="A1.E2.m1.1.1.1.1.2.3.cmml">⁢</mo><mrow id="A1.E2.m1.1.1.1.1.2.2.2" xref="A1.E2.m1.1.1.1.1.2.2.3.cmml"><mo id="A1.E2.m1.1.1.1.1.2.2.2.3" stretchy="false" xref="A1.E2.m1.1.1.1.1.2.2.3.cmml">(</mo><msub id="A1.E2.m1.1.1.1.1.1.1.1.1" xref="A1.E2.m1.1.1.1.1.1.1.1.1.cmml"><mi id="A1.E2.m1.1.1.1.1.1.1.1.1.2" xref="A1.E2.m1.1.1.1.1.1.1.1.1.2.cmml">m</mi><mi id="A1.E2.m1.1.1.1.1.1.1.1.1.3" xref="A1.E2.m1.1.1.1.1.1.1.1.1.3.cmml">k</mi></msub><mo id="A1.E2.m1.1.1.1.1.2.2.2.4" xref="A1.E2.m1.1.1.1.1.2.2.3.cmml">,</mo><msub id="A1.E2.m1.1.1.1.1.2.2.2.2" xref="A1.E2.m1.1.1.1.1.2.2.2.2.cmml"><mi id="A1.E2.m1.1.1.1.1.2.2.2.2.2" xref="A1.E2.m1.1.1.1.1.2.2.2.2.2.cmml">p</mi><mtext id="A1.E2.m1.1.1.1.1.2.2.2.2.3" xref="A1.E2.m1.1.1.1.1.2.2.2.2.3a.cmml">desc</mtext></msub><mo id="A1.E2.m1.1.1.1.1.2.2.2.5" stretchy="false" xref="A1.E2.m1.1.1.1.1.2.2.3.cmml">)</mo></mrow></mrow><mo id="A1.E2.m1.1.1.1.1.5" xref="A1.E2.m1.1.1.1.1.5.cmml">=</mo><mrow id="A1.E2.m1.1.1.1.1.4" xref="A1.E2.m1.1.1.1.1.4.cmml"><mtext id="A1.E2.m1.1.1.1.1.4.4" xref="A1.E2.m1.1.1.1.1.4.4a.cmml">BLEU</mtext><mo id="A1.E2.m1.1.1.1.1.4.3" xref="A1.E2.m1.1.1.1.1.4.3.cmml">⁢</mo><msup id="A1.E2.m1.1.1.1.1.4.2" xref="A1.E2.m1.1.1.1.1.4.2.cmml"><mrow id="A1.E2.m1.1.1.1.1.4.2.2.2" xref="A1.E2.m1.1.1.1.1.4.2.2.3.cmml"><mo id="A1.E2.m1.1.1.1.1.4.2.2.2.3" stretchy="false" xref="A1.E2.m1.1.1.1.1.4.2.2.3.cmml">(</mo><msub id="A1.E2.m1.1.1.1.1.3.1.1.1.1" xref="A1.E2.m1.1.1.1.1.3.1.1.1.1.cmml"><mi id="A1.E2.m1.1.1.1.1.3.1.1.1.1.2" xref="A1.E2.m1.1.1.1.1.3.1.1.1.1.2.cmml">m</mi><msub id="A1.E2.m1.1.1.1.1.3.1.1.1.1.3" xref="A1.E2.m1.1.1.1.1.3.1.1.1.1.3.cmml"><mi id="A1.E2.m1.1.1.1.1.3.1.1.1.1.3.2" xref="A1.E2.m1.1.1.1.1.3.1.1.1.1.3.2.cmml">p</mi><mtext id="A1.E2.m1.1.1.1.1.3.1.1.1.1.3.3" xref="A1.E2.m1.1.1.1.1.3.1.1.1.1.3.3a.cmml">mol</mtext></msub></msub><mo id="A1.E2.m1.1.1.1.1.4.2.2.2.4" xref="A1.E2.m1.1.1.1.1.4.2.2.3.cmml">,</mo><msub id="A1.E2.m1.1.1.1.1.4.2.2.2.2" xref="A1.E2.m1.1.1.1.1.4.2.2.2.2.cmml"><mi id="A1.E2.m1.1.1.1.1.4.2.2.2.2.2" xref="A1.E2.m1.1.1.1.1.4.2.2.2.2.2.cmml">m</mi><mi id="A1.E2.m1.1.1.1.1.4.2.2.2.2.3" xref="A1.E2.m1.1.1.1.1.4.2.2.2.2.3.cmml">k</mi></msub><mo id="A1.E2.m1.1.1.1.1.4.2.2.2.5" stretchy="false" xref="A1.E2.m1.1.1.1.1.4.2.2.3.cmml">)</mo></mrow><mi id="A1.E2.m1.1.1.1.1.4.2.4" xref="A1.E2.m1.1.1.1.1.4.2.4.cmml">α</mi></msup></mrow></mrow><mo id="A1.E2.m1.1.1.1.2" xref="A1.E2.m1.1.1.1.1.cmml">,</mo></mrow><annotation-xml encoding="MathML-Content" id="A1.E2.m1.1b"><apply id="A1.E2.m1.1.1.1.1.cmml" xref="A1.E2.m1.1.1.1"><eq id="A1.E2.m1.1.1.1.1.5.cmml" xref="A1.E2.m1.1.1.1.1.5"></eq><apply id="A1.E2.m1.1.1.1.1.2.cmml" xref="A1.E2.m1.1.1.1.1.2"><times id="A1.E2.m1.1.1.1.1.2.3.cmml" xref="A1.E2.m1.1.1.1.1.2.3"></times><apply id="A1.E2.m1.1.1.1.1.2.4.cmml" xref="A1.E2.m1.1.1.1.1.2.4"><csymbol cd="ambiguous" id="A1.E2.m1.1.1.1.1.2.4.1.cmml" xref="A1.E2.m1.1.1.1.1.2.4">subscript</csymbol><ci id="A1.E2.m1.1.1.1.1.2.4.2.cmml" xref="A1.E2.m1.1.1.1.1.2.4.2">𝑟</ci><ci id="A1.E2.m1.1.1.1.1.2.4.3a.cmml" xref="A1.E2.m1.1.1.1.1.2.4.3"><mtext id="A1.E2.m1.1.1.1.1.2.4.3.cmml" mathsize="70%" xref="A1.E2.m1.1.1.1.1.2.4.3">match</mtext></ci></apply><interval closure="open" id="A1.E2.m1.1.1.1.1.2.2.3.cmml" xref="A1.E2.m1.1.1.1.1.2.2.2"><apply id="A1.E2.m1.1.1.1.1.1.1.1.1.cmml" xref="A1.E2.m1.1.1.1.1.1.1.1.1"><csymbol cd="ambiguous" id="A1.E2.m1.1.1.1.1.1.1.1.1.1.cmml" xref="A1.E2.m1.1.1.1.1.1.1.1.1">subscript</csymbol><ci id="A1.E2.m1.1.1.1.1.1.1.1.1.2.cmml" xref="A1.E2.m1.1.1.1.1.1.1.1.1.2">𝑚</ci><ci id="A1.E2.m1.1.1.1.1.1.1.1.1.3.cmml" xref="A1.E2.m1.1.1.1.1.1.1.1.1.3">𝑘</ci></apply><apply id="A1.E2.m1.1.1.1.1.2.2.2.2.cmml" xref="A1.E2.m1.1.1.1.1.2.2.2.2"><csymbol cd="ambiguous" id="A1.E2.m1.1.1.1.1.2.2.2.2.1.cmml" xref="A1.E2.m1.1.1.1.1.2.2.2.2">subscript</csymbol><ci id="A1.E2.m1.1.1.1.1.2.2.2.2.2.cmml" xref="A1.E2.m1.1.1.1.1.2.2.2.2.2">𝑝</ci><ci id="A1.E2.m1.1.1.1.1.2.2.2.2.3a.cmml" xref="A1.E2.m1.1.1.1.1.2.2.2.2.3"><mtext id="A1.E2.m1.1.1.1.1.2.2.2.2.3.cmml" mathsize="70%" xref="A1.E2.m1.1.1.1.1.2.2.2.2.3">desc</mtext></ci></apply></interval></apply><apply id="A1.E2.m1.1.1.1.1.4.cmml" xref="A1.E2.m1.1.1.1.1.4"><times id="A1.E2.m1.1.1.1.1.4.3.cmml" xref="A1.E2.m1.1.1.1.1.4.3"></times><ci id="A1.E2.m1.1.1.1.1.4.4a.cmml" xref="A1.E2.m1.1.1.1.1.4.4"><mtext id="A1.E2.m1.1.1.1.1.4.4.cmml" xref="A1.E2.m1.1.1.1.1.4.4">BLEU</mtext></ci><apply id="A1.E2.m1.1.1.1.1.4.2.cmml" xref="A1.E2.m1.1.1.1.1.4.2"><csymbol cd="ambiguous" id="A1.E2.m1.1.1.1.1.4.2.3.cmml" xref="A1.E2.m1.1.1.1.1.4.2">superscript</csymbol><interval closure="open" id="A1.E2.m1.1.1.1.1.4.2.2.3.cmml" xref="A1.E2.m1.1.1.1.1.4.2.2.2"><apply id="A1.E2.m1.1.1.1.1.3.1.1.1.1.cmml" xref="A1.E2.m1.1.1.1.1.3.1.1.1.1"><csymbol cd="ambiguous" id="A1.E2.m1.1.1.1.1.3.1.1.1.1.1.cmml" xref="A1.E2.m1.1.1.1.1.3.1.1.1.1">subscript</csymbol><ci id="A1.E2.m1.1.1.1.1.3.1.1.1.1.2.cmml" xref="A1.E2.m1.1.1.1.1.3.1.1.1.1.2">𝑚</ci><apply id="A1.E2.m1.1.1.1.1.3.1.1.1.1.3.cmml" xref="A1.E2.m1.1.1.1.1.3.1.1.1.1.3"><csymbol cd="ambiguous" id="A1.E2.m1.1.1.1.1.3.1.1.1.1.3.1.cmml" xref="A1.E2.m1.1.1.1.1.3.1.1.1.1.3">subscript</csymbol><ci id="A1.E2.m1.1.1.1.1.3.1.1.1.1.3.2.cmml" xref="A1.E2.m1.1.1.1.1.3.1.1.1.1.3.2">𝑝</ci><ci id="A1.E2.m1.1.1.1.1.3.1.1.1.1.3.3a.cmml" xref="A1.E2.m1.1.1.1.1.3.1.1.1.1.3.3"><mtext id="A1.E2.m1.1.1.1.1.3.1.1.1.1.3.3.cmml" mathsize="50%" xref="A1.E2.m1.1.1.1.1.3.1.1.1.1.3.3">mol</mtext></ci></apply></apply><apply id="A1.E2.m1.1.1.1.1.4.2.2.2.2.cmml" xref="A1.E2.m1.1.1.1.1.4.2.2.2.2"><csymbol cd="ambiguous" id="A1.E2.m1.1.1.1.1.4.2.2.2.2.1.cmml" xref="A1.E2.m1.1.1.1.1.4.2.2.2.2">subscript</csymbol><ci id="A1.E2.m1.1.1.1.1.4.2.2.2.2.2.cmml" xref="A1.E2.m1.1.1.1.1.4.2.2.2.2.2">𝑚</ci><ci id="A1.E2.m1.1.1.1.1.4.2.2.2.2.3.cmml" xref="A1.E2.m1.1.1.1.1.4.2.2.2.2.3">𝑘</ci></apply></interval><ci id="A1.E2.m1.1.1.1.1.4.2.4.cmml" xref="A1.E2.m1.1.1.1.1.4.2.4">𝛼</ci></apply></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="A1.E2.m1.1c">r_{\text{match}}(m_{k},p_{\text{desc}})=\text{BLEU}(m_{p_{\text{mol}}},m_{k})^%
{\alpha},</annotation><annotation encoding="application/x-llamapun" id="A1.E2.m1.1d">italic_r start_POSTSUBSCRIPT match end_POSTSUBSCRIPT ( italic_m start_POSTSUBSCRIPT italic_k end_POSTSUBSCRIPT , italic_p start_POSTSUBSCRIPT desc end_POSTSUBSCRIPT ) = BLEU ( italic_m start_POSTSUBSCRIPT italic_p start_POSTSUBSCRIPT mol end_POSTSUBSCRIPT end_POSTSUBSCRIPT , italic_m start_POSTSUBSCRIPT italic_k end_POSTSUBSCRIPT ) start_POSTSUPERSCRIPT italic_α end_POSTSUPERSCRIPT ,</annotation></semantics></math></td>
<td class="ltx_eqn_cell ltx_eqn_center_padright"></td>
<td class="ltx_eqn_cell ltx_eqn_eqno ltx_align_middle ltx_align_right" rowspan="1"><span class="ltx_tag ltx_tag_equation ltx_align_right">(2)</span></td>
</tr></tbody>
</table>
<p class="ltx_p" id="A1.SS2.p1.4">where <math alttext="m_{p_{\text{mol}}}" class="ltx_Math" display="inline" id="A1.SS2.p1.2.m1.1"><semantics id="A1.SS2.p1.2.m1.1a"><msub id="A1.SS2.p1.2.m1.1.1" xref="A1.SS2.p1.2.m1.1.1.cmml"><mi id="A1.SS2.p1.2.m1.1.1.2" xref="A1.SS2.p1.2.m1.1.1.2.cmml">m</mi><msub id="A1.SS2.p1.2.m1.1.1.3" xref="A1.SS2.p1.2.m1.1.1.3.cmml"><mi id="A1.SS2.p1.2.m1.1.1.3.2" xref="A1.SS2.p1.2.m1.1.1.3.2.cmml">p</mi><mtext id="A1.SS2.p1.2.m1.1.1.3.3" xref="A1.SS2.p1.2.m1.1.1.3.3a.cmml">mol</mtext></msub></msub><annotation-xml encoding="MathML-Content" id="A1.SS2.p1.2.m1.1b"><apply id="A1.SS2.p1.2.m1.1.1.cmml" xref="A1.SS2.p1.2.m1.1.1"><csymbol cd="ambiguous" id="A1.SS2.p1.2.m1.1.1.1.cmml" xref="A1.SS2.p1.2.m1.1.1">subscript</csymbol><ci id="A1.SS2.p1.2.m1.1.1.2.cmml" xref="A1.SS2.p1.2.m1.1.1.2">𝑚</ci><apply id="A1.SS2.p1.2.m1.1.1.3.cmml" xref="A1.SS2.p1.2.m1.1.1.3"><csymbol cd="ambiguous" id="A1.SS2.p1.2.m1.1.1.3.1.cmml" xref="A1.SS2.p1.2.m1.1.1.3">subscript</csymbol><ci id="A1.SS2.p1.2.m1.1.1.3.2.cmml" xref="A1.SS2.p1.2.m1.1.1.3.2">𝑝</ci><ci id="A1.SS2.p1.2.m1.1.1.3.3a.cmml" xref="A1.SS2.p1.2.m1.1.1.3.3"><mtext id="A1.SS2.p1.2.m1.1.1.3.3.cmml" mathsize="50%" xref="A1.SS2.p1.2.m1.1.1.3.3">mol</mtext></ci></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="A1.SS2.p1.2.m1.1c">m_{p_{\text{mol}}}</annotation><annotation encoding="application/x-llamapun" id="A1.SS2.p1.2.m1.1d">italic_m start_POSTSUBSCRIPT italic_p start_POSTSUBSCRIPT mol end_POSTSUBSCRIPT end_POSTSUBSCRIPT</annotation></semantics></math> is a ground-truth molecule satisfying the given description. The ChEBI-20 dataset involves a set of pairs <math alttext="(p_{\text{desc}},m_{p_{\text{mol}}})" class="ltx_Math" display="inline" id="A1.SS2.p1.3.m2.2"><semantics id="A1.SS2.p1.3.m2.2a"><mrow id="A1.SS2.p1.3.m2.2.2.2" xref="A1.SS2.p1.3.m2.2.2.3.cmml"><mo id="A1.SS2.p1.3.m2.2.2.2.3" stretchy="false" xref="A1.SS2.p1.3.m2.2.2.3.cmml">(</mo><msub id="A1.SS2.p1.3.m2.1.1.1.1" xref="A1.SS2.p1.3.m2.1.1.1.1.cmml"><mi id="A1.SS2.p1.3.m2.1.1.1.1.2" xref="A1.SS2.p1.3.m2.1.1.1.1.2.cmml">p</mi><mtext id="A1.SS2.p1.3.m2.1.1.1.1.3" xref="A1.SS2.p1.3.m2.1.1.1.1.3a.cmml">desc</mtext></msub><mo id="A1.SS2.p1.3.m2.2.2.2.4" xref="A1.SS2.p1.3.m2.2.2.3.cmml">,</mo><msub id="A1.SS2.p1.3.m2.2.2.2.2" xref="A1.SS2.p1.3.m2.2.2.2.2.cmml"><mi id="A1.SS2.p1.3.m2.2.2.2.2.2" xref="A1.SS2.p1.3.m2.2.2.2.2.2.cmml">m</mi><msub id="A1.SS2.p1.3.m2.2.2.2.2.3" xref="A1.SS2.p1.3.m2.2.2.2.2.3.cmml"><mi id="A1.SS2.p1.3.m2.2.2.2.2.3.2" xref="A1.SS2.p1.3.m2.2.2.2.2.3.2.cmml">p</mi><mtext id="A1.SS2.p1.3.m2.2.2.2.2.3.3" xref="A1.SS2.p1.3.m2.2.2.2.2.3.3a.cmml">mol</mtext></msub></msub><mo id="A1.SS2.p1.3.m2.2.2.2.5" stretchy="false" xref="A1.SS2.p1.3.m2.2.2.3.cmml">)</mo></mrow><annotation-xml encoding="MathML-Content" id="A1.SS2.p1.3.m2.2b"><interval closure="open" id="A1.SS2.p1.3.m2.2.2.3.cmml" xref="A1.SS2.p1.3.m2.2.2.2"><apply id="A1.SS2.p1.3.m2.1.1.1.1.cmml" xref="A1.SS2.p1.3.m2.1.1.1.1"><csymbol cd="ambiguous" id="A1.SS2.p1.3.m2.1.1.1.1.1.cmml" xref="A1.SS2.p1.3.m2.1.1.1.1">subscript</csymbol><ci id="A1.SS2.p1.3.m2.1.1.1.1.2.cmml" xref="A1.SS2.p1.3.m2.1.1.1.1.2">𝑝</ci><ci id="A1.SS2.p1.3.m2.1.1.1.1.3a.cmml" xref="A1.SS2.p1.3.m2.1.1.1.1.3"><mtext id="A1.SS2.p1.3.m2.1.1.1.1.3.cmml" mathsize="70%" xref="A1.SS2.p1.3.m2.1.1.1.1.3">desc</mtext></ci></apply><apply id="A1.SS2.p1.3.m2.2.2.2.2.cmml" xref="A1.SS2.p1.3.m2.2.2.2.2"><csymbol cd="ambiguous" id="A1.SS2.p1.3.m2.2.2.2.2.1.cmml" xref="A1.SS2.p1.3.m2.2.2.2.2">subscript</csymbol><ci id="A1.SS2.p1.3.m2.2.2.2.2.2.cmml" xref="A1.SS2.p1.3.m2.2.2.2.2.2">𝑚</ci><apply id="A1.SS2.p1.3.m2.2.2.2.2.3.cmml" xref="A1.SS2.p1.3.m2.2.2.2.2.3"><csymbol cd="ambiguous" id="A1.SS2.p1.3.m2.2.2.2.2.3.1.cmml" xref="A1.SS2.p1.3.m2.2.2.2.2.3">subscript</csymbol><ci id="A1.SS2.p1.3.m2.2.2.2.2.3.2.cmml" xref="A1.SS2.p1.3.m2.2.2.2.2.3.2">𝑝</ci><ci id="A1.SS2.p1.3.m2.2.2.2.2.3.3a.cmml" xref="A1.SS2.p1.3.m2.2.2.2.2.3.3"><mtext id="A1.SS2.p1.3.m2.2.2.2.2.3.3.cmml" mathsize="50%" xref="A1.SS2.p1.3.m2.2.2.2.2.3.3">mol</mtext></ci></apply></apply></interval></annotation-xml><annotation encoding="application/x-tex" id="A1.SS2.p1.3.m2.2c">(p_{\text{desc}},m_{p_{\text{mol}}})</annotation><annotation encoding="application/x-llamapun" id="A1.SS2.p1.3.m2.2d">( italic_p start_POSTSUBSCRIPT desc end_POSTSUBSCRIPT , italic_m start_POSTSUBSCRIPT italic_p start_POSTSUBSCRIPT mol end_POSTSUBSCRIPT end_POSTSUBSCRIPT )</annotation></semantics></math>. Note that <math alttext="\alpha" class="ltx_Math" display="inline" id="A1.SS2.p1.4.m3.1"><semantics id="A1.SS2.p1.4.m3.1a"><mi id="A1.SS2.p1.4.m3.1.1" xref="A1.SS2.p1.4.m3.1.1.cmml">α</mi><annotation-xml encoding="MathML-Content" id="A1.SS2.p1.4.m3.1b"><ci id="A1.SS2.p1.4.m3.1.1.cmml" xref="A1.SS2.p1.4.m3.1.1">𝛼</ci></annotation-xml><annotation encoding="application/x-tex" id="A1.SS2.p1.4.m3.1c">\alpha</annotation><annotation encoding="application/x-llamapun" id="A1.SS2.p1.4.m3.1d">italic_α</annotation></semantics></math> is a hyper-parameter.</p>
</div>
<div class="ltx_para ltx_noindent" id="A1.SS2.p2">
<p class="ltx_p" id="A1.SS2.p2.3">For experiments with DrugAssist (<a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.SS3" title="4.3 Fine-tuning generalist LLMs ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Section</span> <span class="ltx_text ltx_ref_tag">4.3</span></a>), the property-matching reward yields <math alttext="1" class="ltx_Math" display="inline" id="A1.SS2.p2.1.m1.1"><semantics id="A1.SS2.p2.1.m1.1a"><mn id="A1.SS2.p2.1.m1.1.1" xref="A1.SS2.p2.1.m1.1.1.cmml">1</mn><annotation-xml encoding="MathML-Content" id="A1.SS2.p2.1.m1.1b"><cn id="A1.SS2.p2.1.m1.1.1.cmml" type="integer" xref="A1.SS2.p2.1.m1.1.1">1</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.SS2.p2.1.m1.1c">1</annotation><annotation encoding="application/x-llamapun" id="A1.SS2.p2.1.m1.1d">1</annotation></semantics></math> if the molecule satisfies the quantitative properties described in <math alttext="p_{\text{desc}}" class="ltx_Math" display="inline" id="A1.SS2.p2.2.m2.1"><semantics id="A1.SS2.p2.2.m2.1a"><msub id="A1.SS2.p2.2.m2.1.1" xref="A1.SS2.p2.2.m2.1.1.cmml"><mi id="A1.SS2.p2.2.m2.1.1.2" xref="A1.SS2.p2.2.m2.1.1.2.cmml">p</mi><mtext id="A1.SS2.p2.2.m2.1.1.3" xref="A1.SS2.p2.2.m2.1.1.3a.cmml">desc</mtext></msub><annotation-xml encoding="MathML-Content" id="A1.SS2.p2.2.m2.1b"><apply id="A1.SS2.p2.2.m2.1.1.cmml" xref="A1.SS2.p2.2.m2.1.1"><csymbol cd="ambiguous" id="A1.SS2.p2.2.m2.1.1.1.cmml" xref="A1.SS2.p2.2.m2.1.1">subscript</csymbol><ci id="A1.SS2.p2.2.m2.1.1.2.cmml" xref="A1.SS2.p2.2.m2.1.1.2">𝑝</ci><ci id="A1.SS2.p2.2.m2.1.1.3a.cmml" xref="A1.SS2.p2.2.m2.1.1.3"><mtext id="A1.SS2.p2.2.m2.1.1.3.cmml" mathsize="70%" xref="A1.SS2.p2.2.m2.1.1.3">desc</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="A1.SS2.p2.2.m2.1c">p_{\text{desc}}</annotation><annotation encoding="application/x-llamapun" id="A1.SS2.p2.2.m2.1d">italic_p start_POSTSUBSCRIPT desc end_POSTSUBSCRIPT</annotation></semantics></math>, e.g., HB donors, HB acceptors, and Bertz complexity, and <math alttext="0" class="ltx_Math" display="inline" id="A1.SS2.p2.3.m3.1"><semantics id="A1.SS2.p2.3.m3.1a"><mn id="A1.SS2.p2.3.m3.1.1" xref="A1.SS2.p2.3.m3.1.1.cmml">0</mn><annotation-xml encoding="MathML-Content" id="A1.SS2.p2.3.m3.1b"><cn id="A1.SS2.p2.3.m3.1.1.cmml" type="integer" xref="A1.SS2.p2.3.m3.1.1">0</cn></annotation-xml></semantics></math> otherwise. The properties are evaluated with the RDKit package <cite class="ltx_cite ltx_citemacro_citep">(Landrum, <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib18" title="">2016</a>)</cite>.</p>
</div>
<div class="ltx_para ltx_noindent" id="A1.SS2.p3">
<p class="ltx_p" id="A1.SS2.p3.1">Next, the diversity reward, <math alttext="r_{\text{div}}" class="ltx_Math" display="inline" id="A1.SS2.p3.1.m1.1"><semantics id="A1.SS2.p3.1.m1.1a"><msub id="A1.SS2.p3.1.m1.1.1" xref="A1.SS2.p3.1.m1.1.1.cmml"><mi id="A1.SS2.p3.1.m1.1.1.2" xref="A1.SS2.p3.1.m1.1.1.2.cmml">r</mi><mtext id="A1.SS2.p3.1.m1.1.1.3" xref="A1.SS2.p3.1.m1.1.1.3a.cmml">div</mtext></msub><annotation-xml encoding="MathML-Content" id="A1.SS2.p3.1.m1.1b"><apply id="A1.SS2.p3.1.m1.1.1.cmml" xref="A1.SS2.p3.1.m1.1.1"><csymbol cd="ambiguous" id="A1.SS2.p3.1.m1.1.1.1.cmml" xref="A1.SS2.p3.1.m1.1.1">subscript</csymbol><ci id="A1.SS2.p3.1.m1.1.1.2.cmml" xref="A1.SS2.p3.1.m1.1.1.2">𝑟</ci><ci id="A1.SS2.p3.1.m1.1.1.3a.cmml" xref="A1.SS2.p3.1.m1.1.1.3"><mtext id="A1.SS2.p3.1.m1.1.1.3.cmml" mathsize="70%" xref="A1.SS2.p3.1.m1.1.1.3">div</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="A1.SS2.p3.1.m1.1c">r_{\text{div}}</annotation><annotation encoding="application/x-llamapun" id="A1.SS2.p3.1.m1.1d">italic_r start_POSTSUBSCRIPT div end_POSTSUBSCRIPT</annotation></semantics></math>, is defined to consider molecular structural diversity as follows:</p>
<table class="ltx_equation ltx_eqn_table" id="A1.E3">
<tbody><tr class="ltx_equation ltx_eqn_row ltx_align_baseline">
<td class="ltx_eqn_cell ltx_eqn_center_padleft"></td>
<td class="ltx_eqn_cell ltx_align_center"><math alttext="r_{\text{div}}(m_{k},\{m_{i}\}^{k-1}_{i=1})=1-\text{max}_{m\in\{m_{i}\}^{k-1}_%
{i=1}}T(m_{k},m)^{\beta}," class="ltx_Math" display="block" id="A1.E3.m1.3"><semantics id="A1.E3.m1.3a"><mrow id="A1.E3.m1.3.3.1" xref="A1.E3.m1.3.3.1.1.cmml"><mrow id="A1.E3.m1.3.3.1.1" xref="A1.E3.m1.3.3.1.1.cmml"><mrow id="A1.E3.m1.3.3.1.1.2" xref="A1.E3.m1.3.3.1.1.2.cmml"><msub id="A1.E3.m1.3.3.1.1.2.4" xref="A1.E3.m1.3.3.1.1.2.4.cmml"><mi id="A1.E3.m1.3.3.1.1.2.4.2" xref="A1.E3.m1.3.3.1.1.2.4.2.cmml">r</mi><mtext id="A1.E3.m1.3.3.1.1.2.4.3" xref="A1.E3.m1.3.3.1.1.2.4.3a.cmml">div</mtext></msub><mo id="A1.E3.m1.3.3.1.1.2.3" xref="A1.E3.m1.3.3.1.1.2.3.cmml">⁢</mo><mrow id="A1.E3.m1.3.3.1.1.2.2.2" xref="A1.E3.m1.3.3.1.1.2.2.3.cmml"><mo id="A1.E3.m1.3.3.1.1.2.2.2.3" stretchy="false" xref="A1.E3.m1.3.3.1.1.2.2.3.cmml">(</mo><msub id="A1.E3.m1.3.3.1.1.1.1.1.1" xref="A1.E3.m1.3.3.1.1.1.1.1.1.cmml"><mi id="A1.E3.m1.3.3.1.1.1.1.1.1.2" xref="A1.E3.m1.3.3.1.1.1.1.1.1.2.cmml">m</mi><mi id="A1.E3.m1.3.3.1.1.1.1.1.1.3" xref="A1.E3.m1.3.3.1.1.1.1.1.1.3.cmml">k</mi></msub><mo id="A1.E3.m1.3.3.1.1.2.2.2.4" xref="A1.E3.m1.3.3.1.1.2.2.3.cmml">,</mo><msubsup id="A1.E3.m1.3.3.1.1.2.2.2.2" xref="A1.E3.m1.3.3.1.1.2.2.2.2.cmml"><mrow id="A1.E3.m1.3.3.1.1.2.2.2.2.1.1.1" xref="A1.E3.m1.3.3.1.1.2.2.2.2.1.1.2.cmml"><mo id="A1.E3.m1.3.3.1.1.2.2.2.2.1.1.1.2" stretchy="false" xref="A1.E3.m1.3.3.1.1.2.2.2.2.1.1.2.cmml">{</mo><msub id="A1.E3.m1.3.3.1.1.2.2.2.2.1.1.1.1" xref="A1.E3.m1.3.3.1.1.2.2.2.2.1.1.1.1.cmml"><mi id="A1.E3.m1.3.3.1.1.2.2.2.2.1.1.1.1.2" xref="A1.E3.m1.3.3.1.1.2.2.2.2.1.1.1.1.2.cmml">m</mi><mi id="A1.E3.m1.3.3.1.1.2.2.2.2.1.1.1.1.3" xref="A1.E3.m1.3.3.1.1.2.2.2.2.1.1.1.1.3.cmml">i</mi></msub><mo id="A1.E3.m1.3.3.1.1.2.2.2.2.1.1.1.3" stretchy="false" xref="A1.E3.m1.3.3.1.1.2.2.2.2.1.1.2.cmml">}</mo></mrow><mrow id="A1.E3.m1.3.3.1.1.2.2.2.2.3" xref="A1.E3.m1.3.3.1.1.2.2.2.2.3.cmml"><mi id="A1.E3.m1.3.3.1.1.2.2.2.2.3.2" xref="A1.E3.m1.3.3.1.1.2.2.2.2.3.2.cmml">i</mi><mo id="A1.E3.m1.3.3.1.1.2.2.2.2.3.1" xref="A1.E3.m1.3.3.1.1.2.2.2.2.3.1.cmml">=</mo><mn id="A1.E3.m1.3.3.1.1.2.2.2.2.3.3" xref="A1.E3.m1.3.3.1.1.2.2.2.2.3.3.cmml">1</mn></mrow><mrow id="A1.E3.m1.3.3.1.1.2.2.2.2.1.3" xref="A1.E3.m1.3.3.1.1.2.2.2.2.1.3.cmml"><mi id="A1.E3.m1.3.3.1.1.2.2.2.2.1.3.2" xref="A1.E3.m1.3.3.1.1.2.2.2.2.1.3.2.cmml">k</mi><mo id="A1.E3.m1.3.3.1.1.2.2.2.2.1.3.1" xref="A1.E3.m1.3.3.1.1.2.2.2.2.1.3.1.cmml">−</mo><mn id="A1.E3.m1.3.3.1.1.2.2.2.2.1.3.3" xref="A1.E3.m1.3.3.1.1.2.2.2.2.1.3.3.cmml">1</mn></mrow></msubsup><mo id="A1.E3.m1.3.3.1.1.2.2.2.5" stretchy="false" xref="A1.E3.m1.3.3.1.1.2.2.3.cmml">)</mo></mrow></mrow><mo id="A1.E3.m1.3.3.1.1.4" xref="A1.E3.m1.3.3.1.1.4.cmml">=</mo><mrow id="A1.E3.m1.3.3.1.1.3" xref="A1.E3.m1.3.3.1.1.3.cmml"><mn id="A1.E3.m1.3.3.1.1.3.3" xref="A1.E3.m1.3.3.1.1.3.3.cmml">1</mn><mo id="A1.E3.m1.3.3.1.1.3.2" xref="A1.E3.m1.3.3.1.1.3.2.cmml">−</mo><mrow id="A1.E3.m1.3.3.1.1.3.1" xref="A1.E3.m1.3.3.1.1.3.1.cmml"><msub id="A1.E3.m1.3.3.1.1.3.1.3" xref="A1.E3.m1.3.3.1.1.3.1.3.cmml"><mtext id="A1.E3.m1.3.3.1.1.3.1.3.2" xref="A1.E3.m1.3.3.1.1.3.1.3.2a.cmml">max</mtext><mrow id="A1.E3.m1.1.1.1" xref="A1.E3.m1.1.1.1.cmml"><mi id="A1.E3.m1.1.1.1.3" xref="A1.E3.m1.1.1.1.3.cmml">m</mi><mo id="A1.E3.m1.1.1.1.2" xref="A1.E3.m1.1.1.1.2.cmml">∈</mo><msubsup id="A1.E3.m1.1.1.1.1" xref="A1.E3.m1.1.1.1.1.cmml"><mrow id="A1.E3.m1.1.1.1.1.1.1.1" xref="A1.E3.m1.1.1.1.1.1.1.2.cmml"><mo id="A1.E3.m1.1.1.1.1.1.1.1.2" stretchy="false" xref="A1.E3.m1.1.1.1.1.1.1.2.cmml">{</mo><msub id="A1.E3.m1.1.1.1.1.1.1.1.1" xref="A1.E3.m1.1.1.1.1.1.1.1.1.cmml"><mi id="A1.E3.m1.1.1.1.1.1.1.1.1.2" xref="A1.E3.m1.1.1.1.1.1.1.1.1.2.cmml">m</mi><mi id="A1.E3.m1.1.1.1.1.1.1.1.1.3" xref="A1.E3.m1.1.1.1.1.1.1.1.1.3.cmml">i</mi></msub><mo id="A1.E3.m1.1.1.1.1.1.1.1.3" stretchy="false" xref="A1.E3.m1.1.1.1.1.1.1.2.cmml">}</mo></mrow><mrow id="A1.E3.m1.1.1.1.1.3" xref="A1.E3.m1.1.1.1.1.3.cmml"><mi id="A1.E3.m1.1.1.1.1.3.2" xref="A1.E3.m1.1.1.1.1.3.2.cmml">i</mi><mo id="A1.E3.m1.1.1.1.1.3.1" xref="A1.E3.m1.1.1.1.1.3.1.cmml">=</mo><mn id="A1.E3.m1.1.1.1.1.3.3" xref="A1.E3.m1.1.1.1.1.3.3.cmml">1</mn></mrow><mrow id="A1.E3.m1.1.1.1.1.1.3" xref="A1.E3.m1.1.1.1.1.1.3.cmml"><mi id="A1.E3.m1.1.1.1.1.1.3.2" xref="A1.E3.m1.1.1.1.1.1.3.2.cmml">k</mi><mo id="A1.E3.m1.1.1.1.1.1.3.1" xref="A1.E3.m1.1.1.1.1.1.3.1.cmml">−</mo><mn id="A1.E3.m1.1.1.1.1.1.3.3" xref="A1.E3.m1.1.1.1.1.1.3.3.cmml">1</mn></mrow></msubsup></mrow></msub><mo id="A1.E3.m1.3.3.1.1.3.1.2" xref="A1.E3.m1.3.3.1.1.3.1.2.cmml">⁢</mo><mi id="A1.E3.m1.3.3.1.1.3.1.4" xref="A1.E3.m1.3.3.1.1.3.1.4.cmml">T</mi><mo id="A1.E3.m1.3.3.1.1.3.1.2a" xref="A1.E3.m1.3.3.1.1.3.1.2.cmml">⁢</mo><msup id="A1.E3.m1.3.3.1.1.3.1.1" xref="A1.E3.m1.3.3.1.1.3.1.1.cmml"><mrow id="A1.E3.m1.3.3.1.1.3.1.1.1.1" xref="A1.E3.m1.3.3.1.1.3.1.1.1.2.cmml"><mo id="A1.E3.m1.3.3.1.1.3.1.1.1.1.2" stretchy="false" xref="A1.E3.m1.3.3.1.1.3.1.1.1.2.cmml">(</mo><msub id="A1.E3.m1.3.3.1.1.3.1.1.1.1.1" xref="A1.E3.m1.3.3.1.1.3.1.1.1.1.1.cmml"><mi id="A1.E3.m1.3.3.1.1.3.1.1.1.1.1.2" xref="A1.E3.m1.3.3.1.1.3.1.1.1.1.1.2.cmml">m</mi><mi id="A1.E3.m1.3.3.1.1.3.1.1.1.1.1.3" xref="A1.E3.m1.3.3.1.1.3.1.1.1.1.1.3.cmml">k</mi></msub><mo id="A1.E3.m1.3.3.1.1.3.1.1.1.1.3" xref="A1.E3.m1.3.3.1.1.3.1.1.1.2.cmml">,</mo><mi id="A1.E3.m1.2.2" xref="A1.E3.m1.2.2.cmml">m</mi><mo id="A1.E3.m1.3.3.1.1.3.1.1.1.1.4" stretchy="false" xref="A1.E3.m1.3.3.1.1.3.1.1.1.2.cmml">)</mo></mrow><mi id="A1.E3.m1.3.3.1.1.3.1.1.3" xref="A1.E3.m1.3.3.1.1.3.1.1.3.cmml">β</mi></msup></mrow></mrow></mrow><mo id="A1.E3.m1.3.3.1.2" xref="A1.E3.m1.3.3.1.1.cmml">,</mo></mrow><annotation-xml encoding="MathML-Content" id="A1.E3.m1.3b"><apply id="A1.E3.m1.3.3.1.1.cmml" xref="A1.E3.m1.3.3.1"><eq id="A1.E3.m1.3.3.1.1.4.cmml" xref="A1.E3.m1.3.3.1.1.4"></eq><apply id="A1.E3.m1.3.3.1.1.2.cmml" xref="A1.E3.m1.3.3.1.1.2"><times id="A1.E3.m1.3.3.1.1.2.3.cmml" xref="A1.E3.m1.3.3.1.1.2.3"></times><apply id="A1.E3.m1.3.3.1.1.2.4.cmml" xref="A1.E3.m1.3.3.1.1.2.4"><csymbol cd="ambiguous" id="A1.E3.m1.3.3.1.1.2.4.1.cmml" xref="A1.E3.m1.3.3.1.1.2.4">subscript</csymbol><ci id="A1.E3.m1.3.3.1.1.2.4.2.cmml" xref="A1.E3.m1.3.3.1.1.2.4.2">𝑟</ci><ci id="A1.E3.m1.3.3.1.1.2.4.3a.cmml" xref="A1.E3.m1.3.3.1.1.2.4.3"><mtext id="A1.E3.m1.3.3.1.1.2.4.3.cmml" mathsize="70%" xref="A1.E3.m1.3.3.1.1.2.4.3">div</mtext></ci></apply><interval closure="open" id="A1.E3.m1.3.3.1.1.2.2.3.cmml" xref="A1.E3.m1.3.3.1.1.2.2.2"><apply id="A1.E3.m1.3.3.1.1.1.1.1.1.cmml" xref="A1.E3.m1.3.3.1.1.1.1.1.1"><csymbol cd="ambiguous" id="A1.E3.m1.3.3.1.1.1.1.1.1.1.cmml" xref="A1.E3.m1.3.3.1.1.1.1.1.1">subscript</csymbol><ci id="A1.E3.m1.3.3.1.1.1.1.1.1.2.cmml" xref="A1.E3.m1.3.3.1.1.1.1.1.1.2">𝑚</ci><ci id="A1.E3.m1.3.3.1.1.1.1.1.1.3.cmml" xref="A1.E3.m1.3.3.1.1.1.1.1.1.3">𝑘</ci></apply><apply id="A1.E3.m1.3.3.1.1.2.2.2.2.cmml" xref="A1.E3.m1.3.3.1.1.2.2.2.2"><csymbol cd="ambiguous" id="A1.E3.m1.3.3.1.1.2.2.2.2.2.cmml" xref="A1.E3.m1.3.3.1.1.2.2.2.2">subscript</csymbol><apply id="A1.E3.m1.3.3.1.1.2.2.2.2.1.cmml" xref="A1.E3.m1.3.3.1.1.2.2.2.2"><csymbol cd="ambiguous" id="A1.E3.m1.3.3.1.1.2.2.2.2.1.2.cmml" xref="A1.E3.m1.3.3.1.1.2.2.2.2">superscript</csymbol><set id="A1.E3.m1.3.3.1.1.2.2.2.2.1.1.2.cmml" xref="A1.E3.m1.3.3.1.1.2.2.2.2.1.1.1"><apply id="A1.E3.m1.3.3.1.1.2.2.2.2.1.1.1.1.cmml" xref="A1.E3.m1.3.3.1.1.2.2.2.2.1.1.1.1"><csymbol cd="ambiguous" id="A1.E3.m1.3.3.1.1.2.2.2.2.1.1.1.1.1.cmml" xref="A1.E3.m1.3.3.1.1.2.2.2.2.1.1.1.1">subscript</csymbol><ci id="A1.E3.m1.3.3.1.1.2.2.2.2.1.1.1.1.2.cmml" xref="A1.E3.m1.3.3.1.1.2.2.2.2.1.1.1.1.2">𝑚</ci><ci id="A1.E3.m1.3.3.1.1.2.2.2.2.1.1.1.1.3.cmml" xref="A1.E3.m1.3.3.1.1.2.2.2.2.1.1.1.1.3">𝑖</ci></apply></set><apply id="A1.E3.m1.3.3.1.1.2.2.2.2.1.3.cmml" xref="A1.E3.m1.3.3.1.1.2.2.2.2.1.3"><minus id="A1.E3.m1.3.3.1.1.2.2.2.2.1.3.1.cmml" xref="A1.E3.m1.3.3.1.1.2.2.2.2.1.3.1"></minus><ci id="A1.E3.m1.3.3.1.1.2.2.2.2.1.3.2.cmml" xref="A1.E3.m1.3.3.1.1.2.2.2.2.1.3.2">𝑘</ci><cn id="A1.E3.m1.3.3.1.1.2.2.2.2.1.3.3.cmml" type="integer" xref="A1.E3.m1.3.3.1.1.2.2.2.2.1.3.3">1</cn></apply></apply><apply id="A1.E3.m1.3.3.1.1.2.2.2.2.3.cmml" xref="A1.E3.m1.3.3.1.1.2.2.2.2.3"><eq id="A1.E3.m1.3.3.1.1.2.2.2.2.3.1.cmml" xref="A1.E3.m1.3.3.1.1.2.2.2.2.3.1"></eq><ci id="A1.E3.m1.3.3.1.1.2.2.2.2.3.2.cmml" xref="A1.E3.m1.3.3.1.1.2.2.2.2.3.2">𝑖</ci><cn id="A1.E3.m1.3.3.1.1.2.2.2.2.3.3.cmml" type="integer" xref="A1.E3.m1.3.3.1.1.2.2.2.2.3.3">1</cn></apply></apply></interval></apply><apply id="A1.E3.m1.3.3.1.1.3.cmml" xref="A1.E3.m1.3.3.1.1.3"><minus id="A1.E3.m1.3.3.1.1.3.2.cmml" xref="A1.E3.m1.3.3.1.1.3.2"></minus><cn id="A1.E3.m1.3.3.1.1.3.3.cmml" type="integer" xref="A1.E3.m1.3.3.1.1.3.3">1</cn><apply id="A1.E3.m1.3.3.1.1.3.1.cmml" xref="A1.E3.m1.3.3.1.1.3.1"><times id="A1.E3.m1.3.3.1.1.3.1.2.cmml" xref="A1.E3.m1.3.3.1.1.3.1.2"></times><apply id="A1.E3.m1.3.3.1.1.3.1.3.cmml" xref="A1.E3.m1.3.3.1.1.3.1.3"><csymbol cd="ambiguous" id="A1.E3.m1.3.3.1.1.3.1.3.1.cmml" xref="A1.E3.m1.3.3.1.1.3.1.3">subscript</csymbol><ci id="A1.E3.m1.3.3.1.1.3.1.3.2a.cmml" xref="A1.E3.m1.3.3.1.1.3.1.3.2"><mtext id="A1.E3.m1.3.3.1.1.3.1.3.2.cmml" xref="A1.E3.m1.3.3.1.1.3.1.3.2">max</mtext></ci><apply id="A1.E3.m1.1.1.1.cmml" xref="A1.E3.m1.1.1.1"><in id="A1.E3.m1.1.1.1.2.cmml" xref="A1.E3.m1.1.1.1.2"></in><ci id="A1.E3.m1.1.1.1.3.cmml" xref="A1.E3.m1.1.1.1.3">𝑚</ci><apply id="A1.E3.m1.1.1.1.1.cmml" xref="A1.E3.m1.1.1.1.1"><csymbol cd="ambiguous" id="A1.E3.m1.1.1.1.1.2.cmml" xref="A1.E3.m1.1.1.1.1">subscript</csymbol><apply id="A1.E3.m1.1.1.1.1.1.cmml" xref="A1.E3.m1.1.1.1.1"><csymbol cd="ambiguous" id="A1.E3.m1.1.1.1.1.1.2.cmml" xref="A1.E3.m1.1.1.1.1">superscript</csymbol><set id="A1.E3.m1.1.1.1.1.1.1.2.cmml" xref="A1.E3.m1.1.1.1.1.1.1.1"><apply id="A1.E3.m1.1.1.1.1.1.1.1.1.cmml" xref="A1.E3.m1.1.1.1.1.1.1.1.1"><csymbol cd="ambiguous" id="A1.E3.m1.1.1.1.1.1.1.1.1.1.cmml" xref="A1.E3.m1.1.1.1.1.1.1.1.1">subscript</csymbol><ci id="A1.E3.m1.1.1.1.1.1.1.1.1.2.cmml" xref="A1.E3.m1.1.1.1.1.1.1.1.1.2">𝑚</ci><ci id="A1.E3.m1.1.1.1.1.1.1.1.1.3.cmml" xref="A1.E3.m1.1.1.1.1.1.1.1.1.3">𝑖</ci></apply></set><apply id="A1.E3.m1.1.1.1.1.1.3.cmml" xref="A1.E3.m1.1.1.1.1.1.3"><minus id="A1.E3.m1.1.1.1.1.1.3.1.cmml" xref="A1.E3.m1.1.1.1.1.1.3.1"></minus><ci id="A1.E3.m1.1.1.1.1.1.3.2.cmml" xref="A1.E3.m1.1.1.1.1.1.3.2">𝑘</ci><cn id="A1.E3.m1.1.1.1.1.1.3.3.cmml" type="integer" xref="A1.E3.m1.1.1.1.1.1.3.3">1</cn></apply></apply><apply id="A1.E3.m1.1.1.1.1.3.cmml" xref="A1.E3.m1.1.1.1.1.3"><eq id="A1.E3.m1.1.1.1.1.3.1.cmml" xref="A1.E3.m1.1.1.1.1.3.1"></eq><ci id="A1.E3.m1.1.1.1.1.3.2.cmml" xref="A1.E3.m1.1.1.1.1.3.2">𝑖</ci><cn id="A1.E3.m1.1.1.1.1.3.3.cmml" type="integer" xref="A1.E3.m1.1.1.1.1.3.3">1</cn></apply></apply></apply></apply><ci id="A1.E3.m1.3.3.1.1.3.1.4.cmml" xref="A1.E3.m1.3.3.1.1.3.1.4">𝑇</ci><apply id="A1.E3.m1.3.3.1.1.3.1.1.cmml" xref="A1.E3.m1.3.3.1.1.3.1.1"><csymbol cd="ambiguous" id="A1.E3.m1.3.3.1.1.3.1.1.2.cmml" xref="A1.E3.m1.3.3.1.1.3.1.1">superscript</csymbol><interval closure="open" id="A1.E3.m1.3.3.1.1.3.1.1.1.2.cmml" xref="A1.E3.m1.3.3.1.1.3.1.1.1.1"><apply id="A1.E3.m1.3.3.1.1.3.1.1.1.1.1.cmml" xref="A1.E3.m1.3.3.1.1.3.1.1.1.1.1"><csymbol cd="ambiguous" id="A1.E3.m1.3.3.1.1.3.1.1.1.1.1.1.cmml" xref="A1.E3.m1.3.3.1.1.3.1.1.1.1.1">subscript</csymbol><ci id="A1.E3.m1.3.3.1.1.3.1.1.1.1.1.2.cmml" xref="A1.E3.m1.3.3.1.1.3.1.1.1.1.1.2">𝑚</ci><ci id="A1.E3.m1.3.3.1.1.3.1.1.1.1.1.3.cmml" xref="A1.E3.m1.3.3.1.1.3.1.1.1.1.1.3">𝑘</ci></apply><ci id="A1.E3.m1.2.2.cmml" xref="A1.E3.m1.2.2">𝑚</ci></interval><ci id="A1.E3.m1.3.3.1.1.3.1.1.3.cmml" xref="A1.E3.m1.3.3.1.1.3.1.1.3">𝛽</ci></apply></apply></apply></apply></annotation-xml><annotation encoding="application/x-tex" id="A1.E3.m1.3c">r_{\text{div}}(m_{k},\{m_{i}\}^{k-1}_{i=1})=1-\text{max}_{m\in\{m_{i}\}^{k-1}_%
{i=1}}T(m_{k},m)^{\beta},</annotation><annotation encoding="application/x-llamapun" id="A1.E3.m1.3d">italic_r start_POSTSUBSCRIPT div end_POSTSUBSCRIPT ( italic_m start_POSTSUBSCRIPT italic_k end_POSTSUBSCRIPT , { italic_m start_POSTSUBSCRIPT italic_i end_POSTSUBSCRIPT } start_POSTSUPERSCRIPT italic_k - 1 end_POSTSUPERSCRIPT start_POSTSUBSCRIPT italic_i = 1 end_POSTSUBSCRIPT ) = 1 - max start_POSTSUBSCRIPT italic_m ∈ { italic_m start_POSTSUBSCRIPT italic_i end_POSTSUBSCRIPT } start_POSTSUPERSCRIPT italic_k - 1 end_POSTSUPERSCRIPT start_POSTSUBSCRIPT italic_i = 1 end_POSTSUBSCRIPT end_POSTSUBSCRIPT italic_T ( italic_m start_POSTSUBSCRIPT italic_k end_POSTSUBSCRIPT , italic_m ) start_POSTSUPERSCRIPT italic_β end_POSTSUPERSCRIPT ,</annotation></semantics></math></td>
<td class="ltx_eqn_cell ltx_eqn_center_padright"></td>
<td class="ltx_eqn_cell ltx_eqn_eqno ltx_align_middle ltx_align_right" rowspan="1"><span class="ltx_tag ltx_tag_equation ltx_align_right">(3)</span></td>
</tr></tbody>
</table>
<p class="ltx_p" id="A1.SS2.p3.7">where <math alttext="T(m_{i},m_{j})" class="ltx_Math" display="inline" id="A1.SS2.p3.2.m1.2"><semantics id="A1.SS2.p3.2.m1.2a"><mrow id="A1.SS2.p3.2.m1.2.2" xref="A1.SS2.p3.2.m1.2.2.cmml"><mi id="A1.SS2.p3.2.m1.2.2.4" xref="A1.SS2.p3.2.m1.2.2.4.cmml">T</mi><mo id="A1.SS2.p3.2.m1.2.2.3" xref="A1.SS2.p3.2.m1.2.2.3.cmml">⁢</mo><mrow id="A1.SS2.p3.2.m1.2.2.2.2" xref="A1.SS2.p3.2.m1.2.2.2.3.cmml"><mo id="A1.SS2.p3.2.m1.2.2.2.2.3" stretchy="false" xref="A1.SS2.p3.2.m1.2.2.2.3.cmml">(</mo><msub id="A1.SS2.p3.2.m1.1.1.1.1.1" xref="A1.SS2.p3.2.m1.1.1.1.1.1.cmml"><mi id="A1.SS2.p3.2.m1.1.1.1.1.1.2" xref="A1.SS2.p3.2.m1.1.1.1.1.1.2.cmml">m</mi><mi id="A1.SS2.p3.2.m1.1.1.1.1.1.3" xref="A1.SS2.p3.2.m1.1.1.1.1.1.3.cmml">i</mi></msub><mo id="A1.SS2.p3.2.m1.2.2.2.2.4" xref="A1.SS2.p3.2.m1.2.2.2.3.cmml">,</mo><msub id="A1.SS2.p3.2.m1.2.2.2.2.2" xref="A1.SS2.p3.2.m1.2.2.2.2.2.cmml"><mi id="A1.SS2.p3.2.m1.2.2.2.2.2.2" xref="A1.SS2.p3.2.m1.2.2.2.2.2.2.cmml">m</mi><mi id="A1.SS2.p3.2.m1.2.2.2.2.2.3" xref="A1.SS2.p3.2.m1.2.2.2.2.2.3.cmml">j</mi></msub><mo id="A1.SS2.p3.2.m1.2.2.2.2.5" stretchy="false" xref="A1.SS2.p3.2.m1.2.2.2.3.cmml">)</mo></mrow></mrow><annotation-xml encoding="MathML-Content" id="A1.SS2.p3.2.m1.2b"><apply id="A1.SS2.p3.2.m1.2.2.cmml" xref="A1.SS2.p3.2.m1.2.2"><times id="A1.SS2.p3.2.m1.2.2.3.cmml" xref="A1.SS2.p3.2.m1.2.2.3"></times><ci id="A1.SS2.p3.2.m1.2.2.4.cmml" xref="A1.SS2.p3.2.m1.2.2.4">𝑇</ci><interval closure="open" id="A1.SS2.p3.2.m1.2.2.2.3.cmml" xref="A1.SS2.p3.2.m1.2.2.2.2"><apply id="A1.SS2.p3.2.m1.1.1.1.1.1.cmml" xref="A1.SS2.p3.2.m1.1.1.1.1.1"><csymbol cd="ambiguous" id="A1.SS2.p3.2.m1.1.1.1.1.1.1.cmml" xref="A1.SS2.p3.2.m1.1.1.1.1.1">subscript</csymbol><ci id="A1.SS2.p3.2.m1.1.1.1.1.1.2.cmml" xref="A1.SS2.p3.2.m1.1.1.1.1.1.2">𝑚</ci><ci id="A1.SS2.p3.2.m1.1.1.1.1.1.3.cmml" xref="A1.SS2.p3.2.m1.1.1.1.1.1.3">𝑖</ci></apply><apply id="A1.SS2.p3.2.m1.2.2.2.2.2.cmml" xref="A1.SS2.p3.2.m1.2.2.2.2.2"><csymbol cd="ambiguous" id="A1.SS2.p3.2.m1.2.2.2.2.2.1.cmml" xref="A1.SS2.p3.2.m1.2.2.2.2.2">subscript</csymbol><ci id="A1.SS2.p3.2.m1.2.2.2.2.2.2.cmml" xref="A1.SS2.p3.2.m1.2.2.2.2.2.2">𝑚</ci><ci id="A1.SS2.p3.2.m1.2.2.2.2.2.3.cmml" xref="A1.SS2.p3.2.m1.2.2.2.2.2.3">𝑗</ci></apply></interval></apply></annotation-xml><annotation encoding="application/x-tex" id="A1.SS2.p3.2.m1.2c">T(m_{i},m_{j})</annotation><annotation encoding="application/x-llamapun" id="A1.SS2.p3.2.m1.2d">italic_T ( italic_m start_POSTSUBSCRIPT italic_i end_POSTSUBSCRIPT , italic_m start_POSTSUBSCRIPT italic_j end_POSTSUBSCRIPT )</annotation></semantics></math> measures the Tanimoto similarity between <math alttext="m_{i}" class="ltx_Math" display="inline" id="A1.SS2.p3.3.m2.1"><semantics id="A1.SS2.p3.3.m2.1a"><msub id="A1.SS2.p3.3.m2.1.1" xref="A1.SS2.p3.3.m2.1.1.cmml"><mi id="A1.SS2.p3.3.m2.1.1.2" xref="A1.SS2.p3.3.m2.1.1.2.cmml">m</mi><mi id="A1.SS2.p3.3.m2.1.1.3" xref="A1.SS2.p3.3.m2.1.1.3.cmml">i</mi></msub><annotation-xml encoding="MathML-Content" id="A1.SS2.p3.3.m2.1b"><apply id="A1.SS2.p3.3.m2.1.1.cmml" xref="A1.SS2.p3.3.m2.1.1"><csymbol cd="ambiguous" id="A1.SS2.p3.3.m2.1.1.1.cmml" xref="A1.SS2.p3.3.m2.1.1">subscript</csymbol><ci id="A1.SS2.p3.3.m2.1.1.2.cmml" xref="A1.SS2.p3.3.m2.1.1.2">𝑚</ci><ci id="A1.SS2.p3.3.m2.1.1.3.cmml" xref="A1.SS2.p3.3.m2.1.1.3">𝑖</ci></apply></annotation-xml><annotation encoding="application/x-tex" id="A1.SS2.p3.3.m2.1c">m_{i}</annotation><annotation encoding="application/x-llamapun" id="A1.SS2.p3.3.m2.1d">italic_m start_POSTSUBSCRIPT italic_i end_POSTSUBSCRIPT</annotation></semantics></math> and <math alttext="m_{j}" class="ltx_Math" display="inline" id="A1.SS2.p3.4.m3.1"><semantics id="A1.SS2.p3.4.m3.1a"><msub id="A1.SS2.p3.4.m3.1.1" xref="A1.SS2.p3.4.m3.1.1.cmml"><mi id="A1.SS2.p3.4.m3.1.1.2" xref="A1.SS2.p3.4.m3.1.1.2.cmml">m</mi><mi id="A1.SS2.p3.4.m3.1.1.3" xref="A1.SS2.p3.4.m3.1.1.3.cmml">j</mi></msub><annotation-xml encoding="MathML-Content" id="A1.SS2.p3.4.m3.1b"><apply id="A1.SS2.p3.4.m3.1.1.cmml" xref="A1.SS2.p3.4.m3.1.1"><csymbol cd="ambiguous" id="A1.SS2.p3.4.m3.1.1.1.cmml" xref="A1.SS2.p3.4.m3.1.1">subscript</csymbol><ci id="A1.SS2.p3.4.m3.1.1.2.cmml" xref="A1.SS2.p3.4.m3.1.1.2">𝑚</ci><ci id="A1.SS2.p3.4.m3.1.1.3.cmml" xref="A1.SS2.p3.4.m3.1.1.3">𝑗</ci></apply></annotation-xml><annotation encoding="application/x-tex" id="A1.SS2.p3.4.m3.1c">m_{j}</annotation><annotation encoding="application/x-llamapun" id="A1.SS2.p3.4.m3.1d">italic_m start_POSTSUBSCRIPT italic_j end_POSTSUBSCRIPT</annotation></semantics></math> by assessing their true molecular structures, i.e., fingerprints. This metric yields a value between <math alttext="0" class="ltx_Math" display="inline" id="A1.SS2.p3.5.m4.1"><semantics id="A1.SS2.p3.5.m4.1a"><mn id="A1.SS2.p3.5.m4.1.1" xref="A1.SS2.p3.5.m4.1.1.cmml">0</mn><annotation-xml encoding="MathML-Content" id="A1.SS2.p3.5.m4.1b"><cn id="A1.SS2.p3.5.m4.1.1.cmml" type="integer" xref="A1.SS2.p3.5.m4.1.1">0</cn></annotation-xml></semantics></math> and <math alttext="1" class="ltx_Math" display="inline" id="A1.SS2.p3.6.m5.1"><semantics id="A1.SS2.p3.6.m5.1a"><mn id="A1.SS2.p3.6.m5.1.1" xref="A1.SS2.p3.6.m5.1.1.cmml">1</mn><annotation-xml encoding="MathML-Content" id="A1.SS2.p3.6.m5.1b"><cn id="A1.SS2.p3.6.m5.1.1.cmml" type="integer" xref="A1.SS2.p3.6.m5.1.1">1</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.SS2.p3.6.m5.1c">1</annotation><annotation encoding="application/x-llamapun" id="A1.SS2.p3.6.m5.1d">1</annotation></semantics></math> and can be obtained using the RDKit package. Note that <math alttext="\beta" class="ltx_Math" display="inline" id="A1.SS2.p3.7.m6.1"><semantics id="A1.SS2.p3.7.m6.1a"><mi id="A1.SS2.p3.7.m6.1.1" xref="A1.SS2.p3.7.m6.1.1.cmml">β</mi><annotation-xml encoding="MathML-Content" id="A1.SS2.p3.7.m6.1b"><ci id="A1.SS2.p3.7.m6.1.1.cmml" xref="A1.SS2.p3.7.m6.1.1">𝛽</ci></annotation-xml><annotation encoding="application/x-tex" id="A1.SS2.p3.7.m6.1c">\beta</annotation><annotation encoding="application/x-llamapun" id="A1.SS2.p3.7.m6.1d">italic_β</annotation></semantics></math> is a hyper-parameter.</p>
</div>
<div class="ltx_para ltx_noindent" id="A1.SS2.p4">
<p class="ltx_p" id="A1.SS2.p4.1"><span class="ltx_text ltx_font_bold" id="A1.SS2.p4.1.1">Policy optimization.</span> We consider four NVIDIA A100 SXM4 40GB for reinforcement learning implemented with proximal policy optimization.</p>
</div>
<div class="ltx_para ltx_noindent" id="A1.SS2.p5">
<ul class="ltx_itemize" id="A1.I2">
<li class="ltx_item" id="A1.I2.i1" style="list-style-type:none;">
<span class="ltx_tag ltx_tag_item">•</span>
<div class="ltx_para" id="A1.I2.i1.p1">
<p class="ltx_p" id="A1.I2.i1.p1.15">For the reinforcement learning of BioT5<sup class="ltx_sup" id="A1.I2.i1.p1.15.1"><span class="ltx_text ltx_font_italic" id="A1.I2.i1.p1.15.1.1">+</span></sup> and MolT5 (<a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.SS1" title="4.1 Comparison with decoding schemes ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Sections</span> <span class="ltx_text ltx_ref_tag">4.1</span></a> and <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.SS2" title="4.2 Comparison with existing LLMs ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">4.2</span></a>), we consider <math alttext="200" class="ltx_Math" display="inline" id="A1.I2.i1.p1.2.m2.1"><semantics id="A1.I2.i1.p1.2.m2.1a"><mn id="A1.I2.i1.p1.2.m2.1.1" xref="A1.I2.i1.p1.2.m2.1.1.cmml">200</mn><annotation-xml encoding="MathML-Content" id="A1.I2.i1.p1.2.m2.1b"><cn id="A1.I2.i1.p1.2.m2.1.1.cmml" type="integer" xref="A1.I2.i1.p1.2.m2.1.1">200</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I2.i1.p1.2.m2.1c">200</annotation><annotation encoding="application/x-llamapun" id="A1.I2.i1.p1.2.m2.1d">200</annotation></semantics></math> PPO iterations, <math alttext="8" class="ltx_Math" display="inline" id="A1.I2.i1.p1.3.m3.1"><semantics id="A1.I2.i1.p1.3.m3.1a"><mn id="A1.I2.i1.p1.3.m3.1.1" xref="A1.I2.i1.p1.3.m3.1.1.cmml">8</mn><annotation-xml encoding="MathML-Content" id="A1.I2.i1.p1.3.m3.1b"><cn id="A1.I2.i1.p1.3.m3.1.1.cmml" type="integer" xref="A1.I2.i1.p1.3.m3.1.1">8</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I2.i1.p1.3.m3.1c">8</annotation><annotation encoding="application/x-llamapun" id="A1.I2.i1.p1.3.m3.1d">8</annotation></semantics></math> mini-batch size, <math alttext="128" class="ltx_Math" display="inline" id="A1.I2.i1.p1.4.m4.1"><semantics id="A1.I2.i1.p1.4.m4.1a"><mn id="A1.I2.i1.p1.4.m4.1.1" xref="A1.I2.i1.p1.4.m4.1.1.cmml">128</mn><annotation-xml encoding="MathML-Content" id="A1.I2.i1.p1.4.m4.1b"><cn id="A1.I2.i1.p1.4.m4.1.1.cmml" type="integer" xref="A1.I2.i1.p1.4.m4.1.1">128</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I2.i1.p1.4.m4.1c">128</annotation><annotation encoding="application/x-llamapun" id="A1.I2.i1.p1.4.m4.1d">128</annotation></semantics></math> batch size, and <math alttext="5\textrm{e}-5" class="ltx_Math" display="inline" id="A1.I2.i1.p1.5.m5.1"><semantics id="A1.I2.i1.p1.5.m5.1a"><mrow id="A1.I2.i1.p1.5.m5.1.1" xref="A1.I2.i1.p1.5.m5.1.1.cmml"><mrow id="A1.I2.i1.p1.5.m5.1.1.2" xref="A1.I2.i1.p1.5.m5.1.1.2.cmml"><mn id="A1.I2.i1.p1.5.m5.1.1.2.2" xref="A1.I2.i1.p1.5.m5.1.1.2.2.cmml">5</mn><mo id="A1.I2.i1.p1.5.m5.1.1.2.1" xref="A1.I2.i1.p1.5.m5.1.1.2.1.cmml">⁢</mo><mtext id="A1.I2.i1.p1.5.m5.1.1.2.3" xref="A1.I2.i1.p1.5.m5.1.1.2.3a.cmml">e</mtext></mrow><mo id="A1.I2.i1.p1.5.m5.1.1.1" xref="A1.I2.i1.p1.5.m5.1.1.1.cmml">−</mo><mn id="A1.I2.i1.p1.5.m5.1.1.3" xref="A1.I2.i1.p1.5.m5.1.1.3.cmml">5</mn></mrow><annotation-xml encoding="MathML-Content" id="A1.I2.i1.p1.5.m5.1b"><apply id="A1.I2.i1.p1.5.m5.1.1.cmml" xref="A1.I2.i1.p1.5.m5.1.1"><minus id="A1.I2.i1.p1.5.m5.1.1.1.cmml" xref="A1.I2.i1.p1.5.m5.1.1.1"></minus><apply id="A1.I2.i1.p1.5.m5.1.1.2.cmml" xref="A1.I2.i1.p1.5.m5.1.1.2"><times id="A1.I2.i1.p1.5.m5.1.1.2.1.cmml" xref="A1.I2.i1.p1.5.m5.1.1.2.1"></times><cn id="A1.I2.i1.p1.5.m5.1.1.2.2.cmml" type="integer" xref="A1.I2.i1.p1.5.m5.1.1.2.2">5</cn><ci id="A1.I2.i1.p1.5.m5.1.1.2.3a.cmml" xref="A1.I2.i1.p1.5.m5.1.1.2.3"><mtext id="A1.I2.i1.p1.5.m5.1.1.2.3.cmml" xref="A1.I2.i1.p1.5.m5.1.1.2.3">e</mtext></ci></apply><cn id="A1.I2.i1.p1.5.m5.1.1.3.cmml" type="integer" xref="A1.I2.i1.p1.5.m5.1.1.3">5</cn></apply></annotation-xml><annotation encoding="application/x-tex" id="A1.I2.i1.p1.5.m5.1c">5\textrm{e}-5</annotation><annotation encoding="application/x-llamapun" id="A1.I2.i1.p1.5.m5.1d">5 e - 5</annotation></semantics></math> learning rate. We also consider <math alttext="0.01" class="ltx_Math" display="inline" id="A1.I2.i1.p1.6.m6.1"><semantics id="A1.I2.i1.p1.6.m6.1a"><mn id="A1.I2.i1.p1.6.m6.1.1" xref="A1.I2.i1.p1.6.m6.1.1.cmml">0.01</mn><annotation-xml encoding="MathML-Content" id="A1.I2.i1.p1.6.m6.1b"><cn id="A1.I2.i1.p1.6.m6.1.1.cmml" type="float" xref="A1.I2.i1.p1.6.m6.1.1">0.01</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I2.i1.p1.6.m6.1c">0.01</annotation><annotation encoding="application/x-llamapun" id="A1.I2.i1.p1.6.m6.1d">0.01</annotation></semantics></math> KL penalty. Note that <math alttext="\alpha" class="ltx_Math" display="inline" id="A1.I2.i1.p1.7.m7.1"><semantics id="A1.I2.i1.p1.7.m7.1a"><mi id="A1.I2.i1.p1.7.m7.1.1" xref="A1.I2.i1.p1.7.m7.1.1.cmml">α</mi><annotation-xml encoding="MathML-Content" id="A1.I2.i1.p1.7.m7.1b"><ci id="A1.I2.i1.p1.7.m7.1.1.cmml" xref="A1.I2.i1.p1.7.m7.1.1">𝛼</ci></annotation-xml><annotation encoding="application/x-tex" id="A1.I2.i1.p1.7.m7.1c">\alpha</annotation><annotation encoding="application/x-llamapun" id="A1.I2.i1.p1.7.m7.1d">italic_α</annotation></semantics></math> and <math alttext="\beta" class="ltx_Math" display="inline" id="A1.I2.i1.p1.8.m8.1"><semantics id="A1.I2.i1.p1.8.m8.1a"><mi id="A1.I2.i1.p1.8.m8.1.1" xref="A1.I2.i1.p1.8.m8.1.1.cmml">β</mi><annotation-xml encoding="MathML-Content" id="A1.I2.i1.p1.8.m8.1b"><ci id="A1.I2.i1.p1.8.m8.1.1.cmml" xref="A1.I2.i1.p1.8.m8.1.1">𝛽</ci></annotation-xml><annotation encoding="application/x-tex" id="A1.I2.i1.p1.8.m8.1c">\beta</annotation><annotation encoding="application/x-llamapun" id="A1.I2.i1.p1.8.m8.1d">italic_β</annotation></semantics></math> in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#A1.E2" title="In A.2 Reinforcement learning ‣ Appendix A Detailed implementations and training ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Equations</span> <span class="ltx_text ltx_ref_tag">2</span></a> and <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#A1.E3" title="Equation 3 ‣ A.2 Reinforcement learning ‣ Appendix A Detailed implementations and training ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">3</span></a> are <math alttext="0.5" class="ltx_Math" display="inline" id="A1.I2.i1.p1.9.m9.1"><semantics id="A1.I2.i1.p1.9.m9.1a"><mn id="A1.I2.i1.p1.9.m9.1.1" xref="A1.I2.i1.p1.9.m9.1.1.cmml">0.5</mn><annotation-xml encoding="MathML-Content" id="A1.I2.i1.p1.9.m9.1b"><cn id="A1.I2.i1.p1.9.m9.1.1.cmml" type="float" xref="A1.I2.i1.p1.9.m9.1.1">0.5</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I2.i1.p1.9.m9.1c">0.5</annotation><annotation encoding="application/x-llamapun" id="A1.I2.i1.p1.9.m9.1d">0.5</annotation></semantics></math> and <math alttext="2.0" class="ltx_Math" display="inline" id="A1.I2.i1.p1.10.m10.1"><semantics id="A1.I2.i1.p1.10.m10.1a"><mn id="A1.I2.i1.p1.10.m10.1.1" xref="A1.I2.i1.p1.10.m10.1.1.cmml">2.0</mn><annotation-xml encoding="MathML-Content" id="A1.I2.i1.p1.10.m10.1b"><cn id="A1.I2.i1.p1.10.m10.1.1.cmml" type="float" xref="A1.I2.i1.p1.10.m10.1.1">2.0</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I2.i1.p1.10.m10.1c">2.0</annotation><annotation encoding="application/x-llamapun" id="A1.I2.i1.p1.10.m10.1d">2.0</annotation></semantics></math>, respectively. The reward signal is amplified by multiplying by a value of <math alttext="8.0" class="ltx_Math" display="inline" id="A1.I2.i1.p1.11.m11.1"><semantics id="A1.I2.i1.p1.11.m11.1a"><mn id="A1.I2.i1.p1.11.m11.1.1" xref="A1.I2.i1.p1.11.m11.1.1.cmml">8.0</mn><annotation-xml encoding="MathML-Content" id="A1.I2.i1.p1.11.m11.1b"><cn id="A1.I2.i1.p1.11.m11.1.1.cmml" type="float" xref="A1.I2.i1.p1.11.m11.1.1">8.0</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I2.i1.p1.11.m11.1c">8.0</annotation><annotation encoding="application/x-llamapun" id="A1.I2.i1.p1.11.m11.1d">8.0</annotation></semantics></math>. The maximum sequence length in reinforcement learning is limited to <math alttext="2560" class="ltx_Math" display="inline" id="A1.I2.i1.p1.12.m12.1"><semantics id="A1.I2.i1.p1.12.m12.1a"><mn id="A1.I2.i1.p1.12.m12.1.1" xref="A1.I2.i1.p1.12.m12.1.1.cmml">2560</mn><annotation-xml encoding="MathML-Content" id="A1.I2.i1.p1.12.m12.1b"><cn id="A1.I2.i1.p1.12.m12.1.1.cmml" type="integer" xref="A1.I2.i1.p1.12.m12.1.1">2560</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I2.i1.p1.12.m12.1c">2560</annotation><annotation encoding="application/x-llamapun" id="A1.I2.i1.p1.12.m12.1d">2560</annotation></semantics></math> due to memory limitations. Here, we also apply LoRA <cite class="ltx_cite ltx_citemacro_citep">(Hu et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib14" title="">2022</a>)</cite> where the rank and alpha are <math alttext="16" class="ltx_Math" display="inline" id="A1.I2.i1.p1.13.m13.1"><semantics id="A1.I2.i1.p1.13.m13.1a"><mn id="A1.I2.i1.p1.13.m13.1.1" xref="A1.I2.i1.p1.13.m13.1.1.cmml">16</mn><annotation-xml encoding="MathML-Content" id="A1.I2.i1.p1.13.m13.1b"><cn id="A1.I2.i1.p1.13.m13.1.1.cmml" type="integer" xref="A1.I2.i1.p1.13.m13.1.1">16</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I2.i1.p1.13.m13.1c">16</annotation><annotation encoding="application/x-llamapun" id="A1.I2.i1.p1.13.m13.1d">16</annotation></semantics></math> and <math alttext="32" class="ltx_Math" display="inline" id="A1.I2.i1.p1.14.m14.1"><semantics id="A1.I2.i1.p1.14.m14.1a"><mn id="A1.I2.i1.p1.14.m14.1.1" xref="A1.I2.i1.p1.14.m14.1.1.cmml">32</mn><annotation-xml encoding="MathML-Content" id="A1.I2.i1.p1.14.m14.1b"><cn id="A1.I2.i1.p1.14.m14.1.1.cmml" type="integer" xref="A1.I2.i1.p1.14.m14.1.1">32</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I2.i1.p1.14.m14.1c">32</annotation><annotation encoding="application/x-llamapun" id="A1.I2.i1.p1.14.m14.1d">32</annotation></semantics></math>, respectively. We save the model every <math alttext="25" class="ltx_Math" display="inline" id="A1.I2.i1.p1.15.m15.1"><semantics id="A1.I2.i1.p1.15.m15.1a"><mn id="A1.I2.i1.p1.15.m15.1.1" xref="A1.I2.i1.p1.15.m15.1.1.cmml">25</mn><annotation-xml encoding="MathML-Content" id="A1.I2.i1.p1.15.m15.1b"><cn id="A1.I2.i1.p1.15.m15.1.1.cmml" type="integer" xref="A1.I2.i1.p1.15.m15.1.1">25</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I2.i1.p1.15.m15.1c">25</annotation><annotation encoding="application/x-llamapun" id="A1.I2.i1.p1.15.m15.1d">25</annotation></semantics></math> PPO iteration and select the model yielding the highest rewards for the training prompts.</p>
</div>
</li>
<li class="ltx_item" id="A1.I2.i2" style="list-style-type:none;">
<span class="ltx_tag ltx_tag_item">•</span>
<div class="ltx_para ltx_noindent" id="A1.I2.i2.p1">
<p class="ltx_p" id="A1.I2.i2.p1.12">For the reinforcement learning of DrugAssist (<a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.SS3" title="4.3 Fine-tuning generalist LLMs ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Section</span> <span class="ltx_text ltx_ref_tag">4.3</span></a>), we consider <math alttext="200" class="ltx_Math" display="inline" id="A1.I2.i2.p1.1.m1.1"><semantics id="A1.I2.i2.p1.1.m1.1a"><mn id="A1.I2.i2.p1.1.m1.1.1" xref="A1.I2.i2.p1.1.m1.1.1.cmml">200</mn><annotation-xml encoding="MathML-Content" id="A1.I2.i2.p1.1.m1.1b"><cn id="A1.I2.i2.p1.1.m1.1.1.cmml" type="integer" xref="A1.I2.i2.p1.1.m1.1.1">200</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I2.i2.p1.1.m1.1c">200</annotation><annotation encoding="application/x-llamapun" id="A1.I2.i2.p1.1.m1.1d">200</annotation></semantics></math> PPO iterations, <math alttext="4" class="ltx_Math" display="inline" id="A1.I2.i2.p1.2.m2.1"><semantics id="A1.I2.i2.p1.2.m2.1a"><mn id="A1.I2.i2.p1.2.m2.1.1" xref="A1.I2.i2.p1.2.m2.1.1.cmml">4</mn><annotation-xml encoding="MathML-Content" id="A1.I2.i2.p1.2.m2.1b"><cn id="A1.I2.i2.p1.2.m2.1.1.cmml" type="integer" xref="A1.I2.i2.p1.2.m2.1.1">4</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I2.i2.p1.2.m2.1c">4</annotation><annotation encoding="application/x-llamapun" id="A1.I2.i2.p1.2.m2.1d">4</annotation></semantics></math> mini-batch size, <math alttext="64" class="ltx_Math" display="inline" id="A1.I2.i2.p1.3.m3.1"><semantics id="A1.I2.i2.p1.3.m3.1a"><mn id="A1.I2.i2.p1.3.m3.1.1" xref="A1.I2.i2.p1.3.m3.1.1.cmml">64</mn><annotation-xml encoding="MathML-Content" id="A1.I2.i2.p1.3.m3.1b"><cn id="A1.I2.i2.p1.3.m3.1.1.cmml" type="integer" xref="A1.I2.i2.p1.3.m3.1.1">64</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I2.i2.p1.3.m3.1c">64</annotation><annotation encoding="application/x-llamapun" id="A1.I2.i2.p1.3.m3.1d">64</annotation></semantics></math> batch size, and <math alttext="3\textrm{e}-6" class="ltx_Math" display="inline" id="A1.I2.i2.p1.4.m4.1"><semantics id="A1.I2.i2.p1.4.m4.1a"><mrow id="A1.I2.i2.p1.4.m4.1.1" xref="A1.I2.i2.p1.4.m4.1.1.cmml"><mrow id="A1.I2.i2.p1.4.m4.1.1.2" xref="A1.I2.i2.p1.4.m4.1.1.2.cmml"><mn id="A1.I2.i2.p1.4.m4.1.1.2.2" xref="A1.I2.i2.p1.4.m4.1.1.2.2.cmml">3</mn><mo id="A1.I2.i2.p1.4.m4.1.1.2.1" xref="A1.I2.i2.p1.4.m4.1.1.2.1.cmml">⁢</mo><mtext id="A1.I2.i2.p1.4.m4.1.1.2.3" xref="A1.I2.i2.p1.4.m4.1.1.2.3a.cmml">e</mtext></mrow><mo id="A1.I2.i2.p1.4.m4.1.1.1" xref="A1.I2.i2.p1.4.m4.1.1.1.cmml">−</mo><mn id="A1.I2.i2.p1.4.m4.1.1.3" xref="A1.I2.i2.p1.4.m4.1.1.3.cmml">6</mn></mrow><annotation-xml encoding="MathML-Content" id="A1.I2.i2.p1.4.m4.1b"><apply id="A1.I2.i2.p1.4.m4.1.1.cmml" xref="A1.I2.i2.p1.4.m4.1.1"><minus id="A1.I2.i2.p1.4.m4.1.1.1.cmml" xref="A1.I2.i2.p1.4.m4.1.1.1"></minus><apply id="A1.I2.i2.p1.4.m4.1.1.2.cmml" xref="A1.I2.i2.p1.4.m4.1.1.2"><times id="A1.I2.i2.p1.4.m4.1.1.2.1.cmml" xref="A1.I2.i2.p1.4.m4.1.1.2.1"></times><cn id="A1.I2.i2.p1.4.m4.1.1.2.2.cmml" type="integer" xref="A1.I2.i2.p1.4.m4.1.1.2.2">3</cn><ci id="A1.I2.i2.p1.4.m4.1.1.2.3a.cmml" xref="A1.I2.i2.p1.4.m4.1.1.2.3"><mtext id="A1.I2.i2.p1.4.m4.1.1.2.3.cmml" xref="A1.I2.i2.p1.4.m4.1.1.2.3">e</mtext></ci></apply><cn id="A1.I2.i2.p1.4.m4.1.1.3.cmml" type="integer" xref="A1.I2.i2.p1.4.m4.1.1.3">6</cn></apply></annotation-xml><annotation encoding="application/x-tex" id="A1.I2.i2.p1.4.m4.1c">3\textrm{e}-6</annotation><annotation encoding="application/x-llamapun" id="A1.I2.i2.p1.4.m4.1d">3 e - 6</annotation></semantics></math> learning rate. We also consider <math alttext="0.1" class="ltx_Math" display="inline" id="A1.I2.i2.p1.5.m5.1"><semantics id="A1.I2.i2.p1.5.m5.1a"><mn id="A1.I2.i2.p1.5.m5.1.1" xref="A1.I2.i2.p1.5.m5.1.1.cmml">0.1</mn><annotation-xml encoding="MathML-Content" id="A1.I2.i2.p1.5.m5.1b"><cn id="A1.I2.i2.p1.5.m5.1.1.cmml" type="float" xref="A1.I2.i2.p1.5.m5.1.1">0.1</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I2.i2.p1.5.m5.1c">0.1</annotation><annotation encoding="application/x-llamapun" id="A1.I2.i2.p1.5.m5.1d">0.1</annotation></semantics></math> KL penalty. Note that <math alttext="\beta" class="ltx_Math" display="inline" id="A1.I2.i2.p1.6.m6.1"><semantics id="A1.I2.i2.p1.6.m6.1a"><mi id="A1.I2.i2.p1.6.m6.1.1" xref="A1.I2.i2.p1.6.m6.1.1.cmml">β</mi><annotation-xml encoding="MathML-Content" id="A1.I2.i2.p1.6.m6.1b"><ci id="A1.I2.i2.p1.6.m6.1.1.cmml" xref="A1.I2.i2.p1.6.m6.1.1">𝛽</ci></annotation-xml><annotation encoding="application/x-tex" id="A1.I2.i2.p1.6.m6.1c">\beta</annotation><annotation encoding="application/x-llamapun" id="A1.I2.i2.p1.6.m6.1d">italic_β</annotation></semantics></math> in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#A1.E3" title="In A.2 Reinforcement learning ‣ Appendix A Detailed implementations and training ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Equation</span> <span class="ltx_text ltx_ref_tag">3</span></a> is <math alttext="2.0" class="ltx_Math" display="inline" id="A1.I2.i2.p1.7.m7.1"><semantics id="A1.I2.i2.p1.7.m7.1a"><mn id="A1.I2.i2.p1.7.m7.1.1" xref="A1.I2.i2.p1.7.m7.1.1.cmml">2.0</mn><annotation-xml encoding="MathML-Content" id="A1.I2.i2.p1.7.m7.1b"><cn id="A1.I2.i2.p1.7.m7.1.1.cmml" type="float" xref="A1.I2.i2.p1.7.m7.1.1">2.0</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I2.i2.p1.7.m7.1c">2.0</annotation><annotation encoding="application/x-llamapun" id="A1.I2.i2.p1.7.m7.1d">2.0</annotation></semantics></math>. The reward signal is amplified by multiplying by a value of <math alttext="4.0" class="ltx_Math" display="inline" id="A1.I2.i2.p1.8.m8.1"><semantics id="A1.I2.i2.p1.8.m8.1a"><mn id="A1.I2.i2.p1.8.m8.1.1" xref="A1.I2.i2.p1.8.m8.1.1.cmml">4.0</mn><annotation-xml encoding="MathML-Content" id="A1.I2.i2.p1.8.m8.1b"><cn id="A1.I2.i2.p1.8.m8.1.1.cmml" type="float" xref="A1.I2.i2.p1.8.m8.1.1">4.0</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I2.i2.p1.8.m8.1c">4.0</annotation><annotation encoding="application/x-llamapun" id="A1.I2.i2.p1.8.m8.1d">4.0</annotation></semantics></math>. The maximum sequence length in reinforcement learning is limited to <math alttext="1280" class="ltx_Math" display="inline" id="A1.I2.i2.p1.9.m9.1"><semantics id="A1.I2.i2.p1.9.m9.1a"><mn id="A1.I2.i2.p1.9.m9.1.1" xref="A1.I2.i2.p1.9.m9.1.1.cmml">1280</mn><annotation-xml encoding="MathML-Content" id="A1.I2.i2.p1.9.m9.1b"><cn id="A1.I2.i2.p1.9.m9.1.1.cmml" type="integer" xref="A1.I2.i2.p1.9.m9.1.1">1280</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I2.i2.p1.9.m9.1c">1280</annotation><annotation encoding="application/x-llamapun" id="A1.I2.i2.p1.9.m9.1d">1280</annotation></semantics></math>. We also apply LoRA <cite class="ltx_cite ltx_citemacro_citep">(Hu et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib14" title="">2022</a>)</cite> where the rank and alpha are <math alttext="64" class="ltx_Math" display="inline" id="A1.I2.i2.p1.10.m10.1"><semantics id="A1.I2.i2.p1.10.m10.1a"><mn id="A1.I2.i2.p1.10.m10.1.1" xref="A1.I2.i2.p1.10.m10.1.1.cmml">64</mn><annotation-xml encoding="MathML-Content" id="A1.I2.i2.p1.10.m10.1b"><cn id="A1.I2.i2.p1.10.m10.1.1.cmml" type="integer" xref="A1.I2.i2.p1.10.m10.1.1">64</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I2.i2.p1.10.m10.1c">64</annotation><annotation encoding="application/x-llamapun" id="A1.I2.i2.p1.10.m10.1d">64</annotation></semantics></math> and <math alttext="128" class="ltx_Math" display="inline" id="A1.I2.i2.p1.11.m11.1"><semantics id="A1.I2.i2.p1.11.m11.1a"><mn id="A1.I2.i2.p1.11.m11.1.1" xref="A1.I2.i2.p1.11.m11.1.1.cmml">128</mn><annotation-xml encoding="MathML-Content" id="A1.I2.i2.p1.11.m11.1b"><cn id="A1.I2.i2.p1.11.m11.1.1.cmml" type="integer" xref="A1.I2.i2.p1.11.m11.1.1">128</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I2.i2.p1.11.m11.1c">128</annotation><annotation encoding="application/x-llamapun" id="A1.I2.i2.p1.11.m11.1d">128</annotation></semantics></math>, respectively. We save the model every <math alttext="25" class="ltx_Math" display="inline" id="A1.I2.i2.p1.12.m12.1"><semantics id="A1.I2.i2.p1.12.m12.1a"><mn id="A1.I2.i2.p1.12.m12.1.1" xref="A1.I2.i2.p1.12.m12.1.1.cmml">25</mn><annotation-xml encoding="MathML-Content" id="A1.I2.i2.p1.12.m12.1b"><cn id="A1.I2.i2.p1.12.m12.1.1.cmml" type="integer" xref="A1.I2.i2.p1.12.m12.1.1">25</cn></annotation-xml><annotation encoding="application/x-tex" id="A1.I2.i2.p1.12.m12.1c">25</annotation><annotation encoding="application/x-llamapun" id="A1.I2.i2.p1.12.m12.1d">25</annotation></semantics></math> PPO iteration and select the model yielding the highest rewards for the training prompts.</p>
</div>
</li>
</ul>
</div>
<div class="ltx_pagination ltx_role_newpage"></div>
</section>
</section>
<section class="ltx_appendix" id="A2">
<h2 class="ltx_title ltx_title_appendix">
<span class="ltx_tag ltx_tag_appendix">Appendix B </span>Detailed experimental settings</h2>
<div class="ltx_para ltx_noindent" id="A2.p1">
<p class="ltx_p" id="A2.p1.5"><span class="ltx_text ltx_font_bold" id="A2.p1.5.1">Comparison with decoding schemes (<a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.SS1" title="4.1 Comparison with decoding schemes ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Section</span> <span class="ltx_text ltx_ref_tag">4.1</span></a>).</span> In this experiment, we first consider random sampling with different temperatures <math alttext="\{0.7,1.0,1.5\}" class="ltx_Math" display="inline" id="A2.p1.1.m1.3"><semantics id="A2.p1.1.m1.3a"><mrow id="A2.p1.1.m1.3.4.2" xref="A2.p1.1.m1.3.4.1.cmml"><mo id="A2.p1.1.m1.3.4.2.1" stretchy="false" xref="A2.p1.1.m1.3.4.1.cmml">{</mo><mn id="A2.p1.1.m1.1.1" xref="A2.p1.1.m1.1.1.cmml">0.7</mn><mo id="A2.p1.1.m1.3.4.2.2" xref="A2.p1.1.m1.3.4.1.cmml">,</mo><mn id="A2.p1.1.m1.2.2" xref="A2.p1.1.m1.2.2.cmml">1.0</mn><mo id="A2.p1.1.m1.3.4.2.3" xref="A2.p1.1.m1.3.4.1.cmml">,</mo><mn id="A2.p1.1.m1.3.3" xref="A2.p1.1.m1.3.3.cmml">1.5</mn><mo id="A2.p1.1.m1.3.4.2.4" stretchy="false" xref="A2.p1.1.m1.3.4.1.cmml">}</mo></mrow><annotation-xml encoding="MathML-Content" id="A2.p1.1.m1.3b"><set id="A2.p1.1.m1.3.4.1.cmml" xref="A2.p1.1.m1.3.4.2"><cn id="A2.p1.1.m1.1.1.cmml" type="float" xref="A2.p1.1.m1.1.1">0.7</cn><cn id="A2.p1.1.m1.2.2.cmml" type="float" xref="A2.p1.1.m1.2.2">1.0</cn><cn id="A2.p1.1.m1.3.3.cmml" type="float" xref="A2.p1.1.m1.3.3">1.5</cn></set></annotation-xml><annotation encoding="application/x-tex" id="A2.p1.1.m1.3c">\{0.7,1.0,1.5\}</annotation><annotation encoding="application/x-llamapun" id="A2.p1.1.m1.3d">{ 0.7 , 1.0 , 1.5 }</annotation></semantics></math>. For the other decoding schemes, we consider conventional configurations: nucleus sampling with top-p <math alttext="0.8" class="ltx_Math" display="inline" id="A2.p1.2.m2.1"><semantics id="A2.p1.2.m2.1a"><mn id="A2.p1.2.m2.1.1" xref="A2.p1.2.m2.1.1.cmml">0.8</mn><annotation-xml encoding="MathML-Content" id="A2.p1.2.m2.1b"><cn id="A2.p1.2.m2.1.1.cmml" type="float" xref="A2.p1.2.m2.1.1">0.8</cn></annotation-xml><annotation encoding="application/x-tex" id="A2.p1.2.m2.1c">0.8</annotation><annotation encoding="application/x-llamapun" id="A2.p1.2.m2.1d">0.8</annotation></semantics></math>, beam search, diverse beam search with a diversity penalty of <math alttext="0.5" class="ltx_Math" display="inline" id="A2.p1.3.m3.1"><semantics id="A2.p1.3.m3.1a"><mn id="A2.p1.3.m3.1.1" xref="A2.p1.3.m3.1.1.cmml">0.5</mn><annotation-xml encoding="MathML-Content" id="A2.p1.3.m3.1b"><cn id="A2.p1.3.m3.1.1.cmml" type="float" xref="A2.p1.3.m3.1.1">0.5</cn></annotation-xml><annotation encoding="application/x-tex" id="A2.p1.3.m3.1c">0.5</annotation><annotation encoding="application/x-llamapun" id="A2.p1.3.m3.1d">0.5</annotation></semantics></math>, and contrastive beam search with a penalty alpha of <math alttext="0.5" class="ltx_Math" display="inline" id="A2.p1.4.m4.1"><semantics id="A2.p1.4.m4.1a"><mn id="A2.p1.4.m4.1.1" xref="A2.p1.4.m4.1.1.cmml">0.5</mn><annotation-xml encoding="MathML-Content" id="A2.p1.4.m4.1b"><cn id="A2.p1.4.m4.1.1.cmml" type="float" xref="A2.p1.4.m4.1.1">0.5</cn></annotation-xml><annotation encoding="application/x-tex" id="A2.p1.4.m4.1c">0.5</annotation><annotation encoding="application/x-llamapun" id="A2.p1.4.m4.1d">0.5</annotation></semantics></math>. We apply greedy decoding for our approach. The prompts for BioT5<sup class="ltx_sup" id="A2.p1.5.2"><span class="ltx_text ltx_font_italic" id="A2.p1.5.2.1">+</span></sup> are described in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#A2.T5" title="In Appendix B Detailed experimental settings ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Table</span> <span class="ltx_text ltx_ref_tag">5</span></a>. The prompts for MolT5 are considered as a molecular description without any additional comments.</p>
</div>
<figure class="ltx_table" id="A2.T5">
<figcaption class="ltx_caption"><span class="ltx_tag ltx_tag_table">Table 5: </span><span class="ltx_text ltx_font_bold" id="A2.T5.2.1">Prompts for BioT5<sup class="ltx_sup" id="A2.T5.2.1.1"><span class="ltx_text ltx_font_medium ltx_font_italic" id="A2.T5.2.1.1.1">+</span></sup> <cite class="ltx_cite ltx_citemacro_citep">(Pei et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib26" title="">2024</a>)</cite>.</span></figcaption>
<table class="ltx_tabular ltx_centering ltx_align_middle" id="A2.T5.4">
<tr class="ltx_tr" id="A2.T5.4.3">
<td class="ltx_td ltx_align_left ltx_border_tt" id="A2.T5.4.3.1">Prompt</td>
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_tt" id="A2.T5.4.3.2" style="width:278.2pt;">
<span class="ltx_inline-block ltx_align_top" id="A2.T5.4.3.2.1">
<span class="ltx_p" id="A2.T5.4.3.2.1.1">Contents</span>
</span>
</td>
</tr>
<tr class="ltx_tr" id="A2.T5.3.1">
<td class="ltx_td ltx_align_left ltx_border_t" id="A2.T5.3.1.1"><math alttext="p_{\text{desc}}" class="ltx_Math" display="inline" id="A2.T5.3.1.1.m1.1"><semantics id="A2.T5.3.1.1.m1.1a"><msub id="A2.T5.3.1.1.m1.1.1" xref="A2.T5.3.1.1.m1.1.1.cmml"><mi id="A2.T5.3.1.1.m1.1.1.2" xref="A2.T5.3.1.1.m1.1.1.2.cmml">p</mi><mtext id="A2.T5.3.1.1.m1.1.1.3" xref="A2.T5.3.1.1.m1.1.1.3a.cmml">desc</mtext></msub><annotation-xml encoding="MathML-Content" id="A2.T5.3.1.1.m1.1b"><apply id="A2.T5.3.1.1.m1.1.1.cmml" xref="A2.T5.3.1.1.m1.1.1"><csymbol cd="ambiguous" id="A2.T5.3.1.1.m1.1.1.1.cmml" xref="A2.T5.3.1.1.m1.1.1">subscript</csymbol><ci id="A2.T5.3.1.1.m1.1.1.2.cmml" xref="A2.T5.3.1.1.m1.1.1.2">𝑝</ci><ci id="A2.T5.3.1.1.m1.1.1.3a.cmml" xref="A2.T5.3.1.1.m1.1.1.3"><mtext id="A2.T5.3.1.1.m1.1.1.3.cmml" mathsize="70%" xref="A2.T5.3.1.1.m1.1.1.3">desc</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="A2.T5.3.1.1.m1.1c">p_{\text{desc}}</annotation><annotation encoding="application/x-llamapun" id="A2.T5.3.1.1.m1.1d">italic_p start_POSTSUBSCRIPT desc end_POSTSUBSCRIPT</annotation></semantics></math></td>
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_t" id="A2.T5.3.1.2" style="width:278.2pt;">
<span class="ltx_inline-block ltx_align_top" id="A2.T5.3.1.2.1">
<span class="ltx_p" id="A2.T5.3.1.2.1.1">“Definition: You are given a molecule description in English. Your job is to generate the molecule SELFIES that fits the description.</span>
<span class="ltx_p" id="A2.T5.3.1.2.1.2">Now complete the following example -</span>
<span class="ltx_p" id="A2.T5.3.1.2.1.3">Input: &lt;molecular description&gt;</span>
<span class="ltx_p" id="A2.T5.3.1.2.1.4">Output: ”</span>
</span>
</td>
</tr>
<tr class="ltx_tr" id="A2.T5.4.2">
<td class="ltx_td ltx_align_left ltx_border_bb ltx_border_t" id="A2.T5.4.2.1">
<math alttext="p_{\text{desc+div}}" class="ltx_Math" display="inline" id="A2.T5.4.2.1.m1.1"><semantics id="A2.T5.4.2.1.m1.1a"><msub id="A2.T5.4.2.1.m1.1.1" xref="A2.T5.4.2.1.m1.1.1.cmml"><mi id="A2.T5.4.2.1.m1.1.1.2" xref="A2.T5.4.2.1.m1.1.1.2.cmml">p</mi><mtext id="A2.T5.4.2.1.m1.1.1.3" xref="A2.T5.4.2.1.m1.1.1.3a.cmml">desc+div</mtext></msub><annotation-xml encoding="MathML-Content" id="A2.T5.4.2.1.m1.1b"><apply id="A2.T5.4.2.1.m1.1.1.cmml" xref="A2.T5.4.2.1.m1.1.1"><csymbol cd="ambiguous" id="A2.T5.4.2.1.m1.1.1.1.cmml" xref="A2.T5.4.2.1.m1.1.1">subscript</csymbol><ci id="A2.T5.4.2.1.m1.1.1.2.cmml" xref="A2.T5.4.2.1.m1.1.1.2">𝑝</ci><ci id="A2.T5.4.2.1.m1.1.1.3a.cmml" xref="A2.T5.4.2.1.m1.1.1.3"><mtext id="A2.T5.4.2.1.m1.1.1.3.cmml" mathsize="70%" xref="A2.T5.4.2.1.m1.1.1.3">desc+div</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="A2.T5.4.2.1.m1.1c">p_{\text{desc+div}}</annotation><annotation encoding="application/x-llamapun" id="A2.T5.4.2.1.m1.1d">italic_p start_POSTSUBSCRIPT desc+div end_POSTSUBSCRIPT</annotation></semantics></math> (fine-tuning)</td>
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_bb ltx_border_t" id="A2.T5.4.2.2" style="width:278.2pt;">
<span class="ltx_inline-block ltx_align_top" id="A2.T5.4.2.2.1">
<span class="ltx_p" id="A2.T5.4.2.2.1.1">“Definition: You are given a molecule description in English. Your job is to generate the molecule SELFIES that fits the description.</span>
<span class="ltx_p" id="A2.T5.4.2.2.1.2">Now provide a set of molecules -</span>
<span class="ltx_p" id="A2.T5.4.2.2.1.3">Input: &lt;molecular description&gt;</span>
<span class="ltx_p" id="A2.T5.4.2.2.1.4">Output: ”</span>
</span>
</td>
</tr>
</table>
</figure>
<div class="ltx_para ltx_noindent" id="A2.p2">
<p class="ltx_p" id="A2.p2.8"><span class="ltx_text ltx_font_bold" id="A2.p2.8.1">Comparison with LLMs (<a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S4.SS2" title="4.2 Comparison with existing LLMs ‣ 4 Experiment ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Section</span> <span class="ltx_text ltx_ref_tag">4.2</span></a>).</span> In this experiment, we consider generalist LLMs. For Mol-instructions, and LlasMol, we apply random sampling, beam search, diverse beam search with a diversity penalty of <math alttext="0.5" class="ltx_Math" display="inline" id="A2.p2.1.m1.1"><semantics id="A2.p2.1.m1.1a"><mn id="A2.p2.1.m1.1.1" xref="A2.p2.1.m1.1.1.cmml">0.5</mn><annotation-xml encoding="MathML-Content" id="A2.p2.1.m1.1b"><cn id="A2.p2.1.m1.1.1.cmml" type="float" xref="A2.p2.1.m1.1.1">0.5</cn></annotation-xml><annotation encoding="application/x-tex" id="A2.p2.1.m1.1c">0.5</annotation><annotation encoding="application/x-llamapun" id="A2.p2.1.m1.1d">0.5</annotation></semantics></math>, and contrastive beam search with a penalty alpha of <math alttext="0.5" class="ltx_Math" display="inline" id="A2.p2.2.m2.1"><semantics id="A2.p2.2.m2.1a"><mn id="A2.p2.2.m2.1.1" xref="A2.p2.2.m2.1.1.cmml">0.5</mn><annotation-xml encoding="MathML-Content" id="A2.p2.2.m2.1b"><cn id="A2.p2.2.m2.1.1.cmml" type="float" xref="A2.p2.2.m2.1.1">0.5</cn></annotation-xml><annotation encoding="application/x-tex" id="A2.p2.2.m2.1c">0.5</annotation><annotation encoding="application/x-llamapun" id="A2.p2.2.m2.1d">0.5</annotation></semantics></math>. The prompts are described in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#A2.T6" title="In Appendix B Detailed experimental settings ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Table</span> <span class="ltx_text ltx_ref_tag">6</span></a>. For ChatGPT, we apply either <math alttext="50" class="ltx_Math" display="inline" id="A2.p2.3.m3.1"><semantics id="A2.p2.3.m3.1a"><mn id="A2.p2.3.m3.1.1" xref="A2.p2.3.m3.1.1.cmml">50</mn><annotation-xml encoding="MathML-Content" id="A2.p2.3.m3.1b"><cn id="A2.p2.3.m3.1.1.cmml" type="integer" xref="A2.p2.3.m3.1.1">50</cn></annotation-xml><annotation encoding="application/x-tex" id="A2.p2.3.m3.1c">50</annotation><annotation encoding="application/x-llamapun" id="A2.p2.3.m3.1d">50</annotation></semantics></math> random sampling with <math alttext="p_{\text{desc}}" class="ltx_Math" display="inline" id="A2.p2.4.m4.1"><semantics id="A2.p2.4.m4.1a"><msub id="A2.p2.4.m4.1.1" xref="A2.p2.4.m4.1.1.cmml"><mi id="A2.p2.4.m4.1.1.2" xref="A2.p2.4.m4.1.1.2.cmml">p</mi><mtext id="A2.p2.4.m4.1.1.3" xref="A2.p2.4.m4.1.1.3a.cmml">desc</mtext></msub><annotation-xml encoding="MathML-Content" id="A2.p2.4.m4.1b"><apply id="A2.p2.4.m4.1.1.cmml" xref="A2.p2.4.m4.1.1"><csymbol cd="ambiguous" id="A2.p2.4.m4.1.1.1.cmml" xref="A2.p2.4.m4.1.1">subscript</csymbol><ci id="A2.p2.4.m4.1.1.2.cmml" xref="A2.p2.4.m4.1.1.2">𝑝</ci><ci id="A2.p2.4.m4.1.1.3a.cmml" xref="A2.p2.4.m4.1.1.3"><mtext id="A2.p2.4.m4.1.1.3.cmml" mathsize="70%" xref="A2.p2.4.m4.1.1.3">desc</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="A2.p2.4.m4.1c">p_{\text{desc}}</annotation><annotation encoding="application/x-llamapun" id="A2.p2.4.m4.1d">italic_p start_POSTSUBSCRIPT desc end_POSTSUBSCRIPT</annotation></semantics></math> or greedy decoding with <math alttext="p_{\text{desc+div}}" class="ltx_Math" display="inline" id="A2.p2.5.m5.1"><semantics id="A2.p2.5.m5.1a"><msub id="A2.p2.5.m5.1.1" xref="A2.p2.5.m5.1.1.cmml"><mi id="A2.p2.5.m5.1.1.2" xref="A2.p2.5.m5.1.1.2.cmml">p</mi><mtext id="A2.p2.5.m5.1.1.3" xref="A2.p2.5.m5.1.1.3a.cmml">desc+div</mtext></msub><annotation-xml encoding="MathML-Content" id="A2.p2.5.m5.1b"><apply id="A2.p2.5.m5.1.1.cmml" xref="A2.p2.5.m5.1.1"><csymbol cd="ambiguous" id="A2.p2.5.m5.1.1.1.cmml" xref="A2.p2.5.m5.1.1">subscript</csymbol><ci id="A2.p2.5.m5.1.1.2.cmml" xref="A2.p2.5.m5.1.1.2">𝑝</ci><ci id="A2.p2.5.m5.1.1.3a.cmml" xref="A2.p2.5.m5.1.1.3"><mtext id="A2.p2.5.m5.1.1.3.cmml" mathsize="70%" xref="A2.p2.5.m5.1.1.3">desc+div</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="A2.p2.5.m5.1c">p_{\text{desc+div}}</annotation><annotation encoding="application/x-llamapun" id="A2.p2.5.m5.1d">italic_p start_POSTSUBSCRIPT desc+div end_POSTSUBSCRIPT</annotation></semantics></math>, where both prompts are described in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#A2.T7" title="In Appendix B Detailed experimental settings ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Table</span> <span class="ltx_text ltx_ref_tag">7</span></a>. The prompt design is based on <cite class="ltx_cite ltx_citemacro_citep">(Li et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib19" title="">2024</a>)</cite>. For o1, we use the prompt <math alttext="p_{\text{desc+div}}" class="ltx_Math" display="inline" id="A2.p2.6.m6.1"><semantics id="A2.p2.6.m6.1a"><msub id="A2.p2.6.m6.1.1" xref="A2.p2.6.m6.1.1.cmml"><mi id="A2.p2.6.m6.1.1.2" xref="A2.p2.6.m6.1.1.2.cmml">p</mi><mtext id="A2.p2.6.m6.1.1.3" xref="A2.p2.6.m6.1.1.3a.cmml">desc+div</mtext></msub><annotation-xml encoding="MathML-Content" id="A2.p2.6.m6.1b"><apply id="A2.p2.6.m6.1.1.cmml" xref="A2.p2.6.m6.1.1"><csymbol cd="ambiguous" id="A2.p2.6.m6.1.1.1.cmml" xref="A2.p2.6.m6.1.1">subscript</csymbol><ci id="A2.p2.6.m6.1.1.2.cmml" xref="A2.p2.6.m6.1.1.2">𝑝</ci><ci id="A2.p2.6.m6.1.1.3a.cmml" xref="A2.p2.6.m6.1.1.3"><mtext id="A2.p2.6.m6.1.1.3.cmml" mathsize="70%" xref="A2.p2.6.m6.1.1.3">desc+div</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="A2.p2.6.m6.1c">p_{\text{desc+div}}</annotation><annotation encoding="application/x-llamapun" id="A2.p2.6.m6.1d">italic_p start_POSTSUBSCRIPT desc+div end_POSTSUBSCRIPT</annotation></semantics></math> described in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#A2.T7" title="In Appendix B Detailed experimental settings ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Table</span> <span class="ltx_text ltx_ref_tag">7</span></a>. We report the test score using the decoding scheme or the prompt that yields the highest NCircles<sub class="ltx_sub" id="A2.p2.8.2"><span class="ltx_text ltx_font_italic" id="A2.p2.8.2.1">h=0.65</span></sub> for <math alttext="500" class="ltx_Math" display="inline" id="A2.p2.8.m8.1"><semantics id="A2.p2.8.m8.1a"><mn id="A2.p2.8.m8.1.1" xref="A2.p2.8.m8.1.1.cmml">500</mn><annotation-xml encoding="MathML-Content" id="A2.p2.8.m8.1b"><cn id="A2.p2.8.m8.1.1.cmml" type="integer" xref="A2.p2.8.m8.1.1">500</cn></annotation-xml><annotation encoding="application/x-tex" id="A2.p2.8.m8.1c">500</annotation><annotation encoding="application/x-llamapun" id="A2.p2.8.m8.1d">500</annotation></semantics></math> evaluation molecular description.</p>
</div>
<figure class="ltx_table" id="A2.T6">
<figcaption class="ltx_caption"><span class="ltx_tag ltx_tag_table">Table 6: </span><span class="ltx_text ltx_font_bold" id="A2.T6.3.1">Prompts for generalist LLMs <cite class="ltx_cite ltx_citemacro_citep">(Fang et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib8" title="">2024</a>; Taylor et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib32" title="">2022</a>)</cite>.</span></figcaption>
<table class="ltx_tabular ltx_centering ltx_align_middle" id="A2.T6.1">
<tr class="ltx_tr" id="A2.T6.1.1">
<td class="ltx_td ltx_align_left ltx_border_tt" id="A2.T6.1.1.2">Method</td>
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_tt" id="A2.T6.1.1.1" style="width:306.1pt;">
<span class="ltx_inline-block ltx_align_top" id="A2.T6.1.1.1.1">
<span class="ltx_p" id="A2.T6.1.1.1.1.1"><math alttext="p_{\text{desc}}" class="ltx_Math" display="inline" id="A2.T6.1.1.1.1.1.m1.1"><semantics id="A2.T6.1.1.1.1.1.m1.1a"><msub id="A2.T6.1.1.1.1.1.m1.1.1" xref="A2.T6.1.1.1.1.1.m1.1.1.cmml"><mi id="A2.T6.1.1.1.1.1.m1.1.1.2" xref="A2.T6.1.1.1.1.1.m1.1.1.2.cmml">p</mi><mtext id="A2.T6.1.1.1.1.1.m1.1.1.3" xref="A2.T6.1.1.1.1.1.m1.1.1.3a.cmml">desc</mtext></msub><annotation-xml encoding="MathML-Content" id="A2.T6.1.1.1.1.1.m1.1b"><apply id="A2.T6.1.1.1.1.1.m1.1.1.cmml" xref="A2.T6.1.1.1.1.1.m1.1.1"><csymbol cd="ambiguous" id="A2.T6.1.1.1.1.1.m1.1.1.1.cmml" xref="A2.T6.1.1.1.1.1.m1.1.1">subscript</csymbol><ci id="A2.T6.1.1.1.1.1.m1.1.1.2.cmml" xref="A2.T6.1.1.1.1.1.m1.1.1.2">𝑝</ci><ci id="A2.T6.1.1.1.1.1.m1.1.1.3a.cmml" xref="A2.T6.1.1.1.1.1.m1.1.1.3"><mtext id="A2.T6.1.1.1.1.1.m1.1.1.3.cmml" mathsize="70%" xref="A2.T6.1.1.1.1.1.m1.1.1.3">desc</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="A2.T6.1.1.1.1.1.m1.1c">p_{\text{desc}}</annotation><annotation encoding="application/x-llamapun" id="A2.T6.1.1.1.1.1.m1.1d">italic_p start_POSTSUBSCRIPT desc end_POSTSUBSCRIPT</annotation></semantics></math></span>
</span>
</td>
</tr>
<tr class="ltx_tr" id="A2.T6.1.2">
<td class="ltx_td ltx_align_left ltx_border_t" id="A2.T6.1.2.1">Mol-instructions</td>
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_t" id="A2.T6.1.2.2" style="width:306.1pt;">
<span class="ltx_inline-block ltx_align_top" id="A2.T6.1.2.2.1">
<span class="ltx_p" id="A2.T6.1.2.2.1.1">“Below is an instruction that describes a task, paired with an input that provides further context. Write a response that appropriately completes the request.</span>
<span class="ltx_p" id="A2.T6.1.2.2.1.2">### Instruction:</span>
<span class="ltx_p" id="A2.T6.1.2.2.1.3">Create a molecule with the structure as the one describe.</span>
<span class="ltx_p" id="A2.T6.1.2.2.1.4">### Input:</span>
<span class="ltx_p" id="A2.T6.1.2.2.1.5">&lt;molecular description&gt;</span>
<span class="ltx_p" id="A2.T6.1.2.2.1.6">### Response: ”</span>
</span>
</td>
</tr>
<tr class="ltx_tr" id="A2.T6.1.3">
<td class="ltx_td ltx_align_left ltx_border_bb ltx_border_t" id="A2.T6.1.3.1">LlasMol</td>
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_bb ltx_border_t" id="A2.T6.1.3.2" style="width:306.1pt;">
<span class="ltx_inline-block ltx_align_top" id="A2.T6.1.3.2.1">
<span class="ltx_p" id="A2.T6.1.3.2.1.1">“Give me a molecule that satisfies the conditions outlined in the description: &lt;molecular description&gt;”</span>
</span>
</td>
</tr>
</table>
</figure>
<div class="ltx_pagination ltx_role_newpage"></div>
<figure class="ltx_table" id="A2.T7">
<figcaption class="ltx_caption"><span class="ltx_tag ltx_tag_table">Table 7: </span><span class="ltx_text ltx_font_bold" id="A2.T7.6.1">Prompts for ChatGPT and o1 <cite class="ltx_cite ltx_citemacro_citep">(Li et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib19" title="">2024</a>)</cite>.</span></figcaption>
<table class="ltx_tabular ltx_centering ltx_align_middle" id="A2.T7.4">
<tr class="ltx_tr" id="A2.T7.4.5">
<td class="ltx_td ltx_align_left ltx_border_tt" id="A2.T7.4.5.1">Input</td>
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_tt" id="A2.T7.4.5.2" style="width:278.2pt;">
<span class="ltx_inline-block ltx_align_top" id="A2.T7.4.5.2.1">
<span class="ltx_p" id="A2.T7.4.5.2.1.1">Contents</span>
</span>
</td>
</tr>
<tr class="ltx_tr" id="A2.T7.1.1">
<td class="ltx_td ltx_align_left ltx_border_t" id="A2.T7.1.1.1">System <math alttext="p_{\text{desc}}" class="ltx_Math" display="inline" id="A2.T7.1.1.1.m1.1"><semantics id="A2.T7.1.1.1.m1.1a"><msub id="A2.T7.1.1.1.m1.1.1" xref="A2.T7.1.1.1.m1.1.1.cmml"><mi id="A2.T7.1.1.1.m1.1.1.2" xref="A2.T7.1.1.1.m1.1.1.2.cmml">p</mi><mtext id="A2.T7.1.1.1.m1.1.1.3" xref="A2.T7.1.1.1.m1.1.1.3a.cmml">desc</mtext></msub><annotation-xml encoding="MathML-Content" id="A2.T7.1.1.1.m1.1b"><apply id="A2.T7.1.1.1.m1.1.1.cmml" xref="A2.T7.1.1.1.m1.1.1"><csymbol cd="ambiguous" id="A2.T7.1.1.1.m1.1.1.1.cmml" xref="A2.T7.1.1.1.m1.1.1">subscript</csymbol><ci id="A2.T7.1.1.1.m1.1.1.2.cmml" xref="A2.T7.1.1.1.m1.1.1.2">𝑝</ci><ci id="A2.T7.1.1.1.m1.1.1.3a.cmml" xref="A2.T7.1.1.1.m1.1.1.3"><mtext id="A2.T7.1.1.1.m1.1.1.3.cmml" mathsize="70%" xref="A2.T7.1.1.1.m1.1.1.3">desc</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="A2.T7.1.1.1.m1.1c">p_{\text{desc}}</annotation><annotation encoding="application/x-llamapun" id="A2.T7.1.1.1.m1.1d">italic_p start_POSTSUBSCRIPT desc end_POSTSUBSCRIPT</annotation></semantics></math>
</td>
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_t" id="A2.T7.1.1.2" style="width:278.2pt;">
<span class="ltx_inline-block ltx_align_top" id="A2.T7.1.1.2.1">
<span class="ltx_p" id="A2.T7.1.1.2.1.1">“You are now working as an excellent expert in chemistry and drug discovery. Given the caption of a molecule, your job is to predict the SMILES representation of the molecule. The molecule caption is a sentence that describes the molecule, which mainly describes the molecule’s structures, properties, and production. You can generate the molecule SMILES representation from the caption.</span>
<span class="ltx_p" id="A2.T7.1.1.2.1.2">Example:</span>
<span class="ltx_p" id="A2.T7.1.1.2.1.3">Instruction: Given the caption of a molecule, predict the SMILES representation of the molecule.</span>
<span class="ltx_p" id="A2.T7.1.1.2.1.4">Input: &lt;molecular description&gt;.</span>
<span class="ltx_p" id="A2.T7.1.1.2.1.5">Your output should be:</span>
<span class="ltx_p" id="A2.T7.1.1.2.1.6">{"molecule": "&lt;SMILES&gt;"}</span>
<span class="ltx_p" id="A2.T7.1.1.2.1.7">Your response should only be in the JSON format above; THERE SHOULD BE NO OTHER CONTENT INCLUDED IN YOUR RESPONSE.”</span>
</span>
</td>
</tr>
<tr class="ltx_tr" id="A2.T7.2.2">
<td class="ltx_td ltx_align_left ltx_border_t" id="A2.T7.2.2.1">User <math alttext="p_{\text{desc}}" class="ltx_Math" display="inline" id="A2.T7.2.2.1.m1.1"><semantics id="A2.T7.2.2.1.m1.1a"><msub id="A2.T7.2.2.1.m1.1.1" xref="A2.T7.2.2.1.m1.1.1.cmml"><mi id="A2.T7.2.2.1.m1.1.1.2" xref="A2.T7.2.2.1.m1.1.1.2.cmml">p</mi><mtext id="A2.T7.2.2.1.m1.1.1.3" xref="A2.T7.2.2.1.m1.1.1.3a.cmml">desc</mtext></msub><annotation-xml encoding="MathML-Content" id="A2.T7.2.2.1.m1.1b"><apply id="A2.T7.2.2.1.m1.1.1.cmml" xref="A2.T7.2.2.1.m1.1.1"><csymbol cd="ambiguous" id="A2.T7.2.2.1.m1.1.1.1.cmml" xref="A2.T7.2.2.1.m1.1.1">subscript</csymbol><ci id="A2.T7.2.2.1.m1.1.1.2.cmml" xref="A2.T7.2.2.1.m1.1.1.2">𝑝</ci><ci id="A2.T7.2.2.1.m1.1.1.3a.cmml" xref="A2.T7.2.2.1.m1.1.1.3"><mtext id="A2.T7.2.2.1.m1.1.1.3.cmml" mathsize="70%" xref="A2.T7.2.2.1.m1.1.1.3">desc</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="A2.T7.2.2.1.m1.1c">p_{\text{desc}}</annotation><annotation encoding="application/x-llamapun" id="A2.T7.2.2.1.m1.1d">italic_p start_POSTSUBSCRIPT desc end_POSTSUBSCRIPT</annotation></semantics></math>
</td>
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_t" id="A2.T7.2.2.2" style="width:278.2pt;">
<span class="ltx_inline-block ltx_align_top" id="A2.T7.2.2.2.1">
<span class="ltx_p" id="A2.T7.2.2.2.1.1">“Input: &lt;molecular description&gt;”</span>
</span>
</td>
</tr>
<tr class="ltx_tr" id="A2.T7.3.3">
<td class="ltx_td ltx_align_left ltx_border_t" id="A2.T7.3.3.1">System <math alttext="p_{\text{desc+div}}" class="ltx_Math" display="inline" id="A2.T7.3.3.1.m1.1"><semantics id="A2.T7.3.3.1.m1.1a"><msub id="A2.T7.3.3.1.m1.1.1" xref="A2.T7.3.3.1.m1.1.1.cmml"><mi id="A2.T7.3.3.1.m1.1.1.2" xref="A2.T7.3.3.1.m1.1.1.2.cmml">p</mi><mtext id="A2.T7.3.3.1.m1.1.1.3" xref="A2.T7.3.3.1.m1.1.1.3a.cmml">desc+div</mtext></msub><annotation-xml encoding="MathML-Content" id="A2.T7.3.3.1.m1.1b"><apply id="A2.T7.3.3.1.m1.1.1.cmml" xref="A2.T7.3.3.1.m1.1.1"><csymbol cd="ambiguous" id="A2.T7.3.3.1.m1.1.1.1.cmml" xref="A2.T7.3.3.1.m1.1.1">subscript</csymbol><ci id="A2.T7.3.3.1.m1.1.1.2.cmml" xref="A2.T7.3.3.1.m1.1.1.2">𝑝</ci><ci id="A2.T7.3.3.1.m1.1.1.3a.cmml" xref="A2.T7.3.3.1.m1.1.1.3"><mtext id="A2.T7.3.3.1.m1.1.1.3.cmml" mathsize="70%" xref="A2.T7.3.3.1.m1.1.1.3">desc+div</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="A2.T7.3.3.1.m1.1c">p_{\text{desc+div}}</annotation><annotation encoding="application/x-llamapun" id="A2.T7.3.3.1.m1.1d">italic_p start_POSTSUBSCRIPT desc+div end_POSTSUBSCRIPT</annotation></semantics></math>
</td>
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_t" id="A2.T7.3.3.2" style="width:278.2pt;">
<span class="ltx_inline-block ltx_align_top" id="A2.T7.3.3.2.1">
<span class="ltx_p" id="A2.T7.3.3.2.1.1">“You are now working as an excellent expert in chemistry and drug discovery. Given the caption of a molecule, your job is to generate diverse molecules. The molecule caption is a sentence that describes the molecule, which mainly describes the molecule’s structures, properties, and production. You can generate the molecule SMILES representation from the caption.</span>
<span class="ltx_p" id="A2.T7.3.3.2.1.2">Example:</span>
<span class="ltx_p" id="A2.T7.3.3.2.1.3">Instruction: Given the caption of a molecule, generate fifty structurally diverse molecules with SMILES representations.</span>
<span class="ltx_p" id="A2.T7.3.3.2.1.4">Input: &lt;molecular description&gt;.</span>
<span class="ltx_p" id="A2.T7.3.3.2.1.5">Your output should be:</span>
<span class="ltx_p" id="A2.T7.3.3.2.1.6">{"molecule1": "&lt;SMILES&gt;"}</span>
<span class="ltx_p" id="A2.T7.3.3.2.1.7">{"molecule2": "&lt;SMILES&gt;"}</span>
<span class="ltx_p" id="A2.T7.3.3.2.1.8">{"molecule3": "&lt;SMILES&gt;"}</span>
<span class="ltx_p" id="A2.T7.3.3.2.1.9">…</span>
<span class="ltx_p" id="A2.T7.3.3.2.1.10">{"molecule50": "&lt;SMILES&gt;"}</span>
<span class="ltx_p" id="A2.T7.3.3.2.1.11">Your response should only be in the JSON format above; THERE SHOULD BE NO OTHER CONTENT INCLUDED IN YOUR RESPONSE.”</span>
</span>
</td>
</tr>
<tr class="ltx_tr" id="A2.T7.4.4">
<td class="ltx_td ltx_align_left ltx_border_bb ltx_border_t" id="A2.T7.4.4.1">User <math alttext="p_{\text{desc+div}}" class="ltx_Math" display="inline" id="A2.T7.4.4.1.m1.1"><semantics id="A2.T7.4.4.1.m1.1a"><msub id="A2.T7.4.4.1.m1.1.1" xref="A2.T7.4.4.1.m1.1.1.cmml"><mi id="A2.T7.4.4.1.m1.1.1.2" xref="A2.T7.4.4.1.m1.1.1.2.cmml">p</mi><mtext id="A2.T7.4.4.1.m1.1.1.3" xref="A2.T7.4.4.1.m1.1.1.3a.cmml">desc+div</mtext></msub><annotation-xml encoding="MathML-Content" id="A2.T7.4.4.1.m1.1b"><apply id="A2.T7.4.4.1.m1.1.1.cmml" xref="A2.T7.4.4.1.m1.1.1"><csymbol cd="ambiguous" id="A2.T7.4.4.1.m1.1.1.1.cmml" xref="A2.T7.4.4.1.m1.1.1">subscript</csymbol><ci id="A2.T7.4.4.1.m1.1.1.2.cmml" xref="A2.T7.4.4.1.m1.1.1.2">𝑝</ci><ci id="A2.T7.4.4.1.m1.1.1.3a.cmml" xref="A2.T7.4.4.1.m1.1.1.3"><mtext id="A2.T7.4.4.1.m1.1.1.3.cmml" mathsize="70%" xref="A2.T7.4.4.1.m1.1.1.3">desc+div</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="A2.T7.4.4.1.m1.1c">p_{\text{desc+div}}</annotation><annotation encoding="application/x-llamapun" id="A2.T7.4.4.1.m1.1d">italic_p start_POSTSUBSCRIPT desc+div end_POSTSUBSCRIPT</annotation></semantics></math>
</td>
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_bb ltx_border_t" id="A2.T7.4.4.2" style="width:278.2pt;">
<span class="ltx_inline-block ltx_align_top" id="A2.T7.4.4.2.1">
<span class="ltx_p" id="A2.T7.4.4.2.1.1">“Input: &lt;molecular description&gt;”</span>
</span>
</td>
</tr>
</table>
</figure>
<div class="ltx_pagination ltx_role_newpage"></div>
<div class="ltx_para ltx_noindent" id="A2.p3">
<p class="ltx_p" id="A2.p3.7"><span class="ltx_text ltx_font_bold" id="A2.p3.7.1">Fine-tuning generalist LLMs.</span> We synthesize <math alttext="600" class="ltx_Math" display="inline" id="A2.p3.1.m1.1"><semantics id="A2.p3.1.m1.1a"><mn id="A2.p3.1.m1.1.1" xref="A2.p3.1.m1.1.1.cmml">600</mn><annotation-xml encoding="MathML-Content" id="A2.p3.1.m1.1b"><cn id="A2.p3.1.m1.1.1.cmml" type="integer" xref="A2.p3.1.m1.1.1">600</cn></annotation-xml><annotation encoding="application/x-tex" id="A2.p3.1.m1.1c">600</annotation><annotation encoding="application/x-llamapun" id="A2.p3.1.m1.1d">600</annotation></semantics></math> pairs of training prompts and corresponding sets of molecules. The prompts specify hydrogen bond donors and acceptors ranging from one to four, and a Bertz complexity ranging from <math alttext="0" class="ltx_Math" display="inline" id="A2.p3.2.m2.1"><semantics id="A2.p3.2.m2.1a"><mn id="A2.p3.2.m2.1.1" xref="A2.p3.2.m2.1.1.cmml">0</mn><annotation-xml encoding="MathML-Content" id="A2.p3.2.m2.1b"><cn id="A2.p3.2.m2.1.1.cmml" type="integer" xref="A2.p3.2.m2.1.1">0</cn></annotation-xml></semantics></math> to <math alttext="300" class="ltx_Math" display="inline" id="A2.p3.3.m3.1"><semantics id="A2.p3.3.m3.1a"><mn id="A2.p3.3.m3.1.1" xref="A2.p3.3.m3.1.1.cmml">300</mn><annotation-xml encoding="MathML-Content" id="A2.p3.3.m3.1b"><cn id="A2.p3.3.m3.1.1.cmml" type="integer" xref="A2.p3.3.m3.1.1">300</cn></annotation-xml><annotation encoding="application/x-tex" id="A2.p3.3.m3.1c">300</annotation><annotation encoding="application/x-llamapun" id="A2.p3.3.m3.1d">300</annotation></semantics></math>. The sets of molecules are collected by applying beam search on DrugAssist and then perturbed by shuffling their order. We use the prompts described in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#A2.T8" title="In Appendix B Detailed experimental settings ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Table</span> <span class="ltx_text ltx_ref_tag">8</span></a>. The system prompt follows the default settings from <cite class="ltx_cite ltx_citemacro_citep">(Ye et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib41" title="">2023</a>)</cite>. For evaluation, we use four prompts specifying three hydrogen bond donors, three hydrogen bond acceptors, a Bertz complexity between <math alttext="100" class="ltx_Math" display="inline" id="A2.p3.4.m4.1"><semantics id="A2.p3.4.m4.1a"><mn id="A2.p3.4.m4.1.1" xref="A2.p3.4.m4.1.1.cmml">100</mn><annotation-xml encoding="MathML-Content" id="A2.p3.4.m4.1b"><cn id="A2.p3.4.m4.1.1.cmml" type="integer" xref="A2.p3.4.m4.1.1">100</cn></annotation-xml><annotation encoding="application/x-tex" id="A2.p3.4.m4.1c">100</annotation><annotation encoding="application/x-llamapun" id="A2.p3.4.m4.1d">100</annotation></semantics></math> and <math alttext="200" class="ltx_Math" display="inline" id="A2.p3.5.m5.1"><semantics id="A2.p3.5.m5.1a"><mn id="A2.p3.5.m5.1.1" xref="A2.p3.5.m5.1.1.cmml">200</mn><annotation-xml encoding="MathML-Content" id="A2.p3.5.m5.1b"><cn id="A2.p3.5.m5.1.1.cmml" type="integer" xref="A2.p3.5.m5.1.1">200</cn></annotation-xml><annotation encoding="application/x-tex" id="A2.p3.5.m5.1c">200</annotation><annotation encoding="application/x-llamapun" id="A2.p3.5.m5.1d">200</annotation></semantics></math>, and a QED value between <math alttext="0.4" class="ltx_Math" display="inline" id="A2.p3.6.m6.1"><semantics id="A2.p3.6.m6.1a"><mn id="A2.p3.6.m6.1.1" xref="A2.p3.6.m6.1.1.cmml">0.4</mn><annotation-xml encoding="MathML-Content" id="A2.p3.6.m6.1b"><cn id="A2.p3.6.m6.1.1.cmml" type="float" xref="A2.p3.6.m6.1.1">0.4</cn></annotation-xml><annotation encoding="application/x-tex" id="A2.p3.6.m6.1c">0.4</annotation><annotation encoding="application/x-llamapun" id="A2.p3.6.m6.1d">0.4</annotation></semantics></math> and <math alttext="0.6" class="ltx_Math" display="inline" id="A2.p3.7.m7.1"><semantics id="A2.p3.7.m7.1a"><mn id="A2.p3.7.m7.1.1" xref="A2.p3.7.m7.1.1.cmml">0.6</mn><annotation-xml encoding="MathML-Content" id="A2.p3.7.m7.1b"><cn id="A2.p3.7.m7.1.1.cmml" type="float" xref="A2.p3.7.m7.1.1">0.6</cn></annotation-xml><annotation encoding="application/x-tex" id="A2.p3.7.m7.1c">0.6</annotation><annotation encoding="application/x-llamapun" id="A2.p3.7.m7.1d">0.6</annotation></semantics></math>. Additionally, as shown in <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S1.F3" title="In 1 Introduction ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Figure</span> <span class="ltx_text ltx_ref_tag">3</span></a>, we try to generate diverse molecules by designing prompts without fine-tuning (<a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#A2.T9" title="In Appendix B Detailed experimental settings ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Table</span> <span class="ltx_text ltx_ref_tag">9</span></a>). However, these prompts show lower performance compared to applying beam search.</p>
</div>
<figure class="ltx_table" id="A2.T8">
<figcaption class="ltx_caption"><span class="ltx_tag ltx_tag_table">Table 8: </span><span class="ltx_text ltx_font_bold" id="A2.T8.4.1">Prompts for DrugAssist <cite class="ltx_cite ltx_citemacro_citep">(Ye et al., <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#bib.bib41" title="">2023</a>)</cite>.</span></figcaption>
<table class="ltx_tabular ltx_centering ltx_align_middle" id="A2.T8.2">
<tr class="ltx_tr" id="A2.T8.2.3">
<td class="ltx_td ltx_align_left ltx_border_tt" id="A2.T8.2.3.1">Prompt</td>
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_tt" id="A2.T8.2.3.2" style="width:278.2pt;">
<span class="ltx_inline-block ltx_align_top" id="A2.T8.2.3.2.1">
<span class="ltx_p" id="A2.T8.2.3.2.1.1">Contents</span>
</span>
</td>
</tr>
<tr class="ltx_tr" id="A2.T8.1.1">
<td class="ltx_td ltx_align_left ltx_border_t" id="A2.T8.1.1.1"><math alttext="p_{\text{desc}}" class="ltx_Math" display="inline" id="A2.T8.1.1.1.m1.1"><semantics id="A2.T8.1.1.1.m1.1a"><msub id="A2.T8.1.1.1.m1.1.1" xref="A2.T8.1.1.1.m1.1.1.cmml"><mi id="A2.T8.1.1.1.m1.1.1.2" xref="A2.T8.1.1.1.m1.1.1.2.cmml">p</mi><mtext id="A2.T8.1.1.1.m1.1.1.3" xref="A2.T8.1.1.1.m1.1.1.3a.cmml">desc</mtext></msub><annotation-xml encoding="MathML-Content" id="A2.T8.1.1.1.m1.1b"><apply id="A2.T8.1.1.1.m1.1.1.cmml" xref="A2.T8.1.1.1.m1.1.1"><csymbol cd="ambiguous" id="A2.T8.1.1.1.m1.1.1.1.cmml" xref="A2.T8.1.1.1.m1.1.1">subscript</csymbol><ci id="A2.T8.1.1.1.m1.1.1.2.cmml" xref="A2.T8.1.1.1.m1.1.1.2">𝑝</ci><ci id="A2.T8.1.1.1.m1.1.1.3a.cmml" xref="A2.T8.1.1.1.m1.1.1.3"><mtext id="A2.T8.1.1.1.m1.1.1.3.cmml" mathsize="70%" xref="A2.T8.1.1.1.m1.1.1.3">desc</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="A2.T8.1.1.1.m1.1c">p_{\text{desc}}</annotation><annotation encoding="application/x-llamapun" id="A2.T8.1.1.1.m1.1d">italic_p start_POSTSUBSCRIPT desc end_POSTSUBSCRIPT</annotation></semantics></math></td>
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_t" id="A2.T8.1.1.2" style="width:278.2pt;">
<span class="ltx_inline-block ltx_align_top" id="A2.T8.1.1.2.1">
<span class="ltx_p" id="A2.T8.1.1.2.1.1">Hydrogen bond donors and acceptors: “Can you generate a molecule with &lt;value&gt; &lt;property&gt;? Print it in SMILES format.”</span>
<span class="ltx_p" id="A2.T8.1.1.2.1.2">QED and Bertz complexity: “Can you generate a molecule with &lt;property&gt; below &lt;value1&gt; but at least &lt;value2&gt;? Print it in SMILES format.”</span>
</span>
</td>
</tr>
<tr class="ltx_tr" id="A2.T8.2.2">
<td class="ltx_td ltx_align_left ltx_border_bb ltx_border_t" id="A2.T8.2.2.1">
<math alttext="p_{\text{desc+div}}" class="ltx_Math" display="inline" id="A2.T8.2.2.1.m1.1"><semantics id="A2.T8.2.2.1.m1.1a"><msub id="A2.T8.2.2.1.m1.1.1" xref="A2.T8.2.2.1.m1.1.1.cmml"><mi id="A2.T8.2.2.1.m1.1.1.2" xref="A2.T8.2.2.1.m1.1.1.2.cmml">p</mi><mtext id="A2.T8.2.2.1.m1.1.1.3" xref="A2.T8.2.2.1.m1.1.1.3a.cmml">desc+div</mtext></msub><annotation-xml encoding="MathML-Content" id="A2.T8.2.2.1.m1.1b"><apply id="A2.T8.2.2.1.m1.1.1.cmml" xref="A2.T8.2.2.1.m1.1.1"><csymbol cd="ambiguous" id="A2.T8.2.2.1.m1.1.1.1.cmml" xref="A2.T8.2.2.1.m1.1.1">subscript</csymbol><ci id="A2.T8.2.2.1.m1.1.1.2.cmml" xref="A2.T8.2.2.1.m1.1.1.2">𝑝</ci><ci id="A2.T8.2.2.1.m1.1.1.3a.cmml" xref="A2.T8.2.2.1.m1.1.1.3"><mtext id="A2.T8.2.2.1.m1.1.1.3.cmml" mathsize="70%" xref="A2.T8.2.2.1.m1.1.1.3">desc+div</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="A2.T8.2.2.1.m1.1c">p_{\text{desc+div}}</annotation><annotation encoding="application/x-llamapun" id="A2.T8.2.2.1.m1.1d">italic_p start_POSTSUBSCRIPT desc+div end_POSTSUBSCRIPT</annotation></semantics></math> (fine-tuning)</td>
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_bb ltx_border_t" id="A2.T8.2.2.2" style="width:278.2pt;">
<span class="ltx_inline-block ltx_align_top" id="A2.T8.2.2.2.1">
<span class="ltx_p" id="A2.T8.2.2.2.1.1">Hydrogen bond donors and acceptors: “Can you generate a set of molecules that have &lt;value&gt; &lt;property&gt;? Print each of them in SMILES format.”</span>
<span class="ltx_p" id="A2.T8.2.2.2.1.2">QED and Bertz complexity: “Can you generate a set of molecules that have &lt;property&gt; below &lt;value1&gt; but at least &lt;value2&gt;? Print each of them in SMILES format.”</span>
</span>
</td>
</tr>
</table>
</figure>
<figure class="ltx_table" id="A2.T9">
<figcaption class="ltx_caption"><span class="ltx_tag ltx_tag_table">Table 9: </span><span class="ltx_text ltx_font_bold" id="A2.T9.3.1">Prompts for DrugAssist (without fine-tuning, <a class="ltx_ref" href="https://arxiv.org/html/2410.03138v1#S1.F3" title="In 1 Introduction ‣ Can LLMs Generate Diverse Molecules? Towards Alignment with Structural Diversity"><span class="ltx_text ltx_ref_tag">Figure</span> <span class="ltx_text ltx_ref_tag">3</span></a>).</span></figcaption>
<div class="ltx_inline-block ltx_align_center ltx_transformed_outer" id="A2.T9.1" style="width:38.4pt;height:15016.8pt;vertical-align:-0.0pt;"><span class="ltx_transformed_inner" style="transform:translate(-2.1pt,834.3pt) scale(0.9,0.9) ;">
<table class="ltx_tabular ltx_align_middle" id="A2.T9.1.1">
<tr class="ltx_tr" id="A2.T9.1.1.1">
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_tt" id="A2.T9.1.1.1.1" style="width:385.6pt;">
<span class="ltx_inline-block ltx_align_top" id="A2.T9.1.1.1.1.1">
<span class="ltx_p" id="A2.T9.1.1.1.1.1.1"><math alttext="p_{\text{desc+div}}" class="ltx_Math" display="inline" id="A2.T9.1.1.1.1.1.1.m1.1"><semantics id="A2.T9.1.1.1.1.1.1.m1.1a"><msub id="A2.T9.1.1.1.1.1.1.m1.1.1" xref="A2.T9.1.1.1.1.1.1.m1.1.1.cmml"><mi id="A2.T9.1.1.1.1.1.1.m1.1.1.2" xref="A2.T9.1.1.1.1.1.1.m1.1.1.2.cmml">p</mi><mtext id="A2.T9.1.1.1.1.1.1.m1.1.1.3" xref="A2.T9.1.1.1.1.1.1.m1.1.1.3a.cmml">desc+div</mtext></msub><annotation-xml encoding="MathML-Content" id="A2.T9.1.1.1.1.1.1.m1.1b"><apply id="A2.T9.1.1.1.1.1.1.m1.1.1.cmml" xref="A2.T9.1.1.1.1.1.1.m1.1.1"><csymbol cd="ambiguous" id="A2.T9.1.1.1.1.1.1.m1.1.1.1.cmml" xref="A2.T9.1.1.1.1.1.1.m1.1.1">subscript</csymbol><ci id="A2.T9.1.1.1.1.1.1.m1.1.1.2.cmml" xref="A2.T9.1.1.1.1.1.1.m1.1.1.2">𝑝</ci><ci id="A2.T9.1.1.1.1.1.1.m1.1.1.3a.cmml" xref="A2.T9.1.1.1.1.1.1.m1.1.1.3"><mtext id="A2.T9.1.1.1.1.1.1.m1.1.1.3.cmml" mathsize="70%" xref="A2.T9.1.1.1.1.1.1.m1.1.1.3">desc+div</mtext></ci></apply></annotation-xml><annotation encoding="application/x-tex" id="A2.T9.1.1.1.1.1.1.m1.1c">p_{\text{desc+div}}</annotation><annotation encoding="application/x-llamapun" id="A2.T9.1.1.1.1.1.1.m1.1d">italic_p start_POSTSUBSCRIPT desc+div end_POSTSUBSCRIPT</annotation></semantics></math></span>
</span>
</td>
</tr>
<tr class="ltx_tr" id="A2.T9.1.1.2">
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_t" id="A2.T9.1.1.2.1" style="width:385.6pt;">
<span class="ltx_inline-block ltx_align_top" id="A2.T9.1.1.2.1.1">
<span class="ltx_p" id="A2.T9.1.1.2.1.1.1">“Can you generate a set of molecules? Each molecule has &lt;value&gt; &lt;property&gt;. Print each of them in SMILES format.”</span>
</span>
</td>
</tr>
<tr class="ltx_tr" id="A2.T9.1.1.3">
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_t" id="A2.T9.1.1.3.1" style="width:385.6pt;">
<span class="ltx_inline-block ltx_align_top" id="A2.T9.1.1.3.1.1">
<span class="ltx_p" id="A2.T9.1.1.3.1.1.1">“Can you generate a diverse set of molecules? Each molecule has &lt;value&gt; &lt;property&gt;. Print each of them in SMILES format.”</span>
</span>
</td>
</tr>
<tr class="ltx_tr" id="A2.T9.1.1.4">
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_t" id="A2.T9.1.1.4.1" style="width:385.6pt;">
<span class="ltx_inline-block ltx_align_top" id="A2.T9.1.1.4.1.1">
<span class="ltx_p" id="A2.T9.1.1.4.1.1.1">“Can you generate a structurally diverse set of molecules? Each molecule has &lt;value&gt; &lt;property&gt;. Print each of them in SMILES format.”</span>
</span>
</td>
</tr>
<tr class="ltx_tr" id="A2.T9.1.1.5">
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_t" id="A2.T9.1.1.5.1" style="width:385.6pt;">
<span class="ltx_inline-block ltx_align_top" id="A2.T9.1.1.5.1.1">
<span class="ltx_p" id="A2.T9.1.1.5.1.1.1">“Can you generate fifty molecules? Each molecule has &lt;value&gt; &lt;property&gt;. Print each of them in SMILES format.”</span>
</span>
</td>
</tr>
<tr class="ltx_tr" id="A2.T9.1.1.6">
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_t" id="A2.T9.1.1.6.1" style="width:385.6pt;">
<span class="ltx_inline-block ltx_align_top" id="A2.T9.1.1.6.1.1">
<span class="ltx_p" id="A2.T9.1.1.6.1.1.1">“Can you generate fifty diverse molecules? Each molecule has &lt;value&gt; &lt;property&gt;. Print each of them in SMILES format.”</span>
</span>
</td>
</tr>
<tr class="ltx_tr" id="A2.T9.1.1.7">
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_t" id="A2.T9.1.1.7.1" style="width:385.6pt;">
<span class="ltx_inline-block ltx_align_top" id="A2.T9.1.1.7.1.1">
<span class="ltx_p" id="A2.T9.1.1.7.1.1.1">“Can you generate fifty structurally diverse molecules? Each molecule has &lt;value&gt; &lt;property&gt;. Print each of them in SMILES format.”</span>
</span>
</td>
</tr>
<tr class="ltx_tr" id="A2.T9.1.1.8">
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_t" id="A2.T9.1.1.8.1" style="width:385.6pt;">
<span class="ltx_inline-block ltx_align_top" id="A2.T9.1.1.8.1.1">
<span class="ltx_p" id="A2.T9.1.1.8.1.1.1">“Can you generate a set of molecules with &lt;value&gt; &lt;property&gt;? Print each of them in SMILES format.”</span>
</span>
</td>
</tr>
<tr class="ltx_tr" id="A2.T9.1.1.9">
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_t" id="A2.T9.1.1.9.1" style="width:385.6pt;">
<span class="ltx_inline-block ltx_align_top" id="A2.T9.1.1.9.1.1">
<span class="ltx_p" id="A2.T9.1.1.9.1.1.1">“Can you generate a diverse set of molecules with &lt;value&gt; &lt;property&gt;? Print each of them in SMILES format.”</span>
</span>
</td>
</tr>
<tr class="ltx_tr" id="A2.T9.1.1.10">
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_t" id="A2.T9.1.1.10.1" style="width:385.6pt;">
<span class="ltx_inline-block ltx_align_top" id="A2.T9.1.1.10.1.1">
<span class="ltx_p" id="A2.T9.1.1.10.1.1.1">“Can you generate a structurally diverse set of molecules with &lt;value&gt; &lt;property&gt;? Print each of them in SMILES format.”</span>
</span>
</td>
</tr>
<tr class="ltx_tr" id="A2.T9.1.1.11">
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_t" id="A2.T9.1.1.11.1" style="width:385.6pt;">
<span class="ltx_inline-block ltx_align_top" id="A2.T9.1.1.11.1.1">
<span class="ltx_p" id="A2.T9.1.1.11.1.1.1">“Can you generate fifty molecules with &lt;value&gt; &lt;property&gt;? Print each of them in SMILES format.”</span>
</span>
</td>
</tr>
<tr class="ltx_tr" id="A2.T9.1.1.12">
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_t" id="A2.T9.1.1.12.1" style="width:385.6pt;">
<span class="ltx_inline-block ltx_align_top" id="A2.T9.1.1.12.1.1">
<span class="ltx_p" id="A2.T9.1.1.12.1.1.1">“Can you generate fifty diverse molecules with &lt;value&gt; &lt;property&gt;? Print each of them in SMILES format.”</span>
</span>
</td>
</tr>
<tr class="ltx_tr" id="A2.T9.1.1.13">
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_bb ltx_border_t" id="A2.T9.1.1.13.1" style="width:385.6pt;">
<span class="ltx_inline-block ltx_align_top" id="A2.T9.1.1.13.1.1">
<span class="ltx_p" id="A2.T9.1.1.13.1.1.1">“Can you generate fifty structurally diverse molecules with &lt;value&gt; &lt;property&gt;? Print each of them in SMILES format.”</span>
</span>
</td>
</tr>
</table>
</span></div>
</figure>
<div class="ltx_pagination ltx_role_newpage"></div>
</section>
<section class="ltx_appendix" id="A3">
<h2 class="ltx_title ltx_title_appendix">
<span class="ltx_tag ltx_tag_appendix">Appendix C </span>Additional results</h2>
<figure class="ltx_table" id="A3.T10">
<figcaption class="ltx_caption"><span class="ltx_tag ltx_tag_table">Table 10: </span><span class="ltx_text ltx_font_bold" id="A3.T10.6.1">Visualization of the generated molecules from beam search.</span> The <math alttext="48" class="ltx_Math" display="inline" id="A3.T10.2.m1.1"><semantics id="A3.T10.2.m1.1b"><mn id="A3.T10.2.m1.1.1" xref="A3.T10.2.m1.1.1.cmml">48</mn><annotation-xml encoding="MathML-Content" id="A3.T10.2.m1.1c"><cn id="A3.T10.2.m1.1.1.cmml" type="integer" xref="A3.T10.2.m1.1.1">48</cn></annotation-xml><annotation encoding="application/x-tex" id="A3.T10.2.m1.1d">48</annotation><annotation encoding="application/x-llamapun" id="A3.T10.2.m1.1e">48</annotation></semantics></math> molecules are generated with the description: “The molecule is a primary aliphatic ammonium ion which is obtained from streptothricin F by protonation of the guanidino and amino groups. It has a role as an antimicrobial agent. It is a guanidinium ion and a primary aliphatic ammonium ion. It is a conjugate acid of a streptothricin F’.”</figcaption>
<table class="ltx_tabular ltx_align_middle" id="A3.T10.4">
<tr class="ltx_tr" id="A3.T10.3.1">
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_tt" id="A3.T10.3.1.1" style="width:377.6pt;">
<span class="ltx_inline-block ltx_align_top" id="A3.T10.3.1.1.1">
<span class="ltx_p" id="A3.T10.3.1.1.1.1">Example of outputs (IntDiv. is <math alttext="0.53" class="ltx_Math" display="inline" id="A3.T10.3.1.1.1.1.m1.1"><semantics id="A3.T10.3.1.1.1.1.m1.1a"><mn id="A3.T10.3.1.1.1.1.m1.1.1" xref="A3.T10.3.1.1.1.1.m1.1.1.cmml">0.53</mn><annotation-xml encoding="MathML-Content" id="A3.T10.3.1.1.1.1.m1.1b"><cn id="A3.T10.3.1.1.1.1.m1.1.1.cmml" type="float" xref="A3.T10.3.1.1.1.1.m1.1.1">0.53</cn></annotation-xml><annotation encoding="application/x-tex" id="A3.T10.3.1.1.1.1.m1.1c">0.53</annotation><annotation encoding="application/x-llamapun" id="A3.T10.3.1.1.1.1.m1.1d">0.53</annotation></semantics></math>)</span>
</span>
</td>
</tr>
<tr class="ltx_tr" id="A3.T10.4.2">
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_bb ltx_border_t" id="A3.T10.4.2.1" style="width:377.6pt;">
<span class="ltx_inline-block ltx_align_top" id="A3.T10.4.2.1.1"><img alt="[Uncaptioned image]" class="ltx_graphics ltx_img_portrait" height="964" id="A3.T10.4.2.1.1.g1" src="x22.png" width="723"/>
</span>
</td>
</tr>
</table>
</figure>
<figure class="ltx_table" id="A3.T11">
<figcaption class="ltx_caption"><span class="ltx_tag ltx_tag_table">Table 11: </span><span class="ltx_text ltx_font_bold" id="A3.T11.6.1">Visualization of the generated molecules from diverse beam search.</span> The <math alttext="48" class="ltx_Math" display="inline" id="A3.T11.2.m1.1"><semantics id="A3.T11.2.m1.1b"><mn id="A3.T11.2.m1.1.1" xref="A3.T11.2.m1.1.1.cmml">48</mn><annotation-xml encoding="MathML-Content" id="A3.T11.2.m1.1c"><cn id="A3.T11.2.m1.1.1.cmml" type="integer" xref="A3.T11.2.m1.1.1">48</cn></annotation-xml><annotation encoding="application/x-tex" id="A3.T11.2.m1.1d">48</annotation><annotation encoding="application/x-llamapun" id="A3.T11.2.m1.1e">48</annotation></semantics></math> molecules are generated with the description: “The molecule is a primary aliphatic ammonium ion which is obtained from streptothricin F by protonation of the guanidino and amino groups. It has a role as an antimicrobial agent. It is a guanidinium ion and a primary aliphatic ammonium ion. It is a conjugate acid of a streptothricin F’.”</figcaption>
<table class="ltx_tabular ltx_align_middle" id="A3.T11.4">
<tr class="ltx_tr" id="A3.T11.3.1">
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_tt" id="A3.T11.3.1.1" style="width:377.6pt;">
<span class="ltx_inline-block ltx_align_top" id="A3.T11.3.1.1.1">
<span class="ltx_p" id="A3.T11.3.1.1.1.1">Example of outputs (IntDiv. is <math alttext="0.67" class="ltx_Math" display="inline" id="A3.T11.3.1.1.1.1.m1.1"><semantics id="A3.T11.3.1.1.1.1.m1.1a"><mn id="A3.T11.3.1.1.1.1.m1.1.1" xref="A3.T11.3.1.1.1.1.m1.1.1.cmml">0.67</mn><annotation-xml encoding="MathML-Content" id="A3.T11.3.1.1.1.1.m1.1b"><cn id="A3.T11.3.1.1.1.1.m1.1.1.cmml" type="float" xref="A3.T11.3.1.1.1.1.m1.1.1">0.67</cn></annotation-xml><annotation encoding="application/x-tex" id="A3.T11.3.1.1.1.1.m1.1c">0.67</annotation><annotation encoding="application/x-llamapun" id="A3.T11.3.1.1.1.1.m1.1d">0.67</annotation></semantics></math>)</span>
</span>
</td>
</tr>
<tr class="ltx_tr" id="A3.T11.4.2">
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_bb ltx_border_t" id="A3.T11.4.2.1" style="width:377.6pt;">
<span class="ltx_inline-block ltx_align_top" id="A3.T11.4.2.1.1"><img alt="[Uncaptioned image]" class="ltx_graphics ltx_img_portrait" height="964" id="A3.T11.4.2.1.1.g1" src="x23.png" width="723"/>
</span>
</td>
</tr>
</table>
</figure>
<figure class="ltx_table" id="A3.T12">
<figcaption class="ltx_caption"><span class="ltx_tag ltx_tag_table">Table 12: </span><span class="ltx_text ltx_font_bold" id="A3.T12.6.1">Visualization of the generated molecules from constrastive beam search.</span> The <math alttext="48" class="ltx_Math" display="inline" id="A3.T12.2.m1.1"><semantics id="A3.T12.2.m1.1b"><mn id="A3.T12.2.m1.1.1" xref="A3.T12.2.m1.1.1.cmml">48</mn><annotation-xml encoding="MathML-Content" id="A3.T12.2.m1.1c"><cn id="A3.T12.2.m1.1.1.cmml" type="integer" xref="A3.T12.2.m1.1.1">48</cn></annotation-xml><annotation encoding="application/x-tex" id="A3.T12.2.m1.1d">48</annotation><annotation encoding="application/x-llamapun" id="A3.T12.2.m1.1e">48</annotation></semantics></math> molecules are generated with the description: “The molecule is a primary aliphatic ammonium ion which is obtained from streptothricin F by protonation of the guanidino and amino groups. It has a role as an antimicrobial agent. It is a guanidinium ion and a primary aliphatic ammonium ion. It is a conjugate acid of a streptothricin F’.”</figcaption>
<table class="ltx_tabular ltx_align_middle" id="A3.T12.4">
<tr class="ltx_tr" id="A3.T12.3.1">
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_tt" id="A3.T12.3.1.1" style="width:377.6pt;">
<span class="ltx_inline-block ltx_align_top" id="A3.T12.3.1.1.1">
<span class="ltx_p" id="A3.T12.3.1.1.1.1">Example of outputs (IntDiv. is <math alttext="0.58" class="ltx_Math" display="inline" id="A3.T12.3.1.1.1.1.m1.1"><semantics id="A3.T12.3.1.1.1.1.m1.1a"><mn id="A3.T12.3.1.1.1.1.m1.1.1" xref="A3.T12.3.1.1.1.1.m1.1.1.cmml">0.58</mn><annotation-xml encoding="MathML-Content" id="A3.T12.3.1.1.1.1.m1.1b"><cn id="A3.T12.3.1.1.1.1.m1.1.1.cmml" type="float" xref="A3.T12.3.1.1.1.1.m1.1.1">0.58</cn></annotation-xml><annotation encoding="application/x-tex" id="A3.T12.3.1.1.1.1.m1.1c">0.58</annotation><annotation encoding="application/x-llamapun" id="A3.T12.3.1.1.1.1.m1.1d">0.58</annotation></semantics></math>)</span>
</span>
</td>
</tr>
<tr class="ltx_tr" id="A3.T12.4.2">
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_bb ltx_border_t" id="A3.T12.4.2.1" style="width:377.6pt;">
<span class="ltx_inline-block ltx_align_top" id="A3.T12.4.2.1.1"><img alt="[Uncaptioned image]" class="ltx_graphics ltx_img_portrait" height="964" id="A3.T12.4.2.1.1.g1" src="x24.png" width="723"/>
</span>
</td>
</tr>
</table>
</figure>
<figure class="ltx_table" id="A3.T13">
<figcaption class="ltx_caption"><span class="ltx_tag ltx_tag_table">Table 13: </span><span class="ltx_text ltx_font_bold" id="A3.T13.6.1">Visualization of the generated molecules from our method.</span> The <math alttext="48" class="ltx_Math" display="inline" id="A3.T13.2.m1.1"><semantics id="A3.T13.2.m1.1b"><mn id="A3.T13.2.m1.1.1" xref="A3.T13.2.m1.1.1.cmml">48</mn><annotation-xml encoding="MathML-Content" id="A3.T13.2.m1.1c"><cn id="A3.T13.2.m1.1.1.cmml" type="integer" xref="A3.T13.2.m1.1.1">48</cn></annotation-xml><annotation encoding="application/x-tex" id="A3.T13.2.m1.1d">48</annotation><annotation encoding="application/x-llamapun" id="A3.T13.2.m1.1e">48</annotation></semantics></math> molecules are generated with the description: “The molecule is a primary aliphatic ammonium ion which is obtained from streptothricin F by protonation of the guanidino and amino groups. It has a role as an antimicrobial agent. It is a guanidinium ion and a primary aliphatic ammonium ion. It is a conjugate acid of a streptothricin F.”</figcaption>
<table class="ltx_tabular ltx_align_middle" id="A3.T13.4">
<tr class="ltx_tr" id="A3.T13.3.1">
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_tt" id="A3.T13.3.1.1" style="width:377.6pt;">
<span class="ltx_inline-block ltx_align_top" id="A3.T13.3.1.1.1">
<span class="ltx_p" id="A3.T13.3.1.1.1.1">Example of outputs (IntDiv. is <math alttext="0.77" class="ltx_Math" display="inline" id="A3.T13.3.1.1.1.1.m1.1"><semantics id="A3.T13.3.1.1.1.1.m1.1a"><mn id="A3.T13.3.1.1.1.1.m1.1.1" xref="A3.T13.3.1.1.1.1.m1.1.1.cmml">0.77</mn><annotation-xml encoding="MathML-Content" id="A3.T13.3.1.1.1.1.m1.1b"><cn id="A3.T13.3.1.1.1.1.m1.1.1.cmml" type="float" xref="A3.T13.3.1.1.1.1.m1.1.1">0.77</cn></annotation-xml><annotation encoding="application/x-tex" id="A3.T13.3.1.1.1.1.m1.1c">0.77</annotation><annotation encoding="application/x-llamapun" id="A3.T13.3.1.1.1.1.m1.1d">0.77</annotation></semantics></math>)</span>
</span>
</td>
</tr>
<tr class="ltx_tr" id="A3.T13.4.2">
<td class="ltx_td ltx_align_justify ltx_align_middle ltx_border_bb ltx_border_t" id="A3.T13.4.2.1" style="width:377.6pt;">
<span class="ltx_inline-block ltx_align_top" id="A3.T13.4.2.1.1"><img alt="[Uncaptioned image]" class="ltx_graphics ltx_img_portrait" height="964" id="A3.T13.4.2.1.1.g1" src="x25.png" width="723"/>
</span>
</td>
</tr>
</table>
</figure>
<div class="ltx_pagination ltx_role_newpage"></div>
</section>
</article>
</div>
<footer class="ltx_page_footer">
<div class="ltx_page_logo">Generated  on Fri Oct  4 04:25:14 2024 by <a class="ltx_LaTeXML_logo" href="http://dlmf.nist.gov/LaTeXML/"><span style="letter-spacing:-0.2em; margin-right:0.1em;">L<span class="ltx_font_smallcaps" style="position:relative; bottom:2.2pt;">a</span>T<span class="ltx_font_smallcaps" style="font-size:120%;position:relative; bottom:-0.2ex;">e</span></span><span style="font-size:90%; position:relative; bottom:-0.2ex;">XML</span><img alt="Mascot Sammy" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAsAAAAOCAYAAAD5YeaVAAAAAXNSR0IArs4c6QAAAAZiS0dEAP8A/wD/oL2nkwAAAAlwSFlzAAALEwAACxMBAJqcGAAAAAd0SU1FB9wKExQZLWTEaOUAAAAddEVYdENvbW1lbnQAQ3JlYXRlZCB3aXRoIFRoZSBHSU1Q72QlbgAAAdpJREFUKM9tkL+L2nAARz9fPZNCKFapUn8kyI0e4iRHSR1Kb8ng0lJw6FYHFwv2LwhOpcWxTjeUunYqOmqd6hEoRDhtDWdA8ApRYsSUCDHNt5ul13vz4w0vWCgUnnEc975arX6ORqN3VqtVZbfbTQC4uEHANM3jSqXymFI6yWazP2KxWAXAL9zCUa1Wy2tXVxheKA9YNoR8Pt+aTqe4FVVVvz05O6MBhqUIBGk8Hn8HAOVy+T+XLJfLS4ZhTiRJgqIoVBRFIoric47jPnmeB1mW/9rr9ZpSSn3Lsmir1fJZlqWlUonKsvwWwD8ymc/nXwVBeLjf7xEKhdBut9Hr9WgmkyGEkJwsy5eHG5vN5g0AKIoCAEgkEkin0wQAfN9/cXPdheu6P33fBwB4ngcAcByHJpPJl+fn54mD3Gg0NrquXxeLRQAAwzAYj8cwTZPwPH9/sVg8PXweDAauqqr2cDjEer1GJBLBZDJBs9mE4zjwfZ85lAGg2+06hmGgXq+j3+/DsixYlgVN03a9Xu8jgCNCyIegIAgx13Vfd7vdu+FweG8YRkjXdWy329+dTgeSJD3ieZ7RNO0VAXAPwDEAO5VKndi2fWrb9jWl9Esul6PZbDY9Go1OZ7PZ9z/lyuD3OozU2wAAAABJRU5ErkJggg=="/></a>
</div></footer>
</div>
</body>
</html>
